{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Antaki2021_R_analysis.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "machine_shape": "hm",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "ir",
      "display_name": "R"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/sebatlab/Antaki2021/blob/main/Antaki2021_R_analysis.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QywIWFWURQvm"
      },
      "source": [
        "# Table of Contents\n",
        "-------------------\n",
        "* [Variance Explained by Genetic Risk Factors](#variance_explained_by_genetic_risk_factors)\n",
        "* [Negative Correlations of Rare and Common](#negative_correlations_rare_common)\n",
        "* [Gene-Phenotype Correlations](#gene_phenotype_correlations)"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Jy0RET5LJLFr",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4a66ce4e-b449-41e4-8dad-1d904a3468e5"
      },
      "source": [
        "# Install packages\n",
        "install.packages(\"gplots\")\n",
        "install.packages(\"basicTrendline\")\n",
        "\n",
        "# Attaching packages\n",
        "library(gplots)\n",
        "library(basicTrendline)\n",
        "\n",
        "\n",
        "# Figure 2A packages\n",
        "install.packages(\"fmsb\")\n",
        "install.packages(\"lmtest\")\n",
        "install.packages(\"boot\")\n",
        "install.packages(\"stringr\")\n",
        "\n",
        "library(fmsb)\n",
        "library(lmtest)\n",
        "library(boot)\n",
        "library(stringr)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "L5iqs7Wj7JCh"
      },
      "source": [
        "# Input filenames and output filenames. I should add which sections take in\n",
        "# which inputs and outputs.\n",
        "\n",
        "filename_table = \"SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv\"\n",
        "filename_phen  = \"master_phen_4.SBayesR.csv\"\n",
        "\n",
        "r2_out = \"R2_estimates.20210715.csv\"\n",
        "responses_out = \"SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv\"\n",
        "asd_genetic_correlations_by_sex_out = \"asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv\"\n",
        "clin_results_out = \"clin_results_eur.SBayesR.20210715.csv\""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "gdLB3RK7zDF3"
      },
      "source": [
        "<a name=\"variance_explained_by_genetic_risk_factors\"></a>\n",
        "## Variance Explained by Genetic Risk Factors (Fig. 2A)"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "aIGYenzFiRpu"
      },
      "source": [
        "# Load table\n",
        "df <- read.table(filename_table, header = TRUE, sep = \",\")\n",
        "df <- df[df$EUR.Ancestry == 1,] # Only use samples with European ancestry\n",
        "df <- df[df$Family.Type == \"trio\",] # Only use trios for this analysis\n",
        "# Drop SV columns (not used in this analysis)\n",
        "drops <- c(\"dnSV\",\n",
        "           \"SV.LoF.Paternal.Transmitted\",\"SV.LoF.Paternal.Untransmitted\",\n",
        "          \"SV.LoF.Maternal.Transmitted\",\"SV.LoF.Maternal.Untransmitted\",\n",
        "          \"CRE.SV.Paternal.Transmitted\",\"CRE.SV.Paternal.Untransmitted\",\n",
        "          \"CRE.SV.Maternal.Transmitted\",\"CRE.SV.Maternal.Untransmitted\",\n",
        "          \"RVRS\",\"PRS\",\"GRS\")\n",
        "df <- df[ , !(names(df) %in% drops)]\n",
        "df <- na.omit(df) # Remove rows with NA values\n",
        "\n",
        "# Make cohort variables\n",
        "df$REACH_cohort = as.numeric(df$Cohort == \"REACH\")\n",
        "df$SPARK_cohort = as.numeric(df$Cohort == \"SPARK\")\n",
        "df$SSC_cohort = as.numeric(df$Cohort == \"SSC\")\n",
        "\n",
        "dfm <- df[df$Sex == 0,] # Dataframe containing only male samples\n",
        "dff <- df[df$Sex == 1,] # Dataframe containing only female samples\n",
        "\n",
        "# Grab PCs\n",
        "# df[,grep(\"^PC([1-9]|10)$\", names(df), value=TRUE)]\n",
        "# Grab Sex\n",
        "# df[,grep(\"^Sex$\", names(df), value=TRUE)]\n",
        "# Grab Phenotype\n",
        "# df[,grep(\"^Phenotype$\", names(df), value=TRUE)]\n",
        "\n",
        "# Sample rows (without replacement by default?)\n",
        "# df[sample(nrow(df), 1000), ]"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "WXefMeZHyVoN"
      },
      "source": [
        "# Written by James Guevara\n",
        "\n",
        "# The bootstrap is used to estimate a confidence interval for the R2 estimates.\n",
        "# However, the bootstrap procedure seemingly requires a lot of CPUs and takes a lot of time\n",
        "# given the amount of data in the table.\n",
        "# So I ran the procedure on a compute cluster (Comet at UCSD) using the Slurm Workload Manager.\n",
        "# But I've adapted the code here so that the reader can understand how we ran the\n",
        "# bootstrap. \n",
        "getNagelkerkeR2_bootstrap <- function(formula, data, indices) {\n",
        "  d <- data[indices,]\n",
        "  fit_alt  <- glm(formula = formula, data = d, family = binomial)\n",
        "  fit_null <- glm(formula = d$Phenotype~d$Sex+d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort, data=d, family=binomial)\n",
        "  return(NagelkerkeR2(fit_alt)$R2 - NagelkerkeR2(fit_null)$R2)\n",
        "}\n",
        "\n",
        "getNagelkerkeR2 <- function(dataframe, alt_vars_in, null_vars_in, text) {\n",
        "  d <- dataframe\n",
        "  alt_variables <- alt_vars_in\n",
        "  alt_formula <- reformulate(alt_variables, response = \"Phenotype\")\n",
        "  null_variables <- null_vars_in\n",
        "  null_formula <- reformulate(null_variables, response = \"Phenotype\")\n",
        "  fit_alt  <- glm(formula = alt_formula , data = d, family = binomial)\n",
        "  fit_null <- glm(formula = null_formula , data = d, family = binomial)\n",
        "  nagelkerke_R2_val = NagelkerkeR2(fit_alt)$R2 - NagelkerkeR2(fit_null)$R2\n",
        "  lr <- lrtest(fit_alt,fit_null) # Likelihood ratio test\n",
        "  lr_p_val = lr[\"Pr(>Chisq)\"][2,1] # p-value\n",
        "\n",
        "  # cat(c(text,\"\\n\"))\n",
        "  # cat(c(\"R2:\\t\", nagelkerke_R2_val, \"\\t\", \"Likelihood_ratio_test_p-value:\\t\", lr_p_val, \"\\n\"))\n",
        "\n",
        "  # Put into dataframe? (R2, p-value)\n",
        "  df_r2 = data.frame(text, nagelkerke_R2_val, lr_p_val)\n",
        "\n",
        "  # Get the alt_vars_in c vector into a format appropriate for the bootstrap function \n",
        "  alt_vars_formula = paste(unlist(strsplit(alt_vars_in, split=\"d$\", fixed=TRUE)), collapse=\" \")\n",
        "  alt_vars_formula = paste(\"Phenotype ~\", alt_vars_formula, collapse= \" \")\n",
        "  # print(alt_vars_formula)\n",
        "\n",
        "  # Using the bootstrap to get the confidence interval estimates (and standard error of R2)\n",
        "  # r2_stats <- boot(data=d, statistic=getNagelkerkeR2_bootstrap, R=10000, formula=alt_vars_formula, parallel=\"multicore\", ncpus=2)\n",
        "  # ste = sd(r2_stats$t)\n",
        "  # ci = boot.ci(r2_stats, type=\"bca\") # Using the BCa interval (more information here: https://blogs.sas.com/content/iml/2017/07/12/bootstrap-bca-interval.html)\n",
        "  # ci_lower = as.double(ci$bca[1,4])\n",
        "  # ci_upper = as.double(ci$bca[1,5])\n",
        "  # df_r2_tmp = data.frame(text, nagelkerke_R2_val, lr_p_val, ste, ci_lower, ci_upper)\n",
        "\n",
        "\n",
        "  return(df_r2)\n",
        "} \n",
        "\n",
        "\n",
        "cohort_vars <- c(\"d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort\")\n",
        "pc_vars <- c(\"d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10\")\n",
        "sex_var <- c(\"d$Sex\") # Only use with combined models (it doesn't matter)\n",
        "null_vars <- paste(cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "\n",
        "\n",
        "# Full model\n",
        "var <- c(\"d$dnMIS+d$dnLOF+d$inhLoF+d$pTDT.ASD.Dev+d$pTDT.EA.Dev+d$pTDT.SCZ.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_full_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"combined_both\")\n",
        "# Males\n",
        "df_full_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"combined_male\")\n",
        "# Females\n",
        "df_full_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"combined_female\")\n",
        "\n",
        "\n",
        "# PRS model\n",
        "var <- c(\"d$pTDT.ASD.Dev+d$pTDT.EA.Dev+d$pTDT.SCZ.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_prs_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"prs_both\")\n",
        "# Males\n",
        "df_prs_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"prs_male\")\n",
        "# Females\n",
        "df_prs_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"prs_female\")\n",
        "\n",
        "# Rare model\n",
        "var <- c(\"d$dnMIS+d$dnLOF+d$inhLoF\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_rare_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"rare_both\")\n",
        "# Males\n",
        "df_rare_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"rare_male\")\n",
        "# Females\n",
        "df_rare_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"rare_female\")\n",
        "\n",
        "# pTDT.ASD.Dev\n",
        "var <- c(\"d$pTDT.ASD.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_asd_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"ptdt.asd.dev_both\")\n",
        "# Males\n",
        "df_ptdt_asd_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"ptdt.asd.dev_male\")\n",
        "# Females\n",
        "df_ptdt_asd_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"ptdt.asd.dev_female\")\n",
        "\n",
        "# pTDT.EA.Dev\n",
        "var <- c(\"d$pTDT.EA.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_ea_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"ptdt.ea.dev_both\")\n",
        "# Males\n",
        "df_ptdt_ea_r2_male =getNagelkerkeR2(dfm, alt_vars, null_vars, \"ptdt.ea.dev_male\")\n",
        "# Females\n",
        "df_ptdt_ea_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"ptdt.ea.dev_female\")\n",
        "\n",
        "# pTDT.SCZ.Dev\n",
        "var <- c(\"d$pTDT.SCZ.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_scz_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"ptdt.scz.dev_both\")\n",
        "# Males\n",
        "df_ptdt_scz_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"ptdt.scz.dev_male\")\n",
        "# Females\n",
        "df_ptdt_scz_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"ptdt.scz.dev_female\")\n",
        "\n",
        "# dnMIS\n",
        "var <- c(\"d$dnMIS\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_dnmis_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"dnMIS_both\")\n",
        "# Males\n",
        "df_dnmis_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"dnMIS_male\")\n",
        "# Females\n",
        "df_dnmis_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"dnMIS_female\")\n",
        "\n",
        "# dnLOF\n",
        "var <- c(\"d$dnLOF\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_dnlof_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"dnLOF_both\")\n",
        "# Males\n",
        "df_dnlof_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"dnLOF_male\")\n",
        "# Females\n",
        "df_dnlof_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"dnLOF_female\")\n",
        "\n",
        "# inhLoF\n",
        "var <- c(\"d$inhLoF\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_inhlof_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"inhLOF_both\")\n",
        "# Males\n",
        "df_inhlof_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"inhLOF_male\")\n",
        "# Females\n",
        "df_inhlof_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"inhLOF_female\")\n",
        "\n",
        "\n",
        "# Combine all the dataframes\n",
        "df_combined = rbind(df_full_r2, df_full_r2_male, df_full_r2_female,\n",
        "                    df_prs_r2, df_prs_r2_male, df_prs_r2_female,\n",
        "                    df_rare_r2, df_rare_r2_male, df_rare_r2_female,\n",
        "                    df_ptdt_asd_r2, df_ptdt_asd_r2_male, df_ptdt_asd_r2_female,\n",
        "                    df_ptdt_ea_r2, df_ptdt_ea_r2_male, df_ptdt_ea_r2_female,\n",
        "                    df_ptdt_scz_r2, df_ptdt_scz_r2_male, df_ptdt_scz_r2_female,\n",
        "                    df_dnmis_r2, df_dnmis_r2_male, df_dnmis_r2_female,\n",
        "                    df_dnlof_r2, df_dnlof_r2_male, df_dnlof_r2_female,\n",
        "                    df_inhlof_r2, df_inhlof_r2_male, df_inhlof_r2_female)\n",
        "\n",
        "# Add a header to the combined dataframe\n",
        "colnames(df_combined) <- c(\"R2_type\", \"R2\", \"Likelihood_Ratio_p_value\")\n",
        "\n",
        "# Write to file\n",
        "write.csv(df_combined, file=r2_out, row.name=TRUE)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "0KSm0hxZx1-l"
      },
      "source": [
        "# R2 by cohort\n",
        "pc_vars <- c(\"d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10\")\n",
        "sex_var <- c(\"d$Sex\") # Only use with combined models (it doesn't matter)\n",
        "null_vars <- paste(pc_vars, sex_var, sep = \"+\") \n",
        "\n",
        "df_reach <- df[df$REACH_cohort == 1,]\n",
        "df_spark <- df[df$SPARK_cohort == 1,]\n",
        "df_ssc   <- df[df$SSC_cohort == 1,]\n",
        "\n",
        "# pTDT.ASD.Dev\n",
        "var <- c(\"d$pTDT.ASD.Dev\")\n",
        "alt_vars  <- paste(var, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_asd_r2_reach = getNagelkerkeR2(df_reach, alt_vars, null_vars, \"ptdt.asd.dev_REACH\")\n",
        "df_ptdt_asd_r2_spark = getNagelkerkeR2(df_spark, alt_vars, null_vars, \"ptdt.asd.dev_SPARK\")\n",
        "df_ptdt_asd_r2_ssc   = getNagelkerkeR2(df_ssc, alt_vars, null_vars, \"ptdt.asd.dev_SSC\")\n",
        "\n",
        "# Combined cohort (I already did this..)\n",
        "cohort_vars <- c(\"d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort\")\n",
        "null_vars <- paste(cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_asd_r2_combined   = getNagelkerkeR2(df_ssc, alt_vars, null_vars, \"ptdt.asd.dev_Combined\")\n",
        "\n",
        "df_cohorts = rbind(df_ptdt_asd_r2_reach, df_ptdt_asd_r2_ssc, df_ptdt_asd_r2_spark, df_ptdt_asd_r2_combined)\n",
        "\n",
        "# Add a header to the combined dataframe\n",
        "colnames(df_cohorts) <- c(\"Model\", \"R2\", \"P-Value\")\n",
        "\n",
        "# Write to file\n",
        "write.csv(df_cohorts, file = \"SupplementaryTable5.PS_ASD_R2_by_cohort.csv\", row.name = TRUE)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "IxxT0d1O13sJ"
      },
      "source": [
        "# Responses from residuals\n",
        "library(fmsb)\n",
        "options(stringsAsFactors = FALSE);\n",
        "\n",
        "d = df\n",
        "\n",
        "# PRS scores residualized:\n",
        "ptdt_prs_asd_resid <- resid(glm(d$pTDT.ASD.Dev~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d, na.action=\"na.exclude\"))\n",
        "ptdt_prs_ea_resid  <- resid(glm(d$pTDT.EA.Dev  ~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d, na.action=\"na.exclude\"))\n",
        "ptdt_prs_scz_resid <- resid(glm(d$pTDT.SCZ.Dev~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d, na.action=\"na.exclude\"))\n",
        "\n",
        "\n",
        "# Fit combined prs model\n",
        "fit_alt_prs <- glm(d$Phenotype~ptdt_prs_asd_resid+ptdt_prs_ea_resid+ptdt_prs_scz_resid, data=df, family=binomial)\n",
        "response_prs = predict(fit_alt_prs, type = \"response\")\n",
        "\n",
        "# Rare scores residualized\n",
        "snv_dn_lof_resid <- resid(glm(d$dnLOF ~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d,na.action=\"na.exclude\"))\n",
        "snv_dn_mis_resid <- resid(glm(d$dnMIS ~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d,na.action=\"na.exclude\") )\n",
        "snv_lof_t_resid  <- resid(glm(d$inhLoF~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d,na.action=\"na.exclude\") )\n",
        "\n",
        "\n",
        "# Fit combined rare model\n",
        "fit_alt_rare <- glm(d$Phenotype~snv_dn_lof_resid+snv_dn_mis_resid+snv_lof_t_resid, data=d, family=binomial)\n",
        "response_rare = predict(fit_alt_rare, type = \"response\")\n",
        "\n",
        "# Fit rare + prs combined model\n",
        "fit_alt_combined <- glm(d$Phenotype~ptdt_prs_asd_resid+ptdt_prs_ea_resid+ptdt_prs_scz_resid+snv_dn_lof_resid+snv_dn_mis_resid+snv_lof_t_resid, data=d, family=binomial)\n",
        "response_combined = predict(fit_alt_combined, type = \"response\")\n",
        "\n",
        "# Fit individual features\n",
        "# ptdt_prs_asd\n",
        "fit_alt_ptdt_prs_asd <- glm(d$Phenotype~ptdt_prs_asd_resid, data=d, family=binomial)\n",
        "response_ptdt_prs_asd = predict(fit_alt_ptdt_prs_asd, type = \"response\")\n",
        "# ptdt_prs_ea\n",
        "fit_alt_ptdt_prs_ea <- glm(d$Phenotype~ptdt_prs_ea_resid, data=d, family=binomial)\n",
        "response_ptdt_prs_ea = predict(fit_alt_ptdt_prs_ea, type = \"response\")\n",
        "# ptdt_prs_scz\n",
        "fit_alt_ptdt_prs_scz <- glm(d$Phenotype~ptdt_prs_scz_resid, data=d, family=binomial)\n",
        "response_ptdt_prs_scz = predict(fit_alt_ptdt_prs_scz, type = \"response\")\n",
        "\n",
        "# snv_dn_lof\n",
        "fit_alt_snv_dn_lof <- glm(d$Phenotype~snv_dn_lof_resid, data=d, family=binomial)\n",
        "response_snv_dn_lof = predict(fit_alt_snv_dn_lof, type = \"response\")\n",
        "# snv_dn_mis\n",
        "fit_alt_snv_dn_mis <- glm(d$Phenotype~snv_dn_mis_resid, data=d, family=binomial)\n",
        "response_snv_dn_mis = predict(fit_alt_snv_dn_mis, type = \"response\")\n",
        "# snv_lof_t\n",
        "fit_alt_snv_lof_t <- glm(d$Phenotype~snv_lof_t_resid, data=d, family=binomial)\n",
        "response_snv_lof_t = predict(fit_alt_snv_lof_t, type = \"response\")\n",
        "\n",
        "# Responses (not z-scored yet)\n",
        "# d$all_combined_score = response_combined\n",
        "# d$prs_combined_score = response_prs\n",
        "# d$rare_combined_score = response_rare\n",
        "# d$response_ptdt_prs_asd = response_ptdt_prs_asd\n",
        "# d$response_ptdt_prs_asd = response_ptdt_prs_ea\n",
        "# d$response_ptdt_prs_scz = response_ptdt_prs_scz\n",
        "# d$response_snv_dn_lof = response_snv_dn_lof\n",
        "# d$response_snv_dn_mis = response_snv_dn_mis\n",
        "# d$response_snv_lof_t = response_snv_lof_t\n",
        "\n",
        "# Get GRS, PRS, RVRS columns (normalize all_combined_score, prs_combined_score, and rare_combined_score)\n",
        "d$GRS = (response_combined - mean(response_combined))/sd(response_combined)\n",
        "d$PRS = (response_prs - mean(response_prs))/sd(response_prs)\n",
        "d$RVRS = (response_rare - mean(response_rare))/sd(response_rare)\n",
        "\n",
        "write.csv(d, file=responses_out, row.name=TRUE)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JCR9iE2ZRXQk"
      },
      "source": [
        "<a name=\"negative_correlations_rare_common\"></a>\n",
        "## Negative correlations of rare and common (Fig. 3)\n",
        "-------------------------------"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "C8cypSPGag-u",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f23d85f2-1504-481b-e54f-a8f1fecdbc69"
      },
      "source": [
        "# Written by Danny Antaki, edited by James Guevara\n",
        "options(stringsAsFactors = FALSE);\n",
        "# df = read.csv(\"SummaryGeneticData_REACH_SSC_SPARK.csv\",sep=\",\")\n",
        "df = read.csv(responses_out,sep=\",\")\n",
        "df$dnAll = df$dnLOF+df$dnMIS\n",
        "df$cohort_covar <- factor(df$Cohort)\n",
        "df <- df[df$EUR.Ancestry==1,]\n",
        "vars<-c(\n",
        "    \"dnMIS\" ,\"dnLOF\",\"dnAll\",\n",
        "    \"inhLoF\", \"RVRS\" , \n",
        "    \"pTDT.ASD.Dev\" , \"pTDT.SCZ.Dev\" ,\"pTDT.EA.Dev\",\n",
        "    \"PRS\"\n",
        ");\n",
        "cols <- c(\"cohort_covar\",\"Sex\",\"Phenotype\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\",\"PC6\",\"PC7\",\"PC8\",\"PC9\",\"PC10\",vars)\n",
        "df<-na.omit(df[cols])\n",
        "print(nrow(df))\n",
        "###########################################\n",
        "fit_model<- function(df,x,y,add_sex=FALSE){\n",
        "    Y<- df[[x]]\n",
        "    X<- df[[y]]\n",
        "    PC1<-df$PC1\n",
        "    PC2<-df$PC2\n",
        "    PC3<-df$PC3\n",
        "    PC4<-df$PC4\n",
        "    PC5<-df$PC5\n",
        "    PC6<-df$PC6\n",
        "    PC7<-df$PC7\n",
        "    PC8<-df$PC8\n",
        "    PC9<-df$PC9\n",
        "    PC10<-df$PC10\n",
        "    phen<-df$Phenotype\n",
        "    cohort_covar<-df$cohort_covar\n",
        "    sex<-df$Sex\n",
        "    sex_pv<-NA    \n",
        "    if (add_sex==TRUE){\n",
        "        mod<-lm(Y~X+sex+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        nul<-lm(Y~sex+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        #sex interaction\n",
        "        interaction_mod<-lm(Y~X*sex+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        sex_pv<-signif(summary(interaction_mod)$coefficients[17,4],digits=2)\n",
        "    } else {\n",
        "        mod<-lm(Y~X+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        nul<-lm(Y~phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "    }\n",
        "    # slope of fit of X to Y\n",
        "    est<-signif(summary(mod)$coefficients[2,1],digits=2)\n",
        "    xpv<-signif(summary(mod)$coefficients[2,4],digits=2)\n",
        "    #r2 is the difference of the mod with X and without X\n",
        "    r2<-signif(summary(mod)$r.squared-summary(nul)$r.squared,digits=2)\n",
        "    pearsons_cor<-signif(cor(Y,X, use=\"pairwise.complete.obs\"),digits=2)    \n",
        "    if (add_sex==TRUE){\n",
        "        result = paste(est,xpv,r2,pearsons_cor,sex_pv,sep=\"\\t\")\n",
        "    } else {\n",
        "        result = paste(est,xpv,r2,pearsons_cor,sep=\"\\t\")\n",
        "    }\n",
        "    #sprintf(\"%f\\t%f\\t%f\\t%f\\t%f\",est,xpv,r2,pearson_cor,sex_pv)\n",
        "    return(result)\n",
        "}\n",
        "###########################################\n",
        "header = paste(\"y\",\"x\",\n",
        "               \"male_est\",\"male_pval\",\"male_r2\",\"male_cor\",\n",
        "               \"female_est\",\"female_pval\",\"female_r2\",\"female_cor\",\n",
        "               \"combined_est\",\"combined_pval\",\"combined_r2\",\"combined_cor\",\n",
        "               \"sex_interaction_pval\",\n",
        "               sep=\"\\t\"\n",
        "              );\n",
        "output=vector(length = 1+(length(vars)^2)-(length(vars)))\n",
        "output[1]<-header\n",
        "i=2\n",
        "for (x in vars)\n",
        "{\n",
        "  for(y in vars)\n",
        "  {\n",
        "    if (x==y){ next; } \n",
        "    com_results <- fit_model(df,x,y,add_sex=TRUE)\n",
        "    mal_results <- fit_model(df[df$Sex==0,],x,y,add_sex=FALSE)\n",
        "    fem_results <- fit_model(df[df$Sex==1,],x,y,add_sex=FALSE)\n",
        "    # Y - X - mal\n",
        "    output[i] <- paste(x,y,mal_results,fem_results,com_results,sep=\"\\t\")\n",
        "    i=i+1;\n",
        "  }\n",
        "}\n",
        "out = asd_genetic_correlations_by_sex_out\n",
        "ofh<-file(out)\n",
        "writeLines(output,ofh)\n",
        "close(ofh)\n",
        "print(\">>>DONE<<<\")"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "[1] 8301\n",
            "[1] \">>>DONE<<<\"\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JkVtQILix9lT"
      },
      "source": [
        "<a name=\"gene_phenotype_correlations\"></a>\n",
        "## Gene–phenotype correlations (Fig. 5)"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "z1QElvT8xLlr"
      },
      "source": [
        "#Author: Jonathan Sebat http://sebatlab.org March 19, 2021\n",
        "\n",
        "#The following phenotypes were available for only one cohort in either cases or controls: srs.case,vabs.case,srs.cont,scq.cont,vabs.cont,bapq.parents,srs.parents)\n",
        "#for these, age, sex and PCs were included as covariates, but not cohort (a 1st set of functions is written for this test)\n",
        "\n",
        "#SCQ, RBS and DCDQ were available on cases in the multiple cohorts (SSC and SPARK). Parental age and educational attainment were also available on multiple cohorts\n",
        "#for these, \"cohort\" was included as a covariate (a 2nd set of functions are written for this test)\n",
        "\n",
        "\n",
        "\n",
        "#importing the input data table that contains rare variant counts (dnLoF, dnMis, inhLoF), polygenic scores(PSasd, PSsz, PSea) and clinical phenotypes (srs, scq, rbs, vabs, dcdq, parental age, parental educational attainment, parental bapq)\n",
        "\n",
        "# master.gen.phen.table<-read.csv(\"C:\\\\Users\\\\jsebat\\\\sebatlab Dropbox\\\\Autism\\\\Autism_Illumina_WGS\\\\RV_PRS_Analysis_Jun2020\\\\master_phen_4.csv\", header = TRUE)\n",
        "master.gen.phen.table<-read.csv(filename_phen, header = TRUE)\n",
        "\n",
        "# Turn cohort into a factor variable\n",
        "master.gen.phen.table$cohort = factor(master.gen.phen.table$cohort)\n",
        "\n",
        "################################################################################\n",
        "##################################################################################\n",
        "####### First we create the main functions to run linear regression   ###########\n",
        "#######        on y = phenoype, x = genetic factor                    #############\n",
        "#####################################################################################\n",
        "#####################################################################################\n",
        "\n",
        "###############    1st set of scripts ###########################################################3\n",
        "#For phenotypes available on only a single cohort, \"cohort\" is NOT included as a covariate\n",
        "#linear regression controls for age, sex and PCs\n",
        "#and tests for a gene-by-sex interactions\n",
        "#the functions for these are as follows\n",
        "#gen.phen.1 a function that runs a linear regression and tests for gene-by-sex interaction. These are the primary results described in Fig. 5\n",
        "#gen.phen.2 To characterize gene-by sex effects that are significant, gene.phen.2 runs the regression on males and females separately without sex as a covariate\n",
        "#gene.phen.3 runs linear regression and gene-by-sex interaction on PARENTS, then to investigate gene by sex interactions, runs regression on mothers and fathers individually\n",
        "\n",
        "\n",
        "#gen.phen.1 a function that runs a linear regression on both sexes and outputs the result\n",
        "\n",
        "gen.phen.1<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "#define the sexes\n",
        "male<-master.temp$sex==\"Male\"\n",
        "female<-master.temp$sex==\"Female\"\n",
        "  \n",
        "#Test combined sample\n",
        "sex<-master.temp$sex\n",
        "cohort<-master.temp$cohort\n",
        "PC1<-master.temp$PC1\n",
        "PC2<-master.temp$PC2\n",
        "PC3<-master.temp$PC3\n",
        "PC4<-master.temp$PC4\n",
        "PC5<-master.temp$PC5\n",
        "PC6<-master.temp$PC6\n",
        "PC7<-master.temp$PC7\n",
        "PC8<-master.temp$PC8\n",
        "PC9<-master.temp$PC9\n",
        "PC10<-master.temp$PC10\n",
        "\n",
        "#retreive phen data\n",
        "y<-master.temp[,phen.col]\n",
        "age<-master.temp[,age.col]\n",
        "\n",
        "\n",
        "#retreive gen data\n",
        "x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "x.dn<-master.temp[,\"dnMIS\"]\n",
        "x.inh<-master.temp[,\"inhLoF\"]\n",
        "x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "\n",
        "result.dnlof<-lr.1(y,x.dnlof,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.dn<-lr.1(y,x.dn,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.inh<-lr.1(y,x.inh,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.asd<-lr.1(y,x.ps.asd,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.sz<-lr.1(y,x.ps.sz,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.ea<-lr.1(y,x.ps.ea,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "\n",
        "comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "group<-rep(\"combined\",length(variant.type))\n",
        "comb.result<-cbind(group,variant.type,comb.result)\n",
        "\n",
        "#################Test males\n",
        "\n",
        "PC1<-master.temp$PC1[male]\n",
        "PC2<-master.temp$PC2[male]\n",
        "PC3<-master.temp$PC3[male]\n",
        "PC4<-master.temp$PC4[male]\n",
        "PC5<-master.temp$PC5[male]\n",
        "PC6<-master.temp$PC6[male]\n",
        "PC7<-master.temp$PC7[male]\n",
        "PC8<-master.temp$PC8[male]\n",
        "PC9<-master.temp$PC9[male]\n",
        "PC10<-master.temp$PC10[male]\n",
        "\n",
        "#retreive phen data\n",
        "y<-master.temp[male,phen.col]\n",
        "age<-master.temp[male,age.col]\n",
        "\n",
        "\n",
        "#retreive gen data\n",
        "x.dnlof<-master.temp[male,\"dnLoF\"]\n",
        "x.dn<-master.temp[male,\"dnMIS\"]\n",
        "x.inh<-master.temp[male,\"inhLoF\"]\n",
        "x.ps.asd<-master.temp[male,\"PS.ASD.Z\"]\n",
        "x.ps.sz<-master.temp[male,\"PS.SZ.Z\"]\n",
        "x.ps.ea<-master.temp[male,\"PS.EA.Z\"]\n",
        "\n",
        "\n",
        "result.dnlof<-lr.2(y,x.dnlof,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.dn<-lr.2(y,x.dn,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.inh<-lr.2(y,x.inh,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.asd<-lr.2(y,x.ps.asd,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.sz<-lr.2(y,x.ps.sz,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.ea<-lr.2(y,x.ps.ea,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "\n",
        "male.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "group<-rep(\"male\",length(variant.type))\n",
        "male.result<-cbind(group,variant.type,male.result)\n",
        "\n",
        "#####Test females ######################\n",
        "\n",
        "PC1<-master.temp$PC1[female]\n",
        "PC2<-master.temp$PC2[female]\n",
        "PC3<-master.temp$PC3[female]\n",
        "PC4<-master.temp$PC4[female]\n",
        "PC5<-master.temp$PC5[female]\n",
        "PC6<-master.temp$PC6[female]\n",
        "PC7<-master.temp$PC7[female]\n",
        "PC8<-master.temp$PC8[female]\n",
        "PC9<-master.temp$PC9[female]\n",
        "PC10<-master.temp$PC10[female]\n",
        "\n",
        "#retreive phen data\n",
        "y<-master.temp[female,phen.col]\n",
        "age<-master.temp[female,age.col]\n",
        "\n",
        "\n",
        "#retreive gen data\n",
        "x.dnlof<-master.temp[female,\"dnLoF\"]\n",
        "x.dn<-master.temp[female,\"dnMIS\"]\n",
        "x.inh<-master.temp[female,\"inhLoF\"]\n",
        "x.ps.asd<-master.temp[female,\"PS.ASD.Z\"]\n",
        "x.ps.sz<-master.temp[female,\"PS.SZ.Z\"]\n",
        "x.ps.ea<-master.temp[female,\"PS.EA.Z\"]\n",
        "\n",
        "\n",
        "result.dnlof<-lr.2(y,x.dnlof,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.dn<-lr.2(y,x.dn,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.inh<-lr.2(y,x.inh,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.asd<-lr.2(y,x.ps.asd,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.sz<-lr.2(y,x.ps.sz,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.ea<-lr.2(y,x.ps.ea,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "\n",
        "female.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "group<-rep(\"female\",length(variant.type))\n",
        "female.result<-cbind(group,variant.type,female.result)\n",
        "\n",
        "final.result<-rbind(female.result,male.result,comb.result)\n",
        "final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 1 that also tests sex interaction\n",
        "lr.1<-function(y,x,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "\n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[15,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "###############################################################################33\n",
        "#Then we have a function that doesn't test the sex interaction\n",
        "##############################################################################\n",
        "\n",
        "gen.phen.2<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "  #retrieve covariates\n",
        "  PC1<-master.temp$PC1\n",
        "  PC2<-master.temp$PC2\n",
        "  PC3<-master.temp$PC3\n",
        "  PC4<-master.temp$PC4\n",
        "  PC5<-master.temp$PC5\n",
        "  PC6<-master.temp$PC6\n",
        "  PC7<-master.temp$PC7\n",
        "  PC8<-master.temp$PC8\n",
        "  PC9<-master.temp$PC9\n",
        "  PC10<-master.temp$PC10\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[,phen.col]\n",
        "  age<-master.temp[,age.col]\n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  x.dn<-master.temp[,\"dnMIS\"]\n",
        "  x.inh<-master.temp[,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "\n",
        "  \n",
        "  result.dnlof<-lr.2(y,x.dnlof,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2(y,x.dn,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2(y,x.inh,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2(y,x.ps.asd,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2(y,x.ps.sz,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2(y,x.ps.ea,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  final.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  final.result<-cbind(variant.type,final.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "\n",
        "#linear regression script 2 for testing each sex individually\n",
        "lr.2<-function(y,x,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "##################################################################\n",
        "# Then we have a regression for testing the effects of parents\n",
        "#################################################################\n",
        "\n",
        "#Ftests phenotypes in parents and tests sex interaction\n",
        "gen.phen.3<-function(master.temp,phen.col.f,phen.col.m)\n",
        "{\n",
        "  #retrieve father data\n",
        "  ###########################\n",
        "  sexfather<-rep(1,nrow(master.temp))\n",
        "  PC1f<-master.temp$PC1_PAT\n",
        "  PC2f<-master.temp$PC2_PAT\n",
        "  PC3f<-master.temp$PC3_PAT\n",
        "  PC4f<-master.temp$PC4_PAT\n",
        "  PC5f<-master.temp$PC5_PAT\n",
        "  PC6f<-master.temp$PC6_PAT\n",
        "  PC7f<-master.temp$PC7_PAT\n",
        "  PC8f<-master.temp$PC8_PAT\n",
        "  PC9f<-master.temp$PC9_PAT\n",
        "  PC10f<-master.temp$PC10_PAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  yf<-master.temp[,phen.col.f]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  f.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  f.dn<-master.temp[,\"dnMIS\"]\n",
        "  f.inh<-master.temp$InhLoF.Paternal.Transmitted + master.temp$InhLoF.Paternal.Untransmitted\n",
        "  f.ps.asd<-master.temp$Paternal.PS.ASD.Z\n",
        "  f.ps.sz<-master.temp$Paternal.PS.SZ.Z\n",
        "  f.ps.ea<-master.temp$Paternal.PS.EA.Z\n",
        "  \n",
        "  #retrieve mother data\n",
        "  ###########################\n",
        "  \n",
        "  sexmother<-rep(0,nrow(master.temp))\n",
        "  PC1m<-master.temp$PC1_MAT\n",
        "  PC2m<-master.temp$PC2_MAT\n",
        "  PC3m<-master.temp$PC3_MAT\n",
        "  PC4m<-master.temp$PC4_MAT\n",
        "  PC5m<-master.temp$PC5_MAT\n",
        "  PC6m<-master.temp$PC6_MAT\n",
        "  PC7m<-master.temp$PC7_MAT\n",
        "  PC8m<-master.temp$PC8_MAT\n",
        "  PC9m<-master.temp$PC9_MAT\n",
        "  PC10m<-master.temp$PC10_MAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  ym<-master.temp[,phen.col.m]\n",
        "  \n",
        "  #retreive gen data\n",
        "  m.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  m.dn<-master.temp[,\"dnMIS\"]\n",
        "  m.inh<-master.temp$InhLoF.Maternal.Transmitted + master.temp$InhLoF.Maternal.Untransmitted\n",
        "  m.ps.asd<-master.temp$Maternal.PS.ASD.Z\n",
        "  m.ps.sz<-master.temp$Maternal.PS.SZ.Z\n",
        "  m.ps.ea<-master.temp$Maternal.PS.EA.Z\n",
        "  \n",
        "  #merge father and mother variables\n",
        "  sex<-c(sexfather,sexmother)\n",
        "  PC1<-c(PC1f,PC1m)\n",
        "  PC2<-c(PC2f,PC2m)\n",
        "  PC3<-c(PC3f,PC3m)\n",
        "  PC4<-c(PC4f,PC4m)\n",
        "  PC5<-c(PC5f,PC5m)\n",
        "  PC6<-c(PC6f,PC6m)\n",
        "  PC7<-c(PC7f,PC7m)\n",
        "  PC8<-c(PC8f,PC8m)\n",
        "  PC9<-c(PC9f,PC9m)\n",
        "  PC10<-c(PC10f,PC10m)\n",
        "  x.dnlof<-c(f.dnlof,m.dnlof)\n",
        "  x.dn<-c(f.dn,m.dn)\n",
        "  x.inh<-c(f.inh,m.inh)\n",
        "  x.ps.asd<-c(f.ps.asd,m.ps.asd)\n",
        "  x.ps.sz<-c(f.ps.sz,m.ps.sz)\n",
        "  x.ps.ea<-c(f.ps.ea,m.ps.ea)\n",
        "  y<-c(yf,ym)\n",
        "  \n",
        "  result.dnlof<-lr.3(y,x.dnlof,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.3(y,x.dn,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.3(y, x.inh, sex, PC1, PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.3(y,x.ps.asd,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.3(y,x.ps.sz,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.3(y,x.ps.ea,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  #combined result\n",
        "  comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"parents\",length=length(variant.type))\n",
        "  comb.result<-cbind(group,variant.type,comb.result)\n",
        "  \n",
        "  \n",
        "  #father result\n",
        "  result.dnlof<-lr.4(yf,f.dnlof,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.dn<-lr.4(yf,f.dn,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.inh<-lr.4(yf, f.inh, PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.asd<-lr.4(yf,f.ps.asd,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.sz<-lr.4(yf,f.ps.sz,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.ea<-lr.4(yf,f.ps.ea,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  \n",
        "  f.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"father\",length=length(variant.type))\n",
        "  f.result<-cbind(group,variant.type,f.result)\n",
        "  \n",
        "  #mother result\n",
        "  result.dnlof<-lr.4(ym,m.dnlof,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.dn<-lr.4(ym,m.dn,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.inh<-lr.4(ym, m.inh, PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.asd<-lr.4(ym,m.ps.asd,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.sz<-lr.4(ym,m.ps.sz,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.ea<-lr.4(ym,m.ps.ea,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  \n",
        "  m.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"mother\",length=length(variant.type))\n",
        "  m.result<-cbind(group,variant.type,m.result)\n",
        "  \n",
        "  #output final result\n",
        "  final.result<-rbind(m.result,f.result,comb.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 3 tests phenotypes in parents and tests sex interaction\n",
        "lr.3<-function(y,x,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[14,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "#linear regression script 4 tests phenotypes in mother or father without sex variable\n",
        "lr.4<-function(y,x,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cc<-cor(x,y)\n",
        "\n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "#############################     2nd set of scripts with cohort included as a covaraite    ##################\n",
        "#in addition to age, sex and PCs\n",
        "#and tests for a gene-by-sex interactions\n",
        "#the functions for these are as follows\n",
        "#gen.phen.1c a function that runs a linear regression and tests for gene-by-sex interaction. These are the primary results described in Fig. 5\n",
        "#gen.phen.2c To characterize gene-by sex effects that are significant, gene.phen.2 runs the regression on males and females separately without sex as a covariate\n",
        "#gene.phen.3c runs linear regression and gene-by-sex interaction on PARENTS, then to investigate gene by sex interactions, runs regression on mothers and fathers individually\n",
        "\n",
        "\n",
        "#gen.phen.1c a function that runs a linear regression on both sexes and outputs the result\n",
        "\n",
        "gen.phen.1c<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "  #define the sexes\n",
        "  male<-master.temp$sex==\"Male\"\n",
        "  female<-master.temp$sex==\"Female\"\n",
        "  \n",
        "  #Test combined sample\n",
        "  sex<-master.temp$sex\n",
        "  cohort<-master.temp$cohort\n",
        "  PC1<-master.temp$PC1\n",
        "  PC2<-master.temp$PC2\n",
        "  PC3<-master.temp$PC3\n",
        "  PC4<-master.temp$PC4\n",
        "  PC5<-master.temp$PC5\n",
        "  PC6<-master.temp$PC6\n",
        "  PC7<-master.temp$PC7\n",
        "  PC8<-master.temp$PC8\n",
        "  PC9<-master.temp$PC9\n",
        "  PC10<-master.temp$PC10\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[,phen.col]\n",
        "  age<-master.temp[,age.col]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  x.dn<-master.temp[,\"dnMIS\"]\n",
        "  x.inh<-master.temp[,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "  \n",
        "  result.dnlof<-lr.1c(y,x.dnlof,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.1c(y,x.dn,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.1c(y,x.inh,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.1c(y,x.ps.asd,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.1c(y,x.ps.sz,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.1c(y,x.ps.ea,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"combined\",length(variant.type))\n",
        "  comb.result<-cbind(group,variant.type,comb.result)\n",
        "  \n",
        "  #################Test males\n",
        "  \n",
        "  cohort<-master.temp$cohort[male]\n",
        "  PC1<-master.temp$PC1[male]\n",
        "  PC2<-master.temp$PC2[male]\n",
        "  PC3<-master.temp$PC3[male]\n",
        "  PC4<-master.temp$PC4[male]\n",
        "  PC5<-master.temp$PC5[male]\n",
        "  PC6<-master.temp$PC6[male]\n",
        "  PC7<-master.temp$PC7[male]\n",
        "  PC8<-master.temp$PC8[male]\n",
        "  PC9<-master.temp$PC9[male]\n",
        "  PC10<-master.temp$PC10[male]\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[male,phen.col]\n",
        "  age<-master.temp[male,age.col]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[male,\"dnLoF\"]\n",
        "  x.dn<-master.temp[male,\"dnMIS\"]\n",
        "  x.inh<-master.temp[male,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[male,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[male,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[male,\"PS.EA.Z\"]\n",
        "  \n",
        "  \n",
        "  result.dnlof<-lr.2c(y,x.dnlof,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2c(y,x.dn,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2c(y,x.inh,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2c(y,x.ps.asd,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2c(y,x.ps.sz,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2c(y,x.ps.ea,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  male.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"male\",length(variant.type))\n",
        "  male.result<-cbind(group,variant.type,male.result)\n",
        "  \n",
        "  #####Test females ######################\n",
        "  cohort<-master.temp$cohort[female]\n",
        "  PC1<-master.temp$PC1[female]\n",
        "  PC2<-master.temp$PC2[female]\n",
        "  PC3<-master.temp$PC3[female]\n",
        "  PC4<-master.temp$PC4[female]\n",
        "  PC5<-master.temp$PC5[female]\n",
        "  PC6<-master.temp$PC6[female]\n",
        "  PC7<-master.temp$PC7[female]\n",
        "  PC8<-master.temp$PC8[female]\n",
        "  PC9<-master.temp$PC9[female]\n",
        "  PC10<-master.temp$PC10[female]\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[female,phen.col]\n",
        "  age<-master.temp[female,age.col]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[female,\"dnLoF\"]\n",
        "  x.dn<-master.temp[female,\"dnMIS\"]\n",
        "  x.inh<-master.temp[female,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[female,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[female,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[female,\"PS.EA.Z\"]\n",
        "  \n",
        "  \n",
        "  result.dnlof<-lr.2c(y,x.dnlof,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2c(y,x.dn,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2c(y,x.inh,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2c(y,x.ps.asd,cohort,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2c(y,x.ps.sz,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2c(y,x.ps.ea,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  female.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"female\",length(variant.type))\n",
        "  female.result<-cbind(group,variant.type,female.result)\n",
        "  \n",
        "  final.result<-rbind(female.result,male.result,comb.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 1 that also tests sex interaction\n",
        "lr.1c<-function(y,x,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[15,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "###############################################################################33\n",
        "#Then we have a function that doesn't test the sex interaction\n",
        "##############################################################################\n",
        "\n",
        "gen.phen.2c<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "  #retrieve covariates\n",
        "  PC1<-master.temp$PC1\n",
        "  PC2<-master.temp$PC2\n",
        "  PC3<-master.temp$PC3\n",
        "  PC4<-master.temp$PC4\n",
        "  PC5<-master.temp$PC5\n",
        "  PC6<-master.temp$PC6\n",
        "  PC7<-master.temp$PC7\n",
        "  PC8<-master.temp$PC8\n",
        "  PC9<-master.temp$PC9\n",
        "  PC10<-master.temp$PC10\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[,phen.col]\n",
        "  age<-master.temp[,age.col]\n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  x.dn<-master.temp[,\"dnMIS\"]\n",
        "  x.inh<-master.temp[,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "  \n",
        "  \n",
        "  result.dnlof<-lr.2c(y,x.dnlof,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2c(y,x.dn,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2c(y,x.inh,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2c(y,x.ps.asd,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2c(y,x.ps.sz,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2c(y,x.ps.ea,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  final.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  final.result<-cbind(variant.type,final.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "\n",
        "#linear regression script 2 for testing each sex individually\n",
        "lr.2c<-function(y,x,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "##################################################################\n",
        "# Then we have a regression for testing the effects of parents\n",
        "#################################################################\n",
        "\n",
        "#Ftests phenotypes in parents and tests sex interaction\n",
        "gen.phen.3c<-function(master.temp,phen.col.f,phen.col.m)\n",
        "{\n",
        "  #retrieve father data\n",
        "  ###########################\n",
        "  sexfather<-rep(1,nrow(master.temp))\n",
        "  cohort<-master.temp$cohort\n",
        "  PC1f<-master.temp$PC1_PAT\n",
        "  PC2f<-master.temp$PC2_PAT\n",
        "  PC3f<-master.temp$PC3_PAT\n",
        "  PC4f<-master.temp$PC4_PAT\n",
        "  PC5f<-master.temp$PC5_PAT\n",
        "  PC6f<-master.temp$PC6_PAT\n",
        "  PC7f<-master.temp$PC7_PAT\n",
        "  PC8f<-master.temp$PC8_PAT\n",
        "  PC9f<-master.temp$PC9_PAT\n",
        "  PC10f<-master.temp$PC10_PAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  yf<-master.temp[,phen.col.f]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  f.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  f.dn<-master.temp[,\"dnMIS\"]\n",
        "  f.inh<-master.temp$InhLoF.Paternal.Transmitted + master.temp$InhLoF.Paternal.Untransmitted\n",
        "  f.ps.asd<-master.temp$Paternal.PS.ASD.Z\n",
        "  f.ps.sz<-master.temp$Paternal.PS.SZ.Z\n",
        "  f.ps.ea<-master.temp$Paternal.PS.EA.Z\n",
        "  \n",
        "  #retrieve mother data\n",
        "  ###########################\n",
        "  \n",
        "  sexmother<-rep(0,nrow(master.temp))\n",
        "  PC1m<-master.temp$PC1_MAT\n",
        "  PC2m<-master.temp$PC2_MAT\n",
        "  PC3m<-master.temp$PC3_MAT\n",
        "  PC4m<-master.temp$PC4_MAT\n",
        "  PC5m<-master.temp$PC5_MAT\n",
        "  PC6m<-master.temp$PC6_MAT\n",
        "  PC7m<-master.temp$PC7_MAT\n",
        "  PC8m<-master.temp$PC8_MAT\n",
        "  PC9m<-master.temp$PC9_MAT\n",
        "  PC10m<-master.temp$PC10_MAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  ym<-master.temp[,phen.col.m]\n",
        "  \n",
        "  #retreive gen data\n",
        "  m.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  m.dn<-master.temp[,\"dnMIS\"]\n",
        "  m.inh<-master.temp$InhLoF.Maternal.Transmitted + master.temp$InhLoF.Maternal.Untransmitted\n",
        "  m.ps.asd<-master.temp$Maternal.PS.ASD.Z\n",
        "  m.ps.sz<-master.temp$Maternal.PS.SZ.Z\n",
        "  m.ps.ea<-master.temp$Maternal.PS.EA.Z\n",
        "  \n",
        "  #merge father and mother variables\n",
        "  sex<-c(sexfather,sexmother)\n",
        "  comb.cohort<-c(cohort,cohort)\n",
        "  PC1<-c(PC1f,PC1m)\n",
        "  PC2<-c(PC2f,PC2m)\n",
        "  PC3<-c(PC3f,PC3m)\n",
        "  PC4<-c(PC4f,PC4m)\n",
        "  PC5<-c(PC5f,PC5m)\n",
        "  PC6<-c(PC6f,PC6m)\n",
        "  PC7<-c(PC7f,PC7m)\n",
        "  PC8<-c(PC8f,PC8m)\n",
        "  PC9<-c(PC9f,PC9m)\n",
        "  PC10<-c(PC10f,PC10m)\n",
        "  x.dnlof<-c(f.dnlof,m.dnlof)\n",
        "  x.dn<-c(f.dn,m.dn)\n",
        "  x.inh<-c(f.inh,m.inh)\n",
        "  x.ps.asd<-c(f.ps.asd,m.ps.asd)\n",
        "  x.ps.sz<-c(f.ps.sz,m.ps.sz)\n",
        "  x.ps.ea<-c(f.ps.ea,m.ps.ea)\n",
        "  y<-c(yf,ym)\n",
        "  \n",
        "  result.dnlof<-lr.3c(y,x.dnlof,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.3c(y,x.dn,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.3c(y, x.inh, sex,comb.cohort,PC1, PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.3c(y,x.ps.asd,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.3c(y,x.ps.sz,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.3c(y,x.ps.ea,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  #combined result\n",
        "  comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"parents\",length=length(variant.type))\n",
        "  comb.result<-cbind(group,variant.type,comb.result)\n",
        "  \n",
        "  \n",
        "  #father result\n",
        "  result.dnlof<-lr.4c(yf,f.dnlof,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.dn<-lr.4c(yf,f.dn,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.inh<-lr.4c(yf, f.inh,cohort, PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.asd<-lr.4c(yf,f.ps.asd,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.sz<-lr.4c(yf,f.ps.sz,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.ea<-lr.4c(yf,f.ps.ea,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  \n",
        "  f.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"father\",length=length(variant.type))\n",
        "  f.result<-cbind(group,variant.type,f.result)\n",
        "  \n",
        "  #mother result\n",
        "  result.dnlof<-lr.4c(ym,m.dnlof,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.dn<-lr.4c(ym,m.dn,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.inh<-lr.4c(ym, m.inh,cohort, PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.asd<-lr.4c(ym,m.ps.asd,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.sz<-lr.4c(ym,m.ps.sz,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.ea<-lr.4c(ym,m.ps.ea,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  \n",
        "  m.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"mother\",length=length(variant.type))\n",
        "  m.result<-cbind(group,variant.type,m.result)\n",
        "  \n",
        "  #output final result\n",
        "  final.result<-rbind(m.result,f.result,comb.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 3 tests phenotypes in parents and tests sex interaction\n",
        "lr.3c<-function(y,x,sex,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[15,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "#linear regression script 4 tests phenotypes in mother or father without sex variable\n",
        "lr.4c<-function(y,x,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cc<-cor(x,y)\n",
        "  \n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "################################################################################################################\n",
        "#################   Then the analysis of each gene-phenotype correlation is run as follows    ##################\n",
        "###############################################################################################################\n",
        "\n",
        "\n",
        "#identify the subset of case, controls, and european ancesty\n",
        "cont<-master.gen.phen.table$phen==\"CON\"\n",
        "case<-master.gen.phen.table$phen==\"ASD\"\n",
        "eur<-master.gen.phen.table$is_eur==1\n",
        "\n",
        "\n",
        "#The analysis will be restricted to subjects of confirmed european ancestry\n",
        "master.temp.case.eur<-master.gen.phen.table[case & eur,]\n",
        "master.temp.cont.eur<-master.gen.phen.table[cont & eur,]\n",
        "\n",
        "\n",
        "####################################################\n",
        "# Test each phenotype in offspring\n",
        "#####################################################\n",
        "\n",
        "\n",
        "#VABS\n",
        "\n",
        "phen.col<-\"vabs.child.z\"\n",
        "age.col<-\"vabs.child.age\"\n",
        "\n",
        "vine.case.eur<-gen.phen.1(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "vine.cont.eur<-gen.phen.1(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "# vine.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "# vine.cont.eur<-gen.phen.1c(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"VABS\",nrow(vine.case.eur))\n",
        "vine.case.eur<-cbind(phenotype,vine.case.eur)\n",
        "\n",
        "phenotype<-rep(\"VABS\",nrow(vine.cont.eur))\n",
        "vine.cont.eur<-cbind(phenotype,vine.cont.eur)\n",
        "\n",
        "#SRS\n",
        "phen.col<-\"srs.child.z\"\n",
        "age.col<-\"srs.child.age\"\n",
        "\n",
        "srs.case.eur<-gen.phen.1(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "srs.cont.eur<-gen.phen.1(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "# srs.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "# srs.cont.eur<-gen.phen.1c(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"SRS\",nrow(srs.case.eur))\n",
        "srs.case.eur<-cbind(phenotype,srs.case.eur)\n",
        "\n",
        "phenotype<-rep(\"SRS\",nrow(srs.cont.eur))\n",
        "srs.cont.eur<-cbind(phenotype,srs.cont.eur)\n",
        "\n",
        "\n",
        "#SCQ\n",
        "phen.col<-\"scq.child.z\"\n",
        "age.col<-\"scq.child.age\"\n",
        "\n",
        "scq.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "scq.cont.eur<-gen.phen.1(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "# scq.cont.eur<-gen.phen.1c(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"SCQ\",nrow(scq.case.eur))\n",
        "scq.case.eur<-cbind(phenotype,scq.case.eur)\n",
        "phenotype<-rep(\"SCQ\",nrow(scq.cont.eur))\n",
        "scq.cont.eur<-cbind(phenotype,scq.cont.eur)\n",
        "\n",
        "\n",
        "#DCDQ\n",
        "phen.col<-\"dcdq.child.z\"\n",
        "age.col<-\"dcdq.child.age\"\n",
        "\n",
        "dcdq.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "phenotype<-rep(\"DCDQ\",nrow(dcdq.case.eur))\n",
        "dcdq.case.eur<-cbind(phenotype,dcdq.case.eur)\n",
        "\n",
        "\n",
        "#rbs\n",
        "phen.col<-\"rbs.child.z\"\n",
        "age.col<-\"scq.child.age\"\n",
        "# age.col<-\"rbs.child.age\"\n",
        "\n",
        "\n",
        "rbs.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "phenotype<-rep(\"RBS\",nrow(rbs.case.eur))\n",
        "rbs.case.eur<-cbind(phenotype,rbs.case.eur)\n",
        "\n",
        "\n",
        "\n",
        "############################################\n",
        "# Testing Behavioral Phenotypes in Parents\n",
        "############################################\n",
        "\n",
        "cont<-master.gen.phen.table$phen==\"CON\"\n",
        "case<-master.gen.phen.table$phen==\"ASD\"\n",
        "eur<-master.gen.phen.table$is_eur==1\n",
        "\n",
        "\n",
        "#Remove the sibling control trios (so the same parents are not tested twice)\n",
        "master.temp.case.eur<-master.gen.phen.table[case & eur,]\n",
        "\n",
        "#also remove any younger ASD siblings of cases so we have each set of parents represented once\n",
        "#sort the trios by mothers age in ascending order so that the oldest case is on top\n",
        "sort.order<-order(master.temp.case.eur$mother.age.at.birth.z,decreasing=FALSE)\n",
        "master.temp.case.unique<-master.temp.case.eur[sort.order,]\n",
        "#and unique the family ID\n",
        "master.temp.case.unique<-master.temp.case.unique[!duplicated(master.temp.case.unique$fid),]\n",
        "\n",
        "##############BAPQ\n",
        "\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.bapq.z\"\n",
        "phen.col.m<-\"mother.bapq.z\"\n",
        "\n",
        "bapq.parents.eur<-gen.phen.3(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "phenotype<-rep(\"BAPQ\",nrow(bapq.parents.eur))\n",
        "bapq.parents.eur<-cbind(phenotype,bapq.parents.eur)\n",
        "\n",
        "\n",
        "###################SRS\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.srs.z\"\n",
        "phen.col.m<-\"mother.srs.z\"\n",
        "\n",
        "srs.parents.eur<-gen.phen.3(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "phenotype<-rep(\"SRS\",nrow(srs.parents.eur))\n",
        "srs.parents.eur<-cbind(phenotype,srs.parents.eur)\n",
        "\n",
        "\n",
        "###################Parental Age\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.age.at.birth.z\"\n",
        "phen.col.m<-\"mother.age.at.birth.z\"\n",
        "\n",
        "parental.age.eur<-gen.phen.3c(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"Parental Age\",nrow(parental.age.eur))\n",
        "parental.age.eur<-cbind(phenotype,parental.age.eur)\n",
        "\n",
        "###################EA\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.EA.z\"\n",
        "phen.col.m<-\"mother.EA.z\"\n",
        "\n",
        "parental.ea.eur<-gen.phen.3c(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "phenotype<-rep(\"Parental EA\",nrow(parental.ea.eur))\n",
        "parental.ea.eur<-cbind(phenotype,parental.ea.eur)\n",
        "\n",
        "\n",
        "################################################################\n",
        "##### Combine all of the results and write them out to csv file\n",
        "###############################################################\n",
        "\n",
        "case.results.eur<-rbind(srs.case.eur,scq.case.eur,vine.case.eur,dcdq.case.eur,rbs.case.eur)\n",
        "case.results.eur<-cbind(case.status=rep(\"case\",nrow(case.results.eur)),case.results.eur)\n",
        "control.results.eur<-rbind(srs.cont.eur, scq.cont.eur,vine.cont.eur)\n",
        "control.results.eur<-cbind(case.status=rep(\"cont\",nrow(control.results.eur)),control.results.eur)\n",
        "parent.results.eur<-rbind(bapq.parents.eur,srs.parents.eur,parental.age.eur,parental.ea.eur)\n",
        "parent.results.eur<-cbind(case.status=rep(\"parent\",nrow(parent.results.eur)),parent.results.eur)\n",
        "\n",
        "clin.results.eur<-rbind(case.results.eur,control.results.eur,parent.results.eur)\n",
        "# write.csv(clin.results.eur,\"C:\\\\Users\\\\jsebat\\\\sebatlab Dropbox\\\\Autism\\\\Autism_Illumina_WGS\\\\RV_PRS_Analysis_Jun2020\\\\clin_results_eur.csv\")\n",
        "write.csv(clin.results.eur,clin_results_out)\n",
        "\n",
        "\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "yZDj3acpqCXO",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "84d09822-918f-453e-ca79-9d993519be3b"
      },
      "source": [
        "# Installing packages for Figure 5 display\n",
        "install.packages(\"tidyverse\")\n",
        "install.packages(\"dplyr\")\n",
        "install.packages(\"RColorBrewer\")\n",
        "install.packages(\"ggplot2\")\n",
        "install.packages(\"scales\")\n",
        "install.packages(\"ggpubr\", repos = \"https://ftp.osuosl.org/pub/cran/\", lib=\"/usr/lib/R/site-library\")\n",
        "\n",
        "library(tidyverse)\n",
        "library(dplyr)\n",
        "library(RColorBrewer)\n",
        "library(ggplot2)\n",
        "library(scales)\n",
        "library(ggpubr)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "also installing the dependencies ‘matrixStats’, ‘RcppArmadillo’, ‘numDeriv’, ‘SparseM’, ‘MatrixModels’, ‘conquer’, ‘sp’, ‘openxlsx’, ‘minqa’, ‘nloptr’, ‘RcppEigen’, ‘carData’, ‘abind’, ‘pbkrtest’, ‘quantreg’, ‘maptools’, ‘rio’, ‘lme4’, ‘corrplot’, ‘car’, ‘ggrepel’, ‘ggsci’, ‘cowplot’, ‘ggsignif’, ‘gridExtra’, ‘polynom’, ‘rstatix’\n",
            "\n",
            "\n",
            "Warning message in system(\"timedatectl\", intern = TRUE):\n",
            "“running command 'timedatectl' had status 1”\n",
            "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.1 ──\n",
            "\n",
            "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
            "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.2     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
            "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
            "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 1.4.0     \n",
            "\n",
            "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
            "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
            "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
            "\n",
            "\n",
            "Attaching package: ‘scales’\n",
            "\n",
            "\n",
            "The following object is masked from ‘package:purrr’:\n",
            "\n",
            "    discard\n",
            "\n",
            "\n",
            "The following object is masked from ‘package:readr’:\n",
            "\n",
            "    col_factor\n",
            "\n",
            "\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_IyfW2ZVuuST"
      },
      "source": [
        "# library(svglite)\n",
        "# remove.packages(\"svglite\")\n",
        "# install.packages(\"svglite\", type = \"source\")\n",
        "# library(svglite)\n",
        "# install.packages('svglite')\n",
        "# install.packages(\"svglite\", type = \"source\")\n",
        "# library(svglite)\n",
        "\n",
        "# remove.packages(\"svglite\")\n",
        "str(allPackage <- installed.packages())\n",
        "allPackage [, c(1,3:5)]\n",
        "install.packages(\"svglite\", type = \"source\")\n",
        "str(allPackage <- installed.packages())\n",
        "allPackage [, c(1,3:5)]\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 170
        },
        "id": "nlstCyCxutKP",
        "outputId": "754697c0-2894-4c47-fad4-71b6153dbfde"
      },
      "source": [
        "asd_df <- read.csv(file = clin_results_out, header = TRUE, na.string = \"NA\")\n",
        "\n",
        "MASSIVE = 50 + 10 + 10\n",
        "BIG = 30 + 5\n",
        "\n",
        "# Rounding\n",
        "asd_df[195,10] = round(asd_df[195,10], digits = 2)\n",
        "asd_df[195,11] = round(asd_df[195,11], digits = 2)\n",
        "\n",
        "# Make p-value column (\"logP.2\" will used for scaling circles in figure)\n",
        "asd_df$x.pval_log = -log10(asd_df$x.pval)\n",
        "asd_df <- transform(asd_df, logP.2 = ifelse(x.pval_log >= 5, 5, x.pval_log))\n",
        "\n",
        "# Concatenate the phenotype and case.status variables\n",
        "asd_df$pheno = paste(asd_df$phenotype, \"_\", asd_df$case.status, sep=\"\")\n",
        "\n",
        "# Create a column of significance labels for significant genotype effects\n",
        "asd_df$stars <- cut(asd_df$x.pval, breaks=c(-Inf, 0.0007, 0.05, Inf), label = c(\"**\", \"*\", \"\"))  # Create column of significance labels for main effect\n",
        "\n",
        "# Filter data for panel A (cases & controls)\n",
        "asd_cascon <- asd_df %>%\n",
        "  filter(case.status %in% c(\"case\", \"cont\")) %>%\n",
        "  filter(group %in% c(\"combined\")) %>%\n",
        "  filter(variant.type %in% c( \"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "# Specify the factor levels in the order we want\n",
        "asd_cascon$pheno <- factor(asd_cascon$pheno, levels = c(\"RBS_case\", \"SCQ_case\", \"SCQ_cont\", \"SRS_case\", \"SRS_cont\", \"VABS_case\", \"VABS_cont\", \"DCDQ_case\"))\n",
        "# Reverse the order as follow\n",
        "asd_cascon$pheno <- factor(asd_cascon$pheno, levels = rev(levels(asd_cascon$pheno)))\n",
        "\n",
        "asd_cascon$variant.type <- factor(asd_cascon$variant.type, levels = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "### Panel A ###\n",
        "pdf(\"panelA.test.pdf\", width = 30, height = 30)\n",
        "panelA =\n",
        "ggballoonplot(asd_cascon, x = \"variant.type\", y = \"pheno\", size = \"logP.2\", size.range = c(5, 20), fill = \"est\", ggtheme = theme_bw()) +\n",
        "scale_fill_gradientn(colors = c(\"blue3\", \"blue\", \"white\", \"indianred1\", \"red3\"), values = rescale(c(-0.43, -0.30, 0, 0.10, 0.28)), limits = c(-0.43, 0.28), guide = guide_colorbar(barwidth = 6, barheight = 30, title.position = \"top\", label.theme = element_text(size = MASSIVE))) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0.8, size = BIG - 5) +\n",
        "scale_size(range = c(15*1.53, 60*1.53), breaks =c(1, 2, 3, 4, 5)) +\n",
        "scale_y_discrete(breaks = c(\"RBS_case\", \"SCQ_case\", \"SCQ_cont\", \"SRS_case\", \"SRS_cont\", \"VABS_case\", \"VABS_cont\", \"DCDQ_case\"), labels = c(\"RBS case\", \"SCQ case\", \"SCQ cont\", \"SRS case\", \"SRS cont\", \"VABS case\", \"VABS cont\", \"DCDQ case\")) +\n",
        "scale_x_discrete(breaks = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"), labels=c(\"dnMIS\", \"dnLoF\", \"inhLoF\", expression(PS[ASD]), expression(PS[SCZ]), expression(PS[EA]))) +\n",
        "labs(size = \"-log10(P-value)\") +\n",
        "labs(fill = \"Effect Size\") +\n",
        "theme(panel.grid.major = element_blank()) +\n",
        "theme(panel.grid.minor = element_blank()) +\n",
        "# Text\n",
        "theme(axis.text.x = element_text(color = \"black\", size = MASSIVE)) +\n",
        "theme(axis.text.y = element_text(color = \"black\", size = MASSIVE)) +\n",
        "# Legend\n",
        "theme(legend.title = element_text(size = MASSIVE)) +\n",
        "theme(legend.text = element_text(color = \"black\", size = MASSIVE))\n",
        "panelA\n",
        "dev.off();\n",
        "\n",
        "\n",
        "# Filter data for panel B (parents)\n",
        "asd_parents <- asd_df %>%\n",
        "  filter(group %in% c(\"parents\")) %>%\n",
        "  filter(variant.type %in% c( \"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "# Specify the factor levels in the order you want\n",
        "asd_parents$pheno <- factor(asd_parents$pheno, levels = c(\"SRS_parent\", \"BAPQ_parent\", \"Parental Age_parent\", \"Parental EA_parent\"))\n",
        "# Reverse the order\n",
        "asd_parents$pheno <- factor(asd_parents$pheno, levels = rev(levels(asd_parents$pheno)))\n",
        "\n",
        "asd_parents$variant.type <- factor(asd_parents$variant.type, levels = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "### Panel B ###\n",
        "pdf(\"panelB.test.pdf\", width = 30, height = 30)\n",
        "panelB =\n",
        "ggballoonplot(asd_parents, x = \"variant.type\", y = \"pheno\", size = \"logP.2\", size.range = c(5, 20), fill = \"est\", ggtheme = theme_bw()) +\n",
        "scale_fill_gradientn(colors = c(\"blue3\", \"blue\", \"white\", \"indianred1\", \"red3\"), values = rescale(c(-0.43, -0.30, 0, 0.10, 0.28)), limits = c(-0.43, 0.28)) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0.8, size = BIG - 5) +\n",
        "scale_size(range = c(15*1.53, 60*1.53), breaks =c(1, 2, 3, 4, 5)) +\n",
        "scale_y_discrete(breaks = c(\"SRS_parent\", \"BAPQ_parent\", \"Parental Age_parent\", \"Parental EA_parent\"), labels = c(\"SRS\", \"BAPQ\", \"Parental Age\", \"Parental EA\")) +\n",
        "scale_x_discrete(breaks = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"), labels = c(\"dnMIS\", \"dnLoF\",  \"inhLoF\", expression(PS[ASD]),  expression(PS[SCZ]), expression(PS[EA]))) +\n",
        "labs(size = \"-log10(P-value)\", labels = c(\"1\", \"2\", \"3\", \"4\", \"5\")) +\n",
        "labs(fill = \"Effect Size [sd]\") +\n",
        "theme(panel.grid.major = element_blank()) +\n",
        "theme(panel.grid.minor = element_blank()) +\n",
        "# Text\n",
        "theme(axis.text.x = element_text(color = \"black\", size = MASSIVE)) +\n",
        "theme(axis.text.y = element_text(color = \"black\", size = MASSIVE)) +\n",
        "# Legend\n",
        "theme(legend.title = element_text(size = MASSIVE)) +\n",
        "theme(legend.text = element_text(color = \"black\", size = MASSIVE))\n",
        "panelB\n",
        "dev.off();\n",
        "\n",
        "### PANEL AB ###\n",
        "pdf(\"panelAB.test.pdf\", width = 40, height = 40, onefile = FALSE)\n",
        "panelAB = \n",
        "ggarrange(panelA, panelB,\n",
        "          labels = c(\"A\",\"B\"),\n",
        "          nrow = 2,\n",
        "          ncol = 1,\n",
        "          widths = c(1, 1), \n",
        "          heights = c(1.6, 1),\n",
        "          # vjust = 1.2,\n",
        "          common.legend = TRUE, \n",
        "          legend = \"right\")\n",
        "# theme(plot.margin = unit(c(0,3,0,0), \"cm\")) + \n",
        "panelAB\n",
        "dev.off();\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Scale for 'size' is already present. Adding another scale for 'size', which\n",
            "will replace the existing scale.\n",
            "\n"
          ],
          "name": "stderr"
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "stream",
          "text": [
            "Scale for 'size' is already present. Adding another scale for 'size', which\n",
            "will replace the existing scale.\n",
            "\n"
          ],
          "name": "stderr"
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 85
        },
        "id": "qFzToJk1eSgf",
        "outputId": "f8a3ef34-43cd-4426-f288-6147765db006"
      },
      "source": [
        "asd_df <- read.csv(file = clin_results_out, header = TRUE, na.string = \"NA\")\n",
        "\n",
        "# Create columns of significance labels\n",
        "asd_df$stars.g <- cut(asd_df$x.pval, breaks=c(-Inf, 0.0007, 0.05, Inf), label=c(\"**\", \"*\", \"\"))  # Create column of significance labels for genotype effect\n",
        "asd_df$stars.g.sex <- cut(asd_df$sex.pval, breaks=c(-Inf, 0.0007, 0.05, Inf), label=c(\"**\", \"*\", \"\"))  # Create column of significance labels for genotype by sex effect\n",
        "\n",
        "### PANEL C ###\n",
        "\n",
        "# Get significant sex interactions p-values and save them for later (for displaying in barplots)\n",
        "controls_scq_combined_pssz = subset(x = asd_df,\n",
        "    subset = case.status == \"cont\" & phenotype == \"SCQ\" & group == \"combined\" & variant.type == \"ps.sz\",)\n",
        "controls_scq_combined_pssz_sex_pval = toString(controls_scq_combined_pssz$sex.pval)\n",
        "\n",
        "controls_srs_combined_psasd = subset(x = asd_df,\n",
        "    subset = case.status == \"cont\" & phenotype == \"SRS\" & group == \"combined\" & variant.type == \"ps.asd\",)\n",
        "controls_srs_combined_psasd_sex_pval = toString(controls_srs_combined_psasd$sex.pval)\n",
        "\n",
        "# cases_scq_combined_inhlof = subset(x = asd_df,\n",
        "#     subset = case.status == \"case\" & phenotype == \"SCQ\" & group == \"combined\" & variant.type == \"inhLoF\")\n",
        "# cases_scq_combined_inhlof_sex_pval = toString(cases_scq_combined_inhlof$sex.pval)\n",
        "\n",
        "\n",
        "# Select data for panel C (controls for all g*sex interactions PS-SCZ with SCQ and VABS)\n",
        "asd_C.1 <- asd_df %>%\n",
        "  filter(case.status %in% c(\"cont\")) %>%\n",
        "  filter(phenotype %in% c(\"SRS\")) %>%\n",
        "  filter(group %in% c(\"male\", \"female\")) %>%\n",
        "  filter(variant.type %in% c(\"ps.asd\"))\n",
        "\n",
        "asd_C.2 <- asd_df %>%\n",
        "  filter(case.status %in% c(\"cont\")) %>%\n",
        "  filter(phenotype %in% c(\"SCQ\")) %>%\n",
        "  filter(group %in% c(\"male\", \"female\")) %>%\n",
        "  filter(variant.type %in% c(\"ps.sz\"))\n",
        "\n",
        "# asd_C.3 <- asd_df %>%\n",
        "#   filter(case.status %in% c(\"case\")) %>%\n",
        "#   filter(phenotype %in% c(\"SCQ\")) %>%\n",
        "#   filter(group %in% c(\"male\", \"female\")) %>%\n",
        "#   filter(variant.type %in% c(\"inhLoF\"))\n",
        "\n",
        "asd_C12 <- rbind(asd_C.1, asd_C.2)\n",
        "my_labellerC12 <- as_labeller(c(SCQ = \"SCQ - PS[SCZ]\", SRS = \"SRS - PS[ASD]\"), default = label_parsed) \n",
        "\n",
        "# asd_C3 <- rbind(asd_C.3)\n",
        "# my_labellerC3 <- as_labeller(c(SCQ = \"SCQ - inhLoF\"), default = label_parsed)\n",
        "\n",
        "# # Make panel C12 p-value dataframe\n",
        "# pval_C12_df <- data.frame(\n",
        "#   phenotype = c(\"SCQ\", \"SRS\"),\n",
        "#   start = c(\"Female\", \"Female\"),\n",
        "#   end = c(\"Male\", \"Male\"),\n",
        "#   y.position = c(0.15, 0.15),\n",
        "#   pval = c(paste(\"P = \", controls_scq_combined_pssz_sex_pval,  sep=\"\"), \n",
        "#            paste(\"P = \", controls_srs_combined_psasd_sex_pval, sep=\"\"))\n",
        "# )\n",
        "\n",
        "# # Make panel C3 p-value dataframe\n",
        "# pval_C3_df <- data.frame(\n",
        "#   phenotype = c(\"SCQ\"),\n",
        "#   start = c(\"Female\"),\n",
        "#   end = c(\"Male\"),\n",
        "#   y.position = c(0.15),\n",
        "#   pval = c(paste(\"P = \", cases_scq_combined_inhlof_sex_pval, sep=\"\"))\n",
        "# )\n",
        "\n",
        "asd_C12$phen = factor(asd_C12$phenotype, levels=c('SRS','SCQ'))\n",
        "asd_C12$stars <- cut(asd_C12$x.pval, breaks = c(-Inf, 0.001, 0.01, 0.05, Inf), label = c(\"***\", \"**\", \"*\", \"\"))  # Create column of significance labels for main effect\n",
        "\n",
        "pdf(\"panelC.12.test.pdf\", width = 30, height = 30)\n",
        "panelC12 =\n",
        "ggplot(asd_C12, aes(x = group, y = est, fill = group)) +\n",
        "geom_bar(stat = \"identity\", width = 1, position = position_nudge(x = 0.5)) +\n",
        "facet_wrap(~phenotype, ncol = 2, labeller = my_labellerC12) +\n",
        "# P-value asterisks\n",
        "# geom_text(aes(label = stars), color = \"black\", vjust = -6, size = BIG - 5, hjust = -1.4) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0, size = BIG - 5, hjust = 0) +\n",
        "# Standard error whiskers\n",
        "geom_linerange(aes(ymin = est - est.stderr, ymax = est + est.stderr), position = position_nudge(0.5), size = 1) +\n",
        "scale_fill_manual(values = c(\"coral\", \"dodgerblue\")) +\n",
        "labs(y = expression(paste(\"Effect size of PS [SD]\"))) +\n",
        "labs(x = expression(paste(\"Controls\"))) +\n",
        "# Text\n",
        "# theme(axis.title.x = element_blank()) +\n",
        "theme(axis.title.x = element_text(size = MASSIVE + 5, color = \"black\", vjust = 10.0)) +\n",
        "theme(axis.text.x = element_blank()) +\n",
        "# theme(strip.text.x = element_text(size = MASSIVE - 30, color = \"black\")) +\n",
        "theme(strip.text.x = element_text(size = MASSIVE, color = \"black\")) +\n",
        "# theme(axis.text.y = element_text(size = MASSIVE - 30)) +\n",
        "theme(axis.text.y = element_text(size = MASSIVE, color = \"black\")) +\n",
        "theme(axis.title.y = element_text(size = MASSIVE + 5, color = \"black\")) +\n",
        "# Legend\n",
        "theme(legend.title = element_blank()) +\n",
        "theme(legend.text = element_text(size=MASSIVE)) +\n",
        "theme(legend.key.size = unit(5, \"line\")) +\n",
        "theme(panel.background = element_blank()) +\n",
        "theme(strip.background = element_rect(color = \"white\", fill = \"white\", size = 1.5, linetype = \"solid\")) +\n",
        "ylim(-0.10, 0.15)\n",
        "# p-value significance lines\n",
        "# geom_signif(data = pval_C12_df, aes(xmin = 1.5, xmax = 2.5, annotations = pval, y_position = y.position), manual = TRUE, inherit.aes = FALSE, textsize = BIG - 20)\n",
        "panelC12\n",
        "dev.off();\n",
        "\n",
        "# pdf(\"panelC.3.test.pdf\", width = 30, height = 30)\n",
        "# panelC3 =\n",
        "# ggplot(asd_C3, aes(x = group, y = est, fill = group)) +\n",
        "# geom_bar(stat = \"identity\", width = 1, position = position_nudge(x = 0.5)) +\n",
        "# facet_wrap(~phenotype, ncol = 1, labeller = my_labellerC3) +\n",
        "# geom_linerange(aes(ymin = est - est.stderr, ymax = est + est.stderr), position = position_nudge(0.5), size = 1) +\n",
        "# scale_fill_manual(values = c(\"coral\", \"dodgerblue\")) +\n",
        "# labs(y = expression(paste(\"Effect size of inhLoF [SD]\"))) +\n",
        "# labs(x = expression(paste(\"Cases\"))) +\n",
        "# # Text\n",
        "# # theme(axis.title.x = element_blank()) +\n",
        "# theme(axis.title.x = element_text(size = MASSIVE + 5, color = \"black\", vjust = 10.0)) +\n",
        "# theme(axis.text.x = element_blank()) +\n",
        "# theme(strip.text.x = element_text(size = MASSIVE - 30, color = \"black\")) +\n",
        "# # theme(axis.text.y = element_text(size = MASSIVE - 30)) +\n",
        "# theme(axis.text.y = element_blank()) +\n",
        "# theme(axis.ticks.y = element_blank()) +\n",
        "# theme(axis.title.y = element_text(size = MASSIVE + 5, color = \"black\")) +\n",
        "# # Legend\n",
        "# theme(legend.title = element_blank()) +\n",
        "# theme(legend.text = element_text(size=MASSIVE)) +\n",
        "# theme(legend.key.size = unit(5, \"line\")) +\n",
        "# theme(panel.background = element_blank()) +\n",
        "# theme(strip.background = element_rect(color = \"white\", fill = \"white\", size = 1.5, linetype = \"solid\")) +\n",
        "# ylim(-0.10, 0.15) +\n",
        "# # p-value significance lines\n",
        "# geom_signif(data = pval_C3_df, aes(xmin = 1.5, xmax = 2.5, annotations = pval, y_position = y.position), manual = TRUE, inherit.aes = FALSE, textsize = BIG - 20)\n",
        "# panelC3\n",
        "# dev.off();\n",
        "\n",
        "# # Combine panels C12 and C3 into C123\n",
        "# pdf(\"panelC.123.test.pdf\", width = 30, height = 30, onefile = FALSE)\n",
        "# panelC123 =\n",
        "# ggarrange(panelC12, panelC3,\n",
        "#           labels = c(\"C12\", \"C3\"),\n",
        "#           nrow = 1,\n",
        "#           ncol = 2,\n",
        "#           common.legend = TRUE,\n",
        "#           legend = \"right\",\n",
        "#           widths = c(2, 1))\n",
        "# panelC123\n",
        "# dev.off();\n",
        "\n",
        "\n",
        "### PANEL D ###\n",
        "# Select data for panel D (Parental Age Effect size: dnLoF, dnMIS, inhLoF, PS_SCZ, PS_ASD, PS_EA)\n",
        "asd_D <- asd_df %>%\n",
        "  filter(phenotype %in% c(\"Parental Age\")) %>%\n",
        "  filter(group %in% c(\"mother\", \"father\")) %>%\n",
        "  # filter(group %in% c(\"parents\")) %>%\n",
        "  filter(variant.type %in% c(\"dnlof\", \"inhLoF\", \"dnmis\", \"ps.sz\", \"ps.asd\", \"ps.ea\"))\n",
        "\n",
        "my_labellerD = as_labeller(c(dnlof = \"dnLoF\", inhLoF = \"inhLoF\", dnmis = \"dnMIS\", ps.sz = \"PS[SCZ]\", ps.asd = \"PS[ASD]\", ps.ea = \"PS[EA]\"), default = label_parsed)\n",
        "\n",
        "# Get significant sex interaction p-values\n",
        "parent_pa_parents_dnlof = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"dnlof\",)\n",
        "parent_pa_parents_dnlof_sex_pval = toString(parent_pa_parents_dnlof$sex.pval)\n",
        "\n",
        "parent_pa_parents_inhlof = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"inhLoF\",)\n",
        "parent_pa_parents_inhlof_sex_pval = toString(parent_pa_parents_inhlof$sex.pval)\n",
        "\n",
        "parent_pa_parents_dnmis = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"dnmis\",)\n",
        "parent_pa_parents_dnmis_sex_pval = toString(parent_pa_parents_dnmis$sex.pval)\n",
        "\n",
        "parent_pa_parents_pssz = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"ps.sz\",)\n",
        "parent_pa_parents_pssz_sex_pval = toString(parent_pa_parents_pssz$sex.pval)\n",
        "\n",
        "parent_pa_parents_psasd = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"ps.asd\",)\n",
        "parent_pa_parents_psasd_sex_pval = toString(parent_pa_parents_psasd$sex.pval)\n",
        "\n",
        "parent_pa_parents_psea = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"ps.ea\",)\n",
        "parent_pa_parents_psea_sex_pval = toString(parent_pa_parents_psea$sex.pval)\n",
        "\n",
        "# Make the p-value dataframe for panel D\n",
        "pval_D_df <- data.frame(\n",
        "  variant.type = c(\"dnlof\", \"inhLoF\", \"dnmis\", \"ps.sz\", \"ps.asd\", \"ps.ea\"),\n",
        "  start = c(\"Father\", \"Father\"),\n",
        "  end = c(\"Mother\", \"Mother\"),\n",
        "  y.position = c(0.2, 0.2),\n",
        "  pval = c(paste(\"P = \", parent_pa_parents_dnlof_sex_pval,  sep=\"\"), \n",
        "           paste(\"P = \", parent_pa_parents_inhlof_sex_pval, sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_dnmis_sex_pval,  sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_pssz_sex_pval,   sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_psasd_sex_pval,  sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_psea_sex_pval,   sep=\"\"))\n",
        ")\n",
        "\n",
        "options(repr.plot.width=15, repr.plot.height=8)\n",
        "\n",
        "asd_D$stars <- cut(asd_D$x.pval, breaks = c(-Inf, 0.001, 0.01, 0.05, Inf), label = c(\"***\", \"**\", \"*\", \"\"))  # Create column of significance labels for main effect\n",
        "\n",
        "pdf(\"panelD.test.pdf\", width = 30, height = 30)\n",
        "panelD =\n",
        "ggplot(asd_D, aes(x = group, y = est, fill = group)) +\n",
        "geom_bar(stat = \"identity\", width = 1, position = position_nudge(x = 0.5)) +\n",
        "facet_wrap(~variant.type, ncol = 6, labeller = my_labellerD) +\n",
        "# P-value asterisks\n",
        "# geom_text(aes(label = stars), color = \"black\", vjust = -3, size = BIG - 5, hjust = 0) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0, size = BIG - 5, hjust = 0) +\n",
        "# Standard error whiskers\n",
        "geom_linerange(aes(ymin = est - est.stderr, ymax = est + est.stderr), position = position_nudge(0.5), size = 1) +\n",
        "scale_fill_manual(values = c(\"dodgerblue\", \"coral\")) +\n",
        "labs(y = \"Effect Size [sd]\") +\n",
        "theme(axis.text.x = element_blank()) +\n",
        "# Text\n",
        "theme(axis.title.x = element_blank()) +\n",
        "theme(axis.text.x = element_blank()) +\n",
        "# theme(strip.text.x = element_text(size = MASSIVE - 30, color = \"black\")) +\n",
        "theme(strip.text.x = element_text(size = MASSIVE, color = \"black\")) +\n",
        "# theme(axis.text.y = element_text(size = MASSIVE - 30)) +\n",
        "theme(axis.text.y = element_text(size = MASSIVE, color = \"black\")) +\n",
        "theme(axis.title.y = element_text(size = MASSIVE + 5, color = \"black\")) +\n",
        "# Legend\n",
        "theme(legend.title = element_blank()) +\n",
        "theme(legend.text = element_text(size = MASSIVE)) +\n",
        "theme(legend.key.size = unit(5, \"line\")) +\n",
        "theme(panel.background = element_blank()) +\n",
        "theme(strip.background = element_rect(color = \"white\", fill = \"white\", size = 1.5, linetype = \"solid\")) +\n",
        "# y-axis limits\n",
        "ylim(-0.10, 0.20)\n",
        "# p-value significance lines\n",
        "# geom_signif(data = pval_D_df, aes(xmin = 1.5, xmax = 2.5, annotations = pval, y_position = y.position), manual = TRUE, inherit.aes = FALSE, textsize = BIG - 20)\n",
        "panelD\n",
        "dev.off();\n",
        "\n",
        "### PANEL CD ###\n",
        "\n",
        "pdf(\"panelCD.test.pdf\", width = 40, height = 40)\n",
        "panelCD =\n",
        "ggarrange(panelC12, panelD, # panelC123 -> panelC12\n",
        "          labels = c(\"C\", \"D\"),\n",
        "          nrow = 2,\n",
        "          ncol = 1,\n",
        "          common.legend = FALSE,\n",
        "          legend = \"right\",\n",
        "          widths = c(1, 1))\n",
        "panelCD\n",
        "dev.off();\n",
        "\n",
        "\n",
        "\n",
        "### PANEL ABCD ###\n",
        "\n",
        "# pdf(\"fig5.ABCD.test.pdf\",width=60, height = 40)\n",
        "pdf(\"fig5.pdf\",width=60, height = 40)\n",
        "# svg(\"fig5.svg\", width=60, height=40)\n",
        "# svglite(\"fig5.svglite.svg\", width = 60, height = 40)\n",
        "# ggsave(\"fig5.ggsave.pdf\", plot = panelABCD, width = 120, height = 80, units = \"cm\", limitsize = FALSE)\n",
        "panelABCD =\n",
        "panelABCD = \n",
        "ggarrange(panelAB, panelCD,\n",
        "          nrow = 1,\n",
        "          ncol = 2,\n",
        "          align = \"v\",\n",
        "          widths = c(1.4, 1), heights = c(1, 1),\n",
        "          common.legend = FALSE)\n",
        "# panelABCD$layout$clip = \"off\"\n",
        "panelABCD;\n",
        "dev.off();\n",
        "\n",
        "# ggsave(\"fig5.ggsave.svg\", plot = panelABCD, width = 60, height = 40, units = \"cm\")\n",
        "\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cEAWsAT3vogn",
        "outputId": "520d5e66-77f0-4aaa-dd46-0724118fd78b"
      },
      "source": [
        "# Figure 4B\n",
        "\n",
        "install.packages(\"corrplot2.tar.gz\", repos = NULL)\n",
        "install.packages(\"reshape2\")\n",
        "library(corrplot2)\n",
        "library(reshape2)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "also installing the dependency ‘plyr’\n",
            "\n",
            "\n",
            "corrplot2 0.1 loaded\n",
            "\n",
            "\n",
            "Attaching package: ‘reshape2’\n",
            "\n",
            "\n",
            "The following object is masked from ‘package:tidyr’:\n",
            "\n",
            "    smiths\n",
            "\n",
            "\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 153
        },
        "id": "g-TlUAmAv4Oh",
        "outputId": "b081e303-b7df-44c1-f442-2c403bff7bfe"
      },
      "source": [
        "options(stringsAsFactors = FALSE);\n",
        "df = read.csv(asd_genetic_correlations_by_sex_out, sep = \"\\t\")\n",
        "\n",
        "pal <- colorRampPalette(c(\"#000080\",\"white\",\"#b30000\"))\n",
        "\n",
        "get_mat <- function(df, name) {\n",
        "  mat <- acast(df, y~x, value.var = name, fill = NA)\n",
        "  ord = c(\"dnMIS\", \"dnLOF\", \"inhLoF\",\n",
        "          \"pTDT.ASD.Dev\", \"pTDT.SCZ.Dev\", \"pTDT.EA.Dev\",\n",
        "          \"RVRS\", \"PRS\");\n",
        "  mat <- mat[ord, ord]\n",
        "  \n",
        "  labs = c(\"dnMIS\", \"dnLoF\", \"inhLoF\",\n",
        "           \"$PS[ASD]\", \"$PS[SCZ]\", \"$PS[EA]\",\n",
        "           \"RVRS\", \"CVRS\");\n",
        "\n",
        "\n",
        "\n",
        "  colnames(mat) <- labs\n",
        "  rownames(mat) <- labs\n",
        "  return(mat)\n",
        "}\n",
        "\n",
        "\n",
        "\n",
        "n_pal <- 40\n",
        "rows <- c(\"dnMIS\",\"dnLoF\",\"inhLoF\",\"RVRS\");\n",
        "cols <- c(\"$PS[ASD]\",\"$PS[SCZ]\",\"$PS[EA]\",\"CVRS\");\n",
        "\n",
        "\n",
        "# Males\n",
        "m_co <- get_mat(df,\"male_cor\")\n",
        "m_pv <- get_mat(df,\"male_pval\")\n",
        "f_co <- get_mat(df,\"female_cor\")\n",
        "f_pv <- get_mat(df,\"female_pval\")\n",
        "\n",
        "all_co <- get_mat(df,\"combined_cor\")\n",
        "all_pv <- get_mat(df,\"combined_pval\")\n",
        "\n",
        "m_co <- m_co[rows,cols]\n",
        "f_co <- f_co[rows,cols]\n",
        "all_co <- all_co[rows,cols]\n",
        "\n",
        "m_pv <- m_pv[rows,cols]\n",
        "f_pv <- f_pv[rows,cols]\n",
        "all_pv <- all_pv[rows,cols]\n",
        "\n",
        "# Rebrand female row names\n",
        "row.names(f_co) <- c(1,2,3,4)\n",
        "row.names(f_pv) <- c(1,2,3,4)\n",
        "# rbind the matrices\n",
        "sex_co <- rbind(m_co,f_co)\n",
        "sex_pv <- rbind(m_pv,f_pv)\n",
        "# Reorder rows\n",
        "rows <- c(\"dnMIS\",1,\"dnLoF\",2,\"inhLoF\",3,\"RVRS\",4);\n",
        "sex_co<-sex_co[rows,]\n",
        "sex_pv<-sex_pv[rows,]\n",
        "\n",
        "\n",
        "dum <- rbind(1, 0.1, 0.049, 0.0099)\n",
        "row.names(dum) <- c(1, 0.1, 0.05, 0.01)\n",
        "\n",
        "sig_lev <- c(.01, .05)\n",
        "\n",
        "# pdf(\"genetic_corr_pval_boxes.pdf\", width = 16, height = 16)\n",
        "pdf(\"fig4b.box.legend.pdf\", width=16, height = 16)\n",
        "# svg(\"fig4b.box.legend.svg\", width=16, height = 16)\n",
        "corrplot2(\n",
        "  dum,\n",
        "  # cbind(dum_cor),\n",
        "  method = \"square\",\n",
        "  outline = TRUE,\n",
        "  is.corr.= FALSE,\n",
        "  col = \"navy\", # color\n",
        "  tl.col = \"black\",\n",
        "  # tl.srt = 30, # rotation \n",
        "  tl.cex = 3,\n",
        "  p.mat = cbind(dum),\n",
        "  insig = \"label_sig\",\n",
        "  cl.pos = FALSE,\n",
        "  sig.level = sig_lev,\n",
        "  pch.cex = 5,\n",
        "  pch.col = \"white\",\n",
        "  norm_mat = dum,\n",
        ");\n",
        "dev.off()\n",
        "\n",
        "\n",
        "dum <- rbind(1, 0.75, 0.5, 0.25, 0.1, 0.049, 0.0249, 0.0099)\n",
        "dum_cor <- matrix(min(m_co, f_co), nrow = length(dum), ncol = 1)\n",
        "row.names(dum_cor) <- c(1, 0.75, 0.5, 0.25, 0.1, 0.05, 0.025, 0.01)\n",
        "sig_lev <- c(.01, .05)\n",
        "\n",
        "# Combined\n",
        "# pdf(\"combined_genetic_corr.pdf\", width=16, height = 16)\n",
        "pdf(\"fig4b.pdf\", width=16, height = 16)\n",
        "# svg(\"fig4b.svg\", width=16, height = 16)\n",
        "corrplot2(\n",
        "  all_co,\n",
        "  method = \"square\",\n",
        "  outline = TRUE,\n",
        "  is.corr = FALSE,\n",
        "  col = pal(n_pal),\n",
        "  tl.col = \"black\",\n",
        "  tl.srt = 0, # 90\n",
        "  tl.cex = 3, # 1.5\n",
        "  tl.offset = 1,\n",
        "  p.mat = all_pv,\n",
        "  sig.level =  sig_lev,\n",
        "  insig = \"label_sig\",\n",
        "  pch.cex = 5, # 5\n",
        "  pch.col = \"white\",\n",
        "  norm_mat = dum,\n",
        "  cl.lim = c(min(m_co, f_co), max(m_co, f_co)),\n",
        "  cor_max = max(m_co, f_co),\n",
        "  cor_min = min(m_co,f_co),\n",
        "  cl.cex = 2,\n",
        "  cl.ratio = 0.2,\n",
        ");\n",
        "dev.off()\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Warning message in text.default(pos.xlabel[, 1], pos.xlabel[, 2], newcolnames, srt = tl.srt, :\n",
            "“\"is.corr.\" is not a graphical parameter”\n",
            "Warning message in text.default(pos.ylabel[, 1], pos.ylabel[, 2], newrownames, col = tl.col, :\n",
            "“\"is.corr.\" is not a graphical parameter”\n",
            "Warning message in title(title, ...):\n",
            "“\"is.corr.\" is not a graphical parameter”\n"
          ],
          "name": "stderr"
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 187
        },
        "id": "VnqbEhAok3yb",
        "outputId": "8d80c2c0-fd70-46f8-874c-2721c0c289dd"
      },
      "source": [
        "system('rm -rf /content/sample_data', intern=TRUE)\n",
        "system('cd /home && pwd && tar -cvzf antaki_analysis.tar.gz /content', intern=TRUE)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "character(0)"
            ],
            "text/latex": "",
            "text/markdown": "",
            "text/html": [
              ""
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              " [1] \"/home\"                                                                     \n",
              " [2] \"/content/\"                                                                 \n",
              " [3] \"/content/.config/\"                                                         \n",
              " [4] \"/content/.config/.last_survey_prompt.yaml\"                                 \n",
              " [5] \"/content/.config/config_sentinel\"                                          \n",
              " [6] \"/content/.config/.last_opt_in_prompt.yaml\"                                 \n",
              " [7] \"/content/.config/active_config\"                                            \n",
              " [8] \"/content/.config/logs/\"                                                    \n",
              " [9] \"/content/.config/logs/2021.07.16/\"                                         \n",
              "[10] \"/content/.config/logs/2021.07.16/13.19.23.269640.log\"                      \n",
              "[11] \"/content/.config/logs/2021.07.16/13.20.21.584309.log\"                      \n",
              "[12] \"/content/.config/logs/2021.07.16/13.20.04.172291.log\"                      \n",
              "[13] \"/content/.config/logs/2021.07.16/13.20.22.328313.log\"                      \n",
              "[14] \"/content/.config/logs/2021.07.16/13.19.42.311157.log\"                      \n",
              "[15] \"/content/.config/logs/2021.07.16/13.19.58.237112.log\"                      \n",
              "[16] \"/content/.config/gce\"                                                      \n",
              "[17] \"/content/.config/.last_update_check.json\"                                  \n",
              "[18] \"/content/.config/configurations/\"                                          \n",
              "[19] \"/content/.config/configurations/config_default\"                            \n",
              "[20] \"/content/combined_genetic_corr.pdf\"                                        \n",
              "[21] \"/content/master_phen_4.SBayesR.csv\"                                        \n",
              "[22] \"/content/panelCD.test.pdf\"                                                 \n",
              "[23] \"/content/all_t1burden_tdt_results.tsv\"                                     \n",
              "[24] \"/content/panelD.test.pdf\"                                                  \n",
              "[25] \"/content/SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv\"\n",
              "[26] \"/content/genetic_corr_pval_boxes.pdf\"                                      \n",
              "[27] \"/content/clin_results_eur.SBayesR.20210715.csv\"                            \n",
              "[28] \"/content/panelAB.test.pdf\"                                                 \n",
              "[29] \"/content/fig5.pdf\"                                                         \n",
              "[30] \"/content/panelC.12.test.pdf\"                                               \n",
              "[31] \"/content/panelB.test.pdf\"                                                  \n",
              "[32] \"/content/fig5.ABCD.test.pdf\"                                               \n",
              "[33] \"/content/r2_table.csv\"                                                     \n",
              "[34] \"/content/fig4b.pdf\"                                                        \n",
              "[35] \"/content/panelA.test.pdf\"                                                  \n",
              "[36] \"/content/corrplot2.tar.gz\"                                                 \n",
              "[37] \"/content/fig4b.box.legend.pdf\"                                             \n",
              "[38] \"/content/asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv\"       \n",
              "[39] \"/content/SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv\"                   "
            ],
            "text/latex": "\\begin{enumerate*}\n\\item '/home'\n\\item '/content/'\n\\item '/content/.config/'\n\\item '/content/.config/.last\\_survey\\_prompt.yaml'\n\\item '/content/.config/config\\_sentinel'\n\\item '/content/.config/.last\\_opt\\_in\\_prompt.yaml'\n\\item '/content/.config/active\\_config'\n\\item '/content/.config/logs/'\n\\item '/content/.config/logs/2021.07.16/'\n\\item '/content/.config/logs/2021.07.16/13.19.23.269640.log'\n\\item '/content/.config/logs/2021.07.16/13.20.21.584309.log'\n\\item '/content/.config/logs/2021.07.16/13.20.04.172291.log'\n\\item '/content/.config/logs/2021.07.16/13.20.22.328313.log'\n\\item '/content/.config/logs/2021.07.16/13.19.42.311157.log'\n\\item '/content/.config/logs/2021.07.16/13.19.58.237112.log'\n\\item '/content/.config/gce'\n\\item '/content/.config/.last\\_update\\_check.json'\n\\item '/content/.config/configurations/'\n\\item '/content/.config/configurations/config\\_default'\n\\item '/content/combined\\_genetic\\_corr.pdf'\n\\item '/content/master\\_phen\\_4.SBayesR.csv'\n\\item '/content/panelCD.test.pdf'\n\\item '/content/all\\_t1burden\\_tdt\\_results.tsv'\n\\item '/content/panelD.test.pdf'\n\\item '/content/SummaryGeneticData\\_REACH\\_SSC\\_SPARK.responses.SBayesR.20210715.csv'\n\\item '/content/genetic\\_corr\\_pval\\_boxes.pdf'\n\\item '/content/clin\\_results\\_eur.SBayesR.20210715.csv'\n\\item '/content/panelAB.test.pdf'\n\\item '/content/fig5.pdf'\n\\item '/content/panelC.12.test.pdf'\n\\item '/content/panelB.test.pdf'\n\\item '/content/fig5.ABCD.test.pdf'\n\\item '/content/r2\\_table.csv'\n\\item '/content/fig4b.pdf'\n\\item '/content/panelA.test.pdf'\n\\item '/content/corrplot2.tar.gz'\n\\item '/content/fig4b.box.legend.pdf'\n\\item '/content/asd\\_genetic\\_correlations\\_by\\_sex\\_final.SBayesR.20210715.tsv'\n\\item '/content/SummaryGeneticData\\_REACH\\_SSC\\_SPARK.SBayesR.csv'\n\\end{enumerate*}\n",
            "text/markdown": "1. '/home'\n2. '/content/'\n3. '/content/.config/'\n4. '/content/.config/.last_survey_prompt.yaml'\n5. '/content/.config/config_sentinel'\n6. '/content/.config/.last_opt_in_prompt.yaml'\n7. '/content/.config/active_config'\n8. '/content/.config/logs/'\n9. '/content/.config/logs/2021.07.16/'\n10. '/content/.config/logs/2021.07.16/13.19.23.269640.log'\n11. '/content/.config/logs/2021.07.16/13.20.21.584309.log'\n12. '/content/.config/logs/2021.07.16/13.20.04.172291.log'\n13. '/content/.config/logs/2021.07.16/13.20.22.328313.log'\n14. '/content/.config/logs/2021.07.16/13.19.42.311157.log'\n15. '/content/.config/logs/2021.07.16/13.19.58.237112.log'\n16. '/content/.config/gce'\n17. '/content/.config/.last_update_check.json'\n18. '/content/.config/configurations/'\n19. '/content/.config/configurations/config_default'\n20. '/content/combined_genetic_corr.pdf'\n21. '/content/master_phen_4.SBayesR.csv'\n22. '/content/panelCD.test.pdf'\n23. '/content/all_t1burden_tdt_results.tsv'\n24. '/content/panelD.test.pdf'\n25. '/content/SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv'\n26. '/content/genetic_corr_pval_boxes.pdf'\n27. '/content/clin_results_eur.SBayesR.20210715.csv'\n28. '/content/panelAB.test.pdf'\n29. '/content/fig5.pdf'\n30. '/content/panelC.12.test.pdf'\n31. '/content/panelB.test.pdf'\n32. '/content/fig5.ABCD.test.pdf'\n33. '/content/r2_table.csv'\n34. '/content/fig4b.pdf'\n35. '/content/panelA.test.pdf'\n36. '/content/corrplot2.tar.gz'\n37. '/content/fig4b.box.legend.pdf'\n38. '/content/asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv'\n39. '/content/SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv'\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'/home'</li><li>'/content/'</li><li>'/content/.config/'</li><li>'/content/.config/.last_survey_prompt.yaml'</li><li>'/content/.config/config_sentinel'</li><li>'/content/.config/.last_opt_in_prompt.yaml'</li><li>'/content/.config/active_config'</li><li>'/content/.config/logs/'</li><li>'/content/.config/logs/2021.07.16/'</li><li>'/content/.config/logs/2021.07.16/13.19.23.269640.log'</li><li>'/content/.config/logs/2021.07.16/13.20.21.584309.log'</li><li>'/content/.config/logs/2021.07.16/13.20.04.172291.log'</li><li>'/content/.config/logs/2021.07.16/13.20.22.328313.log'</li><li>'/content/.config/logs/2021.07.16/13.19.42.311157.log'</li><li>'/content/.config/logs/2021.07.16/13.19.58.237112.log'</li><li>'/content/.config/gce'</li><li>'/content/.config/.last_update_check.json'</li><li>'/content/.config/configurations/'</li><li>'/content/.config/configurations/config_default'</li><li>'/content/combined_genetic_corr.pdf'</li><li>'/content/master_phen_4.SBayesR.csv'</li><li>'/content/panelCD.test.pdf'</li><li>'/content/all_t1burden_tdt_results.tsv'</li><li>'/content/panelD.test.pdf'</li><li>'/content/SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv'</li><li>'/content/genetic_corr_pval_boxes.pdf'</li><li>'/content/clin_results_eur.SBayesR.20210715.csv'</li><li>'/content/panelAB.test.pdf'</li><li>'/content/fig5.pdf'</li><li>'/content/panelC.12.test.pdf'</li><li>'/content/panelB.test.pdf'</li><li>'/content/fig5.ABCD.test.pdf'</li><li>'/content/r2_table.csv'</li><li>'/content/fig4b.pdf'</li><li>'/content/panelA.test.pdf'</li><li>'/content/corrplot2.tar.gz'</li><li>'/content/fig4b.box.legend.pdf'</li><li>'/content/asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv'</li><li>'/content/SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv'</li></ol>\n"
            ]
          },
          "metadata": {
            "tags": []
          }
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "unDh3vD20_z_"
      },
      "source": [
        "Figure 5"
      ]
    }
  ]
}