{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Antaki2021_R_analysis.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "machine_shape": "hm",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "ir",
      "display_name": "R"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/sebatlab/Antaki2021/blob/main/Antaki2021_R_analysis.2022_06_13_update.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QywIWFWURQvm"
      },
      "source": [
        "# Table of Contents\n",
        "-------------------\n",
        "* [Variance Explained by Genetic Risk Factors](#variance_explained_by_genetic_risk_factors)\n",
        "* [Negative Correlations of Rare and Common](#negative_correlations_rare_common)\n",
        "* [Gene-Phenotype Correlations](#gene_phenotype_correlations)"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Jy0RET5LJLFr",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b005812b-fe5a-4044-ee9c-e86b68c101ce"
      },
      "source": [
        "#@title\n",
        "# Install packages\n",
        "install.packages(\"gplots\")\n",
        "install.packages(\"basicTrendline\")\n",
        "\n",
        "# Attaching packages\n",
        "library(gplots)\n",
        "library(basicTrendline)\n",
        "\n",
        "\n",
        "# Figure 2A packages\n",
        "install.packages(\"fmsb\")\n",
        "install.packages(\"lmtest\")\n",
        "install.packages(\"boot\")\n",
        "install.packages(\"stringr\")\n",
        "\n",
        "library(fmsb)\n",
        "library(lmtest)\n",
        "library(boot)\n",
        "library(stringr)\n",
        "\n",
        "install.packages(\"collections\")\n",
        "library(collections)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "L5iqs7Wj7JCh"
      },
      "source": [
        "# Input filenames and output filenames. I should add which sections take in\n",
        "# which inputs and outputs.\n",
        "\n",
        "filename_table = \"SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv\"\n",
        "filename_phen  = \"master_phen_4.SBayesR.csv\"\n",
        "\n",
        "r2_out = \"R2_estimates.20210715.csv\"\n",
        "responses_out = \"SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv\"\n",
        "asd_genetic_correlations_by_sex_out = \"asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv\"\n",
        "clin_results_out = \"clin_results_eur.SBayesR.20210715.csv\""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "gdLB3RK7zDF3"
      },
      "source": [
        "<a name=\"variance_explained_by_genetic_risk_factors\"></a>\n",
        "## Variance Explained by Genetic Risk Factors (Fig. 2A)"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "aIGYenzFiRpu"
      },
      "source": [
        "# Load table\n",
        "df <- read.table(filename_table, header = TRUE, sep = \",\")\n",
        "df <- df[df$EUR.Ancestry == 1,] # Only use samples with European ancestry\n",
        "df <- df[df$Family.Type == \"trio\",] # Only use trios for this analysis\n",
        "# Drop SV columns (not used in this analysis)\n",
        "drops <- c(\"dnSV\",\n",
        "           \"SV.LoF.Paternal.Transmitted\",\"SV.LoF.Paternal.Untransmitted\",\n",
        "          \"SV.LoF.Maternal.Transmitted\",\"SV.LoF.Maternal.Untransmitted\",\n",
        "          \"CRE.SV.Paternal.Transmitted\",\"CRE.SV.Paternal.Untransmitted\",\n",
        "          \"CRE.SV.Maternal.Transmitted\",\"CRE.SV.Maternal.Untransmitted\",\n",
        "          \"RVRS\",\"PRS\",\"GRS\")\n",
        "df <- df[ , !(names(df) %in% drops)]\n",
        "df <- na.omit(df) # Remove rows with NA values\n",
        "\n",
        "# Make cohort variables\n",
        "df$REACH_cohort = as.numeric(df$Cohort == \"REACH\")\n",
        "df$SPARK_cohort = as.numeric(df$Cohort == \"SPARK\")\n",
        "df$SSC_cohort = as.numeric(df$Cohort == \"SSC\")\n",
        "\n",
        "dfm <- df[df$Sex == 0,] # Dataframe containing only male samples\n",
        "dff <- df[df$Sex == 1,] # Dataframe containing only female samples\n",
        "\n",
        "# Grab PCs\n",
        "# df[,grep(\"^PC([1-9]|10)$\", names(df), value=TRUE)]\n",
        "# Grab Sex\n",
        "# df[,grep(\"^Sex$\", names(df), value=TRUE)]\n",
        "# Grab Phenotype\n",
        "# df[,grep(\"^Phenotype$\", names(df), value=TRUE)]\n",
        "\n",
        "# Sample rows (without replacement by default?)\n",
        "# df[sample(nrow(df), 1000), ]"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "WXefMeZHyVoN"
      },
      "source": [
        "# Written by James Guevara\n",
        "\n",
        "# The bootstrap is used to estimate a confidence interval for the R2 estimates.\n",
        "# However, the bootstrap procedure seemingly requires a lot of CPUs and takes a lot of time\n",
        "# given the amount of data in the table.\n",
        "# So I ran the procedure on a compute cluster (Comet at UCSD) using the Slurm Workload Manager.\n",
        "# But I've adapted the code here so that the reader can understand how we ran the\n",
        "# bootstrap. \n",
        "getNagelkerkeR2_bootstrap <- function(formula, data, indices) {\n",
        "  d <- data[indices,]\n",
        "  fit_alt  <- glm(formula = formula, data = d, family = binomial)\n",
        "  fit_null <- glm(formula = d$Phenotype~d$Sex+d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort, data=d, family=binomial)\n",
        "  return(NagelkerkeR2(fit_alt)$R2 - NagelkerkeR2(fit_null)$R2)\n",
        "}\n",
        "\n",
        "getNagelkerkeR2 <- function(dataframe, alt_vars_in, null_vars_in, text) {\n",
        "  d <- dataframe\n",
        "  alt_variables <- alt_vars_in\n",
        "  alt_formula <- reformulate(alt_variables, response = \"Phenotype\")\n",
        "  null_variables <- null_vars_in\n",
        "  null_formula <- reformulate(null_variables, response = \"Phenotype\")\n",
        "  fit_alt  <- glm(formula = alt_formula , data = d, family = binomial)\n",
        "  fit_null <- glm(formula = null_formula , data = d, family = binomial)\n",
        "  nagelkerke_R2_val = NagelkerkeR2(fit_alt)$R2 - NagelkerkeR2(fit_null)$R2\n",
        "  lr <- lrtest(fit_alt,fit_null) # Likelihood ratio test\n",
        "  lr_p_val = lr[\"Pr(>Chisq)\"][2,1] # p-value\n",
        "\n",
        "  # cat(c(text,\"\\n\"))\n",
        "  # cat(c(\"R2:\\t\", nagelkerke_R2_val, \"\\t\", \"Likelihood_ratio_test_p-value:\\t\", lr_p_val, \"\\n\"))\n",
        "\n",
        "  # Put into dataframe? (R2, p-value)\n",
        "  df_r2 = data.frame(text, nagelkerke_R2_val, lr_p_val)\n",
        "\n",
        "  # Get the alt_vars_in c vector into a format appropriate for the bootstrap function \n",
        "  alt_vars_formula = paste(unlist(strsplit(alt_vars_in, split=\"d$\", fixed=TRUE)), collapse=\" \")\n",
        "  alt_vars_formula = paste(\"Phenotype ~\", alt_vars_formula, collapse= \" \")\n",
        "  # print(alt_vars_formula)\n",
        "\n",
        "  # Using the bootstrap to get the confidence interval estimates (and standard error of R2)\n",
        "  # r2_stats <- boot(data=d, statistic=getNagelkerkeR2_bootstrap, R=10000, formula=alt_vars_formula, parallel=\"multicore\", ncpus=2)\n",
        "  # ste = sd(r2_stats$t)\n",
        "  # ci = boot.ci(r2_stats, type=\"bca\") # Using the BCa interval (more information here: https://blogs.sas.com/content/iml/2017/07/12/bootstrap-bca-interval.html)\n",
        "  # ci_lower = as.double(ci$bca[1,4])\n",
        "  # ci_upper = as.double(ci$bca[1,5])\n",
        "  # df_r2_tmp = data.frame(text, nagelkerke_R2_val, lr_p_val, ste, ci_lower, ci_upper)\n",
        "\n",
        "\n",
        "  return(df_r2)\n",
        "} \n",
        "\n",
        "\n",
        "cohort_vars <- c(\"d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort\")\n",
        "pc_vars <- c(\"d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10\")\n",
        "sex_var <- c(\"d$Sex\") # Only use with combined models (it doesn't matter)\n",
        "null_vars <- paste(cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "\n",
        "\n",
        "# Full model\n",
        "var <- c(\"d$dnMIS+d$dnLOF+d$inhLoF+d$pTDT.ASD.Dev+d$pTDT.EA.Dev+d$pTDT.SCZ.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_full_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"combined_both\")\n",
        "# Males\n",
        "df_full_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"combined_male\")\n",
        "# Females\n",
        "df_full_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"combined_female\")\n",
        "\n",
        "\n",
        "# PRS model\n",
        "var <- c(\"d$pTDT.ASD.Dev+d$pTDT.EA.Dev+d$pTDT.SCZ.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_prs_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"prs_both\")\n",
        "# Males\n",
        "df_prs_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"prs_male\")\n",
        "# Females\n",
        "df_prs_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"prs_female\")\n",
        "\n",
        "# Rare model\n",
        "var <- c(\"d$dnMIS+d$dnLOF+d$inhLoF\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_rare_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"rare_both\")\n",
        "# Males\n",
        "df_rare_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"rare_male\")\n",
        "# Females\n",
        "df_rare_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"rare_female\")\n",
        "\n",
        "# pTDT.ASD.Dev\n",
        "var <- c(\"d$pTDT.ASD.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_asd_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"ptdt.asd.dev_both\")\n",
        "# Males\n",
        "df_ptdt_asd_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"ptdt.asd.dev_male\")\n",
        "# Females\n",
        "df_ptdt_asd_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"ptdt.asd.dev_female\")\n",
        "\n",
        "# pTDT.EA.Dev\n",
        "var <- c(\"d$pTDT.EA.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_ea_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"ptdt.ea.dev_both\")\n",
        "# Males\n",
        "df_ptdt_ea_r2_male =getNagelkerkeR2(dfm, alt_vars, null_vars, \"ptdt.ea.dev_male\")\n",
        "# Females\n",
        "df_ptdt_ea_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"ptdt.ea.dev_female\")\n",
        "\n",
        "# pTDT.SCZ.Dev\n",
        "var <- c(\"d$pTDT.SCZ.Dev\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_scz_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"ptdt.scz.dev_both\")\n",
        "# Males\n",
        "df_ptdt_scz_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"ptdt.scz.dev_male\")\n",
        "# Females\n",
        "df_ptdt_scz_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"ptdt.scz.dev_female\")\n",
        "\n",
        "# dnMIS\n",
        "var <- c(\"d$dnMIS\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_dnmis_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"dnMIS_both\")\n",
        "# Males\n",
        "df_dnmis_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"dnMIS_male\")\n",
        "# Females\n",
        "df_dnmis_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"dnMIS_female\")\n",
        "\n",
        "# dnLOF\n",
        "var <- c(\"d$dnLOF\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_dnlof_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"dnLOF_both\")\n",
        "# Males\n",
        "df_dnlof_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"dnLOF_male\")\n",
        "# Females\n",
        "df_dnlof_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"dnLOF_female\")\n",
        "\n",
        "# inhLoF\n",
        "var <- c(\"d$inhLoF\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_inhlof_r2 = getNagelkerkeR2(df, alt_vars, null_vars, \"inhLOF_both\")\n",
        "# Males\n",
        "df_inhlof_r2_male = getNagelkerkeR2(dfm, alt_vars, null_vars, \"inhLOF_male\")\n",
        "# Females\n",
        "df_inhlof_r2_female = getNagelkerkeR2(dff, alt_vars, null_vars, \"inhLOF_female\")\n",
        "\n",
        "\n",
        "# Combine all the dataframes\n",
        "df_combined = rbind(df_full_r2, df_full_r2_male, df_full_r2_female,\n",
        "                    df_prs_r2, df_prs_r2_male, df_prs_r2_female,\n",
        "                    df_rare_r2, df_rare_r2_male, df_rare_r2_female,\n",
        "                    df_ptdt_asd_r2, df_ptdt_asd_r2_male, df_ptdt_asd_r2_female,\n",
        "                    df_ptdt_ea_r2, df_ptdt_ea_r2_male, df_ptdt_ea_r2_female,\n",
        "                    df_ptdt_scz_r2, df_ptdt_scz_r2_male, df_ptdt_scz_r2_female,\n",
        "                    df_dnmis_r2, df_dnmis_r2_male, df_dnmis_r2_female,\n",
        "                    df_dnlof_r2, df_dnlof_r2_male, df_dnlof_r2_female,\n",
        "                    df_inhlof_r2, df_inhlof_r2_male, df_inhlof_r2_female)\n",
        "\n",
        "# Add a header to the combined dataframe\n",
        "colnames(df_combined) <- c(\"R2_type\", \"R2\", \"Likelihood_Ratio_p_value\")\n",
        "\n",
        "# Write to file\n",
        "write.csv(df_combined, file=r2_out, row.name=TRUE)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "0KSm0hxZx1-l"
      },
      "source": [
        "# R2 by cohort\n",
        "pc_vars <- c(\"d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10\")\n",
        "sex_var <- c(\"d$Sex\") # Only use with combined models (it doesn't matter)\n",
        "null_vars <- paste(pc_vars, sex_var, sep = \"+\") \n",
        "\n",
        "df_reach <- df[df$REACH_cohort == 1,]\n",
        "df_spark <- df[df$SPARK_cohort == 1,]\n",
        "df_ssc   <- df[df$SSC_cohort == 1,]\n",
        "\n",
        "# pTDT.ASD.Dev\n",
        "var <- c(\"d$pTDT.ASD.Dev\")\n",
        "alt_vars  <- paste(var, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_asd_r2_reach = getNagelkerkeR2(df_reach, alt_vars, null_vars, \"ptdt.asd.dev_REACH\")\n",
        "df_ptdt_asd_r2_spark = getNagelkerkeR2(df_spark, alt_vars, null_vars, \"ptdt.asd.dev_SPARK\")\n",
        "df_ptdt_asd_r2_ssc   = getNagelkerkeR2(df_ssc, alt_vars, null_vars, \"ptdt.asd.dev_SSC\")\n",
        "\n",
        "# Combined cohort (I already did this..)\n",
        "cohort_vars <- c(\"d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort\")\n",
        "null_vars <- paste(cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "alt_vars  <- paste(var, cohort_vars, pc_vars, sex_var, sep = \"+\")\n",
        "df_ptdt_asd_r2_combined   = getNagelkerkeR2(df_ssc, alt_vars, null_vars, \"ptdt.asd.dev_Combined\")\n",
        "\n",
        "df_cohorts = rbind(df_ptdt_asd_r2_reach, df_ptdt_asd_r2_ssc, df_ptdt_asd_r2_spark, df_ptdt_asd_r2_combined)\n",
        "\n",
        "# Add a header to the combined dataframe\n",
        "colnames(df_cohorts) <- c(\"Model\", \"R2\", \"P-Value\")\n",
        "\n",
        "# Write to file\n",
        "write.csv(df_cohorts, file = \"SupplementaryTable5.PS_ASD_R2_by_cohort.csv\", row.name = TRUE)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "IxxT0d1O13sJ"
      },
      "source": [
        "# Responses from residuals\n",
        "library(fmsb)\n",
        "options(stringsAsFactors = FALSE);\n",
        "\n",
        "d = df\n",
        "\n",
        "# PRS scores residualized:\n",
        "ptdt_prs_asd_resid <- resid(glm(d$pTDT.ASD.Dev~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d, na.action=\"na.exclude\"))\n",
        "ptdt_prs_ea_resid  <- resid(glm(d$pTDT.EA.Dev  ~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d, na.action=\"na.exclude\"))\n",
        "ptdt_prs_scz_resid <- resid(glm(d$pTDT.SCZ.Dev~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d, na.action=\"na.exclude\"))\n",
        "\n",
        "\n",
        "# Fit combined prs model\n",
        "fit_alt_prs <- glm(d$Phenotype~ptdt_prs_asd_resid+ptdt_prs_ea_resid+ptdt_prs_scz_resid, data=df, family=binomial)\n",
        "response_prs = predict(fit_alt_prs, type = \"response\")\n",
        "\n",
        "# Rare scores residualized\n",
        "snv_dn_lof_resid <- resid(glm(d$dnLOF ~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d,na.action=\"na.exclude\"))\n",
        "snv_dn_mis_resid <- resid(glm(d$dnMIS ~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d,na.action=\"na.exclude\") )\n",
        "snv_lof_t_resid  <- resid(glm(d$inhLoF~d$PC1+d$PC2+d$PC3+d$PC4+d$PC5+d$PC6+d$PC7+d$PC8+d$PC9+d$PC10+d$REACH_cohort+d$SPARK_cohort+d$SSC_cohort+d$Sex, data=d,na.action=\"na.exclude\") )\n",
        "\n",
        "\n",
        "# Fit combined rare model\n",
        "fit_alt_rare <- glm(d$Phenotype~snv_dn_lof_resid+snv_dn_mis_resid+snv_lof_t_resid, data=d, family=binomial)\n",
        "response_rare = predict(fit_alt_rare, type = \"response\")\n",
        "\n",
        "# Fit rare + prs combined model\n",
        "fit_alt_combined <- glm(d$Phenotype~ptdt_prs_asd_resid+ptdt_prs_ea_resid+ptdt_prs_scz_resid+snv_dn_lof_resid+snv_dn_mis_resid+snv_lof_t_resid, data=d, family=binomial)\n",
        "response_combined = predict(fit_alt_combined, type = \"response\")\n",
        "\n",
        "# Fit individual features\n",
        "# ptdt_prs_asd\n",
        "fit_alt_ptdt_prs_asd <- glm(d$Phenotype~ptdt_prs_asd_resid, data=d, family=binomial)\n",
        "response_ptdt_prs_asd = predict(fit_alt_ptdt_prs_asd, type = \"response\")\n",
        "# ptdt_prs_ea\n",
        "fit_alt_ptdt_prs_ea <- glm(d$Phenotype~ptdt_prs_ea_resid, data=d, family=binomial)\n",
        "response_ptdt_prs_ea = predict(fit_alt_ptdt_prs_ea, type = \"response\")\n",
        "# ptdt_prs_scz\n",
        "fit_alt_ptdt_prs_scz <- glm(d$Phenotype~ptdt_prs_scz_resid, data=d, family=binomial)\n",
        "response_ptdt_prs_scz = predict(fit_alt_ptdt_prs_scz, type = \"response\")\n",
        "\n",
        "# snv_dn_lof\n",
        "fit_alt_snv_dn_lof <- glm(d$Phenotype~snv_dn_lof_resid, data=d, family=binomial)\n",
        "response_snv_dn_lof = predict(fit_alt_snv_dn_lof, type = \"response\")\n",
        "# snv_dn_mis\n",
        "fit_alt_snv_dn_mis <- glm(d$Phenotype~snv_dn_mis_resid, data=d, family=binomial)\n",
        "response_snv_dn_mis = predict(fit_alt_snv_dn_mis, type = \"response\")\n",
        "# snv_lof_t\n",
        "fit_alt_snv_lof_t <- glm(d$Phenotype~snv_lof_t_resid, data=d, family=binomial)\n",
        "response_snv_lof_t = predict(fit_alt_snv_lof_t, type = \"response\")\n",
        "\n",
        "# Responses (not z-scored yet)\n",
        "# d$all_combined_score = response_combined\n",
        "# d$prs_combined_score = response_prs\n",
        "# d$rare_combined_score = response_rare\n",
        "# d$response_ptdt_prs_asd = response_ptdt_prs_asd\n",
        "# d$response_ptdt_prs_asd = response_ptdt_prs_ea\n",
        "# d$response_ptdt_prs_scz = response_ptdt_prs_scz\n",
        "# d$response_snv_dn_lof = response_snv_dn_lof\n",
        "# d$response_snv_dn_mis = response_snv_dn_mis\n",
        "# d$response_snv_lof_t = response_snv_lof_t\n",
        "\n",
        "# Get GRS, PRS, RVRS columns (normalize all_combined_score, prs_combined_score, and rare_combined_score)\n",
        "d$GRS = (response_combined - mean(response_combined))/sd(response_combined)\n",
        "d$PRS = (response_prs - mean(response_prs))/sd(response_prs)\n",
        "d$RVRS = (response_rare - mean(response_rare))/sd(response_rare)\n",
        "\n",
        "write.csv(d, file = responses_out, row.name=TRUE)"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JCR9iE2ZRXQk"
      },
      "source": [
        "<a name=\"negative_correlations_rare_common\"></a>\n",
        "## Negative correlations of rare and common (Fig. 3)\n",
        "-------------------------------"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "C8cypSPGag-u",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "beed3e17-7255-4b67-d0d3-c67826c4b737"
      },
      "source": [
        "# Written by Danny Antaki, edited by James Guevara\n",
        "options(stringsAsFactors = FALSE);\n",
        "# df = read.csv(\"SummaryGeneticData_REACH_SSC_SPARK.csv\",sep=\",\")\n",
        "df = read.csv(responses_out,sep=\",\")\n",
        "df$dnAll = df$dnLOF+df$dnMIS\n",
        "df$cohort_covar <- factor(df$Cohort)\n",
        "df <- df[df$EUR.Ancestry==1,]\n",
        "vars<-c(\n",
        "    \"dnMIS\" ,\"dnLOF\",\"dnAll\",\n",
        "    \"inhLoF\", \"RVRS\" , \n",
        "    \"pTDT.ASD.Dev\" , \"pTDT.SCZ.Dev\" ,\"pTDT.EA.Dev\",\n",
        "    \"PRS\"\n",
        ");\n",
        "cols <- c(\"cohort_covar\",\"Sex\",\"Phenotype\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\",\"PC6\",\"PC7\",\"PC8\",\"PC9\",\"PC10\",vars)\n",
        "df<-na.omit(df[cols])\n",
        "print(nrow(df))\n",
        "###########################################\n",
        "fit_model<- function(df,x,y,add_sex=FALSE){\n",
        "    Y<- df[[x]]\n",
        "    X<- df[[y]]\n",
        "    PC1<-df$PC1\n",
        "    PC2<-df$PC2\n",
        "    PC3<-df$PC3\n",
        "    PC4<-df$PC4\n",
        "    PC5<-df$PC5\n",
        "    PC6<-df$PC6\n",
        "    PC7<-df$PC7\n",
        "    PC8<-df$PC8\n",
        "    PC9<-df$PC9\n",
        "    PC10<-df$PC10\n",
        "    phen<-df$Phenotype\n",
        "    cohort_covar<-df$cohort_covar\n",
        "    sex<-df$Sex\n",
        "    sex_pv<-NA    \n",
        "    if (add_sex==TRUE){\n",
        "        mod<-lm(Y~X+sex+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        nul<-lm(Y~sex+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        #sex interaction\n",
        "        interaction_mod<-lm(Y~X*sex+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        sex_pv<-signif(summary(interaction_mod)$coefficients[17,4],digits=2)\n",
        "    } else {\n",
        "        mod<-lm(Y~X+phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "        nul<-lm(Y~phen+cohort_covar+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "    }\n",
        "    # slope of fit of X to Y\n",
        "    est<-signif(summary(mod)$coefficients[2,1],digits=2)\n",
        "    xpv<-signif(summary(mod)$coefficients[2,4],digits=2)\n",
        "    #r2 is the difference of the mod with X and without X\n",
        "    r2<-signif(summary(mod)$r.squared-summary(nul)$r.squared,digits=2)\n",
        "    pearsons_cor<-signif(cor(Y,X, use=\"pairwise.complete.obs\"),digits=2)    \n",
        "    if (add_sex==TRUE){\n",
        "        result = paste(est,xpv,r2,pearsons_cor,sex_pv,sep=\"\\t\")\n",
        "    } else {\n",
        "        result = paste(est,xpv,r2,pearsons_cor,sep=\"\\t\")\n",
        "    }\n",
        "    #sprintf(\"%f\\t%f\\t%f\\t%f\\t%f\",est,xpv,r2,pearson_cor,sex_pv)\n",
        "    return(result)\n",
        "}\n",
        "###########################################\n",
        "header = paste(\"y\",\"x\",\n",
        "               \"male_est\",\"male_pval\",\"male_r2\",\"male_cor\",\n",
        "               \"female_est\",\"female_pval\",\"female_r2\",\"female_cor\",\n",
        "               \"combined_est\",\"combined_pval\",\"combined_r2\",\"combined_cor\",\n",
        "               \"sex_interaction_pval\",\n",
        "               sep=\"\\t\"\n",
        "              );\n",
        "output=vector(length = 1+(length(vars)^2)-(length(vars)))\n",
        "output[1]<-header\n",
        "i=2\n",
        "for (x in vars)\n",
        "{\n",
        "  for(y in vars)\n",
        "  {\n",
        "    if (x==y){ next; } \n",
        "    com_results <- fit_model(df,x,y,add_sex=TRUE)\n",
        "    mal_results <- fit_model(df[df$Sex==0,],x,y,add_sex=FALSE)\n",
        "    fem_results <- fit_model(df[df$Sex==1,],x,y,add_sex=FALSE)\n",
        "    # Y - X - mal\n",
        "    output[i] <- paste(x,y,mal_results,fem_results,com_results,sep=\"\\t\")\n",
        "    i=i+1;\n",
        "  }\n",
        "}\n",
        "out = asd_genetic_correlations_by_sex_out\n",
        "ofh<-file(out)\n",
        "writeLines(output,ofh)\n",
        "close(ofh)\n",
        "print(\">>>DONE<<<\")"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] 8301\n",
            "[1] \">>>DONE<<<\"\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JkVtQILix9lT"
      },
      "source": [
        "<a name=\"gene_phenotype_correlations\"></a>\n",
        "## Gene–phenotype correlations (Fig. 5)"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "z1QElvT8xLlr"
      },
      "source": [
        "#Author: Jonathan Sebat http://sebatlab.org March 19, 2021\n",
        "\n",
        "#The following phenotypes were available for only one cohort in either cases or controls: srs.case,vabs.case,srs.cont,scq.cont,vabs.cont,bapq.parents,srs.parents)\n",
        "#for these, age, sex and PCs were included as covariates, but not cohort (a 1st set of functions is written for this test)\n",
        "\n",
        "#SCQ, RBS and DCDQ were available on cases in the multiple cohorts (SSC and SPARK). Parental age and educational attainment were also available on multiple cohorts\n",
        "#for these, \"cohort\" was included as a covariate (a 2nd set of functions are written for this test)\n",
        "\n",
        "\n",
        "\n",
        "#importing the input data table that contains rare variant counts (dnLoF, dnMis, inhLoF), polygenic scores(PSasd, PSsz, PSea) and clinical phenotypes (srs, scq, rbs, vabs, dcdq, parental age, parental educational attainment, parental bapq)\n",
        "\n",
        "# master.gen.phen.table<-read.csv(\"C:\\\\Users\\\\jsebat\\\\sebatlab Dropbox\\\\Autism\\\\Autism_Illumina_WGS\\\\RV_PRS_Analysis_Jun2020\\\\master_phen_4.csv\", header = TRUE)\n",
        "# master.gen.phen.table<-read.csv(filename_phen, header = TRUE)\n",
        "master.gen.phen.table<-read.csv(\"/content/master_phen_4.SBayesR.20210915.csv\", header = TRUE)\n",
        "# Turn cohort into a factor variable\n",
        "master.gen.phen.table$cohort = factor(master.gen.phen.table$cohort)\n",
        "\n",
        "################################################################################\n",
        "##################################################################################\n",
        "####### First we create the main functions to run linear regression   ###########\n",
        "#######        on y = phenoype, x = genetic factor                    #############\n",
        "#####################################################################################\n",
        "#####################################################################################\n",
        "\n",
        "###############    1st set of scripts ###########################################################3\n",
        "#For phenotypes available on only a single cohort, \"cohort\" is NOT included as a covariate\n",
        "#linear regression controls for age, sex and PCs\n",
        "#and tests for a gene-by-sex interactions\n",
        "#the functions for these are as follows\n",
        "#gen.phen.1 a function that runs a linear regression and tests for gene-by-sex interaction. These are the primary results described in Fig. 5\n",
        "#gen.phen.2 To characterize gene-by sex effects that are significant, gene.phen.2 runs the regression on males and females separately without sex as a covariate\n",
        "#gene.phen.3 runs linear regression and gene-by-sex interaction on PARENTS, then to investigate gene by sex interactions, runs regression on mothers and fathers individually\n",
        "\n",
        "\n",
        "#gen.phen.1 a function that runs a linear regression on both sexes and outputs the result\n",
        "\n",
        "gen.phen.1<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "#define the sexes\n",
        "male<-master.temp$sex==\"Male\"\n",
        "female<-master.temp$sex==\"Female\"\n",
        "  \n",
        "#Test combined sample\n",
        "sex<-master.temp$sex\n",
        "cohort<-master.temp$cohort\n",
        "PC1<-master.temp$PC1\n",
        "PC2<-master.temp$PC2\n",
        "PC3<-master.temp$PC3\n",
        "PC4<-master.temp$PC4\n",
        "PC5<-master.temp$PC5\n",
        "PC6<-master.temp$PC6\n",
        "PC7<-master.temp$PC7\n",
        "PC8<-master.temp$PC8\n",
        "PC9<-master.temp$PC9\n",
        "PC10<-master.temp$PC10\n",
        "\n",
        "#retreive phen data\n",
        "y<-master.temp[,phen.col]\n",
        "age<-master.temp[,age.col]\n",
        "\n",
        "\n",
        "#retreive gen data\n",
        "x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "x.dn<-master.temp[,\"dnMIS\"]\n",
        "x.inh<-master.temp[,\"inhLoF\"]\n",
        "x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "\n",
        "result.dnlof<-lr.1(y,x.dnlof,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.dn<-lr.1(y,x.dn,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.inh<-lr.1(y,x.inh,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.asd<-lr.1(y,x.ps.asd,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.sz<-lr.1(y,x.ps.sz,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.ea<-lr.1(y,x.ps.ea,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "\n",
        "comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "group<-rep(\"combined\",length(variant.type))\n",
        "comb.result<-cbind(group,variant.type,comb.result)\n",
        "\n",
        "#################Test males\n",
        "\n",
        "PC1<-master.temp$PC1[male]\n",
        "PC2<-master.temp$PC2[male]\n",
        "PC3<-master.temp$PC3[male]\n",
        "PC4<-master.temp$PC4[male]\n",
        "PC5<-master.temp$PC5[male]\n",
        "PC6<-master.temp$PC6[male]\n",
        "PC7<-master.temp$PC7[male]\n",
        "PC8<-master.temp$PC8[male]\n",
        "PC9<-master.temp$PC9[male]\n",
        "PC10<-master.temp$PC10[male]\n",
        "\n",
        "#retreive phen data\n",
        "y<-master.temp[male,phen.col]\n",
        "age<-master.temp[male,age.col]\n",
        "\n",
        "\n",
        "#retreive gen data\n",
        "x.dnlof<-master.temp[male,\"dnLoF\"]\n",
        "x.dn<-master.temp[male,\"dnMIS\"]\n",
        "x.inh<-master.temp[male,\"inhLoF\"]\n",
        "x.ps.asd<-master.temp[male,\"PS.ASD.Z\"]\n",
        "x.ps.sz<-master.temp[male,\"PS.SZ.Z\"]\n",
        "x.ps.ea<-master.temp[male,\"PS.EA.Z\"]\n",
        "\n",
        "\n",
        "result.dnlof<-lr.2(y,x.dnlof,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.dn<-lr.2(y,x.dn,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.inh<-lr.2(y,x.inh,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.asd<-lr.2(y,x.ps.asd,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.sz<-lr.2(y,x.ps.sz,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.ea<-lr.2(y,x.ps.ea,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "\n",
        "male.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "group<-rep(\"male\",length(variant.type))\n",
        "male.result<-cbind(group,variant.type,male.result)\n",
        "\n",
        "#####Test females ######################\n",
        "\n",
        "PC1<-master.temp$PC1[female]\n",
        "PC2<-master.temp$PC2[female]\n",
        "PC3<-master.temp$PC3[female]\n",
        "PC4<-master.temp$PC4[female]\n",
        "PC5<-master.temp$PC5[female]\n",
        "PC6<-master.temp$PC6[female]\n",
        "PC7<-master.temp$PC7[female]\n",
        "PC8<-master.temp$PC8[female]\n",
        "PC9<-master.temp$PC9[female]\n",
        "PC10<-master.temp$PC10[female]\n",
        "\n",
        "#retreive phen data\n",
        "y<-master.temp[female,phen.col]\n",
        "age<-master.temp[female,age.col]\n",
        "\n",
        "\n",
        "#retreive gen data\n",
        "x.dnlof<-master.temp[female,\"dnLoF\"]\n",
        "x.dn<-master.temp[female,\"dnMIS\"]\n",
        "x.inh<-master.temp[female,\"inhLoF\"]\n",
        "x.ps.asd<-master.temp[female,\"PS.ASD.Z\"]\n",
        "x.ps.sz<-master.temp[female,\"PS.SZ.Z\"]\n",
        "x.ps.ea<-master.temp[female,\"PS.EA.Z\"]\n",
        "\n",
        "\n",
        "result.dnlof<-lr.2(y,x.dnlof,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.dn<-lr.2(y,x.dn,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.inh<-lr.2(y,x.inh,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.asd<-lr.2(y,x.ps.asd,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.sz<-lr.2(y,x.ps.sz,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "result.ps.ea<-lr.2(y,x.ps.ea,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "\n",
        "female.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "group<-rep(\"female\",length(variant.type))\n",
        "female.result<-cbind(group,variant.type,female.result)\n",
        "\n",
        "final.result<-rbind(female.result,male.result,comb.result)\n",
        "final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 1 that also tests sex interaction\n",
        "lr.1<-function(y,x,sex,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "\n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[15,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "###############################################################################33\n",
        "#Then we have a function that doesn't test the sex interaction\n",
        "##############################################################################\n",
        "\n",
        "gen.phen.2<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "  #retrieve covariates\n",
        "  PC1<-master.temp$PC1\n",
        "  PC2<-master.temp$PC2\n",
        "  PC3<-master.temp$PC3\n",
        "  PC4<-master.temp$PC4\n",
        "  PC5<-master.temp$PC5\n",
        "  PC6<-master.temp$PC6\n",
        "  PC7<-master.temp$PC7\n",
        "  PC8<-master.temp$PC8\n",
        "  PC9<-master.temp$PC9\n",
        "  PC10<-master.temp$PC10\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[,phen.col]\n",
        "  age<-master.temp[,age.col]\n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  x.dn<-master.temp[,\"dnMIS\"]\n",
        "  x.inh<-master.temp[,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "\n",
        "  \n",
        "  result.dnlof<-lr.2(y,x.dnlof,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2(y,x.dn,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2(y,x.inh,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2(y,x.ps.asd,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2(y,x.ps.sz,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2(y,x.ps.ea,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  final.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  final.result<-cbind(variant.type,final.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "\n",
        "#linear regression script 2 for testing each sex individually\n",
        "lr.2<-function(y,x,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~age+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "##################################################################\n",
        "# Then we have a regression for testing the effects of parents\n",
        "#################################################################\n",
        "\n",
        "#Ftests phenotypes in parents and tests sex interaction\n",
        "gen.phen.3<-function(master.temp,phen.col.f,phen.col.m)\n",
        "{\n",
        "  #retrieve father data\n",
        "  ###########################\n",
        "  sexfather<-rep(1,nrow(master.temp))\n",
        "  PC1f<-master.temp$PC1_PAT\n",
        "  PC2f<-master.temp$PC2_PAT\n",
        "  PC3f<-master.temp$PC3_PAT\n",
        "  PC4f<-master.temp$PC4_PAT\n",
        "  PC5f<-master.temp$PC5_PAT\n",
        "  PC6f<-master.temp$PC6_PAT\n",
        "  PC7f<-master.temp$PC7_PAT\n",
        "  PC8f<-master.temp$PC8_PAT\n",
        "  PC9f<-master.temp$PC9_PAT\n",
        "  PC10f<-master.temp$PC10_PAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  yf<-master.temp[,phen.col.f]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  f.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  f.dn<-master.temp[,\"dnMIS\"]\n",
        "  f.inh<-master.temp$InhLoF.Paternal.Transmitted + master.temp$InhLoF.Paternal.Untransmitted\n",
        "  f.ps.asd<-master.temp$Paternal.PS.ASD.Z\n",
        "  f.ps.sz<-master.temp$Paternal.PS.SZ.Z\n",
        "  f.ps.ea<-master.temp$Paternal.PS.EA.Z\n",
        "  \n",
        "  #retrieve mother data\n",
        "  ###########################\n",
        "  \n",
        "  sexmother<-rep(0,nrow(master.temp))\n",
        "  PC1m<-master.temp$PC1_MAT\n",
        "  PC2m<-master.temp$PC2_MAT\n",
        "  PC3m<-master.temp$PC3_MAT\n",
        "  PC4m<-master.temp$PC4_MAT\n",
        "  PC5m<-master.temp$PC5_MAT\n",
        "  PC6m<-master.temp$PC6_MAT\n",
        "  PC7m<-master.temp$PC7_MAT\n",
        "  PC8m<-master.temp$PC8_MAT\n",
        "  PC9m<-master.temp$PC9_MAT\n",
        "  PC10m<-master.temp$PC10_MAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  ym<-master.temp[,phen.col.m]\n",
        "  \n",
        "  #retreive gen data\n",
        "  m.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  m.dn<-master.temp[,\"dnMIS\"]\n",
        "  m.inh<-master.temp$InhLoF.Maternal.Transmitted + master.temp$InhLoF.Maternal.Untransmitted\n",
        "  m.ps.asd<-master.temp$Maternal.PS.ASD.Z\n",
        "  m.ps.sz<-master.temp$Maternal.PS.SZ.Z\n",
        "  m.ps.ea<-master.temp$Maternal.PS.EA.Z\n",
        "  \n",
        "  #merge father and mother variables\n",
        "  sex<-c(sexfather,sexmother)\n",
        "  PC1<-c(PC1f,PC1m)\n",
        "  PC2<-c(PC2f,PC2m)\n",
        "  PC3<-c(PC3f,PC3m)\n",
        "  PC4<-c(PC4f,PC4m)\n",
        "  PC5<-c(PC5f,PC5m)\n",
        "  PC6<-c(PC6f,PC6m)\n",
        "  PC7<-c(PC7f,PC7m)\n",
        "  PC8<-c(PC8f,PC8m)\n",
        "  PC9<-c(PC9f,PC9m)\n",
        "  PC10<-c(PC10f,PC10m)\n",
        "  x.dnlof<-c(f.dnlof,m.dnlof)\n",
        "  x.dn<-c(f.dn,m.dn)\n",
        "  x.inh<-c(f.inh,m.inh)\n",
        "  x.ps.asd<-c(f.ps.asd,m.ps.asd)\n",
        "  x.ps.sz<-c(f.ps.sz,m.ps.sz)\n",
        "  x.ps.ea<-c(f.ps.ea,m.ps.ea)\n",
        "  y<-c(yf,ym)\n",
        "  \n",
        "  result.dnlof<-lr.3(y,x.dnlof,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.3(y,x.dn,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.3(y, x.inh, sex, PC1, PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.3(y,x.ps.asd,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.3(y,x.ps.sz,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.3(y,x.ps.ea,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  #combined result\n",
        "  comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"parents\",length=length(variant.type))\n",
        "  comb.result<-cbind(group,variant.type,comb.result)\n",
        "  \n",
        "  \n",
        "  #father result\n",
        "  result.dnlof<-lr.4(yf,f.dnlof,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.dn<-lr.4(yf,f.dn,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.inh<-lr.4(yf, f.inh, PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.asd<-lr.4(yf,f.ps.asd,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.sz<-lr.4(yf,f.ps.sz,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.ea<-lr.4(yf,f.ps.ea,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  \n",
        "  f.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"father\",length=length(variant.type))\n",
        "  f.result<-cbind(group,variant.type,f.result)\n",
        "  \n",
        "  #mother result\n",
        "  result.dnlof<-lr.4(ym,m.dnlof,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.dn<-lr.4(ym,m.dn,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.inh<-lr.4(ym, m.inh, PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.asd<-lr.4(ym,m.ps.asd,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.sz<-lr.4(ym,m.ps.sz,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.ea<-lr.4(ym,m.ps.ea,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  \n",
        "  m.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"mother\",length=length(variant.type))\n",
        "  m.result<-cbind(group,variant.type,m.result)\n",
        "  \n",
        "  #output final result\n",
        "  final.result<-rbind(m.result,f.result,comb.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 3 tests phenotypes in parents and tests sex interaction\n",
        "lr.3<-function(y,x,sex,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[14,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "#linear regression script 4 tests phenotypes in mother or father without sex variable\n",
        "lr.4<-function(y,x,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cc<-cor(x,y)\n",
        "\n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "#############################     2nd set of scripts with cohort included as a covaraite    ##################\n",
        "#in addition to age, sex and PCs\n",
        "#and tests for a gene-by-sex interactions\n",
        "#the functions for these are as follows\n",
        "#gen.phen.1c a function that runs a linear regression and tests for gene-by-sex interaction. These are the primary results described in Fig. 5\n",
        "#gen.phen.2c To characterize gene-by sex effects that are significant, gene.phen.2 runs the regression on males and females separately without sex as a covariate\n",
        "#gene.phen.3c runs linear regression and gene-by-sex interaction on PARENTS, then to investigate gene by sex interactions, runs regression on mothers and fathers individually\n",
        "\n",
        "\n",
        "#gen.phen.1c a function that runs a linear regression on both sexes and outputs the result\n",
        "\n",
        "gen.phen.1c<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "  #define the sexes\n",
        "  male<-master.temp$sex==\"Male\"\n",
        "  female<-master.temp$sex==\"Female\"\n",
        "  \n",
        "  #Test combined sample\n",
        "  sex<-master.temp$sex\n",
        "  cohort<-master.temp$cohort\n",
        "  PC1<-master.temp$PC1\n",
        "  PC2<-master.temp$PC2\n",
        "  PC3<-master.temp$PC3\n",
        "  PC4<-master.temp$PC4\n",
        "  PC5<-master.temp$PC5\n",
        "  PC6<-master.temp$PC6\n",
        "  PC7<-master.temp$PC7\n",
        "  PC8<-master.temp$PC8\n",
        "  PC9<-master.temp$PC9\n",
        "  PC10<-master.temp$PC10\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[,phen.col]\n",
        "  age<-master.temp[,age.col]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  x.dn<-master.temp[,\"dnMIS\"]\n",
        "  x.inh<-master.temp[,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "  \n",
        "  result.dnlof<-lr.1c(y,x.dnlof,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.1c(y,x.dn,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.1c(y,x.inh,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.1c(y,x.ps.asd,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.1c(y,x.ps.sz,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.1c(y,x.ps.ea,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"combined\",length(variant.type))\n",
        "  comb.result<-cbind(group,variant.type,comb.result)\n",
        "  \n",
        "  #################Test males\n",
        "  \n",
        "  cohort<-master.temp$cohort[male]\n",
        "  PC1<-master.temp$PC1[male]\n",
        "  PC2<-master.temp$PC2[male]\n",
        "  PC3<-master.temp$PC3[male]\n",
        "  PC4<-master.temp$PC4[male]\n",
        "  PC5<-master.temp$PC5[male]\n",
        "  PC6<-master.temp$PC6[male]\n",
        "  PC7<-master.temp$PC7[male]\n",
        "  PC8<-master.temp$PC8[male]\n",
        "  PC9<-master.temp$PC9[male]\n",
        "  PC10<-master.temp$PC10[male]\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[male,phen.col]\n",
        "  age<-master.temp[male,age.col]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[male,\"dnLoF\"]\n",
        "  x.dn<-master.temp[male,\"dnMIS\"]\n",
        "  x.inh<-master.temp[male,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[male,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[male,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[male,\"PS.EA.Z\"]\n",
        "  \n",
        "  \n",
        "  result.dnlof<-lr.2c(y,x.dnlof,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2c(y,x.dn,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2c(y,x.inh,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2c(y,x.ps.asd,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2c(y,x.ps.sz,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2c(y,x.ps.ea,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  male.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"male\",length(variant.type))\n",
        "  male.result<-cbind(group,variant.type,male.result)\n",
        "  \n",
        "  #####Test females ######################\n",
        "  cohort<-master.temp$cohort[female]\n",
        "  PC1<-master.temp$PC1[female]\n",
        "  PC2<-master.temp$PC2[female]\n",
        "  PC3<-master.temp$PC3[female]\n",
        "  PC4<-master.temp$PC4[female]\n",
        "  PC5<-master.temp$PC5[female]\n",
        "  PC6<-master.temp$PC6[female]\n",
        "  PC7<-master.temp$PC7[female]\n",
        "  PC8<-master.temp$PC8[female]\n",
        "  PC9<-master.temp$PC9[female]\n",
        "  PC10<-master.temp$PC10[female]\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[female,phen.col]\n",
        "  age<-master.temp[female,age.col]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[female,\"dnLoF\"]\n",
        "  x.dn<-master.temp[female,\"dnMIS\"]\n",
        "  x.inh<-master.temp[female,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[female,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[female,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[female,\"PS.EA.Z\"]\n",
        "  \n",
        "  \n",
        "  result.dnlof<-lr.2c(y,x.dnlof,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2c(y,x.dn,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2c(y,x.inh,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2c(y,x.ps.asd,cohort,age,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2c(y,x.ps.sz,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2c(y,x.ps.ea,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  female.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"female\",length(variant.type))\n",
        "  female.result<-cbind(group,variant.type,female.result)\n",
        "  \n",
        "  final.result<-rbind(female.result,male.result,comb.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 1 that also tests sex interaction\n",
        "lr.1c<-function(y,x,sex,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[15,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "###############################################################################33\n",
        "#Then we have a function that doesn't test the sex interaction\n",
        "##############################################################################\n",
        "\n",
        "gen.phen.2c<-function(master.temp,phen.col,age.col)\n",
        "{\n",
        "  #retrieve covariates\n",
        "  PC1<-master.temp$PC1\n",
        "  PC2<-master.temp$PC2\n",
        "  PC3<-master.temp$PC3\n",
        "  PC4<-master.temp$PC4\n",
        "  PC5<-master.temp$PC5\n",
        "  PC6<-master.temp$PC6\n",
        "  PC7<-master.temp$PC7\n",
        "  PC8<-master.temp$PC8\n",
        "  PC9<-master.temp$PC9\n",
        "  PC10<-master.temp$PC10\n",
        "  \n",
        "  #retreive phen data\n",
        "  y<-master.temp[,phen.col]\n",
        "  age<-master.temp[,age.col]\n",
        "  \n",
        "  #retreive gen data\n",
        "  x.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  x.dn<-master.temp[,\"dnMIS\"]\n",
        "  x.inh<-master.temp[,\"inhLoF\"]\n",
        "  x.ps.asd<-master.temp[,\"PS.ASD.Z\"]\n",
        "  x.ps.sz<-master.temp[,\"PS.SZ.Z\"]\n",
        "  x.ps.ea<-master.temp[,\"PS.EA.Z\"]\n",
        "  \n",
        "  \n",
        "  result.dnlof<-lr.2c(y,x.dnlof,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.2c(y,x.dn,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.2c(y,x.inh,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.2c(y,x.ps.asd,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.2c(y,x.ps.sz,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.2c(y,x.ps.ea,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  final.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  final.result<-cbind(variant.type,final.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "\n",
        "#linear regression script 2 for testing each sex individually\n",
        "lr.2c<-function(y,x,age,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~age+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "##################################################################\n",
        "# Then we have a regression for testing the effects of parents\n",
        "#################################################################\n",
        "\n",
        "#Ftests phenotypes in parents and tests sex interaction\n",
        "gen.phen.3c<-function(master.temp,phen.col.f,phen.col.m)\n",
        "{\n",
        "  #retrieve father data\n",
        "  ###########################\n",
        "  sexfather<-rep(1,nrow(master.temp))\n",
        "  cohort<-master.temp$cohort\n",
        "  PC1f<-master.temp$PC1_PAT\n",
        "  PC2f<-master.temp$PC2_PAT\n",
        "  PC3f<-master.temp$PC3_PAT\n",
        "  PC4f<-master.temp$PC4_PAT\n",
        "  PC5f<-master.temp$PC5_PAT\n",
        "  PC6f<-master.temp$PC6_PAT\n",
        "  PC7f<-master.temp$PC7_PAT\n",
        "  PC8f<-master.temp$PC8_PAT\n",
        "  PC9f<-master.temp$PC9_PAT\n",
        "  PC10f<-master.temp$PC10_PAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  yf<-master.temp[,phen.col.f]\n",
        "  \n",
        "  \n",
        "  #retreive gen data\n",
        "  f.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  f.dn<-master.temp[,\"dnMIS\"]\n",
        "  f.inh<-master.temp$InhLoF.Paternal.Transmitted + master.temp$InhLoF.Paternal.Untransmitted\n",
        "  f.ps.asd<-master.temp$Paternal.PS.ASD.Z\n",
        "  f.ps.sz<-master.temp$Paternal.PS.SZ.Z\n",
        "  f.ps.ea<-master.temp$Paternal.PS.EA.Z\n",
        "  \n",
        "  #retrieve mother data\n",
        "  ###########################\n",
        "  \n",
        "  sexmother<-rep(0,nrow(master.temp))\n",
        "  PC1m<-master.temp$PC1_MAT\n",
        "  PC2m<-master.temp$PC2_MAT\n",
        "  PC3m<-master.temp$PC3_MAT\n",
        "  PC4m<-master.temp$PC4_MAT\n",
        "  PC5m<-master.temp$PC5_MAT\n",
        "  PC6m<-master.temp$PC6_MAT\n",
        "  PC7m<-master.temp$PC7_MAT\n",
        "  PC8m<-master.temp$PC8_MAT\n",
        "  PC9m<-master.temp$PC9_MAT\n",
        "  PC10m<-master.temp$PC10_MAT\n",
        "  \n",
        "  #retreive phen data\n",
        "  ym<-master.temp[,phen.col.m]\n",
        "  \n",
        "  #retreive gen data\n",
        "  m.dnlof<-master.temp[,\"dnLoF\"]\n",
        "  m.dn<-master.temp[,\"dnMIS\"]\n",
        "  m.inh<-master.temp$InhLoF.Maternal.Transmitted + master.temp$InhLoF.Maternal.Untransmitted\n",
        "  m.ps.asd<-master.temp$Maternal.PS.ASD.Z\n",
        "  m.ps.sz<-master.temp$Maternal.PS.SZ.Z\n",
        "  m.ps.ea<-master.temp$Maternal.PS.EA.Z\n",
        "  \n",
        "  #merge father and mother variables\n",
        "  sex<-c(sexfather,sexmother)\n",
        "  comb.cohort<-c(cohort,cohort)\n",
        "  PC1<-c(PC1f,PC1m)\n",
        "  PC2<-c(PC2f,PC2m)\n",
        "  PC3<-c(PC3f,PC3m)\n",
        "  PC4<-c(PC4f,PC4m)\n",
        "  PC5<-c(PC5f,PC5m)\n",
        "  PC6<-c(PC6f,PC6m)\n",
        "  PC7<-c(PC7f,PC7m)\n",
        "  PC8<-c(PC8f,PC8m)\n",
        "  PC9<-c(PC9f,PC9m)\n",
        "  PC10<-c(PC10f,PC10m)\n",
        "  x.dnlof<-c(f.dnlof,m.dnlof)\n",
        "  x.dn<-c(f.dn,m.dn)\n",
        "  x.inh<-c(f.inh,m.inh)\n",
        "  x.ps.asd<-c(f.ps.asd,m.ps.asd)\n",
        "  x.ps.sz<-c(f.ps.sz,m.ps.sz)\n",
        "  x.ps.ea<-c(f.ps.ea,m.ps.ea)\n",
        "  y<-c(yf,ym)\n",
        "  \n",
        "  result.dnlof<-lr.3c(y,x.dnlof,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.dn<-lr.3c(y,x.dn,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.inh<-lr.3c(y, x.inh, sex,comb.cohort,PC1, PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.asd<-lr.3c(y,x.ps.asd,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.sz<-lr.3c(y,x.ps.sz,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  result.ps.ea<-lr.3c(y,x.ps.ea,sex,comb.cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "  \n",
        "  #combined result\n",
        "  comb.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"parents\",length=length(variant.type))\n",
        "  comb.result<-cbind(group,variant.type,comb.result)\n",
        "  \n",
        "  \n",
        "  #father result\n",
        "  result.dnlof<-lr.4c(yf,f.dnlof,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.dn<-lr.4c(yf,f.dn,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.inh<-lr.4c(yf, f.inh,cohort, PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.asd<-lr.4c(yf,f.ps.asd,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.sz<-lr.4c(yf,f.ps.sz,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  result.ps.ea<-lr.4c(yf,f.ps.ea,cohort,PC1f,PC2f,PC3f,PC4f,PC5f,PC6f,PC7f,PC8f,PC9f,PC10f)\n",
        "  \n",
        "  f.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"father\",length=length(variant.type))\n",
        "  f.result<-cbind(group,variant.type,f.result)\n",
        "  \n",
        "  #mother result\n",
        "  result.dnlof<-lr.4c(ym,m.dnlof,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.dn<-lr.4c(ym,m.dn,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.inh<-lr.4c(ym, m.inh,cohort, PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.asd<-lr.4c(ym,m.ps.asd,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.sz<-lr.4c(ym,m.ps.sz,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  result.ps.ea<-lr.4c(ym,m.ps.ea,cohort,PC1m,PC2m,PC3m,PC4m,PC5m,PC6m,PC7m,PC8m,PC9m,PC10m)\n",
        "  \n",
        "  m.result<-rbind(result.dnlof,result.dn,result.inh,result.ps.asd,result.ps.sz,result.ps.ea)\n",
        "  variant.type<-c(\"dnlof\",\"dnmis\",\"inhLoF\", \"ps.asd\",\"ps.sz\",\"ps.ea\")\n",
        "  group<-rep(\"mother\",length=length(variant.type))\n",
        "  m.result<-cbind(group,variant.type,m.result)\n",
        "  \n",
        "  #output final result\n",
        "  final.result<-rbind(m.result,f.result,comb.result)\n",
        "  final.result\n",
        "}\n",
        "\n",
        "\n",
        "#linear regression script 3 tests phenotypes in parents and tests sex interaction\n",
        "lr.3c<-function(y,x,sex,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+sex+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~sex+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cor<-cor.test(x,y)\n",
        "  pearson.cc<-pearson.cor$estimate\n",
        "  \n",
        "  #test sex interaction\n",
        "  int.model<-lm(y~x*sex+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  sex.pval<-signif(summary(int.model)$coefficients[15,4],digits=2)\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "#linear regression script 4 tests phenotypes in mother or father without sex variable\n",
        "lr.4c<-function(y,x,cohort,PC1,PC2,PC3,PC4,PC5,PC6,PC7,PC8,PC9,PC10)\n",
        "{\n",
        "  model<-lm(y~x+cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  n.model<-lm(y~cohort+PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10)\n",
        "  est<-signif(summary(model)$coefficients[2,1],digits=2)\n",
        "  est.stderr<-signif(summary(model)$coefficients[2,2],digits=2)\n",
        "  x.pval<-signif(summary(model)$coefficients[2,4],digits=2)\n",
        "  r.sq<-signif(summary(model)$r.squared-summary(n.model)$r.squared,digits=2)\n",
        "  pearson.cc<-cor(x,y)\n",
        "  \n",
        "  \n",
        "  #test sex interaction\n",
        "  sex.pval<-NA\n",
        "  \n",
        "  result<-cbind(r.sq,est,est.stderr,pearson.cc,x.pval,sex.pval)\n",
        "  result\n",
        "}\n",
        "\n",
        "################################################################################################################\n",
        "#################   Then the analysis of each gene-phenotype correlation is run as follows    ##################\n",
        "###############################################################################################################\n",
        "\n",
        "\n",
        "#identify the subset of case, controls, and european ancesty\n",
        "cont<-master.gen.phen.table$phen==\"CON\"\n",
        "case<-master.gen.phen.table$phen==\"ASD\"\n",
        "eur<-master.gen.phen.table$is_eur==1\n",
        "\n",
        "\n",
        "#The analysis will be restricted to subjects of confirmed european ancestry\n",
        "# master.temp.case.eur<-master.gen.phen.table[case & eur,]\n",
        "# master.temp.cont.eur<-master.gen.phen.table[cont & eur,]\n",
        "master.temp.case.eur<-master.gen.phen.table[case ,]\n",
        "master.temp.cont.eur<-master.gen.phen.table[cont ,]\n",
        "\n",
        "\n",
        "####################################################\n",
        "# Test each phenotype in offspring\n",
        "#####################################################\n",
        "\n",
        "\n",
        "#VABS\n",
        "\n",
        "phen.col<-\"vabs.child.z\"\n",
        "age.col<-\"vabs.child.age\"\n",
        "\n",
        "vine.case.eur<-gen.phen.1(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "vine.cont.eur<-gen.phen.1(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "# vine.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "# vine.cont.eur<-gen.phen.1c(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"VABS\",nrow(vine.case.eur))\n",
        "vine.case.eur<-cbind(phenotype,vine.case.eur)\n",
        "\n",
        "phenotype<-rep(\"VABS\",nrow(vine.cont.eur))\n",
        "vine.cont.eur<-cbind(phenotype,vine.cont.eur)\n",
        "\n",
        "#SRS\n",
        "phen.col<-\"srs.child.z\"\n",
        "age.col<-\"srs.child.age\"\n",
        "\n",
        "srs.case.eur<-gen.phen.1(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "srs.cont.eur<-gen.phen.1(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "# srs.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "# srs.cont.eur<-gen.phen.1c(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"SRS\",nrow(srs.case.eur))\n",
        "srs.case.eur<-cbind(phenotype,srs.case.eur)\n",
        "\n",
        "phenotype<-rep(\"SRS\",nrow(srs.cont.eur))\n",
        "srs.cont.eur<-cbind(phenotype,srs.cont.eur)\n",
        "\n",
        "\n",
        "#SCQ\n",
        "phen.col<-\"scq.child.z\"\n",
        "age.col<-\"scq.child.age\"\n",
        "\n",
        "scq.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "scq.cont.eur<-gen.phen.1(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "# scq.cont.eur<-gen.phen.1c(master.temp = master.temp.cont.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"SCQ\",nrow(scq.case.eur))\n",
        "scq.case.eur<-cbind(phenotype,scq.case.eur)\n",
        "phenotype<-rep(\"SCQ\",nrow(scq.cont.eur))\n",
        "scq.cont.eur<-cbind(phenotype,scq.cont.eur)\n",
        "\n",
        "\n",
        "#DCDQ\n",
        "phen.col<-\"dcdq.child.z\"\n",
        "age.col<-\"dcdq.child.age\"\n",
        "\n",
        "dcdq.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "phenotype<-rep(\"DCDQ\",nrow(dcdq.case.eur))\n",
        "dcdq.case.eur<-cbind(phenotype,dcdq.case.eur)\n",
        "\n",
        "\n",
        "#rbs\n",
        "phen.col<-\"rbs.child.z\"\n",
        "age.col<-\"scq.child.age\"\n",
        "# age.col<-\"rbs.child.age\"\n",
        "\n",
        "\n",
        "rbs.case.eur<-gen.phen.1c(master.temp = master.temp.case.eur, phen.col=phen.col, age.col=age.col)\n",
        "\n",
        "phenotype<-rep(\"RBS\",nrow(rbs.case.eur))\n",
        "rbs.case.eur<-cbind(phenotype,rbs.case.eur)\n",
        "\n",
        "\n",
        "\n",
        "############################################\n",
        "# Testing Behavioral Phenotypes in Parents\n",
        "############################################\n",
        "\n",
        "cont<-master.gen.phen.table$phen==\"CON\"\n",
        "case<-master.gen.phen.table$phen==\"ASD\"\n",
        "eur<-master.gen.phen.table$is_eur==1\n",
        "\n",
        "\n",
        "#Remove the sibling control trios (so the same parents are not tested twice)\n",
        "# master.temp.case.eur<-master.gen.phen.table[case & eur,]\n",
        "master.temp.case.eur<-master.gen.phen.table[case,]\n",
        "\n",
        "\n",
        "#also remove any younger ASD siblings of cases so we have each set of parents represented once\n",
        "#sort the trios by mothers age in ascending order so that the oldest case is on top\n",
        "sort.order<-order(master.temp.case.eur$mother.age.at.birth.z,decreasing=FALSE)\n",
        "master.temp.case.unique<-master.temp.case.eur[sort.order,]\n",
        "#and unique the family ID\n",
        "master.temp.case.unique<-master.temp.case.unique[!duplicated(master.temp.case.unique$fid),]\n",
        "\n",
        "##############BAPQ\n",
        "\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.bapq.z\"\n",
        "phen.col.m<-\"mother.bapq.z\"\n",
        "\n",
        "bapq.parents.eur<-gen.phen.3(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "phenotype<-rep(\"BAPQ\",nrow(bapq.parents.eur))\n",
        "bapq.parents.eur<-cbind(phenotype,bapq.parents.eur)\n",
        "\n",
        "\n",
        "###################SRS\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.srs.z\"\n",
        "phen.col.m<-\"mother.srs.z\"\n",
        "\n",
        "srs.parents.eur<-gen.phen.3(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "phenotype<-rep(\"SRS\",nrow(srs.parents.eur))\n",
        "srs.parents.eur<-cbind(phenotype,srs.parents.eur)\n",
        "\n",
        "\n",
        "###################Parental Age\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.age.at.birth.z\"\n",
        "phen.col.m<-\"mother.age.at.birth.z\"\n",
        "\n",
        "parental.age.eur<-gen.phen.3c(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "\n",
        "phenotype<-rep(\"Parental Age\",nrow(parental.age.eur))\n",
        "parental.age.eur<-cbind(phenotype,parental.age.eur)\n",
        "\n",
        "###################EA\n",
        "#define the phenotypes\n",
        "phen.col.f<-\"father.EA.z\"\n",
        "phen.col.m<-\"mother.EA.z\"\n",
        "\n",
        "parental.ea.eur<-gen.phen.3c(master.temp.case.unique,phen.col.f,phen.col.m)\n",
        "\n",
        "phenotype<-rep(\"Parental EA\",nrow(parental.ea.eur))\n",
        "parental.ea.eur<-cbind(phenotype,parental.ea.eur)\n",
        "\n",
        "\n",
        "################################################################\n",
        "##### Combine all of the results and write them out to csv file\n",
        "###############################################################\n",
        "\n",
        "case.results.eur<-rbind(srs.case.eur,scq.case.eur,vine.case.eur,dcdq.case.eur,rbs.case.eur)\n",
        "case.results.eur<-cbind(case.status=rep(\"case\",nrow(case.results.eur)),case.results.eur)\n",
        "control.results.eur<-rbind(srs.cont.eur, scq.cont.eur,vine.cont.eur)\n",
        "control.results.eur<-cbind(case.status=rep(\"cont\",nrow(control.results.eur)),control.results.eur)\n",
        "parent.results.eur<-rbind(bapq.parents.eur,srs.parents.eur,parental.age.eur,parental.ea.eur)\n",
        "parent.results.eur<-cbind(case.status=rep(\"parent\",nrow(parent.results.eur)),parent.results.eur)\n",
        "\n",
        "clin.results.eur<-rbind(case.results.eur,control.results.eur,parent.results.eur)\n",
        "# write.csv(clin.results.eur,\"C:\\\\Users\\\\jsebat\\\\sebatlab Dropbox\\\\Autism\\\\Autism_Illumina_WGS\\\\RV_PRS_Analysis_Jun2020\\\\clin_results_eur.csv\")\n",
        "# write.csv(clin.results.eur,clin_results_out)\n",
        "write.csv(clin.results.eur, \"clin_results_eur_and_noneur.SBayesR.csv\")\n",
        "\n",
        "\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "yZDj3acpqCXO",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0d9cc0a6-4c22-400f-a90a-35a1491041f8"
      },
      "source": [
        "# Installing packages for Figure 5 display\n",
        "install.packages(\"tidyverse\")\n",
        "install.packages(\"dplyr\")\n",
        "install.packages(\"RColorBrewer\")\n",
        "install.packages(\"ggplot2\")\n",
        "install.packages(\"scales\")\n",
        "install.packages(\"ggpubr\", repos = \"https://ftp.osuosl.org/pub/cran/\", lib=\"/usr/lib/R/site-library\")\n",
        "\n",
        "library(tidyverse)\n",
        "library(dplyr)\n",
        "library(RColorBrewer)\n",
        "library(ggplot2)\n",
        "library(scales)\n",
        "library(ggpubr)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "also installing the dependencies ‘matrixStats’, ‘RcppArmadillo’, ‘numDeriv’, ‘SparseM’, ‘MatrixModels’, ‘conquer’, ‘sp’, ‘openxlsx’, ‘minqa’, ‘nloptr’, ‘RcppEigen’, ‘carData’, ‘abind’, ‘pbkrtest’, ‘quantreg’, ‘maptools’, ‘rio’, ‘lme4’, ‘corrplot’, ‘car’, ‘ggrepel’, ‘ggsci’, ‘cowplot’, ‘ggsignif’, ‘gridExtra’, ‘polynom’, ‘rstatix’\n",
            "\n",
            "\n",
            "Warning message in system(\"timedatectl\", intern = TRUE):\n",
            "“running command 'timedatectl' had status 1”\n",
            "── \u001b[1mAttaching packages\u001b[22m ─────────────────────────────────────── tidyverse 1.3.1 ──\n",
            "\n",
            "\u001b[32m✔\u001b[39m \u001b[34mggplot2\u001b[39m 3.3.5     \u001b[32m✔\u001b[39m \u001b[34mpurrr  \u001b[39m 0.3.4\n",
            "\u001b[32m✔\u001b[39m \u001b[34mtibble \u001b[39m 3.1.4     \u001b[32m✔\u001b[39m \u001b[34mdplyr  \u001b[39m 1.0.7\n",
            "\u001b[32m✔\u001b[39m \u001b[34mtidyr  \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mforcats\u001b[39m 0.5.1\n",
            "\u001b[32m✔\u001b[39m \u001b[34mreadr  \u001b[39m 2.0.1     \n",
            "\n",
            "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
            "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
            "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
            "\n",
            "\n",
            "Attaching package: ‘scales’\n",
            "\n",
            "\n",
            "The following object is masked from ‘package:purrr’:\n",
            "\n",
            "    discard\n",
            "\n",
            "\n",
            "The following object is masked from ‘package:readr’:\n",
            "\n",
            "    col_factor\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_IyfW2ZVuuST"
      },
      "source": [
        "# library(svglite)\n",
        "# remove.packages(\"svglite\")\n",
        "# install.packages(\"svglite\", type = \"source\")\n",
        "# library(svglite)\n",
        "# install.packages('svglite')\n",
        "# install.packages(\"svglite\", type = \"source\")\n",
        "# library(svglite)\n",
        "\n",
        "# remove.packages(\"svglite\")\n",
        "str(allPackage <- installed.packages())\n",
        "allPackage [, c(1,3:5)]\n",
        "install.packages(\"svglite\", type = \"source\")\n",
        "str(allPackage <- installed.packages())\n",
        "allPackage [, c(1,3:5)]\n"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 190
        },
        "id": "nlstCyCxutKP",
        "outputId": "e8c87086-da94-4902-f340-f6b9c09d68c4"
      },
      "source": [
        "asd_df <- read.csv(file = clin_results_out, header = TRUE, na.string = \"NA\")\n",
        "asd_df <- read.csv(file = \"clin_results_eur_and_noneur.SBayesR.csv\", header = TRUE, na.string = \"NA\")\n",
        "\n",
        "MASSIVE = 50 + 10 + 10\n",
        "BIG = 30 + 5\n",
        "\n",
        "# Rounding\n",
        "asd_df[195,10] = round(asd_df[195,10], digits = 2)\n",
        "asd_df[195,11] = round(asd_df[195,11], digits = 2)\n",
        "\n",
        "# Make p-value column (\"logP.2\" will used for scaling circles in figure)\n",
        "asd_df$x.pval_log = -log10(asd_df$x.pval)\n",
        "asd_df <- transform(asd_df, logP.2 = ifelse(x.pval_log >= 5, 5, x.pval_log))\n",
        "\n",
        "# Concatenate the phenotype and case.status variables\n",
        "asd_df$pheno = paste(asd_df$phenotype, \"_\", asd_df$case.status, sep=\"\")\n",
        "\n",
        "# Create a column of significance labels for significant genotype effects\n",
        "asd_df$stars <- cut(asd_df$x.pval, breaks=c(-Inf, 0.0007, 0.05, Inf), label = c(\"**\", \"*\", \"\"))  # Create column of significance labels for main effect\n",
        "\n",
        "# Filter data for panel A (cases & controls)\n",
        "asd_cascon <- asd_df %>%\n",
        "  filter(case.status %in% c(\"case\", \"cont\")) %>%\n",
        "  filter(group %in% c(\"combined\")) %>%\n",
        "  filter(variant.type %in% c( \"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "# Specify the factor levels in the order we want\n",
        "asd_cascon$pheno <- factor(asd_cascon$pheno, levels = c(\"RBS_case\", \"SCQ_case\", \"SCQ_cont\", \"SRS_case\", \"SRS_cont\", \"VABS_case\", \"VABS_cont\", \"DCDQ_case\"))\n",
        "# Reverse the order as follow\n",
        "asd_cascon$pheno <- factor(asd_cascon$pheno, levels = rev(levels(asd_cascon$pheno)))\n",
        "\n",
        "asd_cascon$variant.type <- factor(asd_cascon$variant.type, levels = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "### Panel A ###\n",
        "pdf(\"panelA.test.pdf\", width = 30, height = 30)\n",
        "panelA =\n",
        "ggballoonplot(asd_cascon, x = \"variant.type\", y = \"pheno\", size = \"logP.2\", size.range = c(5, 20), fill = \"est\", ggtheme = theme_bw()) +\n",
        "scale_fill_gradientn(colors = c(\"blue3\", \"blue\", \"white\", \"indianred1\", \"red3\"), values = rescale(c(-0.43, -0.30, 0, 0.10, 0.28)), limits = c(-0.43, 0.28), guide = guide_colorbar(barwidth = 6, barheight = 30, title.position = \"top\", label.theme = element_text(size = MASSIVE))) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0.8, size = BIG - 5) +\n",
        "scale_size(range = c(15*1.53, 60*1.53), breaks =c(1, 2, 3, 4, 5)) +\n",
        "scale_y_discrete(breaks = c(\"RBS_case\", \"SCQ_case\", \"SCQ_cont\", \"SRS_case\", \"SRS_cont\", \"VABS_case\", \"VABS_cont\", \"DCDQ_case\"), labels = c(\"RBS case\", \"SCQ case\", \"SCQ cont\", \"SRS case\", \"SRS cont\", \"VABS case\", \"VABS cont\", \"DCDQ case\")) +\n",
        "scale_x_discrete(breaks = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"), labels=c(\"dnMIS\", \"dnLoF\", \"inhLoF\", expression(PS[ASD]), expression(PS[SCZ]), expression(PS[EA]))) +\n",
        "labs(size = \"-log10(P-value)\") +\n",
        "labs(fill = \"Effect Size\") +\n",
        "theme(panel.grid.major = element_blank()) +\n",
        "theme(panel.grid.minor = element_blank()) +\n",
        "# Text\n",
        "theme(axis.text.x = element_text(color = \"black\", size = MASSIVE)) +\n",
        "theme(axis.text.y = element_text(color = \"black\", size = MASSIVE)) +\n",
        "# Legend\n",
        "theme(legend.title = element_text(size = MASSIVE)) +\n",
        "theme(legend.text = element_text(color = \"black\", size = MASSIVE))\n",
        "panelA\n",
        "dev.off();\n",
        "\n",
        "\n",
        "# Filter data for panel B (parents)\n",
        "asd_parents <- asd_df %>%\n",
        "  filter(group %in% c(\"parents\")) %>%\n",
        "  filter(variant.type %in% c( \"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "# Specify the factor levels in the order you want\n",
        "asd_parents$pheno <- factor(asd_parents$pheno, levels = c(\"SRS_parent\", \"BAPQ_parent\", \"Parental Age_parent\", \"Parental EA_parent\"))\n",
        "# Reverse the order\n",
        "asd_parents$pheno <- factor(asd_parents$pheno, levels = rev(levels(asd_parents$pheno)))\n",
        "\n",
        "asd_parents$variant.type <- factor(asd_parents$variant.type, levels = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"))\n",
        "\n",
        "### Panel B ###\n",
        "pdf(\"panelB.test.pdf\", width = 30, height = 30)\n",
        "panelB =\n",
        "ggballoonplot(asd_parents, x = \"variant.type\", y = \"pheno\", size = \"logP.2\", size.range = c(5, 20), fill = \"est\", ggtheme = theme_bw()) +\n",
        "scale_fill_gradientn(colors = c(\"blue3\", \"blue\", \"white\", \"indianred1\", \"red3\"), values = rescale(c(-0.43, -0.30, 0, 0.10, 0.28)), limits = c(-0.43, 0.28)) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0.8, size = BIG - 5) +\n",
        "scale_size(range = c(15*1.53, 60*1.53), breaks =c(1, 2, 3, 4, 5)) +\n",
        "scale_y_discrete(breaks = c(\"SRS_parent\", \"BAPQ_parent\", \"Parental Age_parent\", \"Parental EA_parent\"), labels = c(\"SRS\", \"BAPQ\", \"Parental Age\", \"Parental EA\")) +\n",
        "scale_x_discrete(breaks = c(\"dnmis\", \"dnlof\", \"inhLoF\", \"ps.asd\", \"ps.sz\", \"ps.ea\"), labels = c(\"dnMIS\", \"dnLoF\",  \"inhLoF\", expression(PS[ASD]),  expression(PS[SCZ]), expression(PS[EA]))) +\n",
        "labs(size = \"-log10(P-value)\", labels = c(\"1\", \"2\", \"3\", \"4\", \"5\")) +\n",
        "labs(fill = \"Effect Size [sd]\") +\n",
        "theme(panel.grid.major = element_blank()) +\n",
        "theme(panel.grid.minor = element_blank()) +\n",
        "# Text\n",
        "theme(axis.text.x = element_text(color = \"black\", size = MASSIVE)) +\n",
        "theme(axis.text.y = element_text(color = \"black\", size = MASSIVE)) +\n",
        "# Legend\n",
        "theme(legend.title = element_text(size = MASSIVE)) +\n",
        "theme(legend.text = element_text(color = \"black\", size = MASSIVE))\n",
        "panelB\n",
        "dev.off();\n",
        "\n",
        "### PANEL AB ###\n",
        "pdf(\"panelAB.test.pdf\", width = 40, height = 40, onefile = FALSE)\n",
        "panelAB = \n",
        "ggarrange(panelA, panelB,\n",
        "          labels = c(\"A\",\"B\"),\n",
        "          nrow = 2,\n",
        "          ncol = 1,\n",
        "          widths = c(1, 1), \n",
        "          heights = c(1.6, 1),\n",
        "          # vjust = 1.2,\n",
        "          common.legend = TRUE, \n",
        "          legend = \"right\")\n",
        "# theme(plot.margin = unit(c(0,3,0,0), \"cm\")) + \n",
        "panelAB\n",
        "dev.off();\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Scale for 'size' is already present. Adding another scale for 'size', which\n",
            "will replace the existing scale.\n",
            "\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Scale for 'size' is already present. Adding another scale for 'size', which\n",
            "will replace the existing scale.\n",
            "\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 85
        },
        "id": "qFzToJk1eSgf",
        "outputId": "9cc618bf-25a4-4585-d2e9-ec98294b9efb"
      },
      "source": [
        "asd_df <- read.csv(file = clin_results_out, header = TRUE, na.string = \"NA\")\n",
        "asd_df <- read.csv(file = \"clin_results_eur_and_noneur.SBayesR.csv\", header = TRUE, na.string = \"NA\")\n",
        "\n",
        "# Create columns of significance labels\n",
        "asd_df$stars.g <- cut(asd_df$x.pval, breaks=c(-Inf, 0.0007, 0.05, Inf), label=c(\"**\", \"*\", \"\"))  # Create column of significance labels for genotype effect\n",
        "asd_df$stars.g.sex <- cut(asd_df$sex.pval, breaks=c(-Inf, 0.0007, 0.05, Inf), label=c(\"**\", \"*\", \"\"))  # Create column of significance labels for genotype by sex effect\n",
        "\n",
        "### PANEL C ###\n",
        "\n",
        "# Get significant sex interactions p-values and save them for later (for displaying in barplots)\n",
        "controls_scq_combined_pssz = subset(x = asd_df,\n",
        "    subset = case.status == \"cont\" & phenotype == \"SCQ\" & group == \"combined\" & variant.type == \"ps.sz\",)\n",
        "controls_scq_combined_pssz_sex_pval = toString(controls_scq_combined_pssz$sex.pval)\n",
        "\n",
        "controls_srs_combined_psasd = subset(x = asd_df,\n",
        "    subset = case.status == \"cont\" & phenotype == \"SRS\" & group == \"combined\" & variant.type == \"ps.asd\",)\n",
        "controls_srs_combined_psasd_sex_pval = toString(controls_srs_combined_psasd$sex.pval)\n",
        "\n",
        "# cases_scq_combined_inhlof = subset(x = asd_df,\n",
        "#     subset = case.status == \"case\" & phenotype == \"SCQ\" & group == \"combined\" & variant.type == \"inhLoF\")\n",
        "# cases_scq_combined_inhlof_sex_pval = toString(cases_scq_combined_inhlof$sex.pval)\n",
        "\n",
        "\n",
        "# Select data for panel C (controls for all g*sex interactions PS-SCZ with SCQ and VABS)\n",
        "asd_C.1 <- asd_df %>%\n",
        "  filter(case.status %in% c(\"cont\")) %>%\n",
        "  filter(phenotype %in% c(\"SRS\")) %>%\n",
        "  filter(group %in% c(\"male\", \"female\")) %>%\n",
        "  filter(variant.type %in% c(\"ps.asd\"))\n",
        "\n",
        "asd_C.2 <- asd_df %>%\n",
        "  filter(case.status %in% c(\"cont\")) %>%\n",
        "  filter(phenotype %in% c(\"SCQ\")) %>%\n",
        "  filter(group %in% c(\"male\", \"female\")) %>%\n",
        "  filter(variant.type %in% c(\"ps.sz\"))\n",
        "\n",
        "# asd_C.3 <- asd_df %>%\n",
        "#   filter(case.status %in% c(\"case\")) %>%\n",
        "#   filter(phenotype %in% c(\"SCQ\")) %>%\n",
        "#   filter(group %in% c(\"male\", \"female\")) %>%\n",
        "#   filter(variant.type %in% c(\"inhLoF\"))\n",
        "\n",
        "asd_C12 <- rbind(asd_C.1, asd_C.2)\n",
        "my_labellerC12 <- as_labeller(c(SCQ = \"SCQ - PS[SCZ]\", SRS = \"SRS - PS[ASD]\"), default = label_parsed) \n",
        "\n",
        "# asd_C3 <- rbind(asd_C.3)\n",
        "# my_labellerC3 <- as_labeller(c(SCQ = \"SCQ - inhLoF\"), default = label_parsed)\n",
        "\n",
        "# # Make panel C12 p-value dataframe\n",
        "# pval_C12_df <- data.frame(\n",
        "#   phenotype = c(\"SCQ\", \"SRS\"),\n",
        "#   start = c(\"Female\", \"Female\"),\n",
        "#   end = c(\"Male\", \"Male\"),\n",
        "#   y.position = c(0.15, 0.15),\n",
        "#   pval = c(paste(\"P = \", controls_scq_combined_pssz_sex_pval,  sep=\"\"), \n",
        "#            paste(\"P = \", controls_srs_combined_psasd_sex_pval, sep=\"\"))\n",
        "# )\n",
        "\n",
        "# # Make panel C3 p-value dataframe\n",
        "# pval_C3_df <- data.frame(\n",
        "#   phenotype = c(\"SCQ\"),\n",
        "#   start = c(\"Female\"),\n",
        "#   end = c(\"Male\"),\n",
        "#   y.position = c(0.15),\n",
        "#   pval = c(paste(\"P = \", cases_scq_combined_inhlof_sex_pval, sep=\"\"))\n",
        "# )\n",
        "\n",
        "asd_C12$phen = factor(asd_C12$phenotype, levels=c('SRS','SCQ'))\n",
        "asd_C12$stars <- cut(asd_C12$x.pval, breaks = c(-Inf, 0.001, 0.01, 0.05, Inf), label = c(\"***\", \"**\", \"*\", \"\"))  # Create column of significance labels for main effect\n",
        "\n",
        "pdf(\"panelC.12.test.pdf\", width = 30, height = 30)\n",
        "panelC12 =\n",
        "ggplot(asd_C12, aes(x = group, y = est, fill = group)) +\n",
        "geom_bar(stat = \"identity\", width = 1, position = position_nudge(x = 0.5)) +\n",
        "facet_wrap(~phenotype, ncol = 2, labeller = my_labellerC12) +\n",
        "# P-value asterisks\n",
        "# geom_text(aes(label = stars), color = \"black\", vjust = -6, size = BIG - 5, hjust = -1.4) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0, size = BIG - 5, hjust = 0) +\n",
        "# Standard error whiskers\n",
        "geom_linerange(aes(ymin = est - est.stderr, ymax = est + est.stderr), position = position_nudge(0.5), size = 1) +\n",
        "scale_fill_manual(values = c(\"coral\", \"dodgerblue\")) +\n",
        "labs(y = expression(paste(\"Effect size of PS [SD]\"))) +\n",
        "labs(x = expression(paste(\"Controls\"))) +\n",
        "# Text\n",
        "# theme(axis.title.x = element_blank()) +\n",
        "theme(axis.title.x = element_text(size = MASSIVE + 5, color = \"black\", vjust = 10.0)) +\n",
        "theme(axis.text.x = element_blank()) +\n",
        "# theme(strip.text.x = element_text(size = MASSIVE - 30, color = \"black\")) +\n",
        "theme(strip.text.x = element_text(size = MASSIVE, color = \"black\")) +\n",
        "# theme(axis.text.y = element_text(size = MASSIVE - 30)) +\n",
        "theme(axis.text.y = element_text(size = MASSIVE, color = \"black\")) +\n",
        "theme(axis.title.y = element_text(size = MASSIVE + 5, color = \"black\")) +\n",
        "# Legend\n",
        "theme(legend.title = element_blank()) +\n",
        "theme(legend.text = element_text(size=MASSIVE)) +\n",
        "theme(legend.key.size = unit(5, \"line\")) +\n",
        "theme(panel.background = element_blank()) +\n",
        "theme(strip.background = element_rect(color = \"white\", fill = \"white\", size = 1.5, linetype = \"solid\")) +\n",
        "ylim(-0.10, 0.15)\n",
        "# p-value significance lines\n",
        "# geom_signif(data = pval_C12_df, aes(xmin = 1.5, xmax = 2.5, annotations = pval, y_position = y.position), manual = TRUE, inherit.aes = FALSE, textsize = BIG - 20)\n",
        "panelC12\n",
        "dev.off();\n",
        "\n",
        "# pdf(\"panelC.3.test.pdf\", width = 30, height = 30)\n",
        "# panelC3 =\n",
        "# ggplot(asd_C3, aes(x = group, y = est, fill = group)) +\n",
        "# geom_bar(stat = \"identity\", width = 1, position = position_nudge(x = 0.5)) +\n",
        "# facet_wrap(~phenotype, ncol = 1, labeller = my_labellerC3) +\n",
        "# geom_linerange(aes(ymin = est - est.stderr, ymax = est + est.stderr), position = position_nudge(0.5), size = 1) +\n",
        "# scale_fill_manual(values = c(\"coral\", \"dodgerblue\")) +\n",
        "# labs(y = expression(paste(\"Effect size of inhLoF [SD]\"))) +\n",
        "# labs(x = expression(paste(\"Cases\"))) +\n",
        "# # Text\n",
        "# # theme(axis.title.x = element_blank()) +\n",
        "# theme(axis.title.x = element_text(size = MASSIVE + 5, color = \"black\", vjust = 10.0)) +\n",
        "# theme(axis.text.x = element_blank()) +\n",
        "# theme(strip.text.x = element_text(size = MASSIVE - 30, color = \"black\")) +\n",
        "# # theme(axis.text.y = element_text(size = MASSIVE - 30)) +\n",
        "# theme(axis.text.y = element_blank()) +\n",
        "# theme(axis.ticks.y = element_blank()) +\n",
        "# theme(axis.title.y = element_text(size = MASSIVE + 5, color = \"black\")) +\n",
        "# # Legend\n",
        "# theme(legend.title = element_blank()) +\n",
        "# theme(legend.text = element_text(size=MASSIVE)) +\n",
        "# theme(legend.key.size = unit(5, \"line\")) +\n",
        "# theme(panel.background = element_blank()) +\n",
        "# theme(strip.background = element_rect(color = \"white\", fill = \"white\", size = 1.5, linetype = \"solid\")) +\n",
        "# ylim(-0.10, 0.15) +\n",
        "# # p-value significance lines\n",
        "# geom_signif(data = pval_C3_df, aes(xmin = 1.5, xmax = 2.5, annotations = pval, y_position = y.position), manual = TRUE, inherit.aes = FALSE, textsize = BIG - 20)\n",
        "# panelC3\n",
        "# dev.off();\n",
        "\n",
        "# # Combine panels C12 and C3 into C123\n",
        "# pdf(\"panelC.123.test.pdf\", width = 30, height = 30, onefile = FALSE)\n",
        "# panelC123 =\n",
        "# ggarrange(panelC12, panelC3,\n",
        "#           labels = c(\"C12\", \"C3\"),\n",
        "#           nrow = 1,\n",
        "#           ncol = 2,\n",
        "#           common.legend = TRUE,\n",
        "#           legend = \"right\",\n",
        "#           widths = c(2, 1))\n",
        "# panelC123\n",
        "# dev.off();\n",
        "\n",
        "\n",
        "### PANEL D ###\n",
        "# Select data for panel D (Parental Age Effect size: dnLoF, dnMIS, inhLoF, PS_SCZ, PS_ASD, PS_EA)\n",
        "asd_D <- asd_df %>%\n",
        "  filter(phenotype %in% c(\"Parental Age\")) %>%\n",
        "  filter(group %in% c(\"mother\", \"father\")) %>%\n",
        "  # filter(group %in% c(\"parents\")) %>%\n",
        "  filter(variant.type %in% c(\"dnlof\", \"inhLoF\", \"dnmis\", \"ps.sz\", \"ps.asd\", \"ps.ea\"))\n",
        "\n",
        "my_labellerD = as_labeller(c(dnlof = \"dnLoF\", inhLoF = \"inhLoF\", dnmis = \"dnMIS\", ps.sz = \"PS[SCZ]\", ps.asd = \"PS[ASD]\", ps.ea = \"PS[EA]\"), default = label_parsed)\n",
        "\n",
        "# Get significant sex interaction p-values\n",
        "parent_pa_parents_dnlof = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"dnlof\",)\n",
        "parent_pa_parents_dnlof_sex_pval = toString(parent_pa_parents_dnlof$sex.pval)\n",
        "\n",
        "parent_pa_parents_inhlof = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"inhLoF\",)\n",
        "parent_pa_parents_inhlof_sex_pval = toString(parent_pa_parents_inhlof$sex.pval)\n",
        "\n",
        "parent_pa_parents_dnmis = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"dnmis\",)\n",
        "parent_pa_parents_dnmis_sex_pval = toString(parent_pa_parents_dnmis$sex.pval)\n",
        "\n",
        "parent_pa_parents_pssz = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"ps.sz\",)\n",
        "parent_pa_parents_pssz_sex_pval = toString(parent_pa_parents_pssz$sex.pval)\n",
        "\n",
        "parent_pa_parents_psasd = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"ps.asd\",)\n",
        "parent_pa_parents_psasd_sex_pval = toString(parent_pa_parents_psasd$sex.pval)\n",
        "\n",
        "parent_pa_parents_psea = subset(x = asd_df,\n",
        "    subset = case.status == \"parent\" & phenotype == \"Parental Age\" & group == \"parents\" & variant.type == \"ps.ea\",)\n",
        "parent_pa_parents_psea_sex_pval = toString(parent_pa_parents_psea$sex.pval)\n",
        "\n",
        "# Make the p-value dataframe for panel D\n",
        "pval_D_df <- data.frame(\n",
        "  variant.type = c(\"dnlof\", \"inhLoF\", \"dnmis\", \"ps.sz\", \"ps.asd\", \"ps.ea\"),\n",
        "  start = c(\"Father\", \"Father\"),\n",
        "  end = c(\"Mother\", \"Mother\"),\n",
        "  y.position = c(0.2, 0.2),\n",
        "  pval = c(paste(\"P = \", parent_pa_parents_dnlof_sex_pval,  sep=\"\"), \n",
        "           paste(\"P = \", parent_pa_parents_inhlof_sex_pval, sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_dnmis_sex_pval,  sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_pssz_sex_pval,   sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_psasd_sex_pval,  sep=\"\"),\n",
        "           paste(\"P = \", parent_pa_parents_psea_sex_pval,   sep=\"\"))\n",
        ")\n",
        "\n",
        "options(repr.plot.width=15, repr.plot.height=8)\n",
        "\n",
        "asd_D$stars <- cut(asd_D$x.pval, breaks = c(-Inf, 0.001, 0.01, 0.05, Inf), label = c(\"***\", \"**\", \"*\", \"\"))  # Create column of significance labels for main effect\n",
        "\n",
        "pdf(\"panelD.test.pdf\", width = 30, height = 30)\n",
        "panelD =\n",
        "ggplot(asd_D, aes(x = group, y = est, fill = group)) +\n",
        "geom_bar(stat = \"identity\", width = 1, position = position_nudge(x = 0.5)) +\n",
        "facet_wrap(~variant.type, ncol = 6, labeller = my_labellerD) +\n",
        "# P-value asterisks\n",
        "# geom_text(aes(label = stars), color = \"black\", vjust = -3, size = BIG - 5, hjust = 0) +\n",
        "geom_text(aes(label = stars), color = \"black\", vjust = 0, size = BIG - 5, hjust = 0) +\n",
        "# Standard error whiskers\n",
        "geom_linerange(aes(ymin = est - est.stderr, ymax = est + est.stderr), position = position_nudge(0.5), size = 1) +\n",
        "scale_fill_manual(values = c(\"dodgerblue\", \"coral\")) +\n",
        "labs(y = \"Effect Size [sd]\") +\n",
        "theme(axis.text.x = element_blank()) +\n",
        "# Text\n",
        "theme(axis.title.x = element_blank()) +\n",
        "theme(axis.text.x = element_blank()) +\n",
        "# theme(strip.text.x = element_text(size = MASSIVE - 30, color = \"black\")) +\n",
        "theme(strip.text.x = element_text(size = MASSIVE, color = \"black\")) +\n",
        "# theme(axis.text.y = element_text(size = MASSIVE - 30)) +\n",
        "theme(axis.text.y = element_text(size = MASSIVE, color = \"black\")) +\n",
        "theme(axis.title.y = element_text(size = MASSIVE + 5, color = \"black\")) +\n",
        "# Legend\n",
        "theme(legend.title = element_blank()) +\n",
        "theme(legend.text = element_text(size = MASSIVE)) +\n",
        "theme(legend.key.size = unit(5, \"line\")) +\n",
        "theme(panel.background = element_blank()) +\n",
        "theme(strip.background = element_rect(color = \"white\", fill = \"white\", size = 1.5, linetype = \"solid\")) +\n",
        "# y-axis limits\n",
        "ylim(-0.10, 0.20)\n",
        "# p-value significance lines\n",
        "# geom_signif(data = pval_D_df, aes(xmin = 1.5, xmax = 2.5, annotations = pval, y_position = y.position), manual = TRUE, inherit.aes = FALSE, textsize = BIG - 20)\n",
        "panelD\n",
        "dev.off();\n",
        "\n",
        "### PANEL CD ###\n",
        "\n",
        "pdf(\"panelCD.test.pdf\", width = 40, height = 40)\n",
        "panelCD =\n",
        "ggarrange(panelC12, panelD, # panelC123 -> panelC12\n",
        "          labels = c(\"C\", \"D\"),\n",
        "          nrow = 2,\n",
        "          ncol = 1,\n",
        "          common.legend = FALSE,\n",
        "          legend = \"right\",\n",
        "          widths = c(1, 1))\n",
        "panelCD\n",
        "dev.off();\n",
        "\n",
        "\n",
        "\n",
        "### PANEL ABCD ###\n",
        "\n",
        "# pdf(\"fig5.ABCD.test.pdf\",width=60, height = 40)\n",
        "# pdf(\"fig5.pdf\",width=60, height = 40)\n",
        "pdf(\"fig5.eur_and_noneur.pdf\",width=60, height = 40)\n",
        "# svg(\"fig5.svg\", width=60, height=40)\n",
        "# svglite(\"fig5.svglite.svg\", width = 60, height = 40)\n",
        "# ggsave(\"fig5.ggsave.pdf\", plot = panelABCD, width = 120, height = 80, units = \"cm\", limitsize = FALSE)\n",
        "panelABCD =\n",
        "panelABCD = \n",
        "ggarrange(panelAB, panelCD,\n",
        "          nrow = 1,\n",
        "          ncol = 2,\n",
        "          align = \"v\",\n",
        "          widths = c(1.4, 1), heights = c(1, 1),\n",
        "          common.legend = FALSE)\n",
        "# panelABCD$layout$clip = \"off\"\n",
        "panelABCD;\n",
        "dev.off();\n",
        "\n",
        "# ggsave(\"fig5.ggsave.svg\", plot = panelABCD, width = 60, height = 40, units = \"cm\")\n",
        "\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cEAWsAT3vogn",
        "outputId": "520d5e66-77f0-4aaa-dd46-0724118fd78b"
      },
      "source": [
        "# Figure 4B\n",
        "\n",
        "install.packages(\"corrplot2.tar.gz\", repos = NULL)\n",
        "install.packages(\"reshape2\")\n",
        "library(corrplot2)\n",
        "library(reshape2)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "Installing package into ‘/usr/local/lib/R/site-library’\n",
            "(as ‘lib’ is unspecified)\n",
            "\n",
            "also installing the dependency ‘plyr’\n",
            "\n",
            "\n",
            "corrplot2 0.1 loaded\n",
            "\n",
            "\n",
            "Attaching package: ‘reshape2’\n",
            "\n",
            "\n",
            "The following object is masked from ‘package:tidyr’:\n",
            "\n",
            "    smiths\n",
            "\n",
            "\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 153
        },
        "id": "g-TlUAmAv4Oh",
        "outputId": "b081e303-b7df-44c1-f442-2c403bff7bfe"
      },
      "source": [
        "options(stringsAsFactors = FALSE);\n",
        "df = read.csv(asd_genetic_correlations_by_sex_out, sep = \"\\t\")\n",
        "\n",
        "pal <- colorRampPalette(c(\"#000080\",\"white\",\"#b30000\"))\n",
        "\n",
        "get_mat <- function(df, name) {\n",
        "  mat <- acast(df, y~x, value.var = name, fill = NA)\n",
        "  ord = c(\"dnMIS\", \"dnLOF\", \"inhLoF\",\n",
        "          \"pTDT.ASD.Dev\", \"pTDT.SCZ.Dev\", \"pTDT.EA.Dev\",\n",
        "          \"RVRS\", \"PRS\");\n",
        "  mat <- mat[ord, ord]\n",
        "  \n",
        "  labs = c(\"dnMIS\", \"dnLoF\", \"inhLoF\",\n",
        "           \"$PS[ASD]\", \"$PS[SCZ]\", \"$PS[EA]\",\n",
        "           \"RVRS\", \"CVRS\");\n",
        "\n",
        "\n",
        "\n",
        "  colnames(mat) <- labs\n",
        "  rownames(mat) <- labs\n",
        "  return(mat)\n",
        "}\n",
        "\n",
        "\n",
        "\n",
        "n_pal <- 40\n",
        "rows <- c(\"dnMIS\",\"dnLoF\",\"inhLoF\",\"RVRS\");\n",
        "cols <- c(\"$PS[ASD]\",\"$PS[SCZ]\",\"$PS[EA]\",\"CVRS\");\n",
        "\n",
        "\n",
        "# Males\n",
        "m_co <- get_mat(df,\"male_cor\")\n",
        "m_pv <- get_mat(df,\"male_pval\")\n",
        "f_co <- get_mat(df,\"female_cor\")\n",
        "f_pv <- get_mat(df,\"female_pval\")\n",
        "\n",
        "all_co <- get_mat(df,\"combined_cor\")\n",
        "all_pv <- get_mat(df,\"combined_pval\")\n",
        "\n",
        "m_co <- m_co[rows,cols]\n",
        "f_co <- f_co[rows,cols]\n",
        "all_co <- all_co[rows,cols]\n",
        "\n",
        "m_pv <- m_pv[rows,cols]\n",
        "f_pv <- f_pv[rows,cols]\n",
        "all_pv <- all_pv[rows,cols]\n",
        "\n",
        "# Rebrand female row names\n",
        "row.names(f_co) <- c(1,2,3,4)\n",
        "row.names(f_pv) <- c(1,2,3,4)\n",
        "# rbind the matrices\n",
        "sex_co <- rbind(m_co,f_co)\n",
        "sex_pv <- rbind(m_pv,f_pv)\n",
        "# Reorder rows\n",
        "rows <- c(\"dnMIS\",1,\"dnLoF\",2,\"inhLoF\",3,\"RVRS\",4);\n",
        "sex_co<-sex_co[rows,]\n",
        "sex_pv<-sex_pv[rows,]\n",
        "\n",
        "\n",
        "dum <- rbind(1, 0.1, 0.049, 0.0099)\n",
        "row.names(dum) <- c(1, 0.1, 0.05, 0.01)\n",
        "\n",
        "sig_lev <- c(.01, .05)\n",
        "\n",
        "# pdf(\"genetic_corr_pval_boxes.pdf\", width = 16, height = 16)\n",
        "pdf(\"fig4b.box.legend.pdf\", width=16, height = 16)\n",
        "# svg(\"fig4b.box.legend.svg\", width=16, height = 16)\n",
        "corrplot2(\n",
        "  dum,\n",
        "  # cbind(dum_cor),\n",
        "  method = \"square\",\n",
        "  outline = TRUE,\n",
        "  is.corr.= FALSE,\n",
        "  col = \"navy\", # color\n",
        "  tl.col = \"black\",\n",
        "  # tl.srt = 30, # rotation \n",
        "  tl.cex = 3,\n",
        "  p.mat = cbind(dum),\n",
        "  insig = \"label_sig\",\n",
        "  cl.pos = FALSE,\n",
        "  sig.level = sig_lev,\n",
        "  pch.cex = 5,\n",
        "  pch.col = \"white\",\n",
        "  norm_mat = dum,\n",
        ");\n",
        "dev.off()\n",
        "\n",
        "\n",
        "dum <- rbind(1, 0.75, 0.5, 0.25, 0.1, 0.049, 0.0249, 0.0099)\n",
        "dum_cor <- matrix(min(m_co, f_co), nrow = length(dum), ncol = 1)\n",
        "row.names(dum_cor) <- c(1, 0.75, 0.5, 0.25, 0.1, 0.05, 0.025, 0.01)\n",
        "sig_lev <- c(.01, .05)\n",
        "\n",
        "# Combined\n",
        "# pdf(\"combined_genetic_corr.pdf\", width=16, height = 16)\n",
        "pdf(\"fig4b.pdf\", width=16, height = 16)\n",
        "# svg(\"fig4b.svg\", width=16, height = 16)\n",
        "corrplot2(\n",
        "  all_co,\n",
        "  method = \"square\",\n",
        "  outline = TRUE,\n",
        "  is.corr = FALSE,\n",
        "  col = pal(n_pal),\n",
        "  tl.col = \"black\",\n",
        "  tl.srt = 0, # 90\n",
        "  tl.cex = 3, # 1.5\n",
        "  tl.offset = 1,\n",
        "  p.mat = all_pv,\n",
        "  sig.level =  sig_lev,\n",
        "  insig = \"label_sig\",\n",
        "  pch.cex = 5, # 5\n",
        "  pch.col = \"white\",\n",
        "  norm_mat = dum,\n",
        "  cl.lim = c(min(m_co, f_co), max(m_co, f_co)),\n",
        "  cor_max = max(m_co, f_co),\n",
        "  cor_min = min(m_co,f_co),\n",
        "  cl.cex = 2,\n",
        "  cl.ratio = 0.2,\n",
        ");\n",
        "dev.off()\n"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Warning message in text.default(pos.xlabel[, 1], pos.xlabel[, 2], newcolnames, srt = tl.srt, :\n",
            "“\"is.corr.\" is not a graphical parameter”\n",
            "Warning message in text.default(pos.ylabel[, 1], pos.ylabel[, 2], newrownames, col = tl.col, :\n",
            "“\"is.corr.\" is not a graphical parameter”\n",
            "Warning message in title(title, ...):\n",
            "“\"is.corr.\" is not a graphical parameter”\n"
          ],
          "name": "stderr"
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "png \n",
              "  2 "
            ],
            "text/latex": "\\textbf{png:} 2",
            "text/markdown": "**png:** 2",
            "text/html": [
              "<strong>png:</strong> 2"
            ]
          },
          "metadata": {
            "tags": []
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 187
        },
        "id": "VnqbEhAok3yb",
        "outputId": "8d80c2c0-fd70-46f8-874c-2721c0c289dd"
      },
      "source": [
        "system('rm -rf /content/sample_data', intern=TRUE)\n",
        "system('cd /home && pwd && tar -cvzf antaki_analysis.tar.gz /content', intern=TRUE)"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "character(0)"
            ],
            "text/latex": "",
            "text/markdown": "",
            "text/html": [
              ""
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              " [1] \"/home\"                                                                     \n",
              " [2] \"/content/\"                                                                 \n",
              " [3] \"/content/.config/\"                                                         \n",
              " [4] \"/content/.config/.last_survey_prompt.yaml\"                                 \n",
              " [5] \"/content/.config/config_sentinel\"                                          \n",
              " [6] \"/content/.config/.last_opt_in_prompt.yaml\"                                 \n",
              " [7] \"/content/.config/active_config\"                                            \n",
              " [8] \"/content/.config/logs/\"                                                    \n",
              " [9] \"/content/.config/logs/2021.07.16/\"                                         \n",
              "[10] \"/content/.config/logs/2021.07.16/13.19.23.269640.log\"                      \n",
              "[11] \"/content/.config/logs/2021.07.16/13.20.21.584309.log\"                      \n",
              "[12] \"/content/.config/logs/2021.07.16/13.20.04.172291.log\"                      \n",
              "[13] \"/content/.config/logs/2021.07.16/13.20.22.328313.log\"                      \n",
              "[14] \"/content/.config/logs/2021.07.16/13.19.42.311157.log\"                      \n",
              "[15] \"/content/.config/logs/2021.07.16/13.19.58.237112.log\"                      \n",
              "[16] \"/content/.config/gce\"                                                      \n",
              "[17] \"/content/.config/.last_update_check.json\"                                  \n",
              "[18] \"/content/.config/configurations/\"                                          \n",
              "[19] \"/content/.config/configurations/config_default\"                            \n",
              "[20] \"/content/combined_genetic_corr.pdf\"                                        \n",
              "[21] \"/content/master_phen_4.SBayesR.csv\"                                        \n",
              "[22] \"/content/panelCD.test.pdf\"                                                 \n",
              "[23] \"/content/all_t1burden_tdt_results.tsv\"                                     \n",
              "[24] \"/content/panelD.test.pdf\"                                                  \n",
              "[25] \"/content/SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv\"\n",
              "[26] \"/content/genetic_corr_pval_boxes.pdf\"                                      \n",
              "[27] \"/content/clin_results_eur.SBayesR.20210715.csv\"                            \n",
              "[28] \"/content/panelAB.test.pdf\"                                                 \n",
              "[29] \"/content/fig5.pdf\"                                                         \n",
              "[30] \"/content/panelC.12.test.pdf\"                                               \n",
              "[31] \"/content/panelB.test.pdf\"                                                  \n",
              "[32] \"/content/fig5.ABCD.test.pdf\"                                               \n",
              "[33] \"/content/r2_table.csv\"                                                     \n",
              "[34] \"/content/fig4b.pdf\"                                                        \n",
              "[35] \"/content/panelA.test.pdf\"                                                  \n",
              "[36] \"/content/corrplot2.tar.gz\"                                                 \n",
              "[37] \"/content/fig4b.box.legend.pdf\"                                             \n",
              "[38] \"/content/asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv\"       \n",
              "[39] \"/content/SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv\"                   "
            ],
            "text/latex": "\\begin{enumerate*}\n\\item '/home'\n\\item '/content/'\n\\item '/content/.config/'\n\\item '/content/.config/.last\\_survey\\_prompt.yaml'\n\\item '/content/.config/config\\_sentinel'\n\\item '/content/.config/.last\\_opt\\_in\\_prompt.yaml'\n\\item '/content/.config/active\\_config'\n\\item '/content/.config/logs/'\n\\item '/content/.config/logs/2021.07.16/'\n\\item '/content/.config/logs/2021.07.16/13.19.23.269640.log'\n\\item '/content/.config/logs/2021.07.16/13.20.21.584309.log'\n\\item '/content/.config/logs/2021.07.16/13.20.04.172291.log'\n\\item '/content/.config/logs/2021.07.16/13.20.22.328313.log'\n\\item '/content/.config/logs/2021.07.16/13.19.42.311157.log'\n\\item '/content/.config/logs/2021.07.16/13.19.58.237112.log'\n\\item '/content/.config/gce'\n\\item '/content/.config/.last\\_update\\_check.json'\n\\item '/content/.config/configurations/'\n\\item '/content/.config/configurations/config\\_default'\n\\item '/content/combined\\_genetic\\_corr.pdf'\n\\item '/content/master\\_phen\\_4.SBayesR.csv'\n\\item '/content/panelCD.test.pdf'\n\\item '/content/all\\_t1burden\\_tdt\\_results.tsv'\n\\item '/content/panelD.test.pdf'\n\\item '/content/SummaryGeneticData\\_REACH\\_SSC\\_SPARK.responses.SBayesR.20210715.csv'\n\\item '/content/genetic\\_corr\\_pval\\_boxes.pdf'\n\\item '/content/clin\\_results\\_eur.SBayesR.20210715.csv'\n\\item '/content/panelAB.test.pdf'\n\\item '/content/fig5.pdf'\n\\item '/content/panelC.12.test.pdf'\n\\item '/content/panelB.test.pdf'\n\\item '/content/fig5.ABCD.test.pdf'\n\\item '/content/r2\\_table.csv'\n\\item '/content/fig4b.pdf'\n\\item '/content/panelA.test.pdf'\n\\item '/content/corrplot2.tar.gz'\n\\item '/content/fig4b.box.legend.pdf'\n\\item '/content/asd\\_genetic\\_correlations\\_by\\_sex\\_final.SBayesR.20210715.tsv'\n\\item '/content/SummaryGeneticData\\_REACH\\_SSC\\_SPARK.SBayesR.csv'\n\\end{enumerate*}\n",
            "text/markdown": "1. '/home'\n2. '/content/'\n3. '/content/.config/'\n4. '/content/.config/.last_survey_prompt.yaml'\n5. '/content/.config/config_sentinel'\n6. '/content/.config/.last_opt_in_prompt.yaml'\n7. '/content/.config/active_config'\n8. '/content/.config/logs/'\n9. '/content/.config/logs/2021.07.16/'\n10. '/content/.config/logs/2021.07.16/13.19.23.269640.log'\n11. '/content/.config/logs/2021.07.16/13.20.21.584309.log'\n12. '/content/.config/logs/2021.07.16/13.20.04.172291.log'\n13. '/content/.config/logs/2021.07.16/13.20.22.328313.log'\n14. '/content/.config/logs/2021.07.16/13.19.42.311157.log'\n15. '/content/.config/logs/2021.07.16/13.19.58.237112.log'\n16. '/content/.config/gce'\n17. '/content/.config/.last_update_check.json'\n18. '/content/.config/configurations/'\n19. '/content/.config/configurations/config_default'\n20. '/content/combined_genetic_corr.pdf'\n21. '/content/master_phen_4.SBayesR.csv'\n22. '/content/panelCD.test.pdf'\n23. '/content/all_t1burden_tdt_results.tsv'\n24. '/content/panelD.test.pdf'\n25. '/content/SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv'\n26. '/content/genetic_corr_pval_boxes.pdf'\n27. '/content/clin_results_eur.SBayesR.20210715.csv'\n28. '/content/panelAB.test.pdf'\n29. '/content/fig5.pdf'\n30. '/content/panelC.12.test.pdf'\n31. '/content/panelB.test.pdf'\n32. '/content/fig5.ABCD.test.pdf'\n33. '/content/r2_table.csv'\n34. '/content/fig4b.pdf'\n35. '/content/panelA.test.pdf'\n36. '/content/corrplot2.tar.gz'\n37. '/content/fig4b.box.legend.pdf'\n38. '/content/asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv'\n39. '/content/SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv'\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'/home'</li><li>'/content/'</li><li>'/content/.config/'</li><li>'/content/.config/.last_survey_prompt.yaml'</li><li>'/content/.config/config_sentinel'</li><li>'/content/.config/.last_opt_in_prompt.yaml'</li><li>'/content/.config/active_config'</li><li>'/content/.config/logs/'</li><li>'/content/.config/logs/2021.07.16/'</li><li>'/content/.config/logs/2021.07.16/13.19.23.269640.log'</li><li>'/content/.config/logs/2021.07.16/13.20.21.584309.log'</li><li>'/content/.config/logs/2021.07.16/13.20.04.172291.log'</li><li>'/content/.config/logs/2021.07.16/13.20.22.328313.log'</li><li>'/content/.config/logs/2021.07.16/13.19.42.311157.log'</li><li>'/content/.config/logs/2021.07.16/13.19.58.237112.log'</li><li>'/content/.config/gce'</li><li>'/content/.config/.last_update_check.json'</li><li>'/content/.config/configurations/'</li><li>'/content/.config/configurations/config_default'</li><li>'/content/combined_genetic_corr.pdf'</li><li>'/content/master_phen_4.SBayesR.csv'</li><li>'/content/panelCD.test.pdf'</li><li>'/content/all_t1burden_tdt_results.tsv'</li><li>'/content/panelD.test.pdf'</li><li>'/content/SummaryGeneticData_REACH_SSC_SPARK.responses.SBayesR.20210715.csv'</li><li>'/content/genetic_corr_pval_boxes.pdf'</li><li>'/content/clin_results_eur.SBayesR.20210715.csv'</li><li>'/content/panelAB.test.pdf'</li><li>'/content/fig5.pdf'</li><li>'/content/panelC.12.test.pdf'</li><li>'/content/panelB.test.pdf'</li><li>'/content/fig5.ABCD.test.pdf'</li><li>'/content/r2_table.csv'</li><li>'/content/fig4b.pdf'</li><li>'/content/panelA.test.pdf'</li><li>'/content/corrplot2.tar.gz'</li><li>'/content/fig4b.box.legend.pdf'</li><li>'/content/asd_genetic_correlations_by_sex_final.SBayesR.20210715.tsv'</li><li>'/content/SummaryGeneticData_REACH_SSC_SPARK.SBayesR.csv'</li></ol>\n"
            ]
          },
          "metadata": {
            "tags": []
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9jKmGPgeZtpP",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "1c15bf52-717c-4031-bffc-3ce0b5f6e9ba"
      },
      "source": [
        "# Author: Jonathan Sebat http://sebatlab.org March 19, 2021\n",
        "\n",
        "# The following phenotypes were available for only one cohort in either cases or controls: srs.case, vabs.case, srs.cont, scq.cont, vabs.cont, bapq.parents, srs.parents)\n",
        "# For these, age, sex and PCs were included as covariates, but not cohort (a 1st set of functions is written for this test)\n",
        "\n",
        "# SCQ, RBS and DCDQ were available on cases in the multiple cohorts (SSC and SPARK). Parental age and educational attainment were also available on multiple cohorts\n",
        "# For these, \"cohort\" was included as a covariate (a 2nd set of functions are written for this test)\n",
        "\n",
        "# Importing the input data table that contains rare variant counts (dnLoF, dnMis, inhLoF), polygenic scores(PSasd, PSsz, PSea) and clinical phenotypes (srs, scq, rbs, vabs, dcdq, parental age, parental educational attainment, parental bapq)\n",
        "master.gen.phen.table <- read.csv(filename_phen, header = TRUE)\n",
        "\n",
        "# Turn cohort into a factor variable\n",
        "# master.gen.phen.table$cohort = factor(master.gen.phen.table$cohort)\n",
        "# Make cohort variables (1 is REACH, 2 is SSC, 3 is SPARK)\n",
        "master.gen.phen.table$REACH_cohort = as.numeric(master.gen.phen.table$cohort == 1)\n",
        "master.gen.phen.table$SSC_cohort   = as.numeric(master.gen.phen.table$cohort == 2)\n",
        "master.gen.phen.table$SPARK_cohort = as.numeric(master.gen.phen.table$cohort == 3)\n",
        "\n",
        "\n",
        "# Analysis will restricted to subjects of confirmed European descent\n",
        "master.gen.phen.table = master.gen.phen.table[master.gen.phen.table$is_eur == 1,]\n",
        "\n",
        "# colnames(master.gen.phen.table)\n",
        "# Phenotypes\n",
        "# 'srs.child.age'\n",
        "# 'vabs.child.age'\n",
        "# 'scq.child.age'\n",
        "# 'dcdq.child.age'\n",
        "# 'mother.EA'\n",
        "# 'father.EA'\n",
        "# 'srs.mother'\n",
        "# 'srs.father'\n",
        "# 'mother.bapq'\n",
        "# 'father.bapq'\n",
        "# 'srs.child.z'\n",
        "# 'vabs.child.z'\n",
        "# 'rbs.child.z'\n",
        "# 'scq.child.z'\n",
        "# 'dcdq.child.z'\n",
        "# 'father.age.at.birth.z'\n",
        "# 'mother.age.at.birth.z'\n",
        "# 'father.EA.z'\n",
        "# 'mother.EA.z'\n",
        "# 'father.srs.z'\n",
        "# 'mother.srs.z'\n",
        "# 'father.bapq.z'\n",
        "# 'mother.bapq.z'\n",
        "\n",
        "# Genetic factors\n",
        "# 'dnLoF'\n",
        "# 'dnMIS'\n",
        "# 'inhLoF'\n",
        "# 'PS.ASD.Z'\n",
        "# 'PS.EA.Z'\n",
        "# 'PS.SZ.Z'\n",
        "# 'Paternal.PS.ASD.Z'\n",
        "# 'Paternal.PS.EA.Z'\n",
        "# 'Paternal.PS.SZ.Z'\n",
        "# 'Maternal.PS.ASD.Z'\n",
        "# 'Maternal.PS.EA.Z'\n",
        "# 'Maternal.PS.SZ.Z'\n",
        "\n",
        "run_lr <- function(dataframe, case_status, null_variables, gene_var, response_variable) {\n",
        "  df = dataframe\n",
        "  # The model\n",
        "  if (case_status == \"ASD\"){\n",
        "    df = df[df$phen == \"ASD\",]\n",
        "  }\n",
        "  else if (case_status == \"CON\") {\n",
        "    df = df[df$phen == \"CON\",]\n",
        "  }\n",
        "\n",
        "  null_formula <- reformulate(null_variables, response = response_variable)\n",
        "  null_model <- lm(formula = null_formula, data = df, na.action = na.exclude)\n",
        "  model_summary_null = summary(null_model)\n",
        "\n",
        "  gene_var <- c(gene_var)\n",
        "  alt_variables  <- paste(gene_var, null_variables, sep = \"+\")\n",
        "  alt_formula <- reformulate(alt_variables, response = response_variable)\n",
        "  alt_model  <- lm(formula = alt_formula , data = df, na.action = na.exclude)\n",
        "  model_summary_alt = summary(alt_model)\n",
        "\n",
        "  # Get number of observations in each model\n",
        "  # nobs(null_model)\n",
        "  # nobs(alt_model)\n",
        "\n",
        "  corr_equation <- paste(response_variable, gene_var, sep = \"+\")\n",
        "  corr_formula <- reformulate(corr_equation, response = NULL)\n",
        "  pearson.cor = cor.test(formula = corr_formula, data = df, na.action = na.exclude)\n",
        "\n",
        "\n",
        "  # Test sex interaction (all of this just to make a simple string from a character vector...)\n",
        "  null_variables_vector = unlist(strsplit(null_variables, split = \"[+]\"))\n",
        "  null_variables_vector_no_sex_term <- null_variables_vector[1:length(null_variables_vector) - 1]\n",
        "  null_variables_no_sex_term <- paste(null_variables_vector_no_sex_term, collapse = \"+\")\n",
        "  sex_interaction_variable <- paste(gene_var, \"sex\", sep = \"*\")\n",
        "  sex_interaction_terms <- paste(sex_interaction_variable, null_variables_no_sex_term, sep = \"+\")\n",
        "  \n",
        "  sex_interaction_formula <- reformulate(sex_interaction_terms, response = response_variable)\n",
        "  sex_interaction_model <- lm(formula = sex_interaction_formula, data = df, na.action = na.exclude)\n",
        "  model_summary_sex_interaction = summary(sex_interaction_model)\n",
        "\n",
        "\n",
        "  # Sex stratification\n",
        "  df_combined = df\n",
        "  # Sets Males to 1, and Females to 2.\n",
        "  df_combined$sex <- ifelse(df_combined$sex == \"Male\", 1, 2)\n",
        "  df_male   = df_combined[df_combined$sex == 1,]\n",
        "  df_female = df_combined[df_combined$sex == 2,]\n",
        "\n",
        "  null_model_male <- lm(formula = null_formula, data = df_male, na.action = na.exclude)\n",
        "  alt_model_male  <- lm(formula = alt_formula, data = df_male, na.action = na.exclude)\n",
        "\n",
        "  model_summary_null_male <- summary(null_model_male)\n",
        "  model_summary_alt_male <-  summary(alt_model_male)\n",
        "\n",
        "  pearson.cor.male = cor.test(formula = corr_formula, data = df_male, na.action = na.exclude)\n",
        "\n",
        "  null_model_female <- lm(formula = null_formula, data = df_female, na.action = na.exclude)\n",
        "  alt_model_female  <- lm(formula = alt_formula, data = df_female, na.action = na.exclude)\n",
        "\n",
        "  model_summary_null_female <- summary(null_model_female)\n",
        "  model_summary_alt_female <-  summary(alt_model_female)\n",
        "\n",
        "  pearson.cor.female = cor.test(formula = corr_formula, data = df_female, na.action = na.exclude)\n",
        "\n",
        "  output <- list(model_summary_alt, model_summary_null, \n",
        "                 pearson.cor, model_summary_sex_interaction, \n",
        "                 model_summary_alt_male, model_summary_null_male, pearson.cor.male,\n",
        "                 model_summary_alt_female, model_summary_null_female, pearson.cor.female)\n",
        "  return(output)\n",
        "}\n",
        "\n",
        "# Null variables common to all models\n",
        "cohort_vars <- c(\"REACH_cohort+SSC_cohort+SPARK_cohort\")\n",
        "pc_vars <- c(\"PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10\")\n",
        "sex_var <- c(\"sex\") # Only use with combined models (it doesn't matter)\n",
        "null_vars <- paste(cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "\n",
        "# DCDQ models\n",
        "age_var <- c(\"dcdq.child.age\")\n",
        "null_vars <- paste(age_var, cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "response_var = c(\"dcdq.child.z\")\n",
        "# Cases\n",
        "dcdq.dnlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "dcdq.dnmis.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "dcdq.inhlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "dcdq.ps_asd.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "dcdq.ps_ea.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "dcdq.ps_sz.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "# DCDQ data not available for controls\n",
        "# dcdq.dnlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "# dcdq.dnmis.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "# dcdq.inhlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "# dcdq.ps_asd.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "# dcdq.ps_ea.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "# dcdq.ps_sz.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "\n",
        "# RBS models\n",
        "age_var <- c(\"scq.child.age\") # There isn't any rbs.child.age?\n",
        "null_vars <- paste(age_var, cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "response_var = c(\"rbs.child.z\")\n",
        "# Cases\n",
        "rbs.dnlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "rbs.dnmis.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "rbs.inhlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "rbs.ps_asd.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "rbs.ps_ea.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "rbs.ps_sz.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "# RBS data not available for controls\n",
        "# rbs.dnlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "# rbs.dnmis.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "# rbs.inhlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "# rbs.ps_asd.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "# rbs.ps_ea.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "# rbs.ps_sz.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# SCQ models\n",
        "age_var <- c(\"scq.child.age\")\n",
        "null_vars <- paste(age_var, cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "response_var = c(\"scq.child.z\")\n",
        "# Cases\n",
        "scq.dnlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "scq.dnmis.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "scq.inhlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "scq.ps_asd.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "scq.ps_ea.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "scq.ps_sz.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "# Controls\n",
        "scq.dnlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "scq.dnmis.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "scq.inhlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "scq.ps_asd.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "scq.ps_ea.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "scq.ps_sz.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# SRS models\n",
        "age_var <- c(\"srs.child.age\")\n",
        "null_vars <- paste(age_var, cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "response_var = c(\"srs.child.z\")\n",
        "# Cases\n",
        "srs.dnlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "srs.dnmis.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "srs.inhlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "srs.ps_asd.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "srs.ps_ea.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "srs.ps_sz.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "# Controls\n",
        "srs.dnlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "srs.dnmis.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "srs.inhlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "srs.ps_asd.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "srs.ps_ea.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "srs.ps_sz.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# VABS models\n",
        "age_var <- c(\"vabs.child.age\")\n",
        "null_vars <- paste(age_var, cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "response_var = c(\"vabs.child.z\")\n",
        "# Cases\n",
        "vabs.dnlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "vabs.dnmis.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "vabs.inhlof.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "vabs.ps_asd.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "vabs.ps_ea.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "vabs.ps_sz.case.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"ASD\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "# Controls\n",
        "vabs.dnlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "vabs.dnmis.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "vabs.inhlof.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "vabs.ps_asd.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "vabs.ps_ea.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "vabs.ps_sz.control.summary = run_lr(dataframe = master.gen.phen.table, case_status = \"CON\", null_variables = null_vars, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# Parental models\n",
        "run_lr_parental <- function(dataframe, null_variables, gene_var, response_variable, combined = FALSE) {\n",
        "  df = dataframe\n",
        "\n",
        "  null_formula <- reformulate(null_variables, response = response_variable)\n",
        "  null_model <- lm(formula = null_formula, data = df, na.action = na.exclude)\n",
        "  model_summary_null = summary(null_model)\n",
        "\n",
        "  gene_var <- c(gene_var)\n",
        "  alt_variables  <- paste(gene_var, null_variables, sep = \"+\")\n",
        "  alt_formula <- reformulate(alt_variables, response = response_variable)\n",
        "  alt_model  <- lm(formula = alt_formula , data = df, na.action = na.exclude)\n",
        "  model_summary_alt = summary(alt_model)\n",
        "\n",
        "  # Get number of observations in each model\n",
        "  # nobs(null_model)\n",
        "  # nobs(alt_model)\n",
        "\n",
        "  corr_equation <- paste(response_variable, gene_var, sep = \"+\")\n",
        "  corr_formula <- reformulate(corr_equation, response = NULL)\n",
        "  pearson.cor = cor.test(formula = corr_formula, data = df, na.action = na.exclude)\n",
        "\n",
        "  output <- list(model_summary_alt, model_summary_null, pearson.cor)\n",
        "  return(output)\n",
        "}\n",
        "\n",
        "\n",
        "# Remove sibling control trios (so that the same parents are not tested twice)\n",
        "master.gen.phen.table.case = master.gen.phen.table[master.gen.phen.table$phen == \"ASD\",]\n",
        "# Also, remove any younger ASD siblings of cases so that we have each set of parents represented only once\n",
        "sort.order <- order(master.gen.phen.table.case$mother.age.at.birth.z, decreasing = FALSE)\n",
        "# Sort the trios by mother's age in ascending order so that the oldest case is on top\n",
        "master.gen.phen.table.case.unique <- master.gen.phen.table.case[sort.order, ]\n",
        "# And unique family ID\n",
        "master.gen.phen.table.case.unique <- master.gen.phen.table.case.unique[!duplicated(master.gen.phen.table.case.unique$fid), ]\n",
        "\n",
        "# Make new variables for inherited variants\n",
        "master.gen.phen.table.case.unique$father.inh = master.gen.phen.table.case.unique$InhLoF.Paternal.Transmitted + master.gen.phen.table.case.unique$InhLoF.Paternal.Untransmitted\n",
        "master.gen.phen.table.case.unique$mother.inh = master.gen.phen.table.case.unique$InhLoF.Maternal.Transmitted + master.gen.phen.table.case.unique$InhLoF.Maternal.Untransmitted\n",
        "\n",
        "# Null variables common to the parental models\n",
        "# Null model variables\n",
        "cohort_vars <- c(\"REACH_cohort+SSC_cohort+SPARK_cohort\")\n",
        "pc_vars_father <- c(\"PC1_PAT+PC2_PAT+PC3_PAT+PC4_PAT+PC5_PAT+PC6_PAT+PC7_PAT+PC8_PAT+PC9_PAT+PC10_PAT\")\n",
        "pc_vars_mother <- c(\"PC1_MAT+PC2_MAT+PC3_MAT+PC4_MAT+PC5_MAT+PC6_MAT+PC7_MAT+PC8_MAT+PC9_MAT+PC10_MAT\")\n",
        "null_vars_father <- paste(cohort_vars, pc_vars_father, sep = \"+\") \n",
        "null_vars_mother <- paste(cohort_vars, pc_vars_mother, sep = \"+\") \n",
        "\n",
        "# BAPQ\n",
        "# Father\n",
        "response_var = c(\"father.bapq.z\")\n",
        "bapq.dnlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "bapq.dnmis.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "bapq.inhlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"father.inh\", response_variable = response_var)\n",
        "bapq.ps_asd.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.ASD.Z\", response_variable = response_var)\n",
        "bapq.ps_ea.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.EA.Z\", response_variable = response_var)\n",
        "bapq.ps_sz.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.SZ.Z\", response_variable = response_var)\n",
        "# Mother\n",
        "response_var = c(\"mother.bapq.z\")\n",
        "bapq.dnlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "bapq.dnmis.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "bapq.inhlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"mother.inh\", response_variable = response_var)\n",
        "bapq.ps_asd.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.ASD.Z\", response_variable = response_var)\n",
        "bapq.ps_ea.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.EA.Z\", response_variable = response_var)\n",
        "bapq.ps_sz.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# SRS\n",
        "# Father\n",
        "response_var = c(\"father.srs.z\")\n",
        "srs.dnlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "srs.dnmis.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "srs.inhlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"father.inh\", response_variable = response_var)\n",
        "srs.ps_asd.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.ASD.Z\", response_variable = response_var)\n",
        "srs.ps_ea.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.EA.Z\", response_variable = response_var)\n",
        "srs.ps_sz.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.SZ.Z\", response_variable = response_var)\n",
        "# Mother\n",
        "response_var = c(\"mother.srs.z\")\n",
        "srs.dnlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "srs.dnmis.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "srs.inhlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"mother.inh\", response_variable = response_var)\n",
        "srs.ps_asd.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.ASD.Z\", response_variable = response_var)\n",
        "srs.ps_ea.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.EA.Z\", response_variable = response_var)\n",
        "srs.ps_sz.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# Parents' ages at birth\n",
        "# Father\n",
        "response_var = c(\"father.age.at.birth.z\")\n",
        "age_at_birth.dnlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "age_at_birth.dnmis.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "age_at_birth.inhlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"father.inh\", response_variable = response_var)\n",
        "age_at_birth.ps_asd.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.ASD.Z\", response_variable = response_var)\n",
        "age_at_birth.ps_ea.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.EA.Z\", response_variable = response_var)\n",
        "age_at_birth.ps_sz.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.SZ.Z\", response_variable = response_var)\n",
        "# Mother\n",
        "response_var = c(\"mother.age.at.birth.z\")\n",
        "age_at_birth.dnlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "age_at_birth.dnmis.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "age_at_birth.inhlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"mother.inh\", response_variable = response_var)\n",
        "age_at_birth.ps_asd.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.ASD.Z\", response_variable = response_var)\n",
        "age_at_birth.ps_ea.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.EA.Z\", response_variable = response_var)\n",
        "age_at_birth.ps_sz.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# Parents' educational attainment\n",
        "# Father\n",
        "response_var = c(\"father.EA.z\")\n",
        "ea.dnlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "ea.dnmis.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "ea.inhlof.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"father.inh\", response_variable = response_var)\n",
        "ea.ps_asd.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.ASD.Z\", response_variable = response_var)\n",
        "ea.ps_ea.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.EA.Z\", response_variable = response_var)\n",
        "ea.ps_sz.father.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_father, gene_var = \"Paternal.PS.SZ.Z\", response_variable = response_var)\n",
        "# Mother\n",
        "response_var = c(\"mother.EA.z\")\n",
        "ea.dnlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "ea.dnmis.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "ea.inhlof.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"mother.inh\", response_variable = response_var)\n",
        "ea.ps_asd.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.ASD.Z\", response_variable = response_var)\n",
        "ea.ps_ea.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.EA.Z\", response_variable = response_var)\n",
        "ea.ps_sz.mother.summary = run_lr_parental(dataframe = master.gen.phen.table.case.unique, null_variables = null_vars_mother, gene_var = \"Maternal.PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "\n",
        "# Combined parental model\n",
        "# Making father table\n",
        "master.gen.phen.table.case.unique.father = master.gen.phen.table.case.unique[, \n",
        "          c(\"fid\", \"cohort\", \"REACH_cohort\", \"SSC_cohort\", \"SPARK_cohort\", \n",
        "            \"dnLoF\", \"dnMIS\", \"father.inh\",\n",
        "            \"Paternal.PS.ASD.Z\", \"Paternal.PS.EA.Z\", \"Paternal.PS.SZ.Z\",\n",
        "            \"PC1_PAT\", \"PC2_PAT\", \"PC3_PAT\", \"PC4_PAT\", \"PC5_PAT\",\n",
        "            \"PC6_PAT\", \"PC7_PAT\", \"PC8_PAT\", \"PC9_PAT\", \"PC10_PAT\",\n",
        "            \"father.age.at.birth.z\", \"father.EA.z\", \"father.srs.z\", \"father.bapq.z\"), \n",
        "          drop = FALSE]\n",
        "master.gen.phen.table.case.unique.father$Sex = 1\n",
        "\n",
        "colnames(master.gen.phen.table.case.unique.father) <- c(\"FID\", \"Cohort\",\n",
        "              \"REACH_cohort\", \"SSC_cohort\", \"SPARK_cohort\",\n",
        "              \"dnLoF\", \"dnMis\", \"inh\",\n",
        "              \"PS.ASD.Z\", \"PS.EA.Z\", \"PS.SZ.Z\",\n",
        "              \"PC1\", \"PC2\", \"PC3\", \"PC4\", \"PC5\", \n",
        "              \"PC6\", \"PC7\", \"PC8\", \"PC9\", \"PC10\",\n",
        "              \"age.at.birth.z\", \"EA.z\", \"srs.z\", \"bapq.z\",\n",
        "              \"Sex\")\n",
        "row.names(master.gen.phen.table.case.unique.father) <- NULL\n",
        "\n",
        "# Making mother table\n",
        "master.gen.phen.table.case.unique.mother = master.gen.phen.table.case.unique[, \n",
        "          c(\"fid\", \"cohort\", \"REACH_cohort\", \"SSC_cohort\", \"SPARK_cohort\", \n",
        "            \"dnLoF\", \"dnMIS\", \"mother.inh\",\n",
        "            \"Maternal.PS.ASD.Z\", \"Maternal.PS.EA.Z\", \"Maternal.PS.SZ.Z\",\n",
        "            \"PC1_MAT\", \"PC2_MAT\", \"PC3_MAT\", \"PC4_MAT\", \"PC5_MAT\",\n",
        "            \"PC6_MAT\", \"PC7_MAT\", \"PC8_MAT\", \"PC9_MAT\", \"PC10_MAT\",\n",
        "            \"mother.age.at.birth.z\", \"mother.EA.z\", \"mother.srs.z\", \"mother.bapq.z\"), \n",
        "          drop = FALSE]\n",
        "master.gen.phen.table.case.unique.mother$Sex = 2\n",
        "\n",
        "colnames(master.gen.phen.table.case.unique.mother) <- c(\"FID\", \"Cohort\",\n",
        "              \"REACH_cohort\", \"SSC_cohort\", \"SPARK_cohort\",\n",
        "              \"dnLoF\", \"dnMis\", \"inh\",\n",
        "              \"PS.ASD.Z\", \"PS.EA.Z\", \"PS.SZ.Z\",\n",
        "              \"PC1\", \"PC2\", \"PC3\", \"PC4\", \"PC5\", \n",
        "              \"PC6\", \"PC7\", \"PC8\", \"PC9\", \"PC10\",\n",
        "              \"age.at.birth.z\", \"EA.z\", \"srs.z\", \"bapq.z\",\n",
        "              \"Sex\")\n",
        "row.names(master.gen.phen.table.case.unique.mother) <- NULL\n",
        "\n",
        "# Make the final combined parental table\n",
        "master.gen.phen.table.case.unique.combined <- rbind(master.gen.phen.table.case.unique.father, master.gen.phen.table.case.unique.mother)\n",
        "\n",
        "# Combined parental model function\n",
        "run_lr_parental_combined <- function(dataframe, cohort_vars, pc_vars, sex_var, gene_var, response_variable) {\n",
        "  df = dataframe\n",
        "\n",
        "  null_variables <- paste(cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "\n",
        "  null_formula <- reformulate(null_variables, response = response_variable)\n",
        "  null_model <- lm(formula = null_formula, data = df, na.action = na.exclude)\n",
        "  model_summary_null = summary(null_model)\n",
        "\n",
        "  gene_var <- c(gene_var)\n",
        "  alt_variables  <- paste(gene_var, null_variables, sep = \"+\")\n",
        "  alt_formula <- reformulate(alt_variables, response = response_variable)\n",
        "  alt_model  <- lm(formula = alt_formula , data = df, na.action = na.exclude)\n",
        "  model_summary_alt = summary(alt_model)\n",
        "\n",
        "  corr_equation <- paste(response_variable, gene_var, sep = \"+\")\n",
        "  corr_formula <- reformulate(corr_equation, response = NULL)\n",
        "  pearson.cor = cor.test(formula = corr_formula, data = df, na.action = na.exclude)\n",
        "\n",
        "  # Test sex interaction\n",
        "  sex_interaction_variable <- paste(gene_var, sex_var, sep = \"*\")\n",
        "  sex_interaction_terms <- paste(sex_interaction_variable, cohort_vars, pc_vars, sep = \"+\")\n",
        "  \n",
        "  sex_interaction_formula <- reformulate(sex_interaction_terms, response = response_variable)\n",
        "  sex_interaction_model <- lm(formula = sex_interaction_formula, data = df, na.action = na.exclude)\n",
        "  model_summary_sex_interaction = summary(sex_interaction_model)\n",
        "\n",
        "\n",
        "  output <- list(model_summary_alt, model_summary_null, pearson.cor, model_summary_sex_interaction)\n",
        "  return(output)\n",
        "}\n",
        "\n",
        "\n",
        "# Run the combined parental models\n",
        "# Null model variables\n",
        "cohort_vars <- c(\"REACH_cohort+SSC_cohort+SPARK_cohort\")\n",
        "pc_vars <- c(\"PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10\")\n",
        "sex_var <- c(\"Sex\") # Only use with combined models (it doesn't matter)\n",
        "\n",
        "# BAPQ models\n",
        "response_var = c(\"bapq.z\")\n",
        "bapq.dnlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "bapq.dnmis.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMis\", response_variable = response_var)\n",
        "bapq.inhlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inh\", response_variable = response_var)\n",
        "bapq.ps_asd.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "bapq.ps_ea.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "bapq.ps_sz.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# Parental Age models\n",
        "response_var = c(\"age.at.birth.z\")\n",
        "age_at_birth.dnlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "age_at_birth.dnmis.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMis\", response_variable = response_var)\n",
        "age_at_birth.inhlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inh\", response_variable = response_var)\n",
        "age_at_birth.ps_asd.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "age_at_birth.ps_ea.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "age_at_birth.ps_sz.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# Parental EA models\n",
        "response_var = c(\"EA.z\")\n",
        "ea.dnlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "ea.dnmis.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMis\", response_variable = response_var)\n",
        "ea.inhlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inh\", response_variable = response_var)\n",
        "ea.ps_asd.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "ea.ps_ea.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "ea.ps_sz.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "# SRS models\n",
        "response_var = c(\"srs.z\")\n",
        "srs.dnlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLoF\", response_variable = response_var)\n",
        "srs.dnmis.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMis\", response_variable = response_var)\n",
        "srs.inhlof.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inh\", response_variable = response_var)\n",
        "srs.ps_asd.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.ASD.Z\", response_variable = response_var)\n",
        "srs.ps_ea.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.EA.Z\", response_variable = response_var)\n",
        "srs.ps_sz.combined.summary = run_lr_parental_combined(dataframe = master.gen.phen.table.case.unique.combined, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PS.SZ.Z\", response_variable = response_var)\n",
        "\n",
        "\n",
        "\n",
        "# Put model summaries in a list\n",
        "dcdq_case_model_summaries <- list(dcdq.dnlof.case.summary, dcdq.dnmis.case.summary, dcdq.inhlof.case.summary, dcdq.ps_asd.case.summary, dcdq.ps_ea.case.summary, dcdq.ps_sz.case.summary)\n",
        "rbs_case_model_summaries <- list(rbs.dnlof.case.summary, rbs.dnmis.case.summary, rbs.inhlof.case.summary, rbs.ps_asd.case.summary, rbs.ps_ea.case.summary, rbs.ps_sz.case.summary)\n",
        "scq_case_model_summaries <- list(scq.dnlof.case.summary, scq.dnmis.case.summary, scq.inhlof.case.summary, scq.ps_asd.case.summary, scq.ps_ea.case.summary, scq.ps_sz.case.summary)\n",
        "scq_control_model_summaries <- list(scq.dnlof.control.summary, scq.dnmis.control.summary, scq.inhlof.control.summary, scq.ps_asd.control.summary, scq.ps_ea.control.summary, scq.ps_sz.control.summary)\n",
        "srs_case_model_summaries <- list(srs.dnlof.case.summary, srs.dnmis.case.summary, srs.inhlof.case.summary, srs.ps_asd.case.summary, srs.ps_ea.case.summary, srs.ps_sz.case.summary)\n",
        "srs_control_model_summaries <- list(srs.dnlof.control.summary, srs.dnmis.control.summary, srs.inhlof.control.summary, srs.ps_asd.control.summary, srs.ps_ea.control.summary, srs.ps_sz.control.summary)\n",
        "vabs_case_model_summaries <- list(vabs.dnlof.case.summary, vabs.dnmis.case.summary, vabs.inhlof.case.summary, vabs.ps_asd.case.summary, vabs.ps_ea.case.summary, vabs.ps_sz.case.summary)\n",
        "vabs_control_model_summaries <- list(vabs.dnlof.control.summary, vabs.dnmis.control.summary, vabs.inhlof.control.summary, vabs.ps_asd.control.summary, vabs.ps_ea.control.summary, vabs.ps_sz.control.summary)\n",
        "\n",
        "# Combine model outputs to get summary statistics and put in a table\n",
        "model_dict <- ordered_dict(items = NULL, keys = NULL)\n",
        "\n",
        "model_dict$set(\"DCDQ_case\", dcdq_case_model_summaries)\n",
        "model_dict$set(\"RBS_case\", rbs_case_model_summaries)\n",
        "model_dict$set(\"SCQ_case\", scq_case_model_summaries)\n",
        "model_dict$set(\"SCQ_cont\", scq_control_model_summaries)\n",
        "model_dict$set(\"SRS_case\", srs_case_model_summaries)\n",
        "model_dict$set(\"SRS_cont\", srs_control_model_summaries)\n",
        "model_dict$set(\"VABS_case\", vabs_case_model_summaries)\n",
        "model_dict$set(\"VABS_cont\", vabs_control_model_summaries)\n",
        "\n",
        "keys = list(\"DCDQ_case\", \"RBS_case\", \"SCQ_case\", \"SCQ_cont\", \"SRS_case\", \"SRS_cont\", \"VABS_case\", \"VABS_cont\")\n",
        "\n",
        "clinical_results_table_offspring <- data.frame()\n",
        "\n",
        "for (key in keys) {\n",
        "\n",
        "  phenotype = str_split(key, \"_\")[[1]][1]\n",
        "  case.status = str_split(key, \"_\")[[1]][2]\n",
        "\n",
        "  model_summary_list = model_dict$get(key)\n",
        "  for (model_summaries in model_summary_list) {\n",
        "    # Combined\n",
        "    model_summary_alt  = model_summaries[[1]]\n",
        "    model_summary_null = model_summaries[[2]]\n",
        "\n",
        "    x = rownames(coefficients(model_summary_alt))[2]\n",
        "    x_estimate = signif(coefficients(model_summary_alt)[2, 1], digits = 2)\n",
        "    x_stderr = signif(coefficients(model_summary_alt)[2, 2], digits = 2)\n",
        "    x_pval = signif(coefficients(model_summary_alt)[2, 4], digits = 2)\n",
        "    r2 = signif(model_summary_alt$r.squared - model_summary_null$r.squared, digits = 2)\n",
        "\n",
        "    # Correlation\n",
        "    pearson.cor = model_summaries[[3]]\n",
        "    pearson.cc = signif(pearson.cor$estimate, digits = 2)\n",
        "\n",
        "    # Sex interaction\n",
        "    model_summary_sex_interaction = model_summaries[[4]]\n",
        "    sex_interaction_model_coeffs = coefficients(model_summary_sex_interaction)\n",
        "    num_rows = dim(sex_interaction_model_coeffs)[1]\n",
        "    sex_interaction_coeff = sex_interaction_model_coeffs[num_rows, ]\n",
        "    sex_interaction_pval = signif(sex_interaction_coeff[[4]], digits = 2)\n",
        "\n",
        "    # Males\n",
        "    model_summary_alt_male    = model_summaries[[5]]\n",
        "    model_summary_null_male   = model_summaries[[6]]\n",
        "    pearson.cor.male = model_summaries[[7]]\n",
        "    pearson.cc.male = signif(pearson.cor.male$estimate, digits = 2)\n",
        "\n",
        "\n",
        "    x_estimate_male = signif(coefficients(model_summary_alt_male)[2, 1], digits = 2)\n",
        "    x_stderr_male = signif(coefficients(model_summary_alt_male)[2, 2], digits = 2)\n",
        "    x_pval_male = signif(coefficients(model_summary_alt_male)[2, 4], digits = 2)\n",
        "    r2_male = signif(model_summary_alt_male$r.squared - model_summary_null_male$r.squared, digits = 2)\n",
        "\n",
        "    # Females\n",
        "    model_summary_alt_female  = model_summaries[[8]]\n",
        "    model_summary_null_female = model_summaries[[9]]\n",
        "    pearson.cor.female = model_summaries[[10]]\n",
        "    pearson.cc.female = signif(pearson.cor.female$estimate, digits = 2)\n",
        "\n",
        "    x_estimate_female = signif(coefficients(model_summary_alt_female)[2, 1], digits = 2)\n",
        "    x_stderr_female = signif(coefficients(model_summary_alt_female)[2, 2], digits = 2)\n",
        "    x_pval_female = signif(coefficients(model_summary_alt_female)[2, 4], digits = 2)\n",
        "    r2_female = signif(model_summary_alt_female$r.squared - model_summary_null_female$r.squared, digits = 2)\n",
        "\n",
        "    df_tmp_female = data.frame(case.status, phenotype, \"female\",\n",
        "                  x, r2_female, x_estimate_female, x_stderr_female, \n",
        "                  pearson.cc.female, x_pval_female, NA)\n",
        "    \n",
        "    df_tmp_male = data.frame(case.status, phenotype, \"male\",\n",
        "                  x, r2_male, x_estimate_male, x_stderr_male, \n",
        "                  pearson.cc.male, x_pval_male, NA)\n",
        "\n",
        "    df_tmp_combined = data.frame(case.status, phenotype, \"combined\",\n",
        "                  x, r2, x_estimate, x_stderr, \n",
        "                  pearson.cc, x_pval, sex_interaction_pval)\n",
        "\n",
        "    colnames(df_tmp_female) = colnames(df_tmp_male) = colnames(df_tmp_combined)\n",
        "    df_tmp = rbind(df_tmp_combined, df_tmp_male, df_tmp_female)\n",
        "\n",
        "    clinical_results_table_offspring = rbind(clinical_results_table_offspring, df_tmp)\n",
        "  }\n",
        "}\n",
        "\n",
        "row.names(clinical_results_table_offspring) <- NULL\n",
        "\n",
        "# Put model summaries in a list\n",
        "bapq_father_model_summaries <- list(bapq.dnlof.father.summary, bapq.dnmis.father.summary, bapq.inhlof.father.summary, bapq.ps_asd.father.summary, bapq.ps_ea.father.summary, bapq.ps_sz.father.summary)\n",
        "bapq_mother_model_summaries <- list(bapq.dnlof.mother.summary, bapq.dnmis.mother.summary, bapq.inhlof.mother.summary, bapq.ps_asd.mother.summary, bapq.ps_ea.mother.summary, bapq.ps_sz.mother.summary)\n",
        "srs_father_model_summaries <- list(srs.dnlof.father.summary, srs.dnmis.father.summary, srs.inhlof.father.summary, srs.ps_asd.father.summary, srs.ps_ea.father.summary, srs.ps_sz.father.summary)\n",
        "srs_mother_model_summaries <- list(srs.dnlof.mother.summary, srs.dnmis.mother.summary, srs.inhlof.mother.summary, srs.ps_asd.mother.summary, srs.ps_ea.mother.summary, srs.ps_sz.mother.summary)\n",
        "parental.age_father_model_summaries <- list(age_at_birth.dnlof.father.summary, age_at_birth.dnmis.father.summary, age_at_birth.inhlof.father.summary, age_at_birth.ps_asd.father.summary, age_at_birth.ps_ea.father.summary, age_at_birth.ps_sz.father.summary)\n",
        "parental.age_mother_model_summaries <- list(age_at_birth.dnlof.mother.summary, age_at_birth.dnmis.mother.summary, age_at_birth.inhlof.mother.summary, age_at_birth.ps_asd.mother.summary, age_at_birth.ps_ea.mother.summary, age_at_birth.ps_sz.mother.summary)\n",
        "parental.EA_father_model_summaries <- list(ea.dnlof.father.summary, ea.dnmis.father.summary, ea.inhlof.father.summary, ea.ps_asd.father.summary, ea.ps_ea.father.summary, ea.ps_sz.father.summary)\n",
        "parental.EA_mother_model_summaries <- list(ea.dnlof.mother.summary, ea.dnmis.mother.summary, ea.inhlof.mother.summary, ea.ps_asd.mother.summary, ea.ps_ea.mother.summary, ea.ps_sz.mother.summary)\n",
        "\n",
        "# Combine model outputs to get summary statistics and put in a table\n",
        "model_dict_parents <- ordered_dict(items = NULL, keys = NULL)\n",
        "\n",
        "model_dict_parents$set(\"BAPQ_father\", bapq_father_model_summaries)\n",
        "model_dict_parents$set(\"BAPQ_mother\", bapq_mother_model_summaries)\n",
        "model_dict_parents$set(\"SRS_father\", srs_father_model_summaries)\n",
        "model_dict_parents$set(\"SRS_mother\", srs_mother_model_summaries)\n",
        "model_dict_parents$set(\"Parental Age_father\", parental.age_father_model_summaries)\n",
        "model_dict_parents$set(\"Parental Age_mother\", parental.age_mother_model_summaries)\n",
        "model_dict_parents$set(\"Parental EA_father\", parental.EA_father_model_summaries)\n",
        "model_dict_parents$set(\"Parental EA_mother\", parental.EA_mother_model_summaries)\n",
        "\n",
        "keys_parents = list(\"BAPQ_father\", \"BAPQ_mother\", \"SRS_father\", \"SRS_mother\", \"Parental Age_father\", \"Parental Age_mother\", \"Parental EA_father\", \"Parental EA_mother\")\n",
        "\n",
        "clinical_results_table_parents <- data.frame()\n",
        "\n",
        "for (key in keys_parents) {\n",
        "\n",
        "  phenotype = str_split(key, \"_\")[[1]][1]\n",
        "  group = str_split(key, \"_\")[[1]][2]\n",
        "\n",
        "  case.status = \"parent\"\n",
        "\n",
        "  model_summary_list = model_dict_parents$get(key)\n",
        "  for (model_summaries in model_summary_list) {\n",
        "    # Combined\n",
        "    model_summary_alt  = model_summaries[[1]]\n",
        "    model_summary_null = model_summaries[[2]]\n",
        "\n",
        "    x = rownames(coefficients(model_summary_alt))[2]\n",
        "    x_estimate = signif(coefficients(model_summary_alt)[2, 1], digits = 2)\n",
        "    x_stderr = signif(coefficients(model_summary_alt)[2, 2], digits = 2)\n",
        "    x_pval = signif(coefficients(model_summary_alt)[2, 4], digits = 2)\n",
        "    r2 = signif(model_summary_alt$r.squared - model_summary_null$r.squared, digits = 2)\n",
        "\n",
        "    # Correlation\n",
        "    pearson.cor = model_summaries[[3]]\n",
        "    pearson.cc = signif(pearson.cor$estimate, digits = 2)\n",
        "\n",
        "    df_tmp = data.frame(case.status, phenotype, group,\n",
        "                  x, r2, x_estimate, x_stderr, \n",
        "                  NA, x_pval, NA)\n",
        "\n",
        "    clinical_results_table_parents = rbind(clinical_results_table_parents, df_tmp)\n",
        "  }\n",
        "}\n",
        "\n",
        "row.names(clinical_results_table_parents) <- NULL\n",
        "\n",
        "# Put combined parents model summaries in a list\n",
        "bapq_parents_model_summaries <- list(bapq.dnlof.combined.summary, bapq.dnmis.combined.summary, bapq.inhlof.combined.summary, bapq.ps_asd.combined.summary, bapq.ps_ea.combined.summary, bapq.ps_sz.combined.summary)\n",
        "parental.age_parents_model_summaries <- list(age_at_birth.dnlof.combined.summary, age_at_birth.dnmis.combined.summary, age_at_birth.inhlof.combined.summary, age_at_birth.ps_asd.combined.summary, age_at_birth.ps_ea.combined.summary, age_at_birth.ps_sz.combined.summary)\n",
        "parental.EA_parents_model_summaries <- list(ea.dnlof.combined.summary, ea.dnmis.combined.summary, ea.inhlof.combined.summary, ea.ps_asd.combined.summary, ea.ps_ea.combined.summary, ea.ps_sz.combined.summary)\n",
        "SRS_parents_model_summaries <- list(srs.dnlof.combined.summary, srs.dnmis.combined.summary, srs.inhlof.combined.summary, srs.ps_asd.combined.summary, srs.ps_ea.combined.summary, srs.ps_sz.combined.summary)\n",
        "\n",
        "# Combine model outputs to get summary statistics and put in a table\n",
        "model_dict_combined <- ordered_dict(items = NULL, keys = NULL)\n",
        "\n",
        "model_dict_combined$set(\"BAPQ_parents\", bapq_parents_model_summaries)\n",
        "model_dict_combined$set(\"Parental Age_parents\", parental.age_parents_model_summaries)\n",
        "model_dict_combined$set(\"Parental EA_parents\", parental.EA_parents_model_summaries)\n",
        "model_dict_combined$set(\"SRS_parents\", SRS_parents_model_summaries)\n",
        "\n",
        "keys_combined = list(\"BAPQ_parents\", \"Parental Age_parents\", \"Parental EA_parents\", \"SRS_parents\")\n",
        "\n",
        "clinical_results_table_combined <- data.frame()\n",
        "\n",
        "for (key in keys_combined) {\n",
        "\n",
        "  phenotype = str_split(key, \"_\")[[1]][1]\n",
        "  group = str_split(key, \"_\")[[1]][2]\n",
        "\n",
        "  case.status = \"parent\"\n",
        "\n",
        "  model_summary_list = model_dict_combined$get(key)\n",
        "  for (model_summaries in model_summary_list) {\n",
        "    # Combined\n",
        "    model_summary_alt  = model_summaries[[1]]\n",
        "    model_summary_null = model_summaries[[2]]\n",
        "\n",
        "    x = rownames(coefficients(model_summary_alt))[2]\n",
        "    x_estimate = signif(coefficients(model_summary_alt)[2, 1], digits = 2)\n",
        "    x_stderr = signif(coefficients(model_summary_alt)[2, 2], digits = 2)\n",
        "    x_pval = signif(coefficients(model_summary_alt)[2, 4], digits = 2)\n",
        "    r2 = signif(model_summary_alt$r.squared - model_summary_null$r.squared, digits = 2)\n",
        "\n",
        "    # Correlation\n",
        "    pearson.cor = model_summaries[[3]]\n",
        "    pearson.cc = signif(pearson.cor$estimate, digits = 2)\n",
        "\n",
        "    # Sex interaction\n",
        "    model_summary_sex_interaction = model_summaries[[4]]\n",
        "    sex_interaction_model_coeffs = coefficients(model_summary_sex_interaction)\n",
        "    num_rows = dim(sex_interaction_model_coeffs)[1]\n",
        "    sex_interaction_coeff = sex_interaction_model_coeffs[num_rows, ]\n",
        "    sex_interaction_pval = signif(sex_interaction_coeff[[4]], digits = 2)\n",
        "\n",
        "    df_tmp = data.frame(case.status, phenotype, group,\n",
        "                  x, r2, x_estimate, x_stderr, \n",
        "                  pearson.cc, x_pval, sex_interaction_pval)\n",
        "\n",
        "    clinical_results_table_combined = rbind(clinical_results_table_combined, df_tmp)\n",
        "  }\n",
        "}\n",
        "\n",
        "row.names(clinical_results_table_combined) <- NULL\n",
        "\n",
        "colnames(clinical_results_table_offspring) = colnames(clinical_results_table_parents) = colnames(clinical_results_table_combined)\n",
        "df_clinical_results = rbind(clinical_results_table_offspring, clinical_results_table_parents, clinical_results_table_combined)\n",
        "\n",
        "df_clinical_results"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "    case.status phenotype    group    x                 r2       x_estimate\n",
              "1   case        DCDQ         combined dnLoF              1.1e-02 -0.42000  \n",
              "2   case        DCDQ         male     dnLoF              6.6e-03 -0.36000  \n",
              "3   case        DCDQ         female   dnLoF              3.3e-02 -0.52000  \n",
              "4   case        DCDQ         combined dnMIS              5.3e-03 -0.35000  \n",
              "5   case        DCDQ         male     dnMIS              3.5e-03 -0.33000  \n",
              "6   case        DCDQ         female   dnMIS              2.2e-02 -0.39000  \n",
              "7   case        DCDQ         combined inhLoF             1.0e-03 -0.04300  \n",
              "8   case        DCDQ         male     inhLoF             1.2e-03 -0.04500  \n",
              "9   case        DCDQ         female   inhLoF             4.1e-04 -0.02900  \n",
              "10  case        DCDQ         combined PS.ASD.Z          -1.7e-03  0.00094  \n",
              "11  case        DCDQ         male     PS.ASD.Z          -2.7e-03 -0.00820  \n",
              "12  case        DCDQ         female   PS.ASD.Z           1.5e-02  0.05200  \n",
              "13  case        DCDQ         combined PS.EA.Z           -1.7e-03 -0.00840  \n",
              "14  case        DCDQ         male     PS.EA.Z           -2.2e-03 -0.02500  \n",
              "15  case        DCDQ         female   PS.EA.Z            1.5e-02  0.05700  \n",
              "16  case        DCDQ         combined PS.SZ.Z           -1.7e-03  0.00570  \n",
              "17  case        DCDQ         male     PS.SZ.Z           -2.7e-03  0.00830  \n",
              "18  case        DCDQ         female   PS.SZ.Z            1.2e-02 -0.01200  \n",
              "19  case        RBS          combined dnLoF              5.2e-03  0.02800  \n",
              "20  case        RBS          male     dnLoF              6.8e-03  0.01900  \n",
              "21  case        RBS          female   dnLoF              2.1e-03  0.04300  \n",
              "22  case        RBS          combined dnMIS              5.2e-03 -0.03200  \n",
              "23  case        RBS          male     dnMIS              6.9e-03 -0.05100  \n",
              "24  case        RBS          female   dnMIS              2.0e-03  0.03300  \n",
              "25  case        RBS          combined inhLoF             3.3e-05  0.00770  \n",
              "26  case        RBS          male     inhLoF             1.2e-04  0.01500  \n",
              "27  case        RBS          female   inhLoF             2.7e-04 -0.02500  \n",
              "28  case        RBS          combined PS.ASD.Z           6.9e-03  0.00250  \n",
              "29  case        RBS          male     PS.ASD.Z           8.3e-03  0.01400  \n",
              "30  case        RBS          female   PS.ASD.Z           4.7e-03 -0.05400  \n",
              "⋮   ⋮           ⋮            ⋮        ⋮                 ⋮        ⋮         \n",
              "187 parent      Parental EA  mother   dnLoF             6.2e-04   0.0880   \n",
              "188 parent      Parental EA  mother   dnMIS             3.2e-04   0.0720   \n",
              "189 parent      Parental EA  mother   mother.inh        6.0e-05  -0.0093   \n",
              "190 parent      Parental EA  mother   Maternal.PS.ASD.Z 6.9e-04  -0.0250   \n",
              "191 parent      Parental EA  mother   Maternal.PS.EA.Z  7.1e-02   0.2600   \n",
              "192 parent      Parental EA  mother   Maternal.PS.SZ.Z  8.1e-05  -0.0087   \n",
              "193 parent      BAPQ         parents  dnLoF             2.2e-03  -0.1600   \n",
              "194 parent      BAPQ         parents  dnMis             1.2e-04  -0.0410   \n",
              "195 parent      BAPQ         parents  inh               4.6e-05   0.0084   \n",
              "196 parent      BAPQ         parents  PS.ASD.Z          1.7e-03   0.0430   \n",
              "197 parent      BAPQ         parents  PS.EA.Z           1.4e-03  -0.0390   \n",
              "198 parent      BAPQ         parents  PS.SZ.Z           3.7e-04   0.0200   \n",
              "199 parent      Parental Age parents  dnLoF             2.7e-05   0.0190   \n",
              "200 parent      Parental Age parents  dnMis             5.1e-04   0.0940   \n",
              "201 parent      Parental Age parents  inh               3.9e-04  -0.0260   \n",
              "202 parent      Parental Age parents  PS.ASD.Z          1.5e-03   0.0075   \n",
              "203 parent      Parental Age parents  PS.EA.Z           2.0e-02   0.1400   \n",
              "204 parent      Parental Age parents  PS.SZ.Z           1.8e-03   0.0190   \n",
              "205 parent      Parental EA  parents  dnLoF             8.9e-04   0.1100   \n",
              "206 parent      Parental EA  parents  dnMis             1.8e-04   0.0640   \n",
              "207 parent      Parental EA  parents  inh               3.5e-04  -0.0220   \n",
              "208 parent      Parental EA  parents  PS.ASD.Z          2.7e-04  -0.0060   \n",
              "209 parent      Parental EA  parents  PS.EA.Z           7.9e-02   0.2600   \n",
              "210 parent      Parental EA  parents  PS.SZ.Z           3.8e-04  -0.0120   \n",
              "211 parent      SRS          parents  dnLoF             1.4e-04  -0.0380   \n",
              "212 parent      SRS          parents  dnMis             1.8e-03  -0.1900   \n",
              "213 parent      SRS          parents  inh               8.9e-04   0.0360   \n",
              "214 parent      SRS          parents  PS.ASD.Z          5.5e-03   0.0750   \n",
              "215 parent      SRS          parents  PS.EA.Z           1.5e-03  -0.0390   \n",
              "216 parent      SRS          parents  PS.SZ.Z           1.8e-03   0.0440   \n",
              "    x_stderr pearson.cc x_pval   sex_interaction_pval\n",
              "1   0.065    -0.11000   1.5e-10  0.074               \n",
              "2   0.073    -0.09400   9.8e-07     NA               \n",
              "3   0.140    -0.18000   3.7e-04     NA               \n",
              "4   0.074    -0.08100   3.4e-06  0.640               \n",
              "5   0.084    -0.07300   9.2e-05     NA               \n",
              "6   0.160    -0.11000   1.7e-02     NA               \n",
              "7   0.021    -0.03400   3.9e-02  0.910               \n",
              "8   0.022    -0.03600   4.2e-02     NA               \n",
              "9   0.052    -0.02800   5.7e-01     NA               \n",
              "10  0.018    -0.00048   9.6e-01  0.220               \n",
              "11  0.020    -0.01000   6.8e-01     NA               \n",
              "12  0.044     0.04400   2.4e-01     NA               \n",
              "13  0.018    -0.01300   6.5e-01  0.059               \n",
              "14  0.020    -0.02900   2.2e-01     NA               \n",
              "15  0.044     0.06300   1.9e-01     NA               \n",
              "16  0.018     0.00410   7.6e-01  0.840               \n",
              "17  0.020     0.00460   6.8e-01     NA               \n",
              "18  0.045    -0.00700   7.9e-01     NA               \n",
              "19  0.064     0.00700   6.7e-01  0.780               \n",
              "20  0.072     0.01300   8.0e-01     NA               \n",
              "21  0.140    -0.01200   7.6e-01     NA               \n",
              "22  0.071    -0.01500   6.5e-01  0.600               \n",
              "23  0.083    -0.01900   5.4e-01     NA               \n",
              "24  0.140     0.00650   8.2e-01     NA               \n",
              "25  0.020     0.00460   7.0e-01  0.370               \n",
              "26  0.022     0.00960   5.0e-01     NA               \n",
              "27  0.051    -0.01900   6.3e-01     NA               \n",
              "28  0.017     0.00210   8.8e-01  0.150               \n",
              "29  0.019     0.00960   4.6e-01     NA               \n",
              "30  0.041    -0.03700   1.8e-01     NA               \n",
              "⋮   ⋮        ⋮          ⋮        ⋮                   \n",
              "187 0.058          NA    1.3e-01    NA               \n",
              "188 0.066          NA    2.7e-01    NA               \n",
              "189 0.020          NA    6.4e-01    NA               \n",
              "190 0.016          NA    1.1e-01    NA               \n",
              "191 0.015          NA    6.2e-62    NA               \n",
              "192 0.016          NA    5.8e-01    NA               \n",
              "193 0.056    -0.04400    5.9e-03 0.430               \n",
              "194 0.064    -0.00910    5.2e-01 0.900               \n",
              "195 0.021     0.00360    6.9e-01 0.440               \n",
              "196 0.017     0.01500    1.3e-02 0.110               \n",
              "197 0.017    -0.05500    2.6e-02 0.160               \n",
              "198 0.017    -0.01500    2.5e-01 0.350               \n",
              "199 0.038     0.00720    6.1e-01 0.780               \n",
              "200 0.042     0.02900    2.5e-02 0.420               \n",
              "201 0.013    -0.01600    5.1e-02 0.007               \n",
              "202 0.010     0.02500    4.7e-01 0.610               \n",
              "203 0.010     0.16000    3.6e-40 0.043               \n",
              "204 0.011     0.04300    6.7e-02 0.210               \n",
              "205 0.039     0.03700    6.3e-03 0.660               \n",
              "206 0.045     0.02200    1.6e-01 0.940               \n",
              "207 0.014     0.03500    1.0e-01 0.400               \n",
              "208 0.011     0.00440    5.8e-01 0.096               \n",
              "209 0.010     0.31000   8.1e-137 0.270               \n",
              "210 0.011    -0.00081    2.9e-01 0.720               \n",
              "211 0.076    -0.00880    6.2e-01 0.530               \n",
              "212 0.110    -0.04100    7.6e-02 0.790               \n",
              "213 0.028     0.02900    2.1e-01 0.490               \n",
              "214 0.024     0.06400    1.7e-03 0.670               \n",
              "215 0.024    -0.05000    1.1e-01 0.200               \n",
              "216 0.024     0.03800    7.3e-02 0.630               "
            ],
            "text/latex": "A data.frame: 216 × 10\n\\begin{tabular}{llllllllll}\n case.status & phenotype & group & x & r2 & x\\_estimate & x\\_stderr & pearson.cc & x\\_pval & sex\\_interaction\\_pval\\\\\n <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n\\hline\n\t case & DCDQ & combined & dnLoF    &  1.1e-02 & -0.42000 & 0.065 & -0.11000 & 1.5e-10 & 0.074\\\\\n\t case & DCDQ & male     & dnLoF    &  6.6e-03 & -0.36000 & 0.073 & -0.09400 & 9.8e-07 &    NA\\\\\n\t case & DCDQ & female   & dnLoF    &  3.3e-02 & -0.52000 & 0.140 & -0.18000 & 3.7e-04 &    NA\\\\\n\t case & DCDQ & combined & dnMIS    &  5.3e-03 & -0.35000 & 0.074 & -0.08100 & 3.4e-06 & 0.640\\\\\n\t case & DCDQ & male     & dnMIS    &  3.5e-03 & -0.33000 & 0.084 & -0.07300 & 9.2e-05 &    NA\\\\\n\t case & DCDQ & female   & dnMIS    &  2.2e-02 & -0.39000 & 0.160 & -0.11000 & 1.7e-02 &    NA\\\\\n\t case & DCDQ & combined & inhLoF   &  1.0e-03 & -0.04300 & 0.021 & -0.03400 & 3.9e-02 & 0.910\\\\\n\t case & DCDQ & male     & inhLoF   &  1.2e-03 & -0.04500 & 0.022 & -0.03600 & 4.2e-02 &    NA\\\\\n\t case & DCDQ & female   & inhLoF   &  4.1e-04 & -0.02900 & 0.052 & -0.02800 & 5.7e-01 &    NA\\\\\n\t case & DCDQ & combined & PS.ASD.Z & -1.7e-03 &  0.00094 & 0.018 & -0.00048 & 9.6e-01 & 0.220\\\\\n\t case & DCDQ & male     & PS.ASD.Z & -2.7e-03 & -0.00820 & 0.020 & -0.01000 & 6.8e-01 &    NA\\\\\n\t case & DCDQ & female   & PS.ASD.Z &  1.5e-02 &  0.05200 & 0.044 &  0.04400 & 2.4e-01 &    NA\\\\\n\t case & DCDQ & combined & PS.EA.Z  & -1.7e-03 & -0.00840 & 0.018 & -0.01300 & 6.5e-01 & 0.059\\\\\n\t case & DCDQ & male     & PS.EA.Z  & -2.2e-03 & -0.02500 & 0.020 & -0.02900 & 2.2e-01 &    NA\\\\\n\t case & DCDQ & female   & PS.EA.Z  &  1.5e-02 &  0.05700 & 0.044 &  0.06300 & 1.9e-01 &    NA\\\\\n\t case & DCDQ & combined & PS.SZ.Z  & -1.7e-03 &  0.00570 & 0.018 &  0.00410 & 7.6e-01 & 0.840\\\\\n\t case & DCDQ & male     & PS.SZ.Z  & -2.7e-03 &  0.00830 & 0.020 &  0.00460 & 6.8e-01 &    NA\\\\\n\t case & DCDQ & female   & PS.SZ.Z  &  1.2e-02 & -0.01200 & 0.045 & -0.00700 & 7.9e-01 &    NA\\\\\n\t case & RBS  & combined & dnLoF    &  5.2e-03 &  0.02800 & 0.064 &  0.00700 & 6.7e-01 & 0.780\\\\\n\t case & RBS  & male     & dnLoF    &  6.8e-03 &  0.01900 & 0.072 &  0.01300 & 8.0e-01 &    NA\\\\\n\t case & RBS  & female   & dnLoF    &  2.1e-03 &  0.04300 & 0.140 & -0.01200 & 7.6e-01 &    NA\\\\\n\t case & RBS  & combined & dnMIS    &  5.2e-03 & -0.03200 & 0.071 & -0.01500 & 6.5e-01 & 0.600\\\\\n\t case & RBS  & male     & dnMIS    &  6.9e-03 & -0.05100 & 0.083 & -0.01900 & 5.4e-01 &    NA\\\\\n\t case & RBS  & female   & dnMIS    &  2.0e-03 &  0.03300 & 0.140 &  0.00650 & 8.2e-01 &    NA\\\\\n\t case & RBS  & combined & inhLoF   &  3.3e-05 &  0.00770 & 0.020 &  0.00460 & 7.0e-01 & 0.370\\\\\n\t case & RBS  & male     & inhLoF   &  1.2e-04 &  0.01500 & 0.022 &  0.00960 & 5.0e-01 &    NA\\\\\n\t case & RBS  & female   & inhLoF   &  2.7e-04 & -0.02500 & 0.051 & -0.01900 & 6.3e-01 &    NA\\\\\n\t case & RBS  & combined & PS.ASD.Z &  6.9e-03 &  0.00250 & 0.017 &  0.00210 & 8.8e-01 & 0.150\\\\\n\t case & RBS  & male     & PS.ASD.Z &  8.3e-03 &  0.01400 & 0.019 &  0.00960 & 4.6e-01 &    NA\\\\\n\t case & RBS  & female   & PS.ASD.Z &  4.7e-03 & -0.05400 & 0.041 & -0.03700 & 1.8e-01 &    NA\\\\\n\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n\t parent & Parental EA  & mother  & dnLoF             & 6.2e-04 &  0.0880 & 0.058 &       NA &  1.3e-01 &    NA\\\\\n\t parent & Parental EA  & mother  & dnMIS             & 3.2e-04 &  0.0720 & 0.066 &       NA &  2.7e-01 &    NA\\\\\n\t parent & Parental EA  & mother  & mother.inh        & 6.0e-05 & -0.0093 & 0.020 &       NA &  6.4e-01 &    NA\\\\\n\t parent & Parental EA  & mother  & Maternal.PS.ASD.Z & 6.9e-04 & -0.0250 & 0.016 &       NA &  1.1e-01 &    NA\\\\\n\t parent & Parental EA  & mother  & Maternal.PS.EA.Z  & 7.1e-02 &  0.2600 & 0.015 &       NA &  6.2e-62 &    NA\\\\\n\t parent & Parental EA  & mother  & Maternal.PS.SZ.Z  & 8.1e-05 & -0.0087 & 0.016 &       NA &  5.8e-01 &    NA\\\\\n\t parent & BAPQ         & parents & dnLoF             & 2.2e-03 & -0.1600 & 0.056 & -0.04400 &  5.9e-03 & 0.430\\\\\n\t parent & BAPQ         & parents & dnMis             & 1.2e-04 & -0.0410 & 0.064 & -0.00910 &  5.2e-01 & 0.900\\\\\n\t parent & BAPQ         & parents & inh               & 4.6e-05 &  0.0084 & 0.021 &  0.00360 &  6.9e-01 & 0.440\\\\\n\t parent & BAPQ         & parents & PS.ASD.Z          & 1.7e-03 &  0.0430 & 0.017 &  0.01500 &  1.3e-02 & 0.110\\\\\n\t parent & BAPQ         & parents & PS.EA.Z           & 1.4e-03 & -0.0390 & 0.017 & -0.05500 &  2.6e-02 & 0.160\\\\\n\t parent & BAPQ         & parents & PS.SZ.Z           & 3.7e-04 &  0.0200 & 0.017 & -0.01500 &  2.5e-01 & 0.350\\\\\n\t parent & Parental Age & parents & dnLoF             & 2.7e-05 &  0.0190 & 0.038 &  0.00720 &  6.1e-01 & 0.780\\\\\n\t parent & Parental Age & parents & dnMis             & 5.1e-04 &  0.0940 & 0.042 &  0.02900 &  2.5e-02 & 0.420\\\\\n\t parent & Parental Age & parents & inh               & 3.9e-04 & -0.0260 & 0.013 & -0.01600 &  5.1e-02 & 0.007\\\\\n\t parent & Parental Age & parents & PS.ASD.Z          & 1.5e-03 &  0.0075 & 0.010 &  0.02500 &  4.7e-01 & 0.610\\\\\n\t parent & Parental Age & parents & PS.EA.Z           & 2.0e-02 &  0.1400 & 0.010 &  0.16000 &  3.6e-40 & 0.043\\\\\n\t parent & Parental Age & parents & PS.SZ.Z           & 1.8e-03 &  0.0190 & 0.011 &  0.04300 &  6.7e-02 & 0.210\\\\\n\t parent & Parental EA  & parents & dnLoF             & 8.9e-04 &  0.1100 & 0.039 &  0.03700 &  6.3e-03 & 0.660\\\\\n\t parent & Parental EA  & parents & dnMis             & 1.8e-04 &  0.0640 & 0.045 &  0.02200 &  1.6e-01 & 0.940\\\\\n\t parent & Parental EA  & parents & inh               & 3.5e-04 & -0.0220 & 0.014 &  0.03500 &  1.0e-01 & 0.400\\\\\n\t parent & Parental EA  & parents & PS.ASD.Z          & 2.7e-04 & -0.0060 & 0.011 &  0.00440 &  5.8e-01 & 0.096\\\\\n\t parent & Parental EA  & parents & PS.EA.Z           & 7.9e-02 &  0.2600 & 0.010 &  0.31000 & 8.1e-137 & 0.270\\\\\n\t parent & Parental EA  & parents & PS.SZ.Z           & 3.8e-04 & -0.0120 & 0.011 & -0.00081 &  2.9e-01 & 0.720\\\\\n\t parent & SRS          & parents & dnLoF             & 1.4e-04 & -0.0380 & 0.076 & -0.00880 &  6.2e-01 & 0.530\\\\\n\t parent & SRS          & parents & dnMis             & 1.8e-03 & -0.1900 & 0.110 & -0.04100 &  7.6e-02 & 0.790\\\\\n\t parent & SRS          & parents & inh               & 8.9e-04 &  0.0360 & 0.028 &  0.02900 &  2.1e-01 & 0.490\\\\\n\t parent & SRS          & parents & PS.ASD.Z          & 5.5e-03 &  0.0750 & 0.024 &  0.06400 &  1.7e-03 & 0.670\\\\\n\t parent & SRS          & parents & PS.EA.Z           & 1.5e-03 & -0.0390 & 0.024 & -0.05000 &  1.1e-01 & 0.200\\\\\n\t parent & SRS          & parents & PS.SZ.Z           & 1.8e-03 &  0.0440 & 0.024 &  0.03800 &  7.3e-02 & 0.630\\\\\n\\end{tabular}\n",
            "text/markdown": "\nA data.frame: 216 × 10\n\n| case.status &lt;chr&gt; | phenotype &lt;chr&gt; | group &lt;chr&gt; | x &lt;chr&gt; | r2 &lt;dbl&gt; | x_estimate &lt;dbl&gt; | x_stderr &lt;dbl&gt; | pearson.cc &lt;dbl&gt; | x_pval &lt;dbl&gt; | sex_interaction_pval &lt;dbl&gt; |\n|---|---|---|---|---|---|---|---|---|---|\n| case | DCDQ | combined | dnLoF    |  1.1e-02 | -0.42000 | 0.065 | -0.11000 | 1.5e-10 | 0.074 |\n| case | DCDQ | male     | dnLoF    |  6.6e-03 | -0.36000 | 0.073 | -0.09400 | 9.8e-07 |    NA |\n| case | DCDQ | female   | dnLoF    |  3.3e-02 | -0.52000 | 0.140 | -0.18000 | 3.7e-04 |    NA |\n| case | DCDQ | combined | dnMIS    |  5.3e-03 | -0.35000 | 0.074 | -0.08100 | 3.4e-06 | 0.640 |\n| case | DCDQ | male     | dnMIS    |  3.5e-03 | -0.33000 | 0.084 | -0.07300 | 9.2e-05 |    NA |\n| case | DCDQ | female   | dnMIS    |  2.2e-02 | -0.39000 | 0.160 | -0.11000 | 1.7e-02 |    NA |\n| case | DCDQ | combined | inhLoF   |  1.0e-03 | -0.04300 | 0.021 | -0.03400 | 3.9e-02 | 0.910 |\n| case | DCDQ | male     | inhLoF   |  1.2e-03 | -0.04500 | 0.022 | -0.03600 | 4.2e-02 |    NA |\n| case | DCDQ | female   | inhLoF   |  4.1e-04 | -0.02900 | 0.052 | -0.02800 | 5.7e-01 |    NA |\n| case | DCDQ | combined | PS.ASD.Z | -1.7e-03 |  0.00094 | 0.018 | -0.00048 | 9.6e-01 | 0.220 |\n| case | DCDQ | male     | PS.ASD.Z | -2.7e-03 | -0.00820 | 0.020 | -0.01000 | 6.8e-01 |    NA |\n| case | DCDQ | female   | PS.ASD.Z |  1.5e-02 |  0.05200 | 0.044 |  0.04400 | 2.4e-01 |    NA |\n| case | DCDQ | combined | PS.EA.Z  | -1.7e-03 | -0.00840 | 0.018 | -0.01300 | 6.5e-01 | 0.059 |\n| case | DCDQ | male     | PS.EA.Z  | -2.2e-03 | -0.02500 | 0.020 | -0.02900 | 2.2e-01 |    NA |\n| case | DCDQ | female   | PS.EA.Z  |  1.5e-02 |  0.05700 | 0.044 |  0.06300 | 1.9e-01 |    NA |\n| case | DCDQ | combined | PS.SZ.Z  | -1.7e-03 |  0.00570 | 0.018 |  0.00410 | 7.6e-01 | 0.840 |\n| case | DCDQ | male     | PS.SZ.Z  | -2.7e-03 |  0.00830 | 0.020 |  0.00460 | 6.8e-01 |    NA |\n| case | DCDQ | female   | PS.SZ.Z  |  1.2e-02 | -0.01200 | 0.045 | -0.00700 | 7.9e-01 |    NA |\n| case | RBS  | combined | dnLoF    |  5.2e-03 |  0.02800 | 0.064 |  0.00700 | 6.7e-01 | 0.780 |\n| case | RBS  | male     | dnLoF    |  6.8e-03 |  0.01900 | 0.072 |  0.01300 | 8.0e-01 |    NA |\n| case | RBS  | female   | dnLoF    |  2.1e-03 |  0.04300 | 0.140 | -0.01200 | 7.6e-01 |    NA |\n| case | RBS  | combined | dnMIS    |  5.2e-03 | -0.03200 | 0.071 | -0.01500 | 6.5e-01 | 0.600 |\n| case | RBS  | male     | dnMIS    |  6.9e-03 | -0.05100 | 0.083 | -0.01900 | 5.4e-01 |    NA |\n| case | RBS  | female   | dnMIS    |  2.0e-03 |  0.03300 | 0.140 |  0.00650 | 8.2e-01 |    NA |\n| case | RBS  | combined | inhLoF   |  3.3e-05 |  0.00770 | 0.020 |  0.00460 | 7.0e-01 | 0.370 |\n| case | RBS  | male     | inhLoF   |  1.2e-04 |  0.01500 | 0.022 |  0.00960 | 5.0e-01 |    NA |\n| case | RBS  | female   | inhLoF   |  2.7e-04 | -0.02500 | 0.051 | -0.01900 | 6.3e-01 |    NA |\n| case | RBS  | combined | PS.ASD.Z |  6.9e-03 |  0.00250 | 0.017 |  0.00210 | 8.8e-01 | 0.150 |\n| case | RBS  | male     | PS.ASD.Z |  8.3e-03 |  0.01400 | 0.019 |  0.00960 | 4.6e-01 |    NA |\n| case | RBS  | female   | PS.ASD.Z |  4.7e-03 | -0.05400 | 0.041 | -0.03700 | 1.8e-01 |    NA |\n| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n| parent | Parental EA  | mother  | dnLoF             | 6.2e-04 |  0.0880 | 0.058 |       NA |  1.3e-01 |    NA |\n| parent | Parental EA  | mother  | dnMIS             | 3.2e-04 |  0.0720 | 0.066 |       NA |  2.7e-01 |    NA |\n| parent | Parental EA  | mother  | mother.inh        | 6.0e-05 | -0.0093 | 0.020 |       NA |  6.4e-01 |    NA |\n| parent | Parental EA  | mother  | Maternal.PS.ASD.Z | 6.9e-04 | -0.0250 | 0.016 |       NA |  1.1e-01 |    NA |\n| parent | Parental EA  | mother  | Maternal.PS.EA.Z  | 7.1e-02 |  0.2600 | 0.015 |       NA |  6.2e-62 |    NA |\n| parent | Parental EA  | mother  | Maternal.PS.SZ.Z  | 8.1e-05 | -0.0087 | 0.016 |       NA |  5.8e-01 |    NA |\n| parent | BAPQ         | parents | dnLoF             | 2.2e-03 | -0.1600 | 0.056 | -0.04400 |  5.9e-03 | 0.430 |\n| parent | BAPQ         | parents | dnMis             | 1.2e-04 | -0.0410 | 0.064 | -0.00910 |  5.2e-01 | 0.900 |\n| parent | BAPQ         | parents | inh               | 4.6e-05 |  0.0084 | 0.021 |  0.00360 |  6.9e-01 | 0.440 |\n| parent | BAPQ         | parents | PS.ASD.Z          | 1.7e-03 |  0.0430 | 0.017 |  0.01500 |  1.3e-02 | 0.110 |\n| parent | BAPQ         | parents | PS.EA.Z           | 1.4e-03 | -0.0390 | 0.017 | -0.05500 |  2.6e-02 | 0.160 |\n| parent | BAPQ         | parents | PS.SZ.Z           | 3.7e-04 |  0.0200 | 0.017 | -0.01500 |  2.5e-01 | 0.350 |\n| parent | Parental Age | parents | dnLoF             | 2.7e-05 |  0.0190 | 0.038 |  0.00720 |  6.1e-01 | 0.780 |\n| parent | Parental Age | parents | dnMis             | 5.1e-04 |  0.0940 | 0.042 |  0.02900 |  2.5e-02 | 0.420 |\n| parent | Parental Age | parents | inh               | 3.9e-04 | -0.0260 | 0.013 | -0.01600 |  5.1e-02 | 0.007 |\n| parent | Parental Age | parents | PS.ASD.Z          | 1.5e-03 |  0.0075 | 0.010 |  0.02500 |  4.7e-01 | 0.610 |\n| parent | Parental Age | parents | PS.EA.Z           | 2.0e-02 |  0.1400 | 0.010 |  0.16000 |  3.6e-40 | 0.043 |\n| parent | Parental Age | parents | PS.SZ.Z           | 1.8e-03 |  0.0190 | 0.011 |  0.04300 |  6.7e-02 | 0.210 |\n| parent | Parental EA  | parents | dnLoF             | 8.9e-04 |  0.1100 | 0.039 |  0.03700 |  6.3e-03 | 0.660 |\n| parent | Parental EA  | parents | dnMis             | 1.8e-04 |  0.0640 | 0.045 |  0.02200 |  1.6e-01 | 0.940 |\n| parent | Parental EA  | parents | inh               | 3.5e-04 | -0.0220 | 0.014 |  0.03500 |  1.0e-01 | 0.400 |\n| parent | Parental EA  | parents | PS.ASD.Z          | 2.7e-04 | -0.0060 | 0.011 |  0.00440 |  5.8e-01 | 0.096 |\n| parent | Parental EA  | parents | PS.EA.Z           | 7.9e-02 |  0.2600 | 0.010 |  0.31000 | 8.1e-137 | 0.270 |\n| parent | Parental EA  | parents | PS.SZ.Z           | 3.8e-04 | -0.0120 | 0.011 | -0.00081 |  2.9e-01 | 0.720 |\n| parent | SRS          | parents | dnLoF             | 1.4e-04 | -0.0380 | 0.076 | -0.00880 |  6.2e-01 | 0.530 |\n| parent | SRS          | parents | dnMis             | 1.8e-03 | -0.1900 | 0.110 | -0.04100 |  7.6e-02 | 0.790 |\n| parent | SRS          | parents | inh               | 8.9e-04 |  0.0360 | 0.028 |  0.02900 |  2.1e-01 | 0.490 |\n| parent | SRS          | parents | PS.ASD.Z          | 5.5e-03 |  0.0750 | 0.024 |  0.06400 |  1.7e-03 | 0.670 |\n| parent | SRS          | parents | PS.EA.Z           | 1.5e-03 | -0.0390 | 0.024 | -0.05000 |  1.1e-01 | 0.200 |\n| parent | SRS          | parents | PS.SZ.Z           | 1.8e-03 |  0.0440 | 0.024 |  0.03800 |  7.3e-02 | 0.630 |\n\n",
            "text/html": [
              "<table class=\"dataframe\">\n",
              "<caption>A data.frame: 216 × 10</caption>\n",
              "<thead>\n",
              "\t<tr><th scope=col>case.status</th><th scope=col>phenotype</th><th scope=col>group</th><th scope=col>x</th><th scope=col>r2</th><th scope=col>x_estimate</th><th scope=col>x_stderr</th><th scope=col>pearson.cc</th><th scope=col>x_pval</th><th scope=col>sex_interaction_pval</th></tr>\n",
              "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
              "</thead>\n",
              "<tbody>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>combined</td><td>dnLoF   </td><td> 1.1e-02</td><td>-0.42000</td><td>0.065</td><td>-0.11000</td><td>1.5e-10</td><td>0.074</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>male    </td><td>dnLoF   </td><td> 6.6e-03</td><td>-0.36000</td><td>0.073</td><td>-0.09400</td><td>9.8e-07</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>female  </td><td>dnLoF   </td><td> 3.3e-02</td><td>-0.52000</td><td>0.140</td><td>-0.18000</td><td>3.7e-04</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>combined</td><td>dnMIS   </td><td> 5.3e-03</td><td>-0.35000</td><td>0.074</td><td>-0.08100</td><td>3.4e-06</td><td>0.640</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>male    </td><td>dnMIS   </td><td> 3.5e-03</td><td>-0.33000</td><td>0.084</td><td>-0.07300</td><td>9.2e-05</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>female  </td><td>dnMIS   </td><td> 2.2e-02</td><td>-0.39000</td><td>0.160</td><td>-0.11000</td><td>1.7e-02</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>combined</td><td>inhLoF  </td><td> 1.0e-03</td><td>-0.04300</td><td>0.021</td><td>-0.03400</td><td>3.9e-02</td><td>0.910</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>male    </td><td>inhLoF  </td><td> 1.2e-03</td><td>-0.04500</td><td>0.022</td><td>-0.03600</td><td>4.2e-02</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>female  </td><td>inhLoF  </td><td> 4.1e-04</td><td>-0.02900</td><td>0.052</td><td>-0.02800</td><td>5.7e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>combined</td><td>PS.ASD.Z</td><td>-1.7e-03</td><td> 0.00094</td><td>0.018</td><td>-0.00048</td><td>9.6e-01</td><td>0.220</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>male    </td><td>PS.ASD.Z</td><td>-2.7e-03</td><td>-0.00820</td><td>0.020</td><td>-0.01000</td><td>6.8e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>female  </td><td>PS.ASD.Z</td><td> 1.5e-02</td><td> 0.05200</td><td>0.044</td><td> 0.04400</td><td>2.4e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>combined</td><td>PS.EA.Z </td><td>-1.7e-03</td><td>-0.00840</td><td>0.018</td><td>-0.01300</td><td>6.5e-01</td><td>0.059</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>male    </td><td>PS.EA.Z </td><td>-2.2e-03</td><td>-0.02500</td><td>0.020</td><td>-0.02900</td><td>2.2e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>female  </td><td>PS.EA.Z </td><td> 1.5e-02</td><td> 0.05700</td><td>0.044</td><td> 0.06300</td><td>1.9e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>combined</td><td>PS.SZ.Z </td><td>-1.7e-03</td><td> 0.00570</td><td>0.018</td><td> 0.00410</td><td>7.6e-01</td><td>0.840</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>male    </td><td>PS.SZ.Z </td><td>-2.7e-03</td><td> 0.00830</td><td>0.020</td><td> 0.00460</td><td>6.8e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>DCDQ</td><td>female  </td><td>PS.SZ.Z </td><td> 1.2e-02</td><td>-0.01200</td><td>0.045</td><td>-0.00700</td><td>7.9e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>combined</td><td>dnLoF   </td><td> 5.2e-03</td><td> 0.02800</td><td>0.064</td><td> 0.00700</td><td>6.7e-01</td><td>0.780</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>male    </td><td>dnLoF   </td><td> 6.8e-03</td><td> 0.01900</td><td>0.072</td><td> 0.01300</td><td>8.0e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>female  </td><td>dnLoF   </td><td> 2.1e-03</td><td> 0.04300</td><td>0.140</td><td>-0.01200</td><td>7.6e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>combined</td><td>dnMIS   </td><td> 5.2e-03</td><td>-0.03200</td><td>0.071</td><td>-0.01500</td><td>6.5e-01</td><td>0.600</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>male    </td><td>dnMIS   </td><td> 6.9e-03</td><td>-0.05100</td><td>0.083</td><td>-0.01900</td><td>5.4e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>female  </td><td>dnMIS   </td><td> 2.0e-03</td><td> 0.03300</td><td>0.140</td><td> 0.00650</td><td>8.2e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>combined</td><td>inhLoF  </td><td> 3.3e-05</td><td> 0.00770</td><td>0.020</td><td> 0.00460</td><td>7.0e-01</td><td>0.370</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>male    </td><td>inhLoF  </td><td> 1.2e-04</td><td> 0.01500</td><td>0.022</td><td> 0.00960</td><td>5.0e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>female  </td><td>inhLoF  </td><td> 2.7e-04</td><td>-0.02500</td><td>0.051</td><td>-0.01900</td><td>6.3e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>combined</td><td>PS.ASD.Z</td><td> 6.9e-03</td><td> 0.00250</td><td>0.017</td><td> 0.00210</td><td>8.8e-01</td><td>0.150</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>male    </td><td>PS.ASD.Z</td><td> 8.3e-03</td><td> 0.01400</td><td>0.019</td><td> 0.00960</td><td>4.6e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>case</td><td>RBS </td><td>female  </td><td>PS.ASD.Z</td><td> 4.7e-03</td><td>-0.05400</td><td>0.041</td><td>-0.03700</td><td>1.8e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>mother </td><td>dnLoF            </td><td>6.2e-04</td><td> 0.0880</td><td>0.058</td><td>      NA</td><td> 1.3e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>mother </td><td>dnMIS            </td><td>3.2e-04</td><td> 0.0720</td><td>0.066</td><td>      NA</td><td> 2.7e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>mother </td><td>mother.inh       </td><td>6.0e-05</td><td>-0.0093</td><td>0.020</td><td>      NA</td><td> 6.4e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>mother </td><td>Maternal.PS.ASD.Z</td><td>6.9e-04</td><td>-0.0250</td><td>0.016</td><td>      NA</td><td> 1.1e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>mother </td><td>Maternal.PS.EA.Z </td><td>7.1e-02</td><td> 0.2600</td><td>0.015</td><td>      NA</td><td> 6.2e-62</td><td>   NA</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>mother </td><td>Maternal.PS.SZ.Z </td><td>8.1e-05</td><td>-0.0087</td><td>0.016</td><td>      NA</td><td> 5.8e-01</td><td>   NA</td></tr>\n",
              "\t<tr><td>parent</td><td>BAPQ        </td><td>parents</td><td>dnLoF            </td><td>2.2e-03</td><td>-0.1600</td><td>0.056</td><td>-0.04400</td><td> 5.9e-03</td><td>0.430</td></tr>\n",
              "\t<tr><td>parent</td><td>BAPQ        </td><td>parents</td><td>dnMis            </td><td>1.2e-04</td><td>-0.0410</td><td>0.064</td><td>-0.00910</td><td> 5.2e-01</td><td>0.900</td></tr>\n",
              "\t<tr><td>parent</td><td>BAPQ        </td><td>parents</td><td>inh              </td><td>4.6e-05</td><td> 0.0084</td><td>0.021</td><td> 0.00360</td><td> 6.9e-01</td><td>0.440</td></tr>\n",
              "\t<tr><td>parent</td><td>BAPQ        </td><td>parents</td><td>PS.ASD.Z         </td><td>1.7e-03</td><td> 0.0430</td><td>0.017</td><td> 0.01500</td><td> 1.3e-02</td><td>0.110</td></tr>\n",
              "\t<tr><td>parent</td><td>BAPQ        </td><td>parents</td><td>PS.EA.Z          </td><td>1.4e-03</td><td>-0.0390</td><td>0.017</td><td>-0.05500</td><td> 2.6e-02</td><td>0.160</td></tr>\n",
              "\t<tr><td>parent</td><td>BAPQ        </td><td>parents</td><td>PS.SZ.Z          </td><td>3.7e-04</td><td> 0.0200</td><td>0.017</td><td>-0.01500</td><td> 2.5e-01</td><td>0.350</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental Age</td><td>parents</td><td>dnLoF            </td><td>2.7e-05</td><td> 0.0190</td><td>0.038</td><td> 0.00720</td><td> 6.1e-01</td><td>0.780</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental Age</td><td>parents</td><td>dnMis            </td><td>5.1e-04</td><td> 0.0940</td><td>0.042</td><td> 0.02900</td><td> 2.5e-02</td><td>0.420</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental Age</td><td>parents</td><td>inh              </td><td>3.9e-04</td><td>-0.0260</td><td>0.013</td><td>-0.01600</td><td> 5.1e-02</td><td>0.007</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental Age</td><td>parents</td><td>PS.ASD.Z         </td><td>1.5e-03</td><td> 0.0075</td><td>0.010</td><td> 0.02500</td><td> 4.7e-01</td><td>0.610</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental Age</td><td>parents</td><td>PS.EA.Z          </td><td>2.0e-02</td><td> 0.1400</td><td>0.010</td><td> 0.16000</td><td> 3.6e-40</td><td>0.043</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental Age</td><td>parents</td><td>PS.SZ.Z          </td><td>1.8e-03</td><td> 0.0190</td><td>0.011</td><td> 0.04300</td><td> 6.7e-02</td><td>0.210</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>parents</td><td>dnLoF            </td><td>8.9e-04</td><td> 0.1100</td><td>0.039</td><td> 0.03700</td><td> 6.3e-03</td><td>0.660</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>parents</td><td>dnMis            </td><td>1.8e-04</td><td> 0.0640</td><td>0.045</td><td> 0.02200</td><td> 1.6e-01</td><td>0.940</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>parents</td><td>inh              </td><td>3.5e-04</td><td>-0.0220</td><td>0.014</td><td> 0.03500</td><td> 1.0e-01</td><td>0.400</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>parents</td><td>PS.ASD.Z         </td><td>2.7e-04</td><td>-0.0060</td><td>0.011</td><td> 0.00440</td><td> 5.8e-01</td><td>0.096</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>parents</td><td>PS.EA.Z          </td><td>7.9e-02</td><td> 0.2600</td><td>0.010</td><td> 0.31000</td><td>8.1e-137</td><td>0.270</td></tr>\n",
              "\t<tr><td>parent</td><td>Parental EA </td><td>parents</td><td>PS.SZ.Z          </td><td>3.8e-04</td><td>-0.0120</td><td>0.011</td><td>-0.00081</td><td> 2.9e-01</td><td>0.720</td></tr>\n",
              "\t<tr><td>parent</td><td>SRS         </td><td>parents</td><td>dnLoF            </td><td>1.4e-04</td><td>-0.0380</td><td>0.076</td><td>-0.00880</td><td> 6.2e-01</td><td>0.530</td></tr>\n",
              "\t<tr><td>parent</td><td>SRS         </td><td>parents</td><td>dnMis            </td><td>1.8e-03</td><td>-0.1900</td><td>0.110</td><td>-0.04100</td><td> 7.6e-02</td><td>0.790</td></tr>\n",
              "\t<tr><td>parent</td><td>SRS         </td><td>parents</td><td>inh              </td><td>8.9e-04</td><td> 0.0360</td><td>0.028</td><td> 0.02900</td><td> 2.1e-01</td><td>0.490</td></tr>\n",
              "\t<tr><td>parent</td><td>SRS         </td><td>parents</td><td>PS.ASD.Z         </td><td>5.5e-03</td><td> 0.0750</td><td>0.024</td><td> 0.06400</td><td> 1.7e-03</td><td>0.670</td></tr>\n",
              "\t<tr><td>parent</td><td>SRS         </td><td>parents</td><td>PS.EA.Z          </td><td>1.5e-03</td><td>-0.0390</td><td>0.024</td><td>-0.05000</td><td> 1.1e-01</td><td>0.200</td></tr>\n",
              "\t<tr><td>parent</td><td>SRS         </td><td>parents</td><td>PS.SZ.Z          </td><td>1.8e-03</td><td> 0.0440</td><td>0.024</td><td> 0.03800</td><td> 7.3e-02</td><td>0.630</td></tr>\n",
              "</tbody>\n",
              "</table>\n"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "1AH8q_6mpP4B",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "290192cf-c2b1-4c34-8ffc-50f58a2dc149"
      },
      "source": [
        "df = read.csv(responses_out, sep = \",\")\n",
        "df$dnAll = df$dnLOF + df$dnMIS\n",
        "df <- df[df$EUR.Ancestry == 1, ]\n",
        "\n",
        "# Fit models\n",
        "run_lr_genetic_correlations <- function(dataframe, phenotype_var, cohort_vars, pc_vars, sex_var, gene_var, response_variable) {\n",
        "  df = dataframe\n",
        "\n",
        "  null_variables <- paste(phenotype_var, cohort_vars, pc_vars, sex_var, sep = \"+\") \n",
        "\n",
        "  null_formula <- reformulate(null_variables, response = response_variable)\n",
        "  null_model <- lm(formula = null_formula, data = df, na.action = na.exclude)\n",
        "  model_summary_null = summary(null_model)\n",
        "\n",
        "  gene_var <- c(gene_var)\n",
        "  alt_variables  <- paste(gene_var, null_variables, sep = \"+\")\n",
        "  alt_formula <- reformulate(alt_variables, response = response_variable)\n",
        "  alt_model  <- lm(formula = alt_formula , data = df, na.action = na.exclude)\n",
        "  model_summary_alt = summary(alt_model)\n",
        "\n",
        "  corr_equation <- paste(response_variable, gene_var, sep = \"+\")\n",
        "  corr_formula <- reformulate(corr_equation, response = NULL)\n",
        "  pearson.cor = cor.test(formula = corr_formula, data = df, na.action = na.exclude)\n",
        "\n",
        "  # Test sex interaction\n",
        "  sex_interaction_variable <- paste(gene_var, sex_var, sep = \"*\")\n",
        "  sex_interaction_terms <- paste(sex_interaction_variable, phenotype_var, cohort_vars, pc_vars, sep = \"+\")\n",
        "  \n",
        "  sex_interaction_formula <- reformulate(sex_interaction_terms, response = response_variable)\n",
        "  sex_interaction_model <- lm(formula = sex_interaction_formula, data = df, na.action = na.exclude)\n",
        "  model_summary_sex_interaction = summary(sex_interaction_model)\n",
        "\n",
        "  # Sex stratification\n",
        "  # Males are 0, females are 1\n",
        "  df_male   = df[df$Sex == 0,]\n",
        "  df_female = df[df$Sex == 1,]\n",
        "\n",
        "  null_model_male <- lm(formula = null_formula, data = df_male, na.action = na.exclude)\n",
        "  alt_model_male  <- lm(formula = alt_formula, data = df_male, na.action = na.exclude)\n",
        "\n",
        "  model_summary_null_male <- summary(null_model_male)\n",
        "  model_summary_alt_male <-  summary(alt_model_male)\n",
        "\n",
        "  pearson.cor.male = cor.test(formula = corr_formula, data = df_male, na.action = na.exclude)\n",
        "\n",
        "  null_model_female <- lm(formula = null_formula, data = df_female, na.action = na.exclude)\n",
        "  alt_model_female  <- lm(formula = alt_formula, data = df_female, na.action = na.exclude)\n",
        "\n",
        "  model_summary_null_female <- summary(null_model_female)\n",
        "  model_summary_alt_female <-  summary(alt_model_female)\n",
        "\n",
        "  pearson.cor.female = cor.test(formula = corr_formula, data = df_female, na.action = na.exclude)\n",
        "\n",
        "  output <- list(model_summary_alt, model_summary_null, \n",
        "                 pearson.cor, model_summary_sex_interaction, \n",
        "                 model_summary_alt_male, model_summary_null_male, pearson.cor.male,\n",
        "                 model_summary_alt_female, model_summary_null_female, pearson.cor.female)\n",
        "  return(output)\n",
        "}\n",
        "\n",
        "\n",
        "# Null model variables\n",
        "phenotype_var <- c(\"Phenotype\")\n",
        "cohort_vars <- c(\"REACH_cohort+SSC_cohort+SPARK_cohort\")\n",
        "pc_vars <- c(\"PC1+PC2+PC3+PC4+PC5+PC6+PC7+PC8+PC9+PC10\")\n",
        "sex_var <- c(\"Sex\") # Only use with combined models (it doesn't matter)\n",
        "\n",
        "# I can just for loop this, but I prefer this approach...\n",
        "\n",
        "# Responses (response.input.summary)\n",
        "# dnMis\n",
        "response_var = c(\"dnMIS\")\n",
        "# dnmis.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "dnmis.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "dnmis.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "dnmis.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "dnmis.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "dnmis.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "dnmis.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "dnmis.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "dnmis.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# dnLoF\n",
        "response_var = c(\"dnLOF\")\n",
        "dnlof.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "# dnlof.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "dnlof.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "dnlof.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "dnlof.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "dnlof.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "dnlof.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "dnlof.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "dnlof.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# dnAll\n",
        "response_var = c(\"dnAll\")\n",
        "dnall.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "dnall.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "# dnall.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "dnall.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "dnall.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "dnall.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "dnall.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "dnall.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "dnall.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# inhLoF\n",
        "response_var = c(\"inhLoF\")\n",
        "inhlof.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "inhlof.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "inhlof.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "# inhlof.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "inhlof.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "inhlof.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "inhlof.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "inhlof.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "inhlof.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# RVRS (Rare Variant Risk Score)\n",
        "response_var = c(\"RVRS\")\n",
        "rvrs.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "rvrs.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "rvrs.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "rvrs.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "# rvrs.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "rvrs.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "rvrs.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "rvrs.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "rvrs.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# pTDT.ASD.Dev\n",
        "response_var = c(\"pTDT.ASD.Dev\")\n",
        "ptdt_asd_dev.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "ptdt_asd_dev.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "ptdt_asd_dev.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "ptdt_asd_dev.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "ptdt_asd_dev.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "# ptdt_asd_dev.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "ptdt_asd_dev.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "ptdt_asd_dev.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "ptdt_asd_dev.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# pTDT.SCZ.Dev\n",
        "response_var = c(\"pTDT.SCZ.Dev\")\n",
        "ptdt_scz_dev.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "ptdt_scz_dev.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "ptdt_scz_dev.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "ptdt_scz_dev.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "ptdt_scz_dev.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "ptdt_scz_dev.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "# ptdt_scz_dev.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "ptdt_scz_dev.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "ptdt_scz_dev.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# pTDT.EA.Dev\n",
        "response_var = c(\"pTDT.EA.Dev\")\n",
        "ptdt_ea_dev.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "ptdt_ea_dev.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "ptdt_ea_dev.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "ptdt_ea_dev.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "ptdt_ea_dev.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "ptdt_ea_dev.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "ptdt_ea_dev.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "# ptdt_ea_dev.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "ptdt_ea_dev.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "# PRS (a.k.a. Common Variant Risk Score, or CVRS)\n",
        "response_var = c(\"PRS\")\n",
        "prs.dnmis.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnMIS\", response_variable = response_var)\n",
        "prs.dnlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnLOF\", response_variable = response_var)\n",
        "prs.dnall.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"dnAll\", response_variable = response_var)\n",
        "prs.inhlof.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"inhLoF\", response_variable = response_var)\n",
        "prs.rvrs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"RVRS\", response_variable = response_var)\n",
        "prs.ptdt_asd_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.ASD.Dev\", response_variable = response_var)\n",
        "prs.ptdt_scz_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.SCZ.Dev\", response_variable = response_var)\n",
        "prs.ptdt_ea_dev.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"pTDT.EA.Dev\", response_variable = response_var)\n",
        "# prs.prs.summary = run_lr_genetic_correlations(dataframe = df, phenotype_var = phenotype_var, cohort_vars = cohort_vars, pc_vars = pc_vars, sex_var = sex_var, gene_var = \"PRS\", response_variable = response_var)\n",
        "\n",
        "\n",
        "# Put model summaries in a list\n",
        "dnmis_model_summaries <- list(dnmis.dnlof.summary, dnmis.dnall.summary, dnmis.inhlof.summary, dnmis.rvrs.summary, dnmis.ptdt_asd_dev.summary, dnmis.ptdt_scz_dev.summary, dnmis.ptdt_ea_dev.summary, dnmis.prs.summary)\n",
        "dnlof_model_summaries <- list(dnlof.dnmis.summary, dnlof.dnall.summary, dnlof.inhlof.summary, dnlof.rvrs.summary, dnlof.ptdt_asd_dev.summary, dnlof.ptdt_scz_dev.summary, dnlof.ptdt_ea_dev.summary, dnlof.prs.summary)\n",
        "dnall_model_summaries <- list(dnall.dnmis.summary, dnall.dnlof.summary, dnall.inhlof.summary, dnall.rvrs.summary, dnall.ptdt_asd_dev.summary, dnall.ptdt_scz_dev.summary, dnall.ptdt_ea_dev.summary, dnall.prs.summary)\n",
        "inhlof_model_summaries <- list(inhlof.dnmis.summary, inhlof.dnlof.summary, inhlof.dnall.summary, inhlof.rvrs.summary, inhlof.ptdt_asd_dev.summary, inhlof.ptdt_scz_dev.summary, inhlof.ptdt_ea_dev.summary, inhlof.prs.summary)\n",
        "rvrs_model_summaries <- list(rvrs.dnmis.summary, rvrs.dnlof.summary, rvrs.dnall.summary, rvrs.inhlof.summary, rvrs.ptdt_asd_dev.summary, rvrs.ptdt_scz_dev.summary, rvrs.ptdt_ea_dev.summary, rvrs.prs.summary)\n",
        "ptdt.asd.dev_model_summaries <- list(ptdt_asd_dev.dnmis.summary, ptdt_asd_dev.dnlof.summary, ptdt_asd_dev.dnall.summary, ptdt_asd_dev.inhlof.summary, ptdt_asd_dev.rvrs.summary, ptdt_asd_dev.ptdt_scz_dev.summary, ptdt_asd_dev.ptdt_ea_dev.summary, ptdt_asd_dev.prs.summary)\n",
        "ptdt.scz.dev_model_summaries <- list(ptdt_scz_dev.dnmis.summary, ptdt_scz_dev.dnlof.summary, ptdt_scz_dev.dnall.summary, ptdt_scz_dev.inhlof.summary, ptdt_scz_dev.rvrs.summary, ptdt_scz_dev.ptdt_asd_dev.summary, ptdt_scz_dev.ptdt_ea_dev.summary, ptdt_scz_dev.prs.summary)\n",
        "ptdt.ea.dev_model_summaries <- list(ptdt_ea_dev.dnmis.summary, ptdt_ea_dev.dnlof.summary, ptdt_ea_dev.dnall.summary, ptdt_ea_dev.inhlof.summary, ptdt_ea_dev.rvrs.summary, ptdt_ea_dev.ptdt_asd_dev.summary, ptdt_ea_dev.ptdt_scz_dev.summary, ptdt_ea_dev.prs.summary)\n",
        "prs_model_summaries <- list(prs.dnmis.summary, prs.dnlof.summary, prs.dnall.summary, prs.inhlof.summary, prs.rvrs.summary, prs.ptdt_asd_dev.summary, prs.ptdt_scz_dev.summary, prs.ptdt_ea_dev.summary)\n",
        "\n",
        "model_dict <- ordered_dict(items = NULL, keys = NULL)\n",
        "model_dict$set(\"dnMIS\", dnmis_model_summaries)\n",
        "model_dict$set(\"dnLOF\", dnlof_model_summaries)\n",
        "model_dict$set(\"dnAll\", dnall_model_summaries)\n",
        "model_dict$set(\"inhLoF\", inhlof_model_summaries)\n",
        "model_dict$set(\"RVRS\", rvrs_model_summaries)\n",
        "model_dict$set(\"pTDT.ASD.Dev\", ptdt.asd.dev_model_summaries)\n",
        "model_dict$set(\"pTDT.SCZ.Dev\", ptdt.scz.dev_model_summaries)\n",
        "model_dict$set(\"pTDT.EA.Dev\", ptdt.ea.dev_model_summaries)\n",
        "model_dict$set(\"PRS\", prs_model_summaries)\n",
        "\n",
        "keys = list(\n",
        "  \"dnMIS\", \"dnLOF\", \"dnAll\", \"inhLoF\", \"RVRS\",\n",
        "  \"pTDT.ASD.Dev\", \"pTDT.SCZ.Dev\", \"pTDT.EA.Dev\", \"PRS\"\n",
        ")\n",
        "\n",
        "genetic_correlations_table <- data.frame()\n",
        "\n",
        "for (key in keys) {\n",
        "  y = key\n",
        "\n",
        "  model_summary_list = model_dict$get(key)\n",
        "  for (model_summaries in model_summary_list) {\n",
        "    # Combined\n",
        "    model_summary_alt  = model_summaries[[1]]\n",
        "    model_summary_null = model_summaries[[2]]\n",
        "\n",
        "    x = rownames(coefficients(model_summary_alt))[2]\n",
        "\n",
        "    x_estimate = signif(coefficients(model_summary_alt)[2, 1], digits = 2)\n",
        "    x_stderr = signif(coefficients(model_summary_alt)[2, 2], digits = 2)\n",
        "    x_pval = signif(coefficients(model_summary_alt)[2, 4], digits = 2)\n",
        "    r2 = signif(model_summary_alt$r.squared - model_summary_null$r.squared, digits = 2)\n",
        "\n",
        "    # Correlation\n",
        "    pearson.cor = model_summaries[[3]]\n",
        "    pearson.cc = signif(pearson.cor$estimate, digits = 2)\n",
        "\n",
        "    # Sex interaction\n",
        "    model_summary_sex_interaction = model_summaries[[4]]\n",
        "    sex_interaction_model_coeffs = coefficients(model_summary_sex_interaction)\n",
        "    num_rows = dim(sex_interaction_model_coeffs)[1]\n",
        "    sex_interaction_coeff = sex_interaction_model_coeffs[num_rows, ]\n",
        "    sex_interaction_pval = signif(sex_interaction_coeff[[4]], digits = 2)\n",
        "\n",
        "    # Males\n",
        "    model_summary_alt_male    = model_summaries[[5]]\n",
        "    model_summary_null_male   = model_summaries[[6]]\n",
        "    pearson.cor.male = model_summaries[[7]]\n",
        "    pearson.cc.male = signif(pearson.cor.male$estimate, digits = 2)\n",
        "\n",
        "    x_estimate_male = signif(coefficients(model_summary_alt_male)[2, 1], digits = 2)\n",
        "    x_stderr_male = signif(coefficients(model_summary_alt_male)[2, 2], digits = 2)\n",
        "    x_pval_male = signif(coefficients(model_summary_alt_male)[2, 4], digits = 2)\n",
        "    r2_male = signif(model_summary_alt_male$r.squared - model_summary_null_male$r.squared, digits = 2)\n",
        "\n",
        "    # Females\n",
        "    model_summary_alt_female  = model_summaries[[8]]\n",
        "    model_summary_null_female = model_summaries[[9]]\n",
        "    pearson.cor.female = model_summaries[[10]]\n",
        "    pearson.cc.female = signif(pearson.cor.female$estimate, digits = 2)\n",
        "\n",
        "    x_estimate_female = signif(coefficients(model_summary_alt_female)[2, 1], digits = 2)\n",
        "    x_stderr_female = signif(coefficients(model_summary_alt_female)[2, 2], digits = 2)\n",
        "    x_pval_female = signif(coefficients(model_summary_alt_female)[2, 4], digits = 2)\n",
        "    r2_female = signif(model_summary_alt_female$r.squared - model_summary_null_female$r.squared, digits = 2)\n",
        "\n",
        "\n",
        "    df_tmp = data.frame(y, x, \n",
        "                  x_estimate_male, x_pval_male, r2_male, pearson.cc.male,\n",
        "                  x_estimate_female, x_pval_female, r2_female, pearson.cc.female,\n",
        "                  x_estimate, x_pval, r2, pearson.cc,\n",
        "                  sex_interaction_pval)\n",
        "\n",
        "    genetic_correlations_table = rbind(genetic_correlations_table, df_tmp)\n",
        "\n",
        "  }\n",
        "}\n",
        "\n",
        "row.names(genetic_correlations_table) <- NULL\n",
        "header <- c(\"Y\", \"X\", \n",
        "            \"Male Estimate\", \"Male P-value\", \"Male r2\", \"Male Correlation\", \n",
        "            \"Female Estimate\", \"Female P-value\", \"Female r2\", \"Female Correlation\", \n",
        "            \"Combined Estimate\", \"Combined P-value\", \"Combined r2\", \"Combined Correlation\", \n",
        "            \"Gene by Sex Interaction P-value\")\n",
        "colnames(genetic_correlations_table) <- header\n",
        "\n",
        "genetic_correlations_table"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "   Y            X            Male Estimate Male P-value Male r2\n",
              "1  dnMIS        dnLOF         3.4e-02       8.7e-03     1.2e-03\n",
              "2  dnMIS        dnAll         4.9e-01       0.0e+00     5.1e-01\n",
              "3  dnMIS        inhLoF        4.7e-03       2.1e-01     2.7e-04\n",
              "4  dnMIS        RVRS          7.8e-02      2.0e-168     1.2e-01\n",
              "5  dnMIS        pTDT.ASD.Dev -7.8e-03       5.6e-02     6.3e-04\n",
              "6  dnMIS        pTDT.SCZ.Dev -3.9e-05       9.9e-01     1.4e-08\n",
              "7  dnMIS        pTDT.EA.Dev  -1.8e-03       6.8e-01     2.9e-05\n",
              "8  dnMIS        PRS          -4.7e-03       1.0e-01     4.7e-04\n",
              "9  dnLOF        dnMIS         3.5e-02       8.7e-03     1.2e-03\n",
              "10 dnLOF        dnAll         5.1e-01       0.0e+00     5.2e-01\n",
              "11 dnLOF        inhLoF       -4.2e-03       2.7e-01     2.1e-04\n",
              "12 dnLOF        RVRS          2.0e-01       0.0e+00     7.9e-01\n",
              "13 dnLOF        pTDT.ASD.Dev -3.9e-03       3.5e-01     1.5e-04\n",
              "14 dnLOF        pTDT.SCZ.Dev  1.0e-03       8.2e-01     8.8e-06\n",
              "15 dnLOF        pTDT.EA.Dev  -7.4e-03       1.0e-01     4.6e-04\n",
              "16 dnLOF        PRS          -4.3e-03       1.4e-01     3.7e-04\n",
              "17 dnAll        dnMIS         1.0e+00       0.0e+00     5.0e-01\n",
              "18 dnAll        dnLOF         1.0e+00       0.0e+00     5.2e-01\n",
              "19 dnAll        inhLoF        4.7e-04       9.3e-01     1.3e-06\n",
              "20 dnAll        RVRS          2.8e-01       0.0e+00     7.5e-01\n",
              "21 dnAll        pTDT.ASD.Dev -1.2e-02       4.9e-02     6.7e-04\n",
              "22 dnAll        pTDT.SCZ.Dev  9.6e-04       8.8e-01     4.0e-06\n",
              "23 dnAll        pTDT.EA.Dev  -9.2e-03       1.5e-01     3.5e-04\n",
              "24 dnAll        PRS          -9.0e-03       3.1e-02     8.1e-04\n",
              "25 inhLoF       dnMIS         5.9e-02       2.1e-01     2.7e-04\n",
              "26 inhLoF       dnLOF        -5.1e-02       2.7e-01     2.1e-04\n",
              "27 inhLoF       dnAll         2.8e-03       9.3e-01     1.3e-06\n",
              "28 inhLoF       RVRS          2.2e-01      8.5e-107     8.0e-02\n",
              "29 inhLoF       pTDT.ASD.Dev -1.5e-02       2.9e-01     1.9e-04\n",
              "30 inhLoF       pTDT.SCZ.Dev -1.1e-02       4.8e-01     8.7e-05\n",
              "⋮  ⋮            ⋮            ⋮             ⋮            ⋮      \n",
              "43 pTDT.ASD.Dev dnAll        -0.05800       4.9e-02     6.6e-04\n",
              "44 pTDT.ASD.Dev inhLoF       -0.01300       2.9e-01     1.9e-04\n",
              "45 pTDT.ASD.Dev RVRS         -0.01700       7.9e-02     5.3e-04\n",
              "46 pTDT.ASD.Dev pTDT.SCZ.Dev  0.11000       2.9e-14     9.9e-03\n",
              "47 pTDT.ASD.Dev pTDT.EA.Dev   0.09300       5.7e-11     7.3e-03\n",
              "48 pTDT.ASD.Dev PRS           0.58000       0.0e+00     6.7e-01\n",
              "49 pTDT.SCZ.Dev dnMIS        -0.00036       9.9e-01     1.4e-08\n",
              "50 pTDT.SCZ.Dev dnLOF         0.00890       8.2e-01     8.8e-06\n",
              "51 pTDT.SCZ.Dev dnAll         0.00420       8.8e-01     4.0e-06\n",
              "52 pTDT.SCZ.Dev inhLoF       -0.00800       4.8e-01     8.7e-05\n",
              "53 pTDT.SCZ.Dev RVRS         -0.00150       8.6e-01     5.1e-06\n",
              "54 pTDT.SCZ.Dev pTDT.ASD.Dev  0.09500       2.9e-14     1.0e-02\n",
              "55 pTDT.SCZ.Dev pTDT.EA.Dev   0.03900       4.1e-03     1.4e-03\n",
              "56 pTDT.SCZ.Dev PRS           0.24000      2.2e-175     1.3e-01\n",
              "57 pTDT.EA.Dev  dnMIS        -0.01600       6.8e-01     2.9e-05\n",
              "58 pTDT.EA.Dev  dnLOF        -0.06300       1.0e-01     4.6e-04\n",
              "59 pTDT.EA.Dev  dnAll        -0.03900       1.5e-01     3.5e-04\n",
              "60 pTDT.EA.Dev  inhLoF       -0.00560       6.1e-01     4.4e-05\n",
              "61 pTDT.EA.Dev  RVRS         -0.01500       7.6e-02     5.4e-04\n",
              "62 pTDT.EA.Dev  pTDT.ASD.Dev  0.08000       5.7e-11     7.4e-03\n",
              "63 pTDT.EA.Dev  pTDT.SCZ.Dev  0.03700       4.1e-03     1.4e-03\n",
              "64 pTDT.EA.Dev  PRS           0.37000       0.0e+00     3.2e-01\n",
              "65 PRS          dnMIS        -0.10000       1.0e-01     4.7e-04\n",
              "66 PRS          dnLOF        -0.08700       1.4e-01     3.7e-04\n",
              "67 PRS          dnAll        -0.09000       3.1e-02     8.0e-04\n",
              "68 PRS          inhLoF       -0.02100       2.1e-01     2.7e-04\n",
              "69 PRS          RVRS         -0.03000       2.3e-02     8.9e-04\n",
              "70 PRS          pTDT.ASD.Dev  1.20000       0.0e+00     6.7e-01\n",
              "71 PRS          pTDT.SCZ.Dev  0.54000      2.2e-175     1.3e-01\n",
              "72 PRS          pTDT.EA.Dev   0.87000       0.0e+00     3.2e-01\n",
              "   Male Correlation Female Estimate Female P-value Female r2 Female Correlation\n",
              "1   0.0370           0.08900        2.4e-06        8.6e-03    0.11000          \n",
              "2   0.7100           0.47000        0.0e+00        5.1e-01    0.72000          \n",
              "3   0.0160          -0.00570        3.4e-01        3.6e-04   -0.01300          \n",
              "4   0.3500           0.08400        3.5e-87        1.4e-01    0.39000          \n",
              "5  -0.0230           0.00096        8.8e-01        9.0e-06    0.01400          \n",
              "6   0.0010          -0.00170        8.0e-01        2.6e-05   -0.00220          \n",
              "7  -0.0029          -0.01100        1.0e-01        1.0e-03   -0.02100          \n",
              "8  -0.0190          -0.00370        4.2e-01        2.6e-04   -0.00330          \n",
              "9   0.0370           0.09900        2.4e-06        8.5e-03    0.11000          \n",
              "10  0.7300           0.53000        0.0e+00        5.5e-01    0.76000          \n",
              "11 -0.0100          -0.01600        1.3e-02        2.4e-03   -0.04100          \n",
              "12  0.8900           0.21000        0.0e+00        7.8e-01    0.90000          \n",
              "13 -0.0046          -0.00700        3.0e-01        4.2e-04    0.00120          \n",
              "14  0.0071           0.00440        5.3e-01        1.5e-04    0.01700          \n",
              "15 -0.0160          -0.01200        9.9e-02        1.0e-03   -0.01600          \n",
              "16 -0.0094          -0.00730        1.3e-01        8.9e-04   -0.00370          \n",
              "17  0.7100           1.10000        0.0e+00        5.0e-01    0.72000          \n",
              "18  0.7300           1.10000        0.0e+00        5.5e-01    0.76000          \n",
              "19  0.0042          -0.02100        1.9e-02        2.1e-03   -0.03700          \n",
              "20  0.8700           0.29000        0.0e+00        7.3e-01    0.88000          \n",
              "21 -0.0190          -0.00610        5.3e-01        1.5e-04    0.00980          \n",
              "22  0.0057           0.00270        7.9e-01        2.8e-05    0.01100          \n",
              "23 -0.0140          -0.02300        2.6e-02        1.9e-03   -0.02500          \n",
              "24 -0.0200          -0.01100        1.1e-01        9.6e-04   -0.00470          \n",
              "25  0.0160          -0.06300        3.4e-01        3.6e-04   -0.01300          \n",
              "26 -0.0100          -0.16000        1.3e-02        2.4e-03   -0.04100          \n",
              "27  0.0042          -0.10000        1.9e-02        2.2e-03   -0.03700          \n",
              "28  0.2800           0.16000        1.2e-29        4.9e-02    0.22000          \n",
              "29 -0.0110          -0.02000        3.5e-01        3.5e-04   -0.01200          \n",
              "30 -0.0069          -0.00380        8.6e-01        1.2e-05    0.00015          \n",
              "⋮  ⋮                ⋮               ⋮              ⋮         ⋮                 \n",
              "43 -0.0190          -0.0260          5.3e-01       1.5e-04    0.00980          \n",
              "44 -0.0110          -0.0180          3.5e-01       3.5e-04   -0.01200          \n",
              "45 -0.0150          -0.0140          3.0e-01       4.1e-04    0.00580          \n",
              "46  0.1000           0.0660          1.4e-03       4.0e-03    0.06700          \n",
              "47  0.0920           0.0710          9.8e-04       4.2e-03    0.07500          \n",
              "48  0.8300           0.5900          0.0e+00       6.6e-01    0.82000          \n",
              "49  0.0010          -0.0150          8.0e-01       2.6e-05   -0.00220          \n",
              "50  0.0071           0.0350          5.3e-01       1.6e-04    0.01700          \n",
              "51  0.0057           0.0110          7.9e-01       2.8e-05    0.01100          \n",
              "52 -0.0069          -0.0031          8.6e-01       1.2e-05    0.00015          \n",
              "53  0.0020           0.0019          8.8e-01       8.3e-06    0.01000          \n",
              "54  0.1000           0.0610          1.4e-03       4.0e-03    0.06700          \n",
              "55  0.0410           0.0073          7.2e-01       4.9e-05    0.00980          \n",
              "56  0.3600           0.2200          3.7e-63       1.0e-01    0.33000          \n",
              "57 -0.0029          -0.0930          1.0e-01       1.0e-03   -0.02100          \n",
              "58 -0.0160          -0.0890          9.9e-02       1.1e-03   -0.01600          \n",
              "59 -0.0140          -0.0830          2.6e-02       1.9e-03   -0.02500          \n",
              "60 -0.0041          -0.0015          9.3e-01       2.9e-06    0.00060          \n",
              "61 -0.0180          -0.0270          3.1e-02       1.8e-03   -0.02100          \n",
              "62  0.0920           0.0600          9.8e-04       4.2e-03    0.07500          \n",
              "63  0.0410           0.0067          7.2e-01       4.8e-05    0.00980          \n",
              "64  0.5700           0.3600         6.8e-201       3.0e-01    0.56000          \n",
              "65 -0.0190          -0.0710          4.2e-01       2.5e-04   -0.00330          \n",
              "66 -0.0094          -0.1300          1.3e-01       8.9e-04   -0.00370          \n",
              "67 -0.0200          -0.0910          1.1e-01       9.8e-04   -0.00470          \n",
              "68 -0.0130          -0.0200          4.3e-01       2.4e-04   -0.00980          \n",
              "69 -0.0190          -0.0360          5.8e-02       1.4e-03   -0.00550          \n",
              "70  0.8300           1.1000          0.0e+00       6.6e-01    0.82000          \n",
              "71  0.3600           0.4700          3.7e-63       1.0e-01    0.33000          \n",
              "72  0.5700           0.8400         6.8e-201       3.0e-01    0.56000          \n",
              "   Combined Estimate Combined P-value Combined r2 Combined Correlation\n",
              "1   0.0530            4.9e-07         3.0e-03      0.06000            \n",
              "2   0.4800            0.0e+00         5.1e-01      0.72000            \n",
              "3   0.0017            5.9e-01         3.5e-05      0.00750            \n",
              "4   0.0800           5.2e-256         1.3e-01      0.36000            \n",
              "5  -0.0053            1.2e-01         2.8e-04     -0.01200            \n",
              "6  -0.0007            8.5e-01         4.5e-06     -0.00028            \n",
              "7  -0.0045            2.3e-01         1.7e-04     -0.00880            \n",
              "8  -0.0045            6.6e-02         4.1e-04     -0.01400            \n",
              "9   0.0570            4.9e-07         3.0e-03      0.06000            \n",
              "10  0.5200            0.0e+00         5.4e-01      0.74000            \n",
              "11 -0.0076            2.0e-02         6.5e-04     -0.02000            \n",
              "12  0.2000            0.0e+00         7.9e-01      0.89000            \n",
              "13 -0.0047            1.9e-01         2.1e-04     -0.00300            \n",
              "14  0.0018            6.2e-01         2.9e-05      0.01000            \n",
              "15 -0.0086            2.6e-02         5.9e-04     -0.01700            \n",
              "16 -0.0051            4.3e-02         4.9e-04     -0.00760            \n",
              "17  1.1000            0.0e+00         5.1e-01      0.72000            \n",
              "18  1.1000            0.0e+00         5.4e-01      0.74000            \n",
              "19 -0.0059            2.1e-01         1.9e-04     -0.00870            \n",
              "20  0.2800            0.0e+00         7.5e-01      0.87000            \n",
              "21 -0.0099            5.0e-02         4.6e-04     -0.01000            \n",
              "22  0.0011            8.3e-01         5.4e-06      0.00680            \n",
              "23 -0.0130            1.8e-02         6.7e-04     -0.01800            \n",
              "24 -0.0095            7.8e-03         8.4e-04     -0.01500            \n",
              "25  0.0210            5.9e-01         3.5e-05      0.00750            \n",
              "26 -0.0860            2.0e-02         6.5e-04     -0.02000            \n",
              "27 -0.0330            2.1e-01         1.9e-04     -0.00870            \n",
              "28  0.2000           9.1e-133         6.9e-02      0.26000            \n",
              "29 -0.0170            1.5e-01         2.5e-04     -0.01000            \n",
              "30 -0.0081            5.2e-01         5.0e-05     -0.00390            \n",
              "⋮  ⋮                 ⋮                ⋮           ⋮                   \n",
              "43 -0.04700           5.0e-02         4.6e-04     -0.01000            \n",
              "44 -0.01400           1.5e-01         2.4e-04     -0.01000            \n",
              "45 -0.01600           4.4e-02         4.8e-04     -0.00830            \n",
              "46  0.09400           5.1e-16         7.8e-03      0.09400            \n",
              "47  0.08700           2.4e-13         6.4e-03      0.08800            \n",
              "48  0.58000           0.0e+00         6.7e-01      0.82000            \n",
              "49 -0.00650           8.5e-01         4.5e-06     -0.00028            \n",
              "50  0.01600           6.2e-01         2.9e-05      0.01000            \n",
              "51  0.00480           8.3e-01         5.5e-06      0.00680            \n",
              "52 -0.00620           5.2e-01         5.0e-05     -0.00390            \n",
              "53 -0.00082           9.1e-01         1.5e-06      0.00470            \n",
              "54  0.08400           5.1e-16         7.9e-03      0.09400            \n",
              "55  0.03000           8.5e-03         8.3e-04      0.03300            \n",
              "56  0.23000          1.3e-236         1.2e-01      0.35000            \n",
              "57 -0.03900           2.3e-01         1.7e-04     -0.00880            \n",
              "58 -0.07000           2.6e-02         5.9e-04     -0.01700            \n",
              "59 -0.05200           1.8e-02         6.7e-04     -0.01800            \n",
              "60 -0.00420           6.5e-01         2.4e-05     -0.00170            \n",
              "61 -0.01900           9.5e-03         8.0e-04     -0.01900            \n",
              "62  0.07400           2.4e-13         6.4e-03      0.08800            \n",
              "63  0.02800           8.5e-03         8.3e-04      0.03300            \n",
              "64  0.37000           0.0e+00         3.1e-01      0.57000            \n",
              "65 -0.09100           6.6e-02         4.0e-04     -0.01400            \n",
              "66 -0.09700           4.3e-02         4.8e-04     -0.00760            \n",
              "67 -0.08900           7.8e-03         8.4e-04     -0.01500            \n",
              "68 -0.02100           1.4e-01         2.6e-04     -0.01200            \n",
              "69 -0.03200           3.7e-03         1.0e-03     -0.01500            \n",
              "70  1.20000           0.0e+00         6.7e-01      0.82000            \n",
              "71  0.52000          1.3e-236         1.2e-01      0.35000            \n",
              "72  0.86000           0.0e+00         3.1e-01      0.57000            \n",
              "   Gene by Sex Interaction P-value\n",
              "1  0.00560                        \n",
              "2  0.13000                        \n",
              "3  0.19000                        \n",
              "4  0.17000                        \n",
              "5  0.13000                        \n",
              "6  0.94000                        \n",
              "7  0.36000                        \n",
              "8  0.58000                        \n",
              "9  0.00400                        \n",
              "10 0.13000                        \n",
              "11 0.16000                        \n",
              "12 0.00170                        \n",
              "13 0.96000                        \n",
              "14 0.59000                        \n",
              "15 0.82000                        \n",
              "16 0.98000                        \n",
              "17 0.00400                        \n",
              "18 0.00560                        \n",
              "19 0.06300                        \n",
              "20 0.00016                        \n",
              "21 0.29000                        \n",
              "22 0.74000                        \n",
              "23 0.44000                        \n",
              "24 0.72000                        \n",
              "25 0.17000                        \n",
              "26 0.18000                        \n",
              "27 0.06600                        \n",
              "28 0.00054                        \n",
              "29 0.91000                        \n",
              "30 0.74000                        \n",
              "⋮  ⋮                              \n",
              "43 0.470                          \n",
              "44 0.880                          \n",
              "45 0.800                          \n",
              "46 0.120                          \n",
              "47 0.430                          \n",
              "48 0.370                          \n",
              "49 0.760                          \n",
              "50 0.800                          \n",
              "51 0.980                          \n",
              "52 0.790                          \n",
              "53 0.930                          \n",
              "54 0.110                          \n",
              "55 0.180                          \n",
              "56 0.220                          \n",
              "57 0.320                          \n",
              "58 0.830                          \n",
              "59 0.450                          \n",
              "60 0.900                          \n",
              "61 0.600                          \n",
              "62 0.420                          \n",
              "63 0.210                          \n",
              "64 0.690                          \n",
              "65 0.780                          \n",
              "66 0.840                          \n",
              "67 0.910                          \n",
              "68 0.960                          \n",
              "69 0.930                          \n",
              "70 0.190                          \n",
              "71 0.047                          \n",
              "72 0.280                          "
            ],
            "text/latex": "A data.frame: 72 × 15\n\\begin{tabular}{lllllllllllllll}\n Y & X & Male Estimate & Male P-value & Male r2 & Male Correlation & Female Estimate & Female P-value & Female r2 & Female Correlation & Combined Estimate & Combined P-value & Combined r2 & Combined Correlation & Gene by Sex Interaction P-value\\\\\n <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n\\hline\n\t dnMIS  & dnLOF        &  3.4e-02 &  8.7e-03 & 1.2e-03 &  0.0370 &  0.08900 & 2.4e-06 & 8.6e-03 &  0.11000 &  0.0530 &  4.9e-07 & 3.0e-03 &  0.06000 & 0.00560\\\\\n\t dnMIS  & dnAll        &  4.9e-01 &  0.0e+00 & 5.1e-01 &  0.7100 &  0.47000 & 0.0e+00 & 5.1e-01 &  0.72000 &  0.4800 &  0.0e+00 & 5.1e-01 &  0.72000 & 0.13000\\\\\n\t dnMIS  & inhLoF       &  4.7e-03 &  2.1e-01 & 2.7e-04 &  0.0160 & -0.00570 & 3.4e-01 & 3.6e-04 & -0.01300 &  0.0017 &  5.9e-01 & 3.5e-05 &  0.00750 & 0.19000\\\\\n\t dnMIS  & RVRS         &  7.8e-02 & 2.0e-168 & 1.2e-01 &  0.3500 &  0.08400 & 3.5e-87 & 1.4e-01 &  0.39000 &  0.0800 & 5.2e-256 & 1.3e-01 &  0.36000 & 0.17000\\\\\n\t dnMIS  & pTDT.ASD.Dev & -7.8e-03 &  5.6e-02 & 6.3e-04 & -0.0230 &  0.00096 & 8.8e-01 & 9.0e-06 &  0.01400 & -0.0053 &  1.2e-01 & 2.8e-04 & -0.01200 & 0.13000\\\\\n\t dnMIS  & pTDT.SCZ.Dev & -3.9e-05 &  9.9e-01 & 1.4e-08 &  0.0010 & -0.00170 & 8.0e-01 & 2.6e-05 & -0.00220 & -0.0007 &  8.5e-01 & 4.5e-06 & -0.00028 & 0.94000\\\\\n\t dnMIS  & pTDT.EA.Dev  & -1.8e-03 &  6.8e-01 & 2.9e-05 & -0.0029 & -0.01100 & 1.0e-01 & 1.0e-03 & -0.02100 & -0.0045 &  2.3e-01 & 1.7e-04 & -0.00880 & 0.36000\\\\\n\t dnMIS  & PRS          & -4.7e-03 &  1.0e-01 & 4.7e-04 & -0.0190 & -0.00370 & 4.2e-01 & 2.6e-04 & -0.00330 & -0.0045 &  6.6e-02 & 4.1e-04 & -0.01400 & 0.58000\\\\\n\t dnLOF  & dnMIS        &  3.5e-02 &  8.7e-03 & 1.2e-03 &  0.0370 &  0.09900 & 2.4e-06 & 8.5e-03 &  0.11000 &  0.0570 &  4.9e-07 & 3.0e-03 &  0.06000 & 0.00400\\\\\n\t dnLOF  & dnAll        &  5.1e-01 &  0.0e+00 & 5.2e-01 &  0.7300 &  0.53000 & 0.0e+00 & 5.5e-01 &  0.76000 &  0.5200 &  0.0e+00 & 5.4e-01 &  0.74000 & 0.13000\\\\\n\t dnLOF  & inhLoF       & -4.2e-03 &  2.7e-01 & 2.1e-04 & -0.0100 & -0.01600 & 1.3e-02 & 2.4e-03 & -0.04100 & -0.0076 &  2.0e-02 & 6.5e-04 & -0.02000 & 0.16000\\\\\n\t dnLOF  & RVRS         &  2.0e-01 &  0.0e+00 & 7.9e-01 &  0.8900 &  0.21000 & 0.0e+00 & 7.8e-01 &  0.90000 &  0.2000 &  0.0e+00 & 7.9e-01 &  0.89000 & 0.00170\\\\\n\t dnLOF  & pTDT.ASD.Dev & -3.9e-03 &  3.5e-01 & 1.5e-04 & -0.0046 & -0.00700 & 3.0e-01 & 4.2e-04 &  0.00120 & -0.0047 &  1.9e-01 & 2.1e-04 & -0.00300 & 0.96000\\\\\n\t dnLOF  & pTDT.SCZ.Dev &  1.0e-03 &  8.2e-01 & 8.8e-06 &  0.0071 &  0.00440 & 5.3e-01 & 1.5e-04 &  0.01700 &  0.0018 &  6.2e-01 & 2.9e-05 &  0.01000 & 0.59000\\\\\n\t dnLOF  & pTDT.EA.Dev  & -7.4e-03 &  1.0e-01 & 4.6e-04 & -0.0160 & -0.01200 & 9.9e-02 & 1.0e-03 & -0.01600 & -0.0086 &  2.6e-02 & 5.9e-04 & -0.01700 & 0.82000\\\\\n\t dnLOF  & PRS          & -4.3e-03 &  1.4e-01 & 3.7e-04 & -0.0094 & -0.00730 & 1.3e-01 & 8.9e-04 & -0.00370 & -0.0051 &  4.3e-02 & 4.9e-04 & -0.00760 & 0.98000\\\\\n\t dnAll  & dnMIS        &  1.0e+00 &  0.0e+00 & 5.0e-01 &  0.7100 &  1.10000 & 0.0e+00 & 5.0e-01 &  0.72000 &  1.1000 &  0.0e+00 & 5.1e-01 &  0.72000 & 0.00400\\\\\n\t dnAll  & dnLOF        &  1.0e+00 &  0.0e+00 & 5.2e-01 &  0.7300 &  1.10000 & 0.0e+00 & 5.5e-01 &  0.76000 &  1.1000 &  0.0e+00 & 5.4e-01 &  0.74000 & 0.00560\\\\\n\t dnAll  & inhLoF       &  4.7e-04 &  9.3e-01 & 1.3e-06 &  0.0042 & -0.02100 & 1.9e-02 & 2.1e-03 & -0.03700 & -0.0059 &  2.1e-01 & 1.9e-04 & -0.00870 & 0.06300\\\\\n\t dnAll  & RVRS         &  2.8e-01 &  0.0e+00 & 7.5e-01 &  0.8700 &  0.29000 & 0.0e+00 & 7.3e-01 &  0.88000 &  0.2800 &  0.0e+00 & 7.5e-01 &  0.87000 & 0.00016\\\\\n\t dnAll  & pTDT.ASD.Dev & -1.2e-02 &  4.9e-02 & 6.7e-04 & -0.0190 & -0.00610 & 5.3e-01 & 1.5e-04 &  0.00980 & -0.0099 &  5.0e-02 & 4.6e-04 & -0.01000 & 0.29000\\\\\n\t dnAll  & pTDT.SCZ.Dev &  9.6e-04 &  8.8e-01 & 4.0e-06 &  0.0057 &  0.00270 & 7.9e-01 & 2.8e-05 &  0.01100 &  0.0011 &  8.3e-01 & 5.4e-06 &  0.00680 & 0.74000\\\\\n\t dnAll  & pTDT.EA.Dev  & -9.2e-03 &  1.5e-01 & 3.5e-04 & -0.0140 & -0.02300 & 2.6e-02 & 1.9e-03 & -0.02500 & -0.0130 &  1.8e-02 & 6.7e-04 & -0.01800 & 0.44000\\\\\n\t dnAll  & PRS          & -9.0e-03 &  3.1e-02 & 8.1e-04 & -0.0200 & -0.01100 & 1.1e-01 & 9.6e-04 & -0.00470 & -0.0095 &  7.8e-03 & 8.4e-04 & -0.01500 & 0.72000\\\\\n\t inhLoF & dnMIS        &  5.9e-02 &  2.1e-01 & 2.7e-04 &  0.0160 & -0.06300 & 3.4e-01 & 3.6e-04 & -0.01300 &  0.0210 &  5.9e-01 & 3.5e-05 &  0.00750 & 0.17000\\\\\n\t inhLoF & dnLOF        & -5.1e-02 &  2.7e-01 & 2.1e-04 & -0.0100 & -0.16000 & 1.3e-02 & 2.4e-03 & -0.04100 & -0.0860 &  2.0e-02 & 6.5e-04 & -0.02000 & 0.18000\\\\\n\t inhLoF & dnAll        &  2.8e-03 &  9.3e-01 & 1.3e-06 &  0.0042 & -0.10000 & 1.9e-02 & 2.2e-03 & -0.03700 & -0.0330 &  2.1e-01 & 1.9e-04 & -0.00870 & 0.06600\\\\\n\t inhLoF & RVRS         &  2.2e-01 & 8.5e-107 & 8.0e-02 &  0.2800 &  0.16000 & 1.2e-29 & 4.9e-02 &  0.22000 &  0.2000 & 9.1e-133 & 6.9e-02 &  0.26000 & 0.00054\\\\\n\t inhLoF & pTDT.ASD.Dev & -1.5e-02 &  2.9e-01 & 1.9e-04 & -0.0110 & -0.02000 & 3.5e-01 & 3.5e-04 & -0.01200 & -0.0170 &  1.5e-01 & 2.5e-04 & -0.01000 & 0.91000\\\\\n\t inhLoF & pTDT.SCZ.Dev & -1.1e-02 &  4.8e-01 & 8.7e-05 & -0.0069 & -0.00380 & 8.6e-01 & 1.2e-05 &  0.00015 & -0.0081 &  5.2e-01 & 5.0e-05 & -0.00390 & 0.74000\\\\\n\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n\t pTDT.ASD.Dev & dnAll        & -0.05800 &  4.9e-02 & 6.6e-04 & -0.0190 & -0.0260 &  5.3e-01 & 1.5e-04 &  0.00980 & -0.04700 &  5.0e-02 & 4.6e-04 & -0.01000 & 0.470\\\\\n\t pTDT.ASD.Dev & inhLoF       & -0.01300 &  2.9e-01 & 1.9e-04 & -0.0110 & -0.0180 &  3.5e-01 & 3.5e-04 & -0.01200 & -0.01400 &  1.5e-01 & 2.4e-04 & -0.01000 & 0.880\\\\\n\t pTDT.ASD.Dev & RVRS         & -0.01700 &  7.9e-02 & 5.3e-04 & -0.0150 & -0.0140 &  3.0e-01 & 4.1e-04 &  0.00580 & -0.01600 &  4.4e-02 & 4.8e-04 & -0.00830 & 0.800\\\\\n\t pTDT.ASD.Dev & pTDT.SCZ.Dev &  0.11000 &  2.9e-14 & 9.9e-03 &  0.1000 &  0.0660 &  1.4e-03 & 4.0e-03 &  0.06700 &  0.09400 &  5.1e-16 & 7.8e-03 &  0.09400 & 0.120\\\\\n\t pTDT.ASD.Dev & pTDT.EA.Dev  &  0.09300 &  5.7e-11 & 7.3e-03 &  0.0920 &  0.0710 &  9.8e-04 & 4.2e-03 &  0.07500 &  0.08700 &  2.4e-13 & 6.4e-03 &  0.08800 & 0.430\\\\\n\t pTDT.ASD.Dev & PRS          &  0.58000 &  0.0e+00 & 6.7e-01 &  0.8300 &  0.5900 &  0.0e+00 & 6.6e-01 &  0.82000 &  0.58000 &  0.0e+00 & 6.7e-01 &  0.82000 & 0.370\\\\\n\t pTDT.SCZ.Dev & dnMIS        & -0.00036 &  9.9e-01 & 1.4e-08 &  0.0010 & -0.0150 &  8.0e-01 & 2.6e-05 & -0.00220 & -0.00650 &  8.5e-01 & 4.5e-06 & -0.00028 & 0.760\\\\\n\t pTDT.SCZ.Dev & dnLOF        &  0.00890 &  8.2e-01 & 8.8e-06 &  0.0071 &  0.0350 &  5.3e-01 & 1.6e-04 &  0.01700 &  0.01600 &  6.2e-01 & 2.9e-05 &  0.01000 & 0.800\\\\\n\t pTDT.SCZ.Dev & dnAll        &  0.00420 &  8.8e-01 & 4.0e-06 &  0.0057 &  0.0110 &  7.9e-01 & 2.8e-05 &  0.01100 &  0.00480 &  8.3e-01 & 5.5e-06 &  0.00680 & 0.980\\\\\n\t pTDT.SCZ.Dev & inhLoF       & -0.00800 &  4.8e-01 & 8.7e-05 & -0.0069 & -0.0031 &  8.6e-01 & 1.2e-05 &  0.00015 & -0.00620 &  5.2e-01 & 5.0e-05 & -0.00390 & 0.790\\\\\n\t pTDT.SCZ.Dev & RVRS         & -0.00150 &  8.6e-01 & 5.1e-06 &  0.0020 &  0.0019 &  8.8e-01 & 8.3e-06 &  0.01000 & -0.00082 &  9.1e-01 & 1.5e-06 &  0.00470 & 0.930\\\\\n\t pTDT.SCZ.Dev & pTDT.ASD.Dev &  0.09500 &  2.9e-14 & 1.0e-02 &  0.1000 &  0.0610 &  1.4e-03 & 4.0e-03 &  0.06700 &  0.08400 &  5.1e-16 & 7.9e-03 &  0.09400 & 0.110\\\\\n\t pTDT.SCZ.Dev & pTDT.EA.Dev  &  0.03900 &  4.1e-03 & 1.4e-03 &  0.0410 &  0.0073 &  7.2e-01 & 4.9e-05 &  0.00980 &  0.03000 &  8.5e-03 & 8.3e-04 &  0.03300 & 0.180\\\\\n\t pTDT.SCZ.Dev & PRS          &  0.24000 & 2.2e-175 & 1.3e-01 &  0.3600 &  0.2200 &  3.7e-63 & 1.0e-01 &  0.33000 &  0.23000 & 1.3e-236 & 1.2e-01 &  0.35000 & 0.220\\\\\n\t pTDT.EA.Dev  & dnMIS        & -0.01600 &  6.8e-01 & 2.9e-05 & -0.0029 & -0.0930 &  1.0e-01 & 1.0e-03 & -0.02100 & -0.03900 &  2.3e-01 & 1.7e-04 & -0.00880 & 0.320\\\\\n\t pTDT.EA.Dev  & dnLOF        & -0.06300 &  1.0e-01 & 4.6e-04 & -0.0160 & -0.0890 &  9.9e-02 & 1.1e-03 & -0.01600 & -0.07000 &  2.6e-02 & 5.9e-04 & -0.01700 & 0.830\\\\\n\t pTDT.EA.Dev  & dnAll        & -0.03900 &  1.5e-01 & 3.5e-04 & -0.0140 & -0.0830 &  2.6e-02 & 1.9e-03 & -0.02500 & -0.05200 &  1.8e-02 & 6.7e-04 & -0.01800 & 0.450\\\\\n\t pTDT.EA.Dev  & inhLoF       & -0.00560 &  6.1e-01 & 4.4e-05 & -0.0041 & -0.0015 &  9.3e-01 & 2.9e-06 &  0.00060 & -0.00420 &  6.5e-01 & 2.4e-05 & -0.00170 & 0.900\\\\\n\t pTDT.EA.Dev  & RVRS         & -0.01500 &  7.6e-02 & 5.4e-04 & -0.0180 & -0.0270 &  3.1e-02 & 1.8e-03 & -0.02100 & -0.01900 &  9.5e-03 & 8.0e-04 & -0.01900 & 0.600\\\\\n\t pTDT.EA.Dev  & pTDT.ASD.Dev &  0.08000 &  5.7e-11 & 7.4e-03 &  0.0920 &  0.0600 &  9.8e-04 & 4.2e-03 &  0.07500 &  0.07400 &  2.4e-13 & 6.4e-03 &  0.08800 & 0.420\\\\\n\t pTDT.EA.Dev  & pTDT.SCZ.Dev &  0.03700 &  4.1e-03 & 1.4e-03 &  0.0410 &  0.0067 &  7.2e-01 & 4.8e-05 &  0.00980 &  0.02800 &  8.5e-03 & 8.3e-04 &  0.03300 & 0.210\\\\\n\t pTDT.EA.Dev  & PRS          &  0.37000 &  0.0e+00 & 3.2e-01 &  0.5700 &  0.3600 & 6.8e-201 & 3.0e-01 &  0.56000 &  0.37000 &  0.0e+00 & 3.1e-01 &  0.57000 & 0.690\\\\\n\t PRS          & dnMIS        & -0.10000 &  1.0e-01 & 4.7e-04 & -0.0190 & -0.0710 &  4.2e-01 & 2.5e-04 & -0.00330 & -0.09100 &  6.6e-02 & 4.0e-04 & -0.01400 & 0.780\\\\\n\t PRS          & dnLOF        & -0.08700 &  1.4e-01 & 3.7e-04 & -0.0094 & -0.1300 &  1.3e-01 & 8.9e-04 & -0.00370 & -0.09700 &  4.3e-02 & 4.8e-04 & -0.00760 & 0.840\\\\\n\t PRS          & dnAll        & -0.09000 &  3.1e-02 & 8.0e-04 & -0.0200 & -0.0910 &  1.1e-01 & 9.8e-04 & -0.00470 & -0.08900 &  7.8e-03 & 8.4e-04 & -0.01500 & 0.910\\\\\n\t PRS          & inhLoF       & -0.02100 &  2.1e-01 & 2.7e-04 & -0.0130 & -0.0200 &  4.3e-01 & 2.4e-04 & -0.00980 & -0.02100 &  1.4e-01 & 2.6e-04 & -0.01200 & 0.960\\\\\n\t PRS          & RVRS         & -0.03000 &  2.3e-02 & 8.9e-04 & -0.0190 & -0.0360 &  5.8e-02 & 1.4e-03 & -0.00550 & -0.03200 &  3.7e-03 & 1.0e-03 & -0.01500 & 0.930\\\\\n\t PRS          & pTDT.ASD.Dev &  1.20000 &  0.0e+00 & 6.7e-01 &  0.8300 &  1.1000 &  0.0e+00 & 6.6e-01 &  0.82000 &  1.20000 &  0.0e+00 & 6.7e-01 &  0.82000 & 0.190\\\\\n\t PRS          & pTDT.SCZ.Dev &  0.54000 & 2.2e-175 & 1.3e-01 &  0.3600 &  0.4700 &  3.7e-63 & 1.0e-01 &  0.33000 &  0.52000 & 1.3e-236 & 1.2e-01 &  0.35000 & 0.047\\\\\n\t PRS          & pTDT.EA.Dev  &  0.87000 &  0.0e+00 & 3.2e-01 &  0.5700 &  0.8400 & 6.8e-201 & 3.0e-01 &  0.56000 &  0.86000 &  0.0e+00 & 3.1e-01 &  0.57000 & 0.280\\\\\n\\end{tabular}\n",
            "text/markdown": "\nA data.frame: 72 × 15\n\n| Y &lt;chr&gt; | X &lt;chr&gt; | Male Estimate &lt;dbl&gt; | Male P-value &lt;dbl&gt; | Male r2 &lt;dbl&gt; | Male Correlation &lt;dbl&gt; | Female Estimate &lt;dbl&gt; | Female P-value &lt;dbl&gt; | Female r2 &lt;dbl&gt; | Female Correlation &lt;dbl&gt; | Combined Estimate &lt;dbl&gt; | Combined P-value &lt;dbl&gt; | Combined r2 &lt;dbl&gt; | Combined Correlation &lt;dbl&gt; | Gene by Sex Interaction P-value &lt;dbl&gt; |\n|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n| dnMIS  | dnLOF        |  3.4e-02 |  8.7e-03 | 1.2e-03 |  0.0370 |  0.08900 | 2.4e-06 | 8.6e-03 |  0.11000 |  0.0530 |  4.9e-07 | 3.0e-03 |  0.06000 | 0.00560 |\n| dnMIS  | dnAll        |  4.9e-01 |  0.0e+00 | 5.1e-01 |  0.7100 |  0.47000 | 0.0e+00 | 5.1e-01 |  0.72000 |  0.4800 |  0.0e+00 | 5.1e-01 |  0.72000 | 0.13000 |\n| dnMIS  | inhLoF       |  4.7e-03 |  2.1e-01 | 2.7e-04 |  0.0160 | -0.00570 | 3.4e-01 | 3.6e-04 | -0.01300 |  0.0017 |  5.9e-01 | 3.5e-05 |  0.00750 | 0.19000 |\n| dnMIS  | RVRS         |  7.8e-02 | 2.0e-168 | 1.2e-01 |  0.3500 |  0.08400 | 3.5e-87 | 1.4e-01 |  0.39000 |  0.0800 | 5.2e-256 | 1.3e-01 |  0.36000 | 0.17000 |\n| dnMIS  | pTDT.ASD.Dev | -7.8e-03 |  5.6e-02 | 6.3e-04 | -0.0230 |  0.00096 | 8.8e-01 | 9.0e-06 |  0.01400 | -0.0053 |  1.2e-01 | 2.8e-04 | -0.01200 | 0.13000 |\n| dnMIS  | pTDT.SCZ.Dev | -3.9e-05 |  9.9e-01 | 1.4e-08 |  0.0010 | -0.00170 | 8.0e-01 | 2.6e-05 | -0.00220 | -0.0007 |  8.5e-01 | 4.5e-06 | -0.00028 | 0.94000 |\n| dnMIS  | pTDT.EA.Dev  | -1.8e-03 |  6.8e-01 | 2.9e-05 | -0.0029 | -0.01100 | 1.0e-01 | 1.0e-03 | -0.02100 | -0.0045 |  2.3e-01 | 1.7e-04 | -0.00880 | 0.36000 |\n| dnMIS  | PRS          | -4.7e-03 |  1.0e-01 | 4.7e-04 | -0.0190 | -0.00370 | 4.2e-01 | 2.6e-04 | -0.00330 | -0.0045 |  6.6e-02 | 4.1e-04 | -0.01400 | 0.58000 |\n| dnLOF  | dnMIS        |  3.5e-02 |  8.7e-03 | 1.2e-03 |  0.0370 |  0.09900 | 2.4e-06 | 8.5e-03 |  0.11000 |  0.0570 |  4.9e-07 | 3.0e-03 |  0.06000 | 0.00400 |\n| dnLOF  | dnAll        |  5.1e-01 |  0.0e+00 | 5.2e-01 |  0.7300 |  0.53000 | 0.0e+00 | 5.5e-01 |  0.76000 |  0.5200 |  0.0e+00 | 5.4e-01 |  0.74000 | 0.13000 |\n| dnLOF  | inhLoF       | -4.2e-03 |  2.7e-01 | 2.1e-04 | -0.0100 | -0.01600 | 1.3e-02 | 2.4e-03 | -0.04100 | -0.0076 |  2.0e-02 | 6.5e-04 | -0.02000 | 0.16000 |\n| dnLOF  | RVRS         |  2.0e-01 |  0.0e+00 | 7.9e-01 |  0.8900 |  0.21000 | 0.0e+00 | 7.8e-01 |  0.90000 |  0.2000 |  0.0e+00 | 7.9e-01 |  0.89000 | 0.00170 |\n| dnLOF  | pTDT.ASD.Dev | -3.9e-03 |  3.5e-01 | 1.5e-04 | -0.0046 | -0.00700 | 3.0e-01 | 4.2e-04 |  0.00120 | -0.0047 |  1.9e-01 | 2.1e-04 | -0.00300 | 0.96000 |\n| dnLOF  | pTDT.SCZ.Dev |  1.0e-03 |  8.2e-01 | 8.8e-06 |  0.0071 |  0.00440 | 5.3e-01 | 1.5e-04 |  0.01700 |  0.0018 |  6.2e-01 | 2.9e-05 |  0.01000 | 0.59000 |\n| dnLOF  | pTDT.EA.Dev  | -7.4e-03 |  1.0e-01 | 4.6e-04 | -0.0160 | -0.01200 | 9.9e-02 | 1.0e-03 | -0.01600 | -0.0086 |  2.6e-02 | 5.9e-04 | -0.01700 | 0.82000 |\n| dnLOF  | PRS          | -4.3e-03 |  1.4e-01 | 3.7e-04 | -0.0094 | -0.00730 | 1.3e-01 | 8.9e-04 | -0.00370 | -0.0051 |  4.3e-02 | 4.9e-04 | -0.00760 | 0.98000 |\n| dnAll  | dnMIS        |  1.0e+00 |  0.0e+00 | 5.0e-01 |  0.7100 |  1.10000 | 0.0e+00 | 5.0e-01 |  0.72000 |  1.1000 |  0.0e+00 | 5.1e-01 |  0.72000 | 0.00400 |\n| dnAll  | dnLOF        |  1.0e+00 |  0.0e+00 | 5.2e-01 |  0.7300 |  1.10000 | 0.0e+00 | 5.5e-01 |  0.76000 |  1.1000 |  0.0e+00 | 5.4e-01 |  0.74000 | 0.00560 |\n| dnAll  | inhLoF       |  4.7e-04 |  9.3e-01 | 1.3e-06 |  0.0042 | -0.02100 | 1.9e-02 | 2.1e-03 | -0.03700 | -0.0059 |  2.1e-01 | 1.9e-04 | -0.00870 | 0.06300 |\n| dnAll  | RVRS         |  2.8e-01 |  0.0e+00 | 7.5e-01 |  0.8700 |  0.29000 | 0.0e+00 | 7.3e-01 |  0.88000 |  0.2800 |  0.0e+00 | 7.5e-01 |  0.87000 | 0.00016 |\n| dnAll  | pTDT.ASD.Dev | -1.2e-02 |  4.9e-02 | 6.7e-04 | -0.0190 | -0.00610 | 5.3e-01 | 1.5e-04 |  0.00980 | -0.0099 |  5.0e-02 | 4.6e-04 | -0.01000 | 0.29000 |\n| dnAll  | pTDT.SCZ.Dev |  9.6e-04 |  8.8e-01 | 4.0e-06 |  0.0057 |  0.00270 | 7.9e-01 | 2.8e-05 |  0.01100 |  0.0011 |  8.3e-01 | 5.4e-06 |  0.00680 | 0.74000 |\n| dnAll  | pTDT.EA.Dev  | -9.2e-03 |  1.5e-01 | 3.5e-04 | -0.0140 | -0.02300 | 2.6e-02 | 1.9e-03 | -0.02500 | -0.0130 |  1.8e-02 | 6.7e-04 | -0.01800 | 0.44000 |\n| dnAll  | PRS          | -9.0e-03 |  3.1e-02 | 8.1e-04 | -0.0200 | -0.01100 | 1.1e-01 | 9.6e-04 | -0.00470 | -0.0095 |  7.8e-03 | 8.4e-04 | -0.01500 | 0.72000 |\n| inhLoF | dnMIS        |  5.9e-02 |  2.1e-01 | 2.7e-04 |  0.0160 | -0.06300 | 3.4e-01 | 3.6e-04 | -0.01300 |  0.0210 |  5.9e-01 | 3.5e-05 |  0.00750 | 0.17000 |\n| inhLoF | dnLOF        | -5.1e-02 |  2.7e-01 | 2.1e-04 | -0.0100 | -0.16000 | 1.3e-02 | 2.4e-03 | -0.04100 | -0.0860 |  2.0e-02 | 6.5e-04 | -0.02000 | 0.18000 |\n| inhLoF | dnAll        |  2.8e-03 |  9.3e-01 | 1.3e-06 |  0.0042 | -0.10000 | 1.9e-02 | 2.2e-03 | -0.03700 | -0.0330 |  2.1e-01 | 1.9e-04 | -0.00870 | 0.06600 |\n| inhLoF | RVRS         |  2.2e-01 | 8.5e-107 | 8.0e-02 |  0.2800 |  0.16000 | 1.2e-29 | 4.9e-02 |  0.22000 |  0.2000 | 9.1e-133 | 6.9e-02 |  0.26000 | 0.00054 |\n| inhLoF | pTDT.ASD.Dev | -1.5e-02 |  2.9e-01 | 1.9e-04 | -0.0110 | -0.02000 | 3.5e-01 | 3.5e-04 | -0.01200 | -0.0170 |  1.5e-01 | 2.5e-04 | -0.01000 | 0.91000 |\n| inhLoF | pTDT.SCZ.Dev | -1.1e-02 |  4.8e-01 | 8.7e-05 | -0.0069 | -0.00380 | 8.6e-01 | 1.2e-05 |  0.00015 | -0.0081 |  5.2e-01 | 5.0e-05 | -0.00390 | 0.74000 |\n| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n| pTDT.ASD.Dev | dnAll        | -0.05800 |  4.9e-02 | 6.6e-04 | -0.0190 | -0.0260 |  5.3e-01 | 1.5e-04 |  0.00980 | -0.04700 |  5.0e-02 | 4.6e-04 | -0.01000 | 0.470 |\n| pTDT.ASD.Dev | inhLoF       | -0.01300 |  2.9e-01 | 1.9e-04 | -0.0110 | -0.0180 |  3.5e-01 | 3.5e-04 | -0.01200 | -0.01400 |  1.5e-01 | 2.4e-04 | -0.01000 | 0.880 |\n| pTDT.ASD.Dev | RVRS         | -0.01700 |  7.9e-02 | 5.3e-04 | -0.0150 | -0.0140 |  3.0e-01 | 4.1e-04 |  0.00580 | -0.01600 |  4.4e-02 | 4.8e-04 | -0.00830 | 0.800 |\n| pTDT.ASD.Dev | pTDT.SCZ.Dev |  0.11000 |  2.9e-14 | 9.9e-03 |  0.1000 |  0.0660 |  1.4e-03 | 4.0e-03 |  0.06700 |  0.09400 |  5.1e-16 | 7.8e-03 |  0.09400 | 0.120 |\n| pTDT.ASD.Dev | pTDT.EA.Dev  |  0.09300 |  5.7e-11 | 7.3e-03 |  0.0920 |  0.0710 |  9.8e-04 | 4.2e-03 |  0.07500 |  0.08700 |  2.4e-13 | 6.4e-03 |  0.08800 | 0.430 |\n| pTDT.ASD.Dev | PRS          |  0.58000 |  0.0e+00 | 6.7e-01 |  0.8300 |  0.5900 |  0.0e+00 | 6.6e-01 |  0.82000 |  0.58000 |  0.0e+00 | 6.7e-01 |  0.82000 | 0.370 |\n| pTDT.SCZ.Dev | dnMIS        | -0.00036 |  9.9e-01 | 1.4e-08 |  0.0010 | -0.0150 |  8.0e-01 | 2.6e-05 | -0.00220 | -0.00650 |  8.5e-01 | 4.5e-06 | -0.00028 | 0.760 |\n| pTDT.SCZ.Dev | dnLOF        |  0.00890 |  8.2e-01 | 8.8e-06 |  0.0071 |  0.0350 |  5.3e-01 | 1.6e-04 |  0.01700 |  0.01600 |  6.2e-01 | 2.9e-05 |  0.01000 | 0.800 |\n| pTDT.SCZ.Dev | dnAll        |  0.00420 |  8.8e-01 | 4.0e-06 |  0.0057 |  0.0110 |  7.9e-01 | 2.8e-05 |  0.01100 |  0.00480 |  8.3e-01 | 5.5e-06 |  0.00680 | 0.980 |\n| pTDT.SCZ.Dev | inhLoF       | -0.00800 |  4.8e-01 | 8.7e-05 | -0.0069 | -0.0031 |  8.6e-01 | 1.2e-05 |  0.00015 | -0.00620 |  5.2e-01 | 5.0e-05 | -0.00390 | 0.790 |\n| pTDT.SCZ.Dev | RVRS         | -0.00150 |  8.6e-01 | 5.1e-06 |  0.0020 |  0.0019 |  8.8e-01 | 8.3e-06 |  0.01000 | -0.00082 |  9.1e-01 | 1.5e-06 |  0.00470 | 0.930 |\n| pTDT.SCZ.Dev | pTDT.ASD.Dev |  0.09500 |  2.9e-14 | 1.0e-02 |  0.1000 |  0.0610 |  1.4e-03 | 4.0e-03 |  0.06700 |  0.08400 |  5.1e-16 | 7.9e-03 |  0.09400 | 0.110 |\n| pTDT.SCZ.Dev | pTDT.EA.Dev  |  0.03900 |  4.1e-03 | 1.4e-03 |  0.0410 |  0.0073 |  7.2e-01 | 4.9e-05 |  0.00980 |  0.03000 |  8.5e-03 | 8.3e-04 |  0.03300 | 0.180 |\n| pTDT.SCZ.Dev | PRS          |  0.24000 | 2.2e-175 | 1.3e-01 |  0.3600 |  0.2200 |  3.7e-63 | 1.0e-01 |  0.33000 |  0.23000 | 1.3e-236 | 1.2e-01 |  0.35000 | 0.220 |\n| pTDT.EA.Dev  | dnMIS        | -0.01600 |  6.8e-01 | 2.9e-05 | -0.0029 | -0.0930 |  1.0e-01 | 1.0e-03 | -0.02100 | -0.03900 |  2.3e-01 | 1.7e-04 | -0.00880 | 0.320 |\n| pTDT.EA.Dev  | dnLOF        | -0.06300 |  1.0e-01 | 4.6e-04 | -0.0160 | -0.0890 |  9.9e-02 | 1.1e-03 | -0.01600 | -0.07000 |  2.6e-02 | 5.9e-04 | -0.01700 | 0.830 |\n| pTDT.EA.Dev  | dnAll        | -0.03900 |  1.5e-01 | 3.5e-04 | -0.0140 | -0.0830 |  2.6e-02 | 1.9e-03 | -0.02500 | -0.05200 |  1.8e-02 | 6.7e-04 | -0.01800 | 0.450 |\n| pTDT.EA.Dev  | inhLoF       | -0.00560 |  6.1e-01 | 4.4e-05 | -0.0041 | -0.0015 |  9.3e-01 | 2.9e-06 |  0.00060 | -0.00420 |  6.5e-01 | 2.4e-05 | -0.00170 | 0.900 |\n| pTDT.EA.Dev  | RVRS         | -0.01500 |  7.6e-02 | 5.4e-04 | -0.0180 | -0.0270 |  3.1e-02 | 1.8e-03 | -0.02100 | -0.01900 |  9.5e-03 | 8.0e-04 | -0.01900 | 0.600 |\n| pTDT.EA.Dev  | pTDT.ASD.Dev |  0.08000 |  5.7e-11 | 7.4e-03 |  0.0920 |  0.0600 |  9.8e-04 | 4.2e-03 |  0.07500 |  0.07400 |  2.4e-13 | 6.4e-03 |  0.08800 | 0.420 |\n| pTDT.EA.Dev  | pTDT.SCZ.Dev |  0.03700 |  4.1e-03 | 1.4e-03 |  0.0410 |  0.0067 |  7.2e-01 | 4.8e-05 |  0.00980 |  0.02800 |  8.5e-03 | 8.3e-04 |  0.03300 | 0.210 |\n| pTDT.EA.Dev  | PRS          |  0.37000 |  0.0e+00 | 3.2e-01 |  0.5700 |  0.3600 | 6.8e-201 | 3.0e-01 |  0.56000 |  0.37000 |  0.0e+00 | 3.1e-01 |  0.57000 | 0.690 |\n| PRS          | dnMIS        | -0.10000 |  1.0e-01 | 4.7e-04 | -0.0190 | -0.0710 |  4.2e-01 | 2.5e-04 | -0.00330 | -0.09100 |  6.6e-02 | 4.0e-04 | -0.01400 | 0.780 |\n| PRS          | dnLOF        | -0.08700 |  1.4e-01 | 3.7e-04 | -0.0094 | -0.1300 |  1.3e-01 | 8.9e-04 | -0.00370 | -0.09700 |  4.3e-02 | 4.8e-04 | -0.00760 | 0.840 |\n| PRS          | dnAll        | -0.09000 |  3.1e-02 | 8.0e-04 | -0.0200 | -0.0910 |  1.1e-01 | 9.8e-04 | -0.00470 | -0.08900 |  7.8e-03 | 8.4e-04 | -0.01500 | 0.910 |\n| PRS          | inhLoF       | -0.02100 |  2.1e-01 | 2.7e-04 | -0.0130 | -0.0200 |  4.3e-01 | 2.4e-04 | -0.00980 | -0.02100 |  1.4e-01 | 2.6e-04 | -0.01200 | 0.960 |\n| PRS          | RVRS         | -0.03000 |  2.3e-02 | 8.9e-04 | -0.0190 | -0.0360 |  5.8e-02 | 1.4e-03 | -0.00550 | -0.03200 |  3.7e-03 | 1.0e-03 | -0.01500 | 0.930 |\n| PRS          | pTDT.ASD.Dev |  1.20000 |  0.0e+00 | 6.7e-01 |  0.8300 |  1.1000 |  0.0e+00 | 6.6e-01 |  0.82000 |  1.20000 |  0.0e+00 | 6.7e-01 |  0.82000 | 0.190 |\n| PRS          | pTDT.SCZ.Dev |  0.54000 | 2.2e-175 | 1.3e-01 |  0.3600 |  0.4700 |  3.7e-63 | 1.0e-01 |  0.33000 |  0.52000 | 1.3e-236 | 1.2e-01 |  0.35000 | 0.047 |\n| PRS          | pTDT.EA.Dev  |  0.87000 |  0.0e+00 | 3.2e-01 |  0.5700 |  0.8400 | 6.8e-201 | 3.0e-01 |  0.56000 |  0.86000 |  0.0e+00 | 3.1e-01 |  0.57000 | 0.280 |\n\n",
            "text/html": [
              "<table class=\"dataframe\">\n",
              "<caption>A data.frame: 72 × 15</caption>\n",
              "<thead>\n",
              "\t<tr><th scope=col>Y</th><th scope=col>X</th><th scope=col>Male Estimate</th><th scope=col>Male P-value</th><th scope=col>Male r2</th><th scope=col>Male Correlation</th><th scope=col>Female Estimate</th><th scope=col>Female P-value</th><th scope=col>Female r2</th><th scope=col>Female Correlation</th><th scope=col>Combined Estimate</th><th scope=col>Combined P-value</th><th scope=col>Combined r2</th><th scope=col>Combined Correlation</th><th scope=col>Gene by Sex Interaction P-value</th></tr>\n",
              "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
              "</thead>\n",
              "<tbody>\n",
              "\t<tr><td>dnMIS </td><td>dnLOF       </td><td> 3.4e-02</td><td> 8.7e-03</td><td>1.2e-03</td><td> 0.0370</td><td> 0.08900</td><td>2.4e-06</td><td>8.6e-03</td><td> 0.11000</td><td> 0.0530</td><td> 4.9e-07</td><td>3.0e-03</td><td> 0.06000</td><td>0.00560</td></tr>\n",
              "\t<tr><td>dnMIS </td><td>dnAll       </td><td> 4.9e-01</td><td> 0.0e+00</td><td>5.1e-01</td><td> 0.7100</td><td> 0.47000</td><td>0.0e+00</td><td>5.1e-01</td><td> 0.72000</td><td> 0.4800</td><td> 0.0e+00</td><td>5.1e-01</td><td> 0.72000</td><td>0.13000</td></tr>\n",
              "\t<tr><td>dnMIS </td><td>inhLoF      </td><td> 4.7e-03</td><td> 2.1e-01</td><td>2.7e-04</td><td> 0.0160</td><td>-0.00570</td><td>3.4e-01</td><td>3.6e-04</td><td>-0.01300</td><td> 0.0017</td><td> 5.9e-01</td><td>3.5e-05</td><td> 0.00750</td><td>0.19000</td></tr>\n",
              "\t<tr><td>dnMIS </td><td>RVRS        </td><td> 7.8e-02</td><td>2.0e-168</td><td>1.2e-01</td><td> 0.3500</td><td> 0.08400</td><td>3.5e-87</td><td>1.4e-01</td><td> 0.39000</td><td> 0.0800</td><td>5.2e-256</td><td>1.3e-01</td><td> 0.36000</td><td>0.17000</td></tr>\n",
              "\t<tr><td>dnMIS </td><td>pTDT.ASD.Dev</td><td>-7.8e-03</td><td> 5.6e-02</td><td>6.3e-04</td><td>-0.0230</td><td> 0.00096</td><td>8.8e-01</td><td>9.0e-06</td><td> 0.01400</td><td>-0.0053</td><td> 1.2e-01</td><td>2.8e-04</td><td>-0.01200</td><td>0.13000</td></tr>\n",
              "\t<tr><td>dnMIS </td><td>pTDT.SCZ.Dev</td><td>-3.9e-05</td><td> 9.9e-01</td><td>1.4e-08</td><td> 0.0010</td><td>-0.00170</td><td>8.0e-01</td><td>2.6e-05</td><td>-0.00220</td><td>-0.0007</td><td> 8.5e-01</td><td>4.5e-06</td><td>-0.00028</td><td>0.94000</td></tr>\n",
              "\t<tr><td>dnMIS </td><td>pTDT.EA.Dev </td><td>-1.8e-03</td><td> 6.8e-01</td><td>2.9e-05</td><td>-0.0029</td><td>-0.01100</td><td>1.0e-01</td><td>1.0e-03</td><td>-0.02100</td><td>-0.0045</td><td> 2.3e-01</td><td>1.7e-04</td><td>-0.00880</td><td>0.36000</td></tr>\n",
              "\t<tr><td>dnMIS </td><td>PRS         </td><td>-4.7e-03</td><td> 1.0e-01</td><td>4.7e-04</td><td>-0.0190</td><td>-0.00370</td><td>4.2e-01</td><td>2.6e-04</td><td>-0.00330</td><td>-0.0045</td><td> 6.6e-02</td><td>4.1e-04</td><td>-0.01400</td><td>0.58000</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>dnMIS       </td><td> 3.5e-02</td><td> 8.7e-03</td><td>1.2e-03</td><td> 0.0370</td><td> 0.09900</td><td>2.4e-06</td><td>8.5e-03</td><td> 0.11000</td><td> 0.0570</td><td> 4.9e-07</td><td>3.0e-03</td><td> 0.06000</td><td>0.00400</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>dnAll       </td><td> 5.1e-01</td><td> 0.0e+00</td><td>5.2e-01</td><td> 0.7300</td><td> 0.53000</td><td>0.0e+00</td><td>5.5e-01</td><td> 0.76000</td><td> 0.5200</td><td> 0.0e+00</td><td>5.4e-01</td><td> 0.74000</td><td>0.13000</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>inhLoF      </td><td>-4.2e-03</td><td> 2.7e-01</td><td>2.1e-04</td><td>-0.0100</td><td>-0.01600</td><td>1.3e-02</td><td>2.4e-03</td><td>-0.04100</td><td>-0.0076</td><td> 2.0e-02</td><td>6.5e-04</td><td>-0.02000</td><td>0.16000</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>RVRS        </td><td> 2.0e-01</td><td> 0.0e+00</td><td>7.9e-01</td><td> 0.8900</td><td> 0.21000</td><td>0.0e+00</td><td>7.8e-01</td><td> 0.90000</td><td> 0.2000</td><td> 0.0e+00</td><td>7.9e-01</td><td> 0.89000</td><td>0.00170</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>pTDT.ASD.Dev</td><td>-3.9e-03</td><td> 3.5e-01</td><td>1.5e-04</td><td>-0.0046</td><td>-0.00700</td><td>3.0e-01</td><td>4.2e-04</td><td> 0.00120</td><td>-0.0047</td><td> 1.9e-01</td><td>2.1e-04</td><td>-0.00300</td><td>0.96000</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>pTDT.SCZ.Dev</td><td> 1.0e-03</td><td> 8.2e-01</td><td>8.8e-06</td><td> 0.0071</td><td> 0.00440</td><td>5.3e-01</td><td>1.5e-04</td><td> 0.01700</td><td> 0.0018</td><td> 6.2e-01</td><td>2.9e-05</td><td> 0.01000</td><td>0.59000</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>pTDT.EA.Dev </td><td>-7.4e-03</td><td> 1.0e-01</td><td>4.6e-04</td><td>-0.0160</td><td>-0.01200</td><td>9.9e-02</td><td>1.0e-03</td><td>-0.01600</td><td>-0.0086</td><td> 2.6e-02</td><td>5.9e-04</td><td>-0.01700</td><td>0.82000</td></tr>\n",
              "\t<tr><td>dnLOF </td><td>PRS         </td><td>-4.3e-03</td><td> 1.4e-01</td><td>3.7e-04</td><td>-0.0094</td><td>-0.00730</td><td>1.3e-01</td><td>8.9e-04</td><td>-0.00370</td><td>-0.0051</td><td> 4.3e-02</td><td>4.9e-04</td><td>-0.00760</td><td>0.98000</td></tr>\n",
              "\t<tr><td>dnAll </td><td>dnMIS       </td><td> 1.0e+00</td><td> 0.0e+00</td><td>5.0e-01</td><td> 0.7100</td><td> 1.10000</td><td>0.0e+00</td><td>5.0e-01</td><td> 0.72000</td><td> 1.1000</td><td> 0.0e+00</td><td>5.1e-01</td><td> 0.72000</td><td>0.00400</td></tr>\n",
              "\t<tr><td>dnAll </td><td>dnLOF       </td><td> 1.0e+00</td><td> 0.0e+00</td><td>5.2e-01</td><td> 0.7300</td><td> 1.10000</td><td>0.0e+00</td><td>5.5e-01</td><td> 0.76000</td><td> 1.1000</td><td> 0.0e+00</td><td>5.4e-01</td><td> 0.74000</td><td>0.00560</td></tr>\n",
              "\t<tr><td>dnAll </td><td>inhLoF      </td><td> 4.7e-04</td><td> 9.3e-01</td><td>1.3e-06</td><td> 0.0042</td><td>-0.02100</td><td>1.9e-02</td><td>2.1e-03</td><td>-0.03700</td><td>-0.0059</td><td> 2.1e-01</td><td>1.9e-04</td><td>-0.00870</td><td>0.06300</td></tr>\n",
              "\t<tr><td>dnAll </td><td>RVRS        </td><td> 2.8e-01</td><td> 0.0e+00</td><td>7.5e-01</td><td> 0.8700</td><td> 0.29000</td><td>0.0e+00</td><td>7.3e-01</td><td> 0.88000</td><td> 0.2800</td><td> 0.0e+00</td><td>7.5e-01</td><td> 0.87000</td><td>0.00016</td></tr>\n",
              "\t<tr><td>dnAll </td><td>pTDT.ASD.Dev</td><td>-1.2e-02</td><td> 4.9e-02</td><td>6.7e-04</td><td>-0.0190</td><td>-0.00610</td><td>5.3e-01</td><td>1.5e-04</td><td> 0.00980</td><td>-0.0099</td><td> 5.0e-02</td><td>4.6e-04</td><td>-0.01000</td><td>0.29000</td></tr>\n",
              "\t<tr><td>dnAll </td><td>pTDT.SCZ.Dev</td><td> 9.6e-04</td><td> 8.8e-01</td><td>4.0e-06</td><td> 0.0057</td><td> 0.00270</td><td>7.9e-01</td><td>2.8e-05</td><td> 0.01100</td><td> 0.0011</td><td> 8.3e-01</td><td>5.4e-06</td><td> 0.00680</td><td>0.74000</td></tr>\n",
              "\t<tr><td>dnAll </td><td>pTDT.EA.Dev </td><td>-9.2e-03</td><td> 1.5e-01</td><td>3.5e-04</td><td>-0.0140</td><td>-0.02300</td><td>2.6e-02</td><td>1.9e-03</td><td>-0.02500</td><td>-0.0130</td><td> 1.8e-02</td><td>6.7e-04</td><td>-0.01800</td><td>0.44000</td></tr>\n",
              "\t<tr><td>dnAll </td><td>PRS         </td><td>-9.0e-03</td><td> 3.1e-02</td><td>8.1e-04</td><td>-0.0200</td><td>-0.01100</td><td>1.1e-01</td><td>9.6e-04</td><td>-0.00470</td><td>-0.0095</td><td> 7.8e-03</td><td>8.4e-04</td><td>-0.01500</td><td>0.72000</td></tr>\n",
              "\t<tr><td>inhLoF</td><td>dnMIS       </td><td> 5.9e-02</td><td> 2.1e-01</td><td>2.7e-04</td><td> 0.0160</td><td>-0.06300</td><td>3.4e-01</td><td>3.6e-04</td><td>-0.01300</td><td> 0.0210</td><td> 5.9e-01</td><td>3.5e-05</td><td> 0.00750</td><td>0.17000</td></tr>\n",
              "\t<tr><td>inhLoF</td><td>dnLOF       </td><td>-5.1e-02</td><td> 2.7e-01</td><td>2.1e-04</td><td>-0.0100</td><td>-0.16000</td><td>1.3e-02</td><td>2.4e-03</td><td>-0.04100</td><td>-0.0860</td><td> 2.0e-02</td><td>6.5e-04</td><td>-0.02000</td><td>0.18000</td></tr>\n",
              "\t<tr><td>inhLoF</td><td>dnAll       </td><td> 2.8e-03</td><td> 9.3e-01</td><td>1.3e-06</td><td> 0.0042</td><td>-0.10000</td><td>1.9e-02</td><td>2.2e-03</td><td>-0.03700</td><td>-0.0330</td><td> 2.1e-01</td><td>1.9e-04</td><td>-0.00870</td><td>0.06600</td></tr>\n",
              "\t<tr><td>inhLoF</td><td>RVRS        </td><td> 2.2e-01</td><td>8.5e-107</td><td>8.0e-02</td><td> 0.2800</td><td> 0.16000</td><td>1.2e-29</td><td>4.9e-02</td><td> 0.22000</td><td> 0.2000</td><td>9.1e-133</td><td>6.9e-02</td><td> 0.26000</td><td>0.00054</td></tr>\n",
              "\t<tr><td>inhLoF</td><td>pTDT.ASD.Dev</td><td>-1.5e-02</td><td> 2.9e-01</td><td>1.9e-04</td><td>-0.0110</td><td>-0.02000</td><td>3.5e-01</td><td>3.5e-04</td><td>-0.01200</td><td>-0.0170</td><td> 1.5e-01</td><td>2.5e-04</td><td>-0.01000</td><td>0.91000</td></tr>\n",
              "\t<tr><td>inhLoF</td><td>pTDT.SCZ.Dev</td><td>-1.1e-02</td><td> 4.8e-01</td><td>8.7e-05</td><td>-0.0069</td><td>-0.00380</td><td>8.6e-01</td><td>1.2e-05</td><td> 0.00015</td><td>-0.0081</td><td> 5.2e-01</td><td>5.0e-05</td><td>-0.00390</td><td>0.74000</td></tr>\n",
              "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
              "\t<tr><td>pTDT.ASD.Dev</td><td>dnAll       </td><td>-0.05800</td><td> 4.9e-02</td><td>6.6e-04</td><td>-0.0190</td><td>-0.0260</td><td> 5.3e-01</td><td>1.5e-04</td><td> 0.00980</td><td>-0.04700</td><td> 5.0e-02</td><td>4.6e-04</td><td>-0.01000</td><td>0.470</td></tr>\n",
              "\t<tr><td>pTDT.ASD.Dev</td><td>inhLoF      </td><td>-0.01300</td><td> 2.9e-01</td><td>1.9e-04</td><td>-0.0110</td><td>-0.0180</td><td> 3.5e-01</td><td>3.5e-04</td><td>-0.01200</td><td>-0.01400</td><td> 1.5e-01</td><td>2.4e-04</td><td>-0.01000</td><td>0.880</td></tr>\n",
              "\t<tr><td>pTDT.ASD.Dev</td><td>RVRS        </td><td>-0.01700</td><td> 7.9e-02</td><td>5.3e-04</td><td>-0.0150</td><td>-0.0140</td><td> 3.0e-01</td><td>4.1e-04</td><td> 0.00580</td><td>-0.01600</td><td> 4.4e-02</td><td>4.8e-04</td><td>-0.00830</td><td>0.800</td></tr>\n",
              "\t<tr><td>pTDT.ASD.Dev</td><td>pTDT.SCZ.Dev</td><td> 0.11000</td><td> 2.9e-14</td><td>9.9e-03</td><td> 0.1000</td><td> 0.0660</td><td> 1.4e-03</td><td>4.0e-03</td><td> 0.06700</td><td> 0.09400</td><td> 5.1e-16</td><td>7.8e-03</td><td> 0.09400</td><td>0.120</td></tr>\n",
              "\t<tr><td>pTDT.ASD.Dev</td><td>pTDT.EA.Dev </td><td> 0.09300</td><td> 5.7e-11</td><td>7.3e-03</td><td> 0.0920</td><td> 0.0710</td><td> 9.8e-04</td><td>4.2e-03</td><td> 0.07500</td><td> 0.08700</td><td> 2.4e-13</td><td>6.4e-03</td><td> 0.08800</td><td>0.430</td></tr>\n",
              "\t<tr><td>pTDT.ASD.Dev</td><td>PRS         </td><td> 0.58000</td><td> 0.0e+00</td><td>6.7e-01</td><td> 0.8300</td><td> 0.5900</td><td> 0.0e+00</td><td>6.6e-01</td><td> 0.82000</td><td> 0.58000</td><td> 0.0e+00</td><td>6.7e-01</td><td> 0.82000</td><td>0.370</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>dnMIS       </td><td>-0.00036</td><td> 9.9e-01</td><td>1.4e-08</td><td> 0.0010</td><td>-0.0150</td><td> 8.0e-01</td><td>2.6e-05</td><td>-0.00220</td><td>-0.00650</td><td> 8.5e-01</td><td>4.5e-06</td><td>-0.00028</td><td>0.760</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>dnLOF       </td><td> 0.00890</td><td> 8.2e-01</td><td>8.8e-06</td><td> 0.0071</td><td> 0.0350</td><td> 5.3e-01</td><td>1.6e-04</td><td> 0.01700</td><td> 0.01600</td><td> 6.2e-01</td><td>2.9e-05</td><td> 0.01000</td><td>0.800</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>dnAll       </td><td> 0.00420</td><td> 8.8e-01</td><td>4.0e-06</td><td> 0.0057</td><td> 0.0110</td><td> 7.9e-01</td><td>2.8e-05</td><td> 0.01100</td><td> 0.00480</td><td> 8.3e-01</td><td>5.5e-06</td><td> 0.00680</td><td>0.980</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>inhLoF      </td><td>-0.00800</td><td> 4.8e-01</td><td>8.7e-05</td><td>-0.0069</td><td>-0.0031</td><td> 8.6e-01</td><td>1.2e-05</td><td> 0.00015</td><td>-0.00620</td><td> 5.2e-01</td><td>5.0e-05</td><td>-0.00390</td><td>0.790</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>RVRS        </td><td>-0.00150</td><td> 8.6e-01</td><td>5.1e-06</td><td> 0.0020</td><td> 0.0019</td><td> 8.8e-01</td><td>8.3e-06</td><td> 0.01000</td><td>-0.00082</td><td> 9.1e-01</td><td>1.5e-06</td><td> 0.00470</td><td>0.930</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>pTDT.ASD.Dev</td><td> 0.09500</td><td> 2.9e-14</td><td>1.0e-02</td><td> 0.1000</td><td> 0.0610</td><td> 1.4e-03</td><td>4.0e-03</td><td> 0.06700</td><td> 0.08400</td><td> 5.1e-16</td><td>7.9e-03</td><td> 0.09400</td><td>0.110</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>pTDT.EA.Dev </td><td> 0.03900</td><td> 4.1e-03</td><td>1.4e-03</td><td> 0.0410</td><td> 0.0073</td><td> 7.2e-01</td><td>4.9e-05</td><td> 0.00980</td><td> 0.03000</td><td> 8.5e-03</td><td>8.3e-04</td><td> 0.03300</td><td>0.180</td></tr>\n",
              "\t<tr><td>pTDT.SCZ.Dev</td><td>PRS         </td><td> 0.24000</td><td>2.2e-175</td><td>1.3e-01</td><td> 0.3600</td><td> 0.2200</td><td> 3.7e-63</td><td>1.0e-01</td><td> 0.33000</td><td> 0.23000</td><td>1.3e-236</td><td>1.2e-01</td><td> 0.35000</td><td>0.220</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>dnMIS       </td><td>-0.01600</td><td> 6.8e-01</td><td>2.9e-05</td><td>-0.0029</td><td>-0.0930</td><td> 1.0e-01</td><td>1.0e-03</td><td>-0.02100</td><td>-0.03900</td><td> 2.3e-01</td><td>1.7e-04</td><td>-0.00880</td><td>0.320</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>dnLOF       </td><td>-0.06300</td><td> 1.0e-01</td><td>4.6e-04</td><td>-0.0160</td><td>-0.0890</td><td> 9.9e-02</td><td>1.1e-03</td><td>-0.01600</td><td>-0.07000</td><td> 2.6e-02</td><td>5.9e-04</td><td>-0.01700</td><td>0.830</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>dnAll       </td><td>-0.03900</td><td> 1.5e-01</td><td>3.5e-04</td><td>-0.0140</td><td>-0.0830</td><td> 2.6e-02</td><td>1.9e-03</td><td>-0.02500</td><td>-0.05200</td><td> 1.8e-02</td><td>6.7e-04</td><td>-0.01800</td><td>0.450</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>inhLoF      </td><td>-0.00560</td><td> 6.1e-01</td><td>4.4e-05</td><td>-0.0041</td><td>-0.0015</td><td> 9.3e-01</td><td>2.9e-06</td><td> 0.00060</td><td>-0.00420</td><td> 6.5e-01</td><td>2.4e-05</td><td>-0.00170</td><td>0.900</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>RVRS        </td><td>-0.01500</td><td> 7.6e-02</td><td>5.4e-04</td><td>-0.0180</td><td>-0.0270</td><td> 3.1e-02</td><td>1.8e-03</td><td>-0.02100</td><td>-0.01900</td><td> 9.5e-03</td><td>8.0e-04</td><td>-0.01900</td><td>0.600</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>pTDT.ASD.Dev</td><td> 0.08000</td><td> 5.7e-11</td><td>7.4e-03</td><td> 0.0920</td><td> 0.0600</td><td> 9.8e-04</td><td>4.2e-03</td><td> 0.07500</td><td> 0.07400</td><td> 2.4e-13</td><td>6.4e-03</td><td> 0.08800</td><td>0.420</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>pTDT.SCZ.Dev</td><td> 0.03700</td><td> 4.1e-03</td><td>1.4e-03</td><td> 0.0410</td><td> 0.0067</td><td> 7.2e-01</td><td>4.8e-05</td><td> 0.00980</td><td> 0.02800</td><td> 8.5e-03</td><td>8.3e-04</td><td> 0.03300</td><td>0.210</td></tr>\n",
              "\t<tr><td>pTDT.EA.Dev </td><td>PRS         </td><td> 0.37000</td><td> 0.0e+00</td><td>3.2e-01</td><td> 0.5700</td><td> 0.3600</td><td>6.8e-201</td><td>3.0e-01</td><td> 0.56000</td><td> 0.37000</td><td> 0.0e+00</td><td>3.1e-01</td><td> 0.57000</td><td>0.690</td></tr>\n",
              "\t<tr><td>PRS         </td><td>dnMIS       </td><td>-0.10000</td><td> 1.0e-01</td><td>4.7e-04</td><td>-0.0190</td><td>-0.0710</td><td> 4.2e-01</td><td>2.5e-04</td><td>-0.00330</td><td>-0.09100</td><td> 6.6e-02</td><td>4.0e-04</td><td>-0.01400</td><td>0.780</td></tr>\n",
              "\t<tr><td>PRS         </td><td>dnLOF       </td><td>-0.08700</td><td> 1.4e-01</td><td>3.7e-04</td><td>-0.0094</td><td>-0.1300</td><td> 1.3e-01</td><td>8.9e-04</td><td>-0.00370</td><td>-0.09700</td><td> 4.3e-02</td><td>4.8e-04</td><td>-0.00760</td><td>0.840</td></tr>\n",
              "\t<tr><td>PRS         </td><td>dnAll       </td><td>-0.09000</td><td> 3.1e-02</td><td>8.0e-04</td><td>-0.0200</td><td>-0.0910</td><td> 1.1e-01</td><td>9.8e-04</td><td>-0.00470</td><td>-0.08900</td><td> 7.8e-03</td><td>8.4e-04</td><td>-0.01500</td><td>0.910</td></tr>\n",
              "\t<tr><td>PRS         </td><td>inhLoF      </td><td>-0.02100</td><td> 2.1e-01</td><td>2.7e-04</td><td>-0.0130</td><td>-0.0200</td><td> 4.3e-01</td><td>2.4e-04</td><td>-0.00980</td><td>-0.02100</td><td> 1.4e-01</td><td>2.6e-04</td><td>-0.01200</td><td>0.960</td></tr>\n",
              "\t<tr><td>PRS         </td><td>RVRS        </td><td>-0.03000</td><td> 2.3e-02</td><td>8.9e-04</td><td>-0.0190</td><td>-0.0360</td><td> 5.8e-02</td><td>1.4e-03</td><td>-0.00550</td><td>-0.03200</td><td> 3.7e-03</td><td>1.0e-03</td><td>-0.01500</td><td>0.930</td></tr>\n",
              "\t<tr><td>PRS         </td><td>pTDT.ASD.Dev</td><td> 1.20000</td><td> 0.0e+00</td><td>6.7e-01</td><td> 0.8300</td><td> 1.1000</td><td> 0.0e+00</td><td>6.6e-01</td><td> 0.82000</td><td> 1.20000</td><td> 0.0e+00</td><td>6.7e-01</td><td> 0.82000</td><td>0.190</td></tr>\n",
              "\t<tr><td>PRS         </td><td>pTDT.SCZ.Dev</td><td> 0.54000</td><td>2.2e-175</td><td>1.3e-01</td><td> 0.3600</td><td> 0.4700</td><td> 3.7e-63</td><td>1.0e-01</td><td> 0.33000</td><td> 0.52000</td><td>1.3e-236</td><td>1.2e-01</td><td> 0.35000</td><td>0.047</td></tr>\n",
              "\t<tr><td>PRS         </td><td>pTDT.EA.Dev </td><td> 0.87000</td><td> 0.0e+00</td><td>3.2e-01</td><td> 0.5700</td><td> 0.8400</td><td>6.8e-201</td><td>3.0e-01</td><td> 0.56000</td><td> 0.86000</td><td> 0.0e+00</td><td>3.1e-01</td><td> 0.57000</td><td>0.280</td></tr>\n",
              "</tbody>\n",
              "</table>\n"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 272
        },
        "id": "2WaG4CIvX7B3",
        "outputId": "7dd6559a-4dab-4ebe-f5cd-ed6b235ead61"
      },
      "source": [
        "# Read data\n",
        "master_phen <- read.csv(\"master_phen_4.csv\", header = T)\n",
        "colnames(master_phen)\n",
        "# get Fig.5 variables\n",
        "cols <- c(\"phen\",\"sex\",\"cohort\",\"duo\",\"is_eur\",\"dnLoF\",\"dnMIS\",\"inhLoF\",\"PS.ASD.Z\",\n",
        "          \"PS.EA.Z\",\"PS.SZ.Z\",\"rbs.child.z\",\"scq.child.z\",\"srs.child.z\",\n",
        "          \"vabs.child.z\",\"dcdq.child.z\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\",\"PC6\",\n",
        "          \"PC7\",\"PC8\",\"PC9\",\"PC10\")\n",
        "master_phen2 <- master_phen[,cols]\n",
        "master_phen2[,\"is_eur\"] <- as.factor(master_phen2[,\"is_eur\"])\n",
        "# pick cases where is_eur = 1   and   duo = \"trio\"\n",
        "europ <- which(master_phen2[,\"is_eur\"]==1)\n",
        "trio <- which(master_phen2[,\"duo\"]==\"trio\")\n",
        "pick <- intersect(europ,trio)\n",
        "master_phen_filtered <- master_phen2[pick,]\n",
        "# 00:44\n",
        "# > length(which(is.na(master_phen_filtered$rbs.child.z)))\n",
        "# [1] 5086\n",
        "# > dim(master_phen_filtered)\n",
        "# [1] 9354   26\n",
        "# > dim(na.omit(master_phen_filtered))\n",
        "# [1] 819  26\n",
        "\n",
        "\n",
        "length(which(is.na(master_phen_filtered$rbs.child.z)))\n",
        "\n",
        "dim(master_phen_filtered)\n",
        "\n",
        "dim(na.omit(master_phen_filtered))"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              " [1] \"fid\"                           \"iid\"                          \n",
              " [3] \"phen\"                          \"sex\"                          \n",
              " [5] \"cohort\"                        \"duo\"                          \n",
              " [7] \"family\"                        \"is_eur\"                       \n",
              " [9] \"dnLoF\"                         \"dnMIS\"                        \n",
              "[11] \"inhLoF\"                        \"InhLoF.Paternal.Transmitted\"  \n",
              "[13] \"InhLoF.Paternal.Untransmitted\" \"InhLoF.Maternal.Transmitted\"  \n",
              "[15] \"InhLoF.Maternal.Untransmitted\" \"PS.ASD.Z\"                     \n",
              "[17] \"Paternal.PS.ASD.Z\"             \"Maternal.PS.ASD.Z\"            \n",
              "[19] \"PS.EA.Z\"                       \"Paternal.PS.EA.Z\"             \n",
              "[21] \"Maternal.PS.EA.Z\"              \"PS.SZ.Z\"                      \n",
              "[23] \"Paternal.PS.SZ.Z\"              \"Maternal.PS.SZ.Z\"             \n",
              "[25] \"PC1\"                           \"PC2\"                          \n",
              "[27] \"PC3\"                           \"PC4\"                          \n",
              "[29] \"PC5\"                           \"PC6\"                          \n",
              "[31] \"PC7\"                           \"PC8\"                          \n",
              "[33] \"PC9\"                           \"PC10\"                         \n",
              "[35] \"PC1_PAT\"                       \"PC2_PAT\"                      \n",
              "[37] \"PC3_PAT\"                       \"PC4_PAT\"                      \n",
              "[39] \"PC5_PAT\"                       \"PC6_PAT\"                      \n",
              "[41] \"PC7_PAT\"                       \"PC8_PAT\"                      \n",
              "[43] \"PC9_PAT\"                       \"PC10_PAT\"                     \n",
              "[45] \"PC1_MAT\"                       \"PC2_MAT\"                      \n",
              "[47] \"PC3_MAT\"                       \"PC4_MAT\"                      \n",
              "[49] \"PC5_MAT\"                       \"PC6_MAT\"                      \n",
              "[51] \"PC7_MAT\"                       \"PC8_MAT\"                      \n",
              "[53] \"PC9_MAT\"                       \"PC10_MAT\"                     \n",
              "[55] \"srs.child.age\"                 \"vabs.child.age\"               \n",
              "[57] \"scq.child.age\"                 \"dcdq.child.age\"               \n",
              "[59] \"mother.EA\"                     \"father.EA\"                    \n",
              "[61] \"srs.mother\"                    \"srs.father\"                   \n",
              "[63] \"mother.bapq\"                   \"father.bapq\"                  \n",
              "[65] \"srs.child.z\"                   \"vabs.child.z\"                 \n",
              "[67] \"rbs.child.z\"                   \"scq.child.z\"                  \n",
              "[69] \"dcdq.child.z\"                  \"father.age.at.birth.z\"        \n",
              "[71] \"mother.age.at.birth.z\"         \"father.EA.z\"                  \n",
              "[73] \"mother.EA.z\"                   \"father.srs.z\"                 \n",
              "[75] \"mother.srs.z\"                  \"father.bapq.z\"                \n",
              "[77] \"mother.bapq.z\"                "
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 'fid'\n\\item 'iid'\n\\item 'phen'\n\\item 'sex'\n\\item 'cohort'\n\\item 'duo'\n\\item 'family'\n\\item 'is\\_eur'\n\\item 'dnLoF'\n\\item 'dnMIS'\n\\item 'inhLoF'\n\\item 'InhLoF.Paternal.Transmitted'\n\\item 'InhLoF.Paternal.Untransmitted'\n\\item 'InhLoF.Maternal.Transmitted'\n\\item 'InhLoF.Maternal.Untransmitted'\n\\item 'PS.ASD.Z'\n\\item 'Paternal.PS.ASD.Z'\n\\item 'Maternal.PS.ASD.Z'\n\\item 'PS.EA.Z'\n\\item 'Paternal.PS.EA.Z'\n\\item 'Maternal.PS.EA.Z'\n\\item 'PS.SZ.Z'\n\\item 'Paternal.PS.SZ.Z'\n\\item 'Maternal.PS.SZ.Z'\n\\item 'PC1'\n\\item 'PC2'\n\\item 'PC3'\n\\item 'PC4'\n\\item 'PC5'\n\\item 'PC6'\n\\item 'PC7'\n\\item 'PC8'\n\\item 'PC9'\n\\item 'PC10'\n\\item 'PC1\\_PAT'\n\\item 'PC2\\_PAT'\n\\item 'PC3\\_PAT'\n\\item 'PC4\\_PAT'\n\\item 'PC5\\_PAT'\n\\item 'PC6\\_PAT'\n\\item 'PC7\\_PAT'\n\\item 'PC8\\_PAT'\n\\item 'PC9\\_PAT'\n\\item 'PC10\\_PAT'\n\\item 'PC1\\_MAT'\n\\item 'PC2\\_MAT'\n\\item 'PC3\\_MAT'\n\\item 'PC4\\_MAT'\n\\item 'PC5\\_MAT'\n\\item 'PC6\\_MAT'\n\\item 'PC7\\_MAT'\n\\item 'PC8\\_MAT'\n\\item 'PC9\\_MAT'\n\\item 'PC10\\_MAT'\n\\item 'srs.child.age'\n\\item 'vabs.child.age'\n\\item 'scq.child.age'\n\\item 'dcdq.child.age'\n\\item 'mother.EA'\n\\item 'father.EA'\n\\item 'srs.mother'\n\\item 'srs.father'\n\\item 'mother.bapq'\n\\item 'father.bapq'\n\\item 'srs.child.z'\n\\item 'vabs.child.z'\n\\item 'rbs.child.z'\n\\item 'scq.child.z'\n\\item 'dcdq.child.z'\n\\item 'father.age.at.birth.z'\n\\item 'mother.age.at.birth.z'\n\\item 'father.EA.z'\n\\item 'mother.EA.z'\n\\item 'father.srs.z'\n\\item 'mother.srs.z'\n\\item 'father.bapq.z'\n\\item 'mother.bapq.z'\n\\end{enumerate*}\n",
            "text/markdown": "1. 'fid'\n2. 'iid'\n3. 'phen'\n4. 'sex'\n5. 'cohort'\n6. 'duo'\n7. 'family'\n8. 'is_eur'\n9. 'dnLoF'\n10. 'dnMIS'\n11. 'inhLoF'\n12. 'InhLoF.Paternal.Transmitted'\n13. 'InhLoF.Paternal.Untransmitted'\n14. 'InhLoF.Maternal.Transmitted'\n15. 'InhLoF.Maternal.Untransmitted'\n16. 'PS.ASD.Z'\n17. 'Paternal.PS.ASD.Z'\n18. 'Maternal.PS.ASD.Z'\n19. 'PS.EA.Z'\n20. 'Paternal.PS.EA.Z'\n21. 'Maternal.PS.EA.Z'\n22. 'PS.SZ.Z'\n23. 'Paternal.PS.SZ.Z'\n24. 'Maternal.PS.SZ.Z'\n25. 'PC1'\n26. 'PC2'\n27. 'PC3'\n28. 'PC4'\n29. 'PC5'\n30. 'PC6'\n31. 'PC7'\n32. 'PC8'\n33. 'PC9'\n34. 'PC10'\n35. 'PC1_PAT'\n36. 'PC2_PAT'\n37. 'PC3_PAT'\n38. 'PC4_PAT'\n39. 'PC5_PAT'\n40. 'PC6_PAT'\n41. 'PC7_PAT'\n42. 'PC8_PAT'\n43. 'PC9_PAT'\n44. 'PC10_PAT'\n45. 'PC1_MAT'\n46. 'PC2_MAT'\n47. 'PC3_MAT'\n48. 'PC4_MAT'\n49. 'PC5_MAT'\n50. 'PC6_MAT'\n51. 'PC7_MAT'\n52. 'PC8_MAT'\n53. 'PC9_MAT'\n54. 'PC10_MAT'\n55. 'srs.child.age'\n56. 'vabs.child.age'\n57. 'scq.child.age'\n58. 'dcdq.child.age'\n59. 'mother.EA'\n60. 'father.EA'\n61. 'srs.mother'\n62. 'srs.father'\n63. 'mother.bapq'\n64. 'father.bapq'\n65. 'srs.child.z'\n66. 'vabs.child.z'\n67. 'rbs.child.z'\n68. 'scq.child.z'\n69. 'dcdq.child.z'\n70. 'father.age.at.birth.z'\n71. 'mother.age.at.birth.z'\n72. 'father.EA.z'\n73. 'mother.EA.z'\n74. 'father.srs.z'\n75. 'mother.srs.z'\n76. 'father.bapq.z'\n77. 'mother.bapq.z'\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'fid'</li><li>'iid'</li><li>'phen'</li><li>'sex'</li><li>'cohort'</li><li>'duo'</li><li>'family'</li><li>'is_eur'</li><li>'dnLoF'</li><li>'dnMIS'</li><li>'inhLoF'</li><li>'InhLoF.Paternal.Transmitted'</li><li>'InhLoF.Paternal.Untransmitted'</li><li>'InhLoF.Maternal.Transmitted'</li><li>'InhLoF.Maternal.Untransmitted'</li><li>'PS.ASD.Z'</li><li>'Paternal.PS.ASD.Z'</li><li>'Maternal.PS.ASD.Z'</li><li>'PS.EA.Z'</li><li>'Paternal.PS.EA.Z'</li><li>'Maternal.PS.EA.Z'</li><li>'PS.SZ.Z'</li><li>'Paternal.PS.SZ.Z'</li><li>'Maternal.PS.SZ.Z'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'PC8'</li><li>'PC9'</li><li>'PC10'</li><li>'PC1_PAT'</li><li>'PC2_PAT'</li><li>'PC3_PAT'</li><li>'PC4_PAT'</li><li>'PC5_PAT'</li><li>'PC6_PAT'</li><li>'PC7_PAT'</li><li>'PC8_PAT'</li><li>'PC9_PAT'</li><li>'PC10_PAT'</li><li>'PC1_MAT'</li><li>'PC2_MAT'</li><li>'PC3_MAT'</li><li>'PC4_MAT'</li><li>'PC5_MAT'</li><li>'PC6_MAT'</li><li>'PC7_MAT'</li><li>'PC8_MAT'</li><li>'PC9_MAT'</li><li>'PC10_MAT'</li><li>'srs.child.age'</li><li>'vabs.child.age'</li><li>'scq.child.age'</li><li>'dcdq.child.age'</li><li>'mother.EA'</li><li>'father.EA'</li><li>'srs.mother'</li><li>'srs.father'</li><li>'mother.bapq'</li><li>'father.bapq'</li><li>'srs.child.z'</li><li>'vabs.child.z'</li><li>'rbs.child.z'</li><li>'scq.child.z'</li><li>'dcdq.child.z'</li><li>'father.age.at.birth.z'</li><li>'mother.age.at.birth.z'</li><li>'father.EA.z'</li><li>'mother.EA.z'</li><li>'father.srs.z'</li><li>'mother.srs.z'</li><li>'father.bapq.z'</li><li>'mother.bapq.z'</li></ol>\n"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1] 0"
            ],
            "text/latex": "0",
            "text/markdown": "0",
            "text/html": [
              "0"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1]  0 26"
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 0\n\\item 26\n\\end{enumerate*}\n",
            "text/markdown": "1. 0\n2. 26\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>0</li><li>26</li></ol>\n"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1]  0 26"
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 0\n\\item 26\n\\end{enumerate*}\n",
            "text/markdown": "1. 0\n2. 26\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>0</li><li>26</li></ol>\n"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 272
        },
        "id": "MTdSQy0PZX1d",
        "outputId": "31a0a599-d93e-42c3-b1e2-417a0685f7cc"
      },
      "source": [
        "# Read data\n",
        "master_phen <- read.csv(\"master_phen_4.SBayesR.20210915.csv\", header = T)\n",
        "colnames(master_phen)\n",
        "# get Fig.5 variables\n",
        "cols <- c(\"phen\",\"sex\",\"cohort\",\"duo\",\"is_eur\",\"dnLoF\",\"dnMIS\",\"inhLoF\",\"PS.ASD.Z\",\n",
        "          \"PS.EA.Z\",\"PS.SZ.Z\",\"rbs.child.z\",\"scq.child.z\",\"srs.child.z\",\n",
        "          \"vabs.child.z\",\"dcdq.child.z\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\",\"PC6\",\n",
        "          \"PC7\",\"PC8\",\"PC9\",\"PC10\")\n",
        "master_phen2 <- master_phen[,cols]\n",
        "master_phen2[,\"is_eur\"] <- as.factor(master_phen2[,\"is_eur\"])\n",
        "# pick cases where is_eur = 1   and   duo = \"trio\"\n",
        "europ <- which(master_phen2[,\"is_eur\"]==1)\n",
        "trio <- which(master_phen2[,\"duo\"]==\"trio\")\n",
        "pick <- intersect(europ,trio)\n",
        "master_phen_filtered <- master_phen2[pick,]\n",
        "# 00:44\n",
        "# > length(which(is.na(master_phen_filtered$rbs.child.z)))\n",
        "# [1] 5086\n",
        "# > dim(master_phen_filtered)\n",
        "# [1] 9354   26\n",
        "# > dim(na.omit(master_phen_filtered))\n",
        "# [1] 819  26\n",
        "\n",
        "\n",
        "length(which(is.na(master_phen_filtered$rbs.child.z)))\n",
        "\n",
        "dim(master_phen_filtered)\n",
        "\n",
        "dim(na.omit(master_phen_filtered))"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              " [1] \"fid\"                           \"iid\"                          \n",
              " [3] \"phen\"                          \"sex\"                          \n",
              " [5] \"cohort\"                        \"duo\"                          \n",
              " [7] \"family\"                        \"is_eur\"                       \n",
              " [9] \"dnLoF\"                         \"dnMIS\"                        \n",
              "[11] \"inhLoF\"                        \"InhLoF.Paternal.Transmitted\"  \n",
              "[13] \"InhLoF.Paternal.Untransmitted\" \"InhLoF.Maternal.Transmitted\"  \n",
              "[15] \"InhLoF.Maternal.Untransmitted\" \"PC1\"                          \n",
              "[17] \"PC2\"                           \"PC3\"                          \n",
              "[19] \"PC4\"                           \"PC5\"                          \n",
              "[21] \"PC6\"                           \"PC7\"                          \n",
              "[23] \"PC8\"                           \"PC9\"                          \n",
              "[25] \"PC10\"                          \"PC1_PAT\"                      \n",
              "[27] \"PC2_PAT\"                       \"PC3_PAT\"                      \n",
              "[29] \"PC4_PAT\"                       \"PC5_PAT\"                      \n",
              "[31] \"PC6_PAT\"                       \"PC7_PAT\"                      \n",
              "[33] \"PC8_PAT\"                       \"PC9_PAT\"                      \n",
              "[35] \"PC10_PAT\"                      \"PC1_MAT\"                      \n",
              "[37] \"PC2_MAT\"                       \"PC3_MAT\"                      \n",
              "[39] \"PC4_MAT\"                       \"PC5_MAT\"                      \n",
              "[41] \"PC6_MAT\"                       \"PC7_MAT\"                      \n",
              "[43] \"PC8_MAT\"                       \"PC9_MAT\"                      \n",
              "[45] \"PC10_MAT\"                      \"srs.child.age\"                \n",
              "[47] \"vabs.child.age\"                \"scq.child.age\"                \n",
              "[49] \"dcdq.child.age\"                \"mother.EA\"                    \n",
              "[51] \"father.EA\"                     \"srs.mother\"                   \n",
              "[53] \"srs.father\"                    \"mother.bapq\"                  \n",
              "[55] \"father.bapq\"                   \"srs.child.z\"                  \n",
              "[57] \"vabs.child.z\"                  \"rbs.child.z\"                  \n",
              "[59] \"scq.child.z\"                   \"dcdq.child.z\"                 \n",
              "[61] \"father.age.at.birth.z\"         \"mother.age.at.birth.z\"        \n",
              "[63] \"father.EA.z\"                   \"mother.EA.z\"                  \n",
              "[65] \"father.srs.z\"                  \"mother.srs.z\"                 \n",
              "[67] \"father.bapq.z\"                 \"mother.bapq.z\"                \n",
              "[69] \"PS.ASD.Z\"                      \"PS.EA.Z\"                      \n",
              "[71] \"PS.SZ.Z\"                       \"Paternal.PS.ASD.Z\"            \n",
              "[73] \"Paternal.PS.EA.Z\"              \"Paternal.PS.SZ.Z\"             \n",
              "[75] \"Maternal.PS.ASD.Z\"             \"Maternal.PS.EA.Z\"             \n",
              "[77] \"Maternal.PS.SZ.Z\"             "
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 'fid'\n\\item 'iid'\n\\item 'phen'\n\\item 'sex'\n\\item 'cohort'\n\\item 'duo'\n\\item 'family'\n\\item 'is\\_eur'\n\\item 'dnLoF'\n\\item 'dnMIS'\n\\item 'inhLoF'\n\\item 'InhLoF.Paternal.Transmitted'\n\\item 'InhLoF.Paternal.Untransmitted'\n\\item 'InhLoF.Maternal.Transmitted'\n\\item 'InhLoF.Maternal.Untransmitted'\n\\item 'PC1'\n\\item 'PC2'\n\\item 'PC3'\n\\item 'PC4'\n\\item 'PC5'\n\\item 'PC6'\n\\item 'PC7'\n\\item 'PC8'\n\\item 'PC9'\n\\item 'PC10'\n\\item 'PC1\\_PAT'\n\\item 'PC2\\_PAT'\n\\item 'PC3\\_PAT'\n\\item 'PC4\\_PAT'\n\\item 'PC5\\_PAT'\n\\item 'PC6\\_PAT'\n\\item 'PC7\\_PAT'\n\\item 'PC8\\_PAT'\n\\item 'PC9\\_PAT'\n\\item 'PC10\\_PAT'\n\\item 'PC1\\_MAT'\n\\item 'PC2\\_MAT'\n\\item 'PC3\\_MAT'\n\\item 'PC4\\_MAT'\n\\item 'PC5\\_MAT'\n\\item 'PC6\\_MAT'\n\\item 'PC7\\_MAT'\n\\item 'PC8\\_MAT'\n\\item 'PC9\\_MAT'\n\\item 'PC10\\_MAT'\n\\item 'srs.child.age'\n\\item 'vabs.child.age'\n\\item 'scq.child.age'\n\\item 'dcdq.child.age'\n\\item 'mother.EA'\n\\item 'father.EA'\n\\item 'srs.mother'\n\\item 'srs.father'\n\\item 'mother.bapq'\n\\item 'father.bapq'\n\\item 'srs.child.z'\n\\item 'vabs.child.z'\n\\item 'rbs.child.z'\n\\item 'scq.child.z'\n\\item 'dcdq.child.z'\n\\item 'father.age.at.birth.z'\n\\item 'mother.age.at.birth.z'\n\\item 'father.EA.z'\n\\item 'mother.EA.z'\n\\item 'father.srs.z'\n\\item 'mother.srs.z'\n\\item 'father.bapq.z'\n\\item 'mother.bapq.z'\n\\item 'PS.ASD.Z'\n\\item 'PS.EA.Z'\n\\item 'PS.SZ.Z'\n\\item 'Paternal.PS.ASD.Z'\n\\item 'Paternal.PS.EA.Z'\n\\item 'Paternal.PS.SZ.Z'\n\\item 'Maternal.PS.ASD.Z'\n\\item 'Maternal.PS.EA.Z'\n\\item 'Maternal.PS.SZ.Z'\n\\end{enumerate*}\n",
            "text/markdown": "1. 'fid'\n2. 'iid'\n3. 'phen'\n4. 'sex'\n5. 'cohort'\n6. 'duo'\n7. 'family'\n8. 'is_eur'\n9. 'dnLoF'\n10. 'dnMIS'\n11. 'inhLoF'\n12. 'InhLoF.Paternal.Transmitted'\n13. 'InhLoF.Paternal.Untransmitted'\n14. 'InhLoF.Maternal.Transmitted'\n15. 'InhLoF.Maternal.Untransmitted'\n16. 'PC1'\n17. 'PC2'\n18. 'PC3'\n19. 'PC4'\n20. 'PC5'\n21. 'PC6'\n22. 'PC7'\n23. 'PC8'\n24. 'PC9'\n25. 'PC10'\n26. 'PC1_PAT'\n27. 'PC2_PAT'\n28. 'PC3_PAT'\n29. 'PC4_PAT'\n30. 'PC5_PAT'\n31. 'PC6_PAT'\n32. 'PC7_PAT'\n33. 'PC8_PAT'\n34. 'PC9_PAT'\n35. 'PC10_PAT'\n36. 'PC1_MAT'\n37. 'PC2_MAT'\n38. 'PC3_MAT'\n39. 'PC4_MAT'\n40. 'PC5_MAT'\n41. 'PC6_MAT'\n42. 'PC7_MAT'\n43. 'PC8_MAT'\n44. 'PC9_MAT'\n45. 'PC10_MAT'\n46. 'srs.child.age'\n47. 'vabs.child.age'\n48. 'scq.child.age'\n49. 'dcdq.child.age'\n50. 'mother.EA'\n51. 'father.EA'\n52. 'srs.mother'\n53. 'srs.father'\n54. 'mother.bapq'\n55. 'father.bapq'\n56. 'srs.child.z'\n57. 'vabs.child.z'\n58. 'rbs.child.z'\n59. 'scq.child.z'\n60. 'dcdq.child.z'\n61. 'father.age.at.birth.z'\n62. 'mother.age.at.birth.z'\n63. 'father.EA.z'\n64. 'mother.EA.z'\n65. 'father.srs.z'\n66. 'mother.srs.z'\n67. 'father.bapq.z'\n68. 'mother.bapq.z'\n69. 'PS.ASD.Z'\n70. 'PS.EA.Z'\n71. 'PS.SZ.Z'\n72. 'Paternal.PS.ASD.Z'\n73. 'Paternal.PS.EA.Z'\n74. 'Paternal.PS.SZ.Z'\n75. 'Maternal.PS.ASD.Z'\n76. 'Maternal.PS.EA.Z'\n77. 'Maternal.PS.SZ.Z'\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'fid'</li><li>'iid'</li><li>'phen'</li><li>'sex'</li><li>'cohort'</li><li>'duo'</li><li>'family'</li><li>'is_eur'</li><li>'dnLoF'</li><li>'dnMIS'</li><li>'inhLoF'</li><li>'InhLoF.Paternal.Transmitted'</li><li>'InhLoF.Paternal.Untransmitted'</li><li>'InhLoF.Maternal.Transmitted'</li><li>'InhLoF.Maternal.Untransmitted'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'PC8'</li><li>'PC9'</li><li>'PC10'</li><li>'PC1_PAT'</li><li>'PC2_PAT'</li><li>'PC3_PAT'</li><li>'PC4_PAT'</li><li>'PC5_PAT'</li><li>'PC6_PAT'</li><li>'PC7_PAT'</li><li>'PC8_PAT'</li><li>'PC9_PAT'</li><li>'PC10_PAT'</li><li>'PC1_MAT'</li><li>'PC2_MAT'</li><li>'PC3_MAT'</li><li>'PC4_MAT'</li><li>'PC5_MAT'</li><li>'PC6_MAT'</li><li>'PC7_MAT'</li><li>'PC8_MAT'</li><li>'PC9_MAT'</li><li>'PC10_MAT'</li><li>'srs.child.age'</li><li>'vabs.child.age'</li><li>'scq.child.age'</li><li>'dcdq.child.age'</li><li>'mother.EA'</li><li>'father.EA'</li><li>'srs.mother'</li><li>'srs.father'</li><li>'mother.bapq'</li><li>'father.bapq'</li><li>'srs.child.z'</li><li>'vabs.child.z'</li><li>'rbs.child.z'</li><li>'scq.child.z'</li><li>'dcdq.child.z'</li><li>'father.age.at.birth.z'</li><li>'mother.age.at.birth.z'</li><li>'father.EA.z'</li><li>'mother.EA.z'</li><li>'father.srs.z'</li><li>'mother.srs.z'</li><li>'father.bapq.z'</li><li>'mother.bapq.z'</li><li>'PS.ASD.Z'</li><li>'PS.EA.Z'</li><li>'PS.SZ.Z'</li><li>'Paternal.PS.ASD.Z'</li><li>'Paternal.PS.EA.Z'</li><li>'Paternal.PS.SZ.Z'</li><li>'Maternal.PS.ASD.Z'</li><li>'Maternal.PS.EA.Z'</li><li>'Maternal.PS.SZ.Z'</li></ol>\n"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1] 5086"
            ],
            "text/latex": "5086",
            "text/markdown": "5086",
            "text/html": [
              "5086"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1] 9354   26"
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 9354\n\\item 26\n\\end{enumerate*}\n",
            "text/markdown": "1. 9354\n2. 26\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>9354</li><li>26</li></ol>\n"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1] 819  26"
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 819\n\\item 26\n\\end{enumerate*}\n",
            "text/markdown": "1. 819\n2. 26\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>819</li><li>26</li></ol>\n"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 187
        },
        "id": "JrPrhhC5bEjX",
        "outputId": "e90de257-854d-46f8-86ec-4a4ad020e550"
      },
      "source": [
        "# Read data\n",
        "master_phen <- read.csv(\"master_phen_4.SBayesR.csv\", header = T)\n",
        "colnames(master_phen)\n",
        "# get Fig.5 variables\n",
        "cols <- c(\"phen\",\"sex\",\"cohort\",\"duo\",\"is_eur\",\"dnLoF\",\"dnMIS\",\"inhLoF\",\"PS.ASD.Z\",\n",
        "          \"PS.EA.Z\",\"PS.SZ.Z\",\"rbs.child.z\",\"scq.child.z\",\"srs.child.z\",\n",
        "          \"vabs.child.z\",\"dcdq.child.z\",\"PC1\",\"PC2\",\"PC3\",\"PC4\",\"PC5\",\"PC6\",\n",
        "          \"PC7\",\"PC8\",\"PC9\",\"PC10\")\n",
        "master_phen2 <- master_phen[,cols]\n",
        "master_phen2[,\"is_eur\"] <- as.factor(master_phen2[,\"is_eur\"])\n",
        "# pick cases where is_eur = 1   and   duo = \"trio\"\n",
        "europ <- which(master_phen2[,\"is_eur\"]==1)\n",
        "trio <- which(master_phen2[,\"duo\"]==\"trio\")\n",
        "pick <- intersect(europ,trio)\n",
        "master_phen_filtered <- master_phen2[pick,]\n",
        "# 00:44\n",
        "# > length(which(is.na(master_phen_filtered$rbs.child.z)))\n",
        "# [1] 5086\n",
        "# > dim(master_phen_filtered)\n",
        "# [1] 9354   26\n",
        "# > dim(na.omit(master_phen_filtered))\n",
        "# [1] 819  26\n",
        "\n",
        "\n",
        "length(which(is.na(master_phen_filtered$rbs.child.z)))\n",
        "\n",
        "dim(master_phen_filtered)\n",
        "\n",
        "dim(na.omit(master_phen_filtered))"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              " [1] \"fid\"                           \"iid\"                          \n",
              " [3] \"phen\"                          \"sex\"                          \n",
              " [5] \"cohort\"                        \"duo\"                          \n",
              " [7] \"family\"                        \"is_eur\"                       \n",
              " [9] \"dnLoF\"                         \"dnMIS\"                        \n",
              "[11] \"inhLoF\"                        \"InhLoF.Paternal.Transmitted\"  \n",
              "[13] \"InhLoF.Paternal.Untransmitted\" \"InhLoF.Maternal.Transmitted\"  \n",
              "[15] \"InhLoF.Maternal.Untransmitted\" \"PC1\"                          \n",
              "[17] \"PC2\"                           \"PC3\"                          \n",
              "[19] \"PC4\"                           \"PC5\"                          \n",
              "[21] \"PC6\"                           \"PC7\"                          \n",
              "[23] \"PC8\"                           \"PC9\"                          \n",
              "[25] \"PC10\"                          \"PC1_PAT\"                      \n",
              "[27] \"PC2_PAT\"                       \"PC3_PAT\"                      \n",
              "[29] \"PC4_PAT\"                       \"PC5_PAT\"                      \n",
              "[31] \"PC6_PAT\"                       \"PC7_PAT\"                      \n",
              "[33] \"PC8_PAT\"                       \"PC9_PAT\"                      \n",
              "[35] \"PC10_PAT\"                      \"PC1_MAT\"                      \n",
              "[37] \"PC2_MAT\"                       \"PC3_MAT\"                      \n",
              "[39] \"PC4_MAT\"                       \"PC5_MAT\"                      \n",
              "[41] \"PC6_MAT\"                       \"PC7_MAT\"                      \n",
              "[43] \"PC8_MAT\"                       \"PC9_MAT\"                      \n",
              "[45] \"PC10_MAT\"                      \"srs.child.age\"                \n",
              "[47] \"vabs.child.age\"                \"scq.child.age\"                \n",
              "[49] \"dcdq.child.age\"                \"mother.EA\"                    \n",
              "[51] \"father.EA\"                     \"srs.mother\"                   \n",
              "[53] \"srs.father\"                    \"mother.bapq\"                  \n",
              "[55] \"father.bapq\"                   \"srs.child.z\"                  \n",
              "[57] \"vabs.child.z\"                  \"rbs.child.z\"                  \n",
              "[59] \"scq.child.z\"                   \"dcdq.child.z\"                 \n",
              "[61] \"father.age.at.birth.z\"         \"mother.age.at.birth.z\"        \n",
              "[63] \"father.EA.z\"                   \"mother.EA.z\"                  \n",
              "[65] \"father.srs.z\"                  \"mother.srs.z\"                 \n",
              "[67] \"father.bapq.z\"                 \"mother.bapq.z\"                \n",
              "[69] \"PS.ASD.Z\"                      \"PS.EA.Z\"                      \n",
              "[71] \"PS.SZ.Z\"                       \"Paternal.PS.ASD.Z\"            \n",
              "[73] \"Paternal.PS.EA.Z\"              \"Paternal.PS.SZ.Z\"             \n",
              "[75] \"Maternal.PS.ASD.Z\"             \"Maternal.PS.EA.Z\"             \n",
              "[77] \"Maternal.PS.SZ.Z\"             "
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 'fid'\n\\item 'iid'\n\\item 'phen'\n\\item 'sex'\n\\item 'cohort'\n\\item 'duo'\n\\item 'family'\n\\item 'is\\_eur'\n\\item 'dnLoF'\n\\item 'dnMIS'\n\\item 'inhLoF'\n\\item 'InhLoF.Paternal.Transmitted'\n\\item 'InhLoF.Paternal.Untransmitted'\n\\item 'InhLoF.Maternal.Transmitted'\n\\item 'InhLoF.Maternal.Untransmitted'\n\\item 'PC1'\n\\item 'PC2'\n\\item 'PC3'\n\\item 'PC4'\n\\item 'PC5'\n\\item 'PC6'\n\\item 'PC7'\n\\item 'PC8'\n\\item 'PC9'\n\\item 'PC10'\n\\item 'PC1\\_PAT'\n\\item 'PC2\\_PAT'\n\\item 'PC3\\_PAT'\n\\item 'PC4\\_PAT'\n\\item 'PC5\\_PAT'\n\\item 'PC6\\_PAT'\n\\item 'PC7\\_PAT'\n\\item 'PC8\\_PAT'\n\\item 'PC9\\_PAT'\n\\item 'PC10\\_PAT'\n\\item 'PC1\\_MAT'\n\\item 'PC2\\_MAT'\n\\item 'PC3\\_MAT'\n\\item 'PC4\\_MAT'\n\\item 'PC5\\_MAT'\n\\item 'PC6\\_MAT'\n\\item 'PC7\\_MAT'\n\\item 'PC8\\_MAT'\n\\item 'PC9\\_MAT'\n\\item 'PC10\\_MAT'\n\\item 'srs.child.age'\n\\item 'vabs.child.age'\n\\item 'scq.child.age'\n\\item 'dcdq.child.age'\n\\item 'mother.EA'\n\\item 'father.EA'\n\\item 'srs.mother'\n\\item 'srs.father'\n\\item 'mother.bapq'\n\\item 'father.bapq'\n\\item 'srs.child.z'\n\\item 'vabs.child.z'\n\\item 'rbs.child.z'\n\\item 'scq.child.z'\n\\item 'dcdq.child.z'\n\\item 'father.age.at.birth.z'\n\\item 'mother.age.at.birth.z'\n\\item 'father.EA.z'\n\\item 'mother.EA.z'\n\\item 'father.srs.z'\n\\item 'mother.srs.z'\n\\item 'father.bapq.z'\n\\item 'mother.bapq.z'\n\\item 'PS.ASD.Z'\n\\item 'PS.EA.Z'\n\\item 'PS.SZ.Z'\n\\item 'Paternal.PS.ASD.Z'\n\\item 'Paternal.PS.EA.Z'\n\\item 'Paternal.PS.SZ.Z'\n\\item 'Maternal.PS.ASD.Z'\n\\item 'Maternal.PS.EA.Z'\n\\item 'Maternal.PS.SZ.Z'\n\\end{enumerate*}\n",
            "text/markdown": "1. 'fid'\n2. 'iid'\n3. 'phen'\n4. 'sex'\n5. 'cohort'\n6. 'duo'\n7. 'family'\n8. 'is_eur'\n9. 'dnLoF'\n10. 'dnMIS'\n11. 'inhLoF'\n12. 'InhLoF.Paternal.Transmitted'\n13. 'InhLoF.Paternal.Untransmitted'\n14. 'InhLoF.Maternal.Transmitted'\n15. 'InhLoF.Maternal.Untransmitted'\n16. 'PC1'\n17. 'PC2'\n18. 'PC3'\n19. 'PC4'\n20. 'PC5'\n21. 'PC6'\n22. 'PC7'\n23. 'PC8'\n24. 'PC9'\n25. 'PC10'\n26. 'PC1_PAT'\n27. 'PC2_PAT'\n28. 'PC3_PAT'\n29. 'PC4_PAT'\n30. 'PC5_PAT'\n31. 'PC6_PAT'\n32. 'PC7_PAT'\n33. 'PC8_PAT'\n34. 'PC9_PAT'\n35. 'PC10_PAT'\n36. 'PC1_MAT'\n37. 'PC2_MAT'\n38. 'PC3_MAT'\n39. 'PC4_MAT'\n40. 'PC5_MAT'\n41. 'PC6_MAT'\n42. 'PC7_MAT'\n43. 'PC8_MAT'\n44. 'PC9_MAT'\n45. 'PC10_MAT'\n46. 'srs.child.age'\n47. 'vabs.child.age'\n48. 'scq.child.age'\n49. 'dcdq.child.age'\n50. 'mother.EA'\n51. 'father.EA'\n52. 'srs.mother'\n53. 'srs.father'\n54. 'mother.bapq'\n55. 'father.bapq'\n56. 'srs.child.z'\n57. 'vabs.child.z'\n58. 'rbs.child.z'\n59. 'scq.child.z'\n60. 'dcdq.child.z'\n61. 'father.age.at.birth.z'\n62. 'mother.age.at.birth.z'\n63. 'father.EA.z'\n64. 'mother.EA.z'\n65. 'father.srs.z'\n66. 'mother.srs.z'\n67. 'father.bapq.z'\n68. 'mother.bapq.z'\n69. 'PS.ASD.Z'\n70. 'PS.EA.Z'\n71. 'PS.SZ.Z'\n72. 'Paternal.PS.ASD.Z'\n73. 'Paternal.PS.EA.Z'\n74. 'Paternal.PS.SZ.Z'\n75. 'Maternal.PS.ASD.Z'\n76. 'Maternal.PS.EA.Z'\n77. 'Maternal.PS.SZ.Z'\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'fid'</li><li>'iid'</li><li>'phen'</li><li>'sex'</li><li>'cohort'</li><li>'duo'</li><li>'family'</li><li>'is_eur'</li><li>'dnLoF'</li><li>'dnMIS'</li><li>'inhLoF'</li><li>'InhLoF.Paternal.Transmitted'</li><li>'InhLoF.Paternal.Untransmitted'</li><li>'InhLoF.Maternal.Transmitted'</li><li>'InhLoF.Maternal.Untransmitted'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'PC8'</li><li>'PC9'</li><li>'PC10'</li><li>'PC1_PAT'</li><li>'PC2_PAT'</li><li>'PC3_PAT'</li><li>'PC4_PAT'</li><li>'PC5_PAT'</li><li>'PC6_PAT'</li><li>'PC7_PAT'</li><li>'PC8_PAT'</li><li>'PC9_PAT'</li><li>'PC10_PAT'</li><li>'PC1_MAT'</li><li>'PC2_MAT'</li><li>'PC3_MAT'</li><li>'PC4_MAT'</li><li>'PC5_MAT'</li><li>'PC6_MAT'</li><li>'PC7_MAT'</li><li>'PC8_MAT'</li><li>'PC9_MAT'</li><li>'PC10_MAT'</li><li>'srs.child.age'</li><li>'vabs.child.age'</li><li>'scq.child.age'</li><li>'dcdq.child.age'</li><li>'mother.EA'</li><li>'father.EA'</li><li>'srs.mother'</li><li>'srs.father'</li><li>'mother.bapq'</li><li>'father.bapq'</li><li>'srs.child.z'</li><li>'vabs.child.z'</li><li>'rbs.child.z'</li><li>'scq.child.z'</li><li>'dcdq.child.z'</li><li>'father.age.at.birth.z'</li><li>'mother.age.at.birth.z'</li><li>'father.EA.z'</li><li>'mother.EA.z'</li><li>'father.srs.z'</li><li>'mother.srs.z'</li><li>'father.bapq.z'</li><li>'mother.bapq.z'</li><li>'PS.ASD.Z'</li><li>'PS.EA.Z'</li><li>'PS.SZ.Z'</li><li>'Paternal.PS.ASD.Z'</li><li>'Paternal.PS.EA.Z'</li><li>'Paternal.PS.SZ.Z'</li><li>'Maternal.PS.ASD.Z'</li><li>'Maternal.PS.EA.Z'</li><li>'Maternal.PS.SZ.Z'</li></ol>\n"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1] 5086"
            ],
            "text/latex": "5086",
            "text/markdown": "5086",
            "text/html": [
              "5086"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1] 9354   26"
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 9354\n\\item 26\n\\end{enumerate*}\n",
            "text/markdown": "1. 9354\n2. 26\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>9354</li><li>26</li></ol>\n"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "[1] 819  26"
            ],
            "text/latex": "\\begin{enumerate*}\n\\item 819\n\\item 26\n\\end{enumerate*}\n",
            "text/markdown": "1. 819\n2. 26\n\n\n",
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>819</li><li>26</li></ol>\n"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-AFVa37YbkGV"
      },
      "source": [
        ""
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "OqWWYobvH1l8",
        "cellView": "form"
      },
      "source": [
        "#@title Misc\n",
        "\n",
        "# Sys.info()\n",
        "# sysname'Linux'release'5.4.104+'version'#1 SMP Sat Jun 5 09:50:34 PDT 2021'nodename'1cb9b7bc95e9'machine'x86_64'login'unknown'user'root'effective_user'root'\n",
        "\n",
        "# sessionInfo()\n",
        "# R version 4.1.1 (2021-08-10)\n",
        "# Platform: x86_64-pc-linux-gnu (64-bit)\n",
        "# Running under: Ubuntu 18.04.5 LTS\n",
        "\n",
        "# Matrix products: default\n",
        "# BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3\n",
        "# LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so\n",
        "\n",
        "# locale:\n",
        "#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
        "#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
        "#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
        "#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
        "#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
        "# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
        "\n",
        "# attached base packages:\n",
        "# [1] stats     graphics  grDevices utils     datasets  methods   base     \n",
        "\n",
        "# loaded via a namespace (and not attached):\n",
        "#  [1] fansi_0.5.0       digest_0.6.27     utf8_1.2.2        crayon_1.4.1     \n",
        "#  [5] IRdisplay_1.0     repr_1.1.3        lifecycle_1.0.0   jsonlite_1.7.2   \n",
        "#  [9] evaluate_0.14     pillar_1.6.2      rlang_0.4.11      uuid_0.1-4       \n",
        "# [13] vctrs_0.3.8       ellipsis_0.3.2    IRkernel_1.2      tools_4.1.1      \n",
        "# [17] compiler_4.1.1    base64enc_0.1-3   pbdZMQ_0.3-5      htmltools_0.5.1.1\n",
        "\n",
        "\n",
        "# print(installed.packages())\n",
        "#               Package         LibPath                         Version  \n",
        "# IRdisplay     \"IRdisplay\"     \"/usr/local/lib/R/site-library\" \"1.0\"    \n",
        "# IRkernel      \"IRkernel\"      \"/usr/local/lib/R/site-library\" \"1.2\"    \n",
        "# pbdZMQ        \"pbdZMQ\"        \"/usr/local/lib/R/site-library\" \"0.3-5\"  \n",
        "# repr          \"repr\"          \"/usr/local/lib/R/site-library\" \"1.1.3\"  \n",
        "# askpass       \"askpass\"       \"/usr/lib/R/site-library\"       \"1.1\"    \n",
        "# assertthat    \"assertthat\"    \"/usr/lib/R/site-library\"       \"0.2.1\"  \n",
        "# backports     \"backports\"     \"/usr/lib/R/site-library\"       \"1.2.1\"  \n",
        "# base64enc     \"base64enc\"     \"/usr/lib/R/site-library\"       \"0.1-3\"  \n",
        "# bit           \"bit\"           \"/usr/lib/R/site-library\"       \"4.0.4\"  \n",
        "# bit64         \"bit64\"         \"/usr/lib/R/site-library\"       \"4.0.5\"  \n",
        "# blob          \"blob\"          \"/usr/lib/R/site-library\"       \"1.2.2\"  \n",
        "# brew          \"brew\"          \"/usr/lib/R/site-library\"       \"1.0-6\"  \n",
        "# brio          \"brio\"          \"/usr/lib/R/site-library\"       \"1.1.2\"  \n",
        "# broom         \"broom\"         \"/usr/lib/R/site-library\"       \"0.7.9\"  \n",
        "# cachem        \"cachem\"        \"/usr/lib/R/site-library\"       \"1.0.6\"  \n",
        "# callr         \"callr\"         \"/usr/lib/R/site-library\"       \"3.7.0\"  \n",
        "# cellranger    \"cellranger\"    \"/usr/lib/R/site-library\"       \"1.1.0\"  \n",
        "# cli           \"cli\"           \"/usr/lib/R/site-library\"       \"3.0.1\"  \n",
        "# clipr         \"clipr\"         \"/usr/lib/R/site-library\"       \"0.7.1\"  \n",
        "# colorspace    \"colorspace\"    \"/usr/lib/R/site-library\"       \"2.0-2\"  \n",
        "# commonmark    \"commonmark\"    \"/usr/lib/R/site-library\"       \"1.7\"    \n",
        "# cpp11         \"cpp11\"         \"/usr/lib/R/site-library\"       \"0.3.1\"  \n",
        "# crayon        \"crayon\"        \"/usr/lib/R/site-library\"       \"1.4.1\"  \n",
        "# credentials   \"credentials\"   \"/usr/lib/R/site-library\"       \"1.3.1\"  \n",
        "# curl          \"curl\"          \"/usr/lib/R/site-library\"       \"4.3.2\"  \n",
        "# data.table    \"data.table\"    \"/usr/lib/R/site-library\"       \"1.14.0\" \n",
        "# DBI           \"DBI\"           \"/usr/lib/R/site-library\"       \"1.1.1\"  \n",
        "# dbplyr        \"dbplyr\"        \"/usr/lib/R/site-library\"       \"2.1.1\"  \n",
        "# desc          \"desc\"          \"/usr/lib/R/site-library\"       \"1.3.0\"  \n",
        "# devtools      \"devtools\"      \"/usr/lib/R/site-library\"       \"2.4.2\"  \n",
        "# diffobj       \"diffobj\"       \"/usr/lib/R/site-library\"       \"0.3.4\"  \n",
        "# digest        \"digest\"        \"/usr/lib/R/site-library\"       \"0.6.27\" \n",
        "# dplyr         \"dplyr\"         \"/usr/lib/R/site-library\"       \"1.0.7\"  \n",
        "# dtplyr        \"dtplyr\"        \"/usr/lib/R/site-library\"       \"1.1.0\"  \n",
        "# ellipsis      \"ellipsis\"      \"/usr/lib/R/site-library\"       \"0.3.2\"  \n",
        "# evaluate      \"evaluate\"      \"/usr/lib/R/site-library\"       \"0.14\"   \n",
        "# fansi         \"fansi\"         \"/usr/lib/R/site-library\"       \"0.5.0\"  \n",
        "# farver        \"farver\"        \"/usr/lib/R/site-library\"       \"2.1.0\"  \n",
        "# fastmap       \"fastmap\"       \"/usr/lib/R/site-library\"       \"1.1.0\"  \n",
        "# forcats       \"forcats\"       \"/usr/lib/R/site-library\"       \"0.5.1\"  \n",
        "# fs            \"fs\"            \"/usr/lib/R/site-library\"       \"1.5.0\"  \n",
        "# gargle        \"gargle\"        \"/usr/lib/R/site-library\"       \"1.2.0\"  \n",
        "# generics      \"generics\"      \"/usr/lib/R/site-library\"       \"0.1.0\"  \n",
        "# gert          \"gert\"          \"/usr/lib/R/site-library\"       \"1.3.2\"  \n",
        "# ggplot2       \"ggplot2\"       \"/usr/lib/R/site-library\"       \"3.3.5\"  \n",
        "# gh            \"gh\"            \"/usr/lib/R/site-library\"       \"1.3.0\"  \n",
        "# gitcreds      \"gitcreds\"      \"/usr/lib/R/site-library\"       \"0.1.1\"  \n",
        "# glue          \"glue\"          \"/usr/lib/R/site-library\"       \"1.4.2\"  \n",
        "# googledrive   \"googledrive\"   \"/usr/lib/R/site-library\"       \"2.0.0\"  \n",
        "# googlesheets4 \"googlesheets4\" \"/usr/lib/R/site-library\"       \"1.0.0\"  \n",
        "# gtable        \"gtable\"        \"/usr/lib/R/site-library\"       \"0.3.0\"  \n",
        "# haven         \"haven\"         \"/usr/lib/R/site-library\"       \"2.4.3\"  \n",
        "# highr         \"highr\"         \"/usr/lib/R/site-library\"       \"0.9\"    \n",
        "# hms           \"hms\"           \"/usr/lib/R/site-library\"       \"1.1.0\"  \n",
        "# htmltools     \"htmltools\"     \"/usr/lib/R/site-library\"       \"0.5.1.1\"\n",
        "# httr          \"httr\"          \"/usr/lib/R/site-library\"       \"1.4.2\"  \n",
        "# ids           \"ids\"           \"/usr/lib/R/site-library\"       \"1.0.1\"  \n",
        "# ini           \"ini\"           \"/usr/lib/R/site-library\"       \"0.3.1\"  \n",
        "# isoband       \"isoband\"       \"/usr/lib/R/site-library\"       \"0.2.5\"  \n",
        "# jsonlite      \"jsonlite\"      \"/usr/lib/R/site-library\"       \"1.7.2\"  \n",
        "# knitr         \"knitr\"         \"/usr/lib/R/site-library\"       \"1.33\"   \n",
        "# labeling      \"labeling\"      \"/usr/lib/R/site-library\"       \"0.4.2\"  \n",
        "# lifecycle     \"lifecycle\"     \"/usr/lib/R/site-library\"       \"1.0.0\"  \n",
        "# lubridate     \"lubridate\"     \"/usr/lib/R/site-library\"       \"1.7.10\" \n",
        "# magrittr      \"magrittr\"      \"/usr/lib/R/site-library\"       \"2.0.1\"  \n",
        "# markdown      \"markdown\"      \"/usr/lib/R/site-library\"       \"1.1\"    \n",
        "# memoise       \"memoise\"       \"/usr/lib/R/site-library\"       \"2.0.0\"  \n",
        "# mime          \"mime\"          \"/usr/lib/R/site-library\"       \"0.11\"   \n",
        "# modelr        \"modelr\"        \"/usr/lib/R/site-library\"       \"0.1.8\"  \n",
        "# munsell       \"munsell\"       \"/usr/lib/R/site-library\"       \"0.5.0\"  \n",
        "# openssl       \"openssl\"       \"/usr/lib/R/site-library\"       \"1.4.3\"  \n",
        "# pillar        \"pillar\"        \"/usr/lib/R/site-library\"       \"1.6.2\"  \n",
        "# pkgbuild      \"pkgbuild\"      \"/usr/lib/R/site-library\"       \"1.2.0\"  \n",
        "# pkgconfig     \"pkgconfig\"     \"/usr/lib/R/site-library\"       \"2.0.3\"  \n",
        "# pkgload       \"pkgload\"       \"/usr/lib/R/site-library\"       \"1.2.1\"  \n",
        "# praise        \"praise\"        \"/usr/lib/R/site-library\"       \"1.0.0\"  \n",
        "# prettyunits   \"prettyunits\"   \"/usr/lib/R/site-library\"       \"1.1.1\"  \n",
        "# processx      \"processx\"      \"/usr/lib/R/site-library\"       \"3.5.2\"  \n",
        "# progress      \"progress\"      \"/usr/lib/R/site-library\"       \"1.2.2\"  \n",
        "# ps            \"ps\"            \"/usr/lib/R/site-library\"       \"1.6.0\"  \n",
        "# purrr         \"purrr\"         \"/usr/lib/R/site-library\"       \"0.3.4\"  \n",
        "# R6            \"R6\"            \"/usr/lib/R/site-library\"       \"2.5.1\"  \n",
        "# rappdirs      \"rappdirs\"      \"/usr/lib/R/site-library\"       \"0.3.3\"  \n",
        "# rcmdcheck     \"rcmdcheck\"     \"/usr/lib/R/site-library\"       \"1.3.3\"  \n",
        "# RColorBrewer  \"RColorBrewer\"  \"/usr/lib/R/site-library\"       \"1.1-2\"  \n",
        "# Rcpp          \"Rcpp\"          \"/usr/lib/R/site-library\"       \"1.0.7\"  \n",
        "# readr         \"readr\"         \"/usr/lib/R/site-library\"       \"2.0.1\"  \n",
        "# readxl        \"readxl\"        \"/usr/lib/R/site-library\"       \"1.3.1\"  \n",
        "# rematch       \"rematch\"       \"/usr/lib/R/site-library\"       \"1.0.1\"  \n",
        "# rematch2      \"rematch2\"      \"/usr/lib/R/site-library\"       \"2.1.2\"  \n",
        "# remotes       \"remotes\"       \"/usr/lib/R/site-library\"       \"2.4.0\"  \n",
        "# reprex        \"reprex\"        \"/usr/lib/R/site-library\"       \"2.0.1\"  \n",
        "# rlang         \"rlang\"         \"/usr/lib/R/site-library\"       \"0.4.11\" \n",
        "# rmarkdown     \"rmarkdown\"     \"/usr/lib/R/site-library\"       \"2.10\"   \n",
        "# roxygen2      \"roxygen2\"      \"/usr/lib/R/site-library\"       \"7.1.1\"  \n",
        "# rprojroot     \"rprojroot\"     \"/usr/lib/R/site-library\"       \"2.0.2\"  \n",
        "# rstudioapi    \"rstudioapi\"    \"/usr/lib/R/site-library\"       \"0.13\"   \n",
        "# rversions     \"rversions\"     \"/usr/lib/R/site-library\"       \"2.1.1\"  \n",
        "# rvest         \"rvest\"         \"/usr/lib/R/site-library\"       \"1.0.1\"  \n",
        "# scales        \"scales\"        \"/usr/lib/R/site-library\"       \"1.1.1\"  \n",
        "# selectr       \"selectr\"       \"/usr/lib/R/site-library\"       \"0.4-2\"  \n",
        "# sessioninfo   \"sessioninfo\"   \"/usr/lib/R/site-library\"       \"1.1.1\"  \n",
        "# stringi       \"stringi\"       \"/usr/lib/R/site-library\"       \"1.7.3\"  \n",
        "# stringr       \"stringr\"       \"/usr/lib/R/site-library\"       \"1.4.0\"  \n",
        "# svglite       \"svglite\"       \"/usr/lib/R/site-library\"       \"2.0.0\"  \n",
        "# sys           \"sys\"           \"/usr/lib/R/site-library\"       \"3.4\"    \n",
        "# systemfonts   \"systemfonts\"   \"/usr/lib/R/site-library\"       \"1.0.2\"  \n",
        "# testthat      \"testthat\"      \"/usr/lib/R/site-library\"       \"3.0.4\"  \n",
        "# tibble        \"tibble\"        \"/usr/lib/R/site-library\"       \"3.1.3\"  \n",
        "# tidyr         \"tidyr\"         \"/usr/lib/R/site-library\"       \"1.1.3\"  \n",
        "# tidyselect    \"tidyselect\"    \"/usr/lib/R/site-library\"       \"1.1.1\"  \n",
        "# tidyverse     \"tidyverse\"     \"/usr/lib/R/site-library\"       \"1.3.1\"  \n",
        "# tinytex       \"tinytex\"       \"/usr/lib/R/site-library\"       \"0.33\"   \n",
        "# tzdb          \"tzdb\"          \"/usr/lib/R/site-library\"       \"0.1.2\"  \n",
        "# usethis       \"usethis\"       \"/usr/lib/R/site-library\"       \"2.0.1\"  \n",
        "# utf8          \"utf8\"          \"/usr/lib/R/site-library\"       \"1.2.2\"  \n",
        "# uuid          \"uuid\"          \"/usr/lib/R/site-library\"       \"0.1-4\"  \n",
        "# vctrs         \"vctrs\"         \"/usr/lib/R/site-library\"       \"0.3.8\"  \n",
        "# viridisLite   \"viridisLite\"   \"/usr/lib/R/site-library\"       \"0.4.0\"  \n",
        "# vroom         \"vroom\"         \"/usr/lib/R/site-library\"       \"1.5.4\"  \n",
        "# waldo         \"waldo\"         \"/usr/lib/R/site-library\"       \"0.3.0\"  \n",
        "# whisker       \"whisker\"       \"/usr/lib/R/site-library\"       \"0.4\"    \n",
        "# withr         \"withr\"         \"/usr/lib/R/site-library\"       \"2.4.2\"  \n",
        "# xfun          \"xfun\"          \"/usr/lib/R/site-library\"       \"0.25\"   \n",
        "# xml2          \"xml2\"          \"/usr/lib/R/site-library\"       \"1.3.2\"  \n",
        "# xopen         \"xopen\"         \"/usr/lib/R/site-library\"       \"1.0.0\"  \n",
        "# yaml          \"yaml\"          \"/usr/lib/R/site-library\"       \"2.2.1\"  \n",
        "# zip           \"zip\"           \"/usr/lib/R/site-library\"       \"2.2.0\"  \n",
        "# base          \"base\"          \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# boot          \"boot\"          \"/usr/lib/R/library\"            \"1.3-28\" \n",
        "# class         \"class\"         \"/usr/lib/R/library\"            \"7.3-19\" \n",
        "# cluster       \"cluster\"       \"/usr/lib/R/library\"            \"2.1.2\"  \n",
        "# codetools     \"codetools\"     \"/usr/lib/R/library\"            \"0.2-18\" \n",
        "# compiler      \"compiler\"      \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# datasets      \"datasets\"      \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# foreign       \"foreign\"       \"/usr/lib/R/library\"            \"0.8-81\" \n",
        "# graphics      \"graphics\"      \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# grDevices     \"grDevices\"     \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# grid          \"grid\"          \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# KernSmooth    \"KernSmooth\"    \"/usr/lib/R/library\"            \"2.23-20\"\n",
        "# lattice       \"lattice\"       \"/usr/lib/R/library\"            \"0.20-44\"\n",
        "# MASS          \"MASS\"          \"/usr/lib/R/library\"            \"7.3-54\" \n",
        "# Matrix        \"Matrix\"        \"/usr/lib/R/library\"            \"1.3-4\"  \n",
        "# methods       \"methods\"       \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# mgcv          \"mgcv\"          \"/usr/lib/R/library\"            \"1.8-36\" \n",
        "# nlme          \"nlme\"          \"/usr/lib/R/library\"            \"3.1-152\"\n",
        "# nnet          \"nnet\"          \"/usr/lib/R/library\"            \"7.3-16\" \n",
        "# parallel      \"parallel\"      \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# rpart         \"rpart\"         \"/usr/lib/R/library\"            \"4.1-15\" \n",
        "# spatial       \"spatial\"       \"/usr/lib/R/library\"            \"7.3-14\" \n",
        "# splines       \"splines\"       \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# stats         \"stats\"         \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# stats4        \"stats4\"        \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# survival      \"survival\"      \"/usr/lib/R/library\"            \"3.2-13\" \n",
        "# tcltk         \"tcltk\"         \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# tools         \"tools\"         \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "# utils         \"utils\"         \"/usr/lib/R/library\"            \"4.1.1\"  \n",
        "#               Priority     \n",
        "# IRdisplay     NA           \n",
        "# IRkernel      NA           \n",
        "# pbdZMQ        NA           \n",
        "# repr          NA           \n",
        "# askpass       NA           \n",
        "# assertthat    NA           \n",
        "# backports     NA           \n",
        "# base64enc     NA           \n",
        "# bit           NA           \n",
        "# bit64         NA           \n",
        "# blob          NA           \n",
        "# brew          NA           \n",
        "# brio          NA           \n",
        "# broom         NA           \n",
        "# cachem        NA           \n",
        "# callr         NA           \n",
        "# cellranger    NA           \n",
        "# cli           NA           \n",
        "# clipr         NA           \n",
        "# colorspace    NA           \n",
        "# commonmark    NA           \n",
        "# cpp11         NA           \n",
        "# crayon        NA           \n",
        "# credentials   NA           \n",
        "# curl          NA           \n",
        "# data.table    NA           \n",
        "# DBI           NA           \n",
        "# dbplyr        NA           \n",
        "# desc          NA           \n",
        "# devtools      NA           \n",
        "# diffobj       NA           \n",
        "# digest        NA           \n",
        "# dplyr         NA           \n",
        "# dtplyr        NA           \n",
        "# ellipsis      NA           \n",
        "# evaluate      NA           \n",
        "# fansi         NA           \n",
        "# farver        NA           \n",
        "# fastmap       NA           \n",
        "# forcats       NA           \n",
        "# fs            NA           \n",
        "# gargle        NA           \n",
        "# generics      NA           \n",
        "# gert          NA           \n",
        "# ggplot2       NA           \n",
        "# gh            NA           \n",
        "# gitcreds      NA           \n",
        "# glue          NA           \n",
        "# googledrive   NA           \n",
        "# googlesheets4 NA           \n",
        "# gtable        NA           \n",
        "# haven         NA           \n",
        "# highr         NA           \n",
        "# hms           NA           \n",
        "# htmltools     NA           \n",
        "# httr          NA           \n",
        "# ids           NA           \n",
        "# ini           NA           \n",
        "# isoband       NA           \n",
        "# jsonlite      NA           \n",
        "# knitr         NA           \n",
        "# labeling      NA           \n",
        "# lifecycle     NA           \n",
        "# lubridate     NA           \n",
        "# magrittr      NA           \n",
        "# markdown      NA           \n",
        "# memoise       NA           \n",
        "# mime          NA           \n",
        "# modelr        NA           \n",
        "# munsell       NA           \n",
        "# openssl       NA           \n",
        "# pillar        NA           \n",
        "# pkgbuild      NA           \n",
        "# pkgconfig     NA           \n",
        "# pkgload       NA           \n",
        "# praise        NA           \n",
        "# prettyunits   NA           \n",
        "# processx      NA           \n",
        "# progress      NA           \n",
        "# ps            NA           \n",
        "# purrr         NA           \n",
        "# R6            NA           \n",
        "# rappdirs      NA           \n",
        "# rcmdcheck     NA           \n",
        "# RColorBrewer  NA           \n",
        "# Rcpp          NA           \n",
        "# readr         NA           \n",
        "# readxl        NA           \n",
        "# rematch       NA           \n",
        "# rematch2      NA           \n",
        "# remotes       NA           \n",
        "# reprex        NA           \n",
        "# rlang         NA           \n",
        "# rmarkdown     NA           \n",
        "# roxygen2      NA           \n",
        "# rprojroot     NA           \n",
        "# rstudioapi    NA           \n",
        "# rversions     NA           \n",
        "# rvest         NA           \n",
        "# scales        NA           \n",
        "# selectr       NA           \n",
        "# sessioninfo   NA           \n",
        "# stringi       NA           \n",
        "# stringr       NA           \n",
        "# svglite       NA           \n",
        "# sys           NA           \n",
        "# systemfonts   NA           \n",
        "# testthat      NA           \n",
        "# tibble        NA           \n",
        "# tidyr         NA           \n",
        "# tidyselect    NA           \n",
        "# tidyverse     NA           \n",
        "# tinytex       NA           \n",
        "# tzdb          NA           \n",
        "# usethis       NA           \n",
        "# utf8          NA           \n",
        "# uuid          NA           \n",
        "# vctrs         NA           \n",
        "# viridisLite   NA           \n",
        "# vroom         NA           \n",
        "# waldo         NA           \n",
        "# whisker       NA           \n",
        "# withr         NA           \n",
        "# xfun          NA           \n",
        "# xml2          NA           \n",
        "# xopen         NA           \n",
        "# yaml          NA           \n",
        "# zip           NA           \n",
        "# base          \"base\"       \n",
        "# boot          \"recommended\"\n",
        "# class         \"recommended\"\n",
        "# cluster       \"recommended\"\n",
        "# codetools     \"recommended\"\n",
        "# compiler      \"base\"       \n",
        "# datasets      \"base\"       \n",
        "# foreign       \"recommended\"\n",
        "# graphics      \"base\"       \n",
        "# grDevices     \"base\"       \n",
        "# grid          \"base\"       \n",
        "# KernSmooth    \"recommended\"\n",
        "# lattice       \"recommended\"\n",
        "# MASS          \"recommended\"\n",
        "# Matrix        \"recommended\"\n",
        "# methods       \"base\"       \n",
        "# mgcv          \"recommended\"\n",
        "# nlme          \"recommended\"\n",
        "# nnet          \"recommended\"\n",
        "# parallel      \"base\"       \n",
        "# rpart         \"recommended\"\n",
        "# spatial       \"recommended\"\n",
        "# splines       \"base\"       \n",
        "# stats         \"base\"       \n",
        "# stats4        \"base\"       \n",
        "# survival      \"recommended\"\n",
        "# tcltk         \"base\"       \n",
        "# tools         \"base\"       \n",
        "# utils         \"base\"       \n",
        "#               Depends                                              \n",
        "# IRdisplay     \"R (>= 3.0.1)\"                                       \n",
        "# IRkernel      \"R (>= 3.2.0)\"                                       \n",
        "# pbdZMQ        \"R (>= 3.5.0)\"                                       \n",
        "# repr          \"R (>= 3.0.1)\"                                       \n",
        "# askpass       NA                                                   \n",
        "# assertthat    NA                                                   \n",
        "# backports     \"R (>= 3.0.0)\"                                       \n",
        "# base64enc     \"R (>= 2.9.0)\"                                       \n",
        "# bit           \"R (>= 2.9.2)\"                                       \n",
        "# bit64         \"R (>= 3.0.1), bit (>= 4.0.0), utils, methods, stats\"\n",
        "# blob          NA                                                   \n",
        "# brew          NA                                                   \n",
        "# brio          NA                                                   \n",
        "# broom         \"R (>= 3.1)\"                                         \n",
        "# cachem        NA                                                   \n",
        "# callr         NA                                                   \n",
        "# cellranger    \"R (>= 3.0.0)\"                                       \n",
        "# cli           \"R (>= 2.10)\"                                        \n",
        "# clipr         NA                                                   \n",
        "# colorspace    \"R (>= 3.0.0), methods\"                              \n",
        "# commonmark    NA                                                   \n",
        "# cpp11         NA                                                   \n",
        "# crayon        NA                                                   \n",
        "# credentials   NA                                                   \n",
        "# curl          \"R (>= 3.0.0)\"                                       \n",
        "# data.table    \"R (>= 3.1.0)\"                                       \n",
        "# DBI           \"methods, R (>= 3.0.0)\"                              \n",
        "# dbplyr        \"R (>= 3.1)\"                                         \n",
        "# desc          \"R (>= 3.1.0)\"                                       \n",
        "# devtools      \"R (>= 3.0.2), usethis (>= 2.0.1)\"                   \n",
        "# diffobj       \"R (>= 3.1.0)\"                                       \n",
        "# digest        \"R (>= 3.3.0)\"                                       \n",
        "# dplyr         \"R (>= 3.3.0)\"                                       \n",
        "# dtplyr        \"R (>= 3.3)\"                                         \n",
        "# ellipsis      \"R (>= 3.2)\"                                         \n",
        "# evaluate      \"R (>= 3.0.2)\"                                       \n",
        "# fansi         \"R (>= 3.1.0)\"                                       \n",
        "# farver        NA                                                   \n",
        "# fastmap       NA                                                   \n",
        "# forcats       \"R (>= 3.2)\"                                         \n",
        "# fs            \"R (>= 3.1)\"                                         \n",
        "# gargle        \"R (>= 3.3)\"                                         \n",
        "# generics      \"R (>= 3.1)\"                                         \n",
        "# gert          NA                                                   \n",
        "# ggplot2       \"R (>= 3.3)\"                                         \n",
        "# gh            NA                                                   \n",
        "# gitcreds      NA                                                   \n",
        "# glue          \"R (>= 3.2)\"                                         \n",
        "# googledrive   \"R (>= 3.3)\"                                         \n",
        "# googlesheets4 \"R (>= 3.3)\"                                         \n",
        "# gtable        \"R (>= 3.0)\"                                         \n",
        "# haven         \"R (>= 3.2)\"                                         \n",
        "# highr         \"R (>= 3.2.3)\"                                       \n",
        "# hms           NA                                                   \n",
        "# htmltools     \"R (>= 2.14.1)\"                                      \n",
        "# httr          \"R (>= 3.2)\"                                         \n",
        "# ids           NA                                                   \n",
        "# ini           NA                                                   \n",
        "# isoband       NA                                                   \n",
        "# jsonlite      \"methods\"                                            \n",
        "# knitr         \"R (>= 3.2.3)\"                                       \n",
        "# labeling      NA                                                   \n",
        "# lifecycle     \"R (>= 3.3)\"                                         \n",
        "# lubridate     \"methods, R (>= 3.2)\"                                \n",
        "# magrittr      NA                                                   \n",
        "# markdown      \"R (>= 2.11.1)\"                                      \n",
        "# memoise       NA                                                   \n",
        "# mime          NA                                                   \n",
        "# modelr        \"R (>= 3.2)\"                                         \n",
        "# munsell       NA                                                   \n",
        "# openssl       NA                                                   \n",
        "# pillar        NA                                                   \n",
        "# pkgbuild      \"R (>= 3.1)\"                                         \n",
        "# pkgconfig     NA                                                   \n",
        "# pkgload       NA                                                   \n",
        "# praise        NA                                                   \n",
        "# prettyunits   NA                                                   \n",
        "# processx      NA                                                   \n",
        "# progress      NA                                                   \n",
        "# ps            \"R (>= 3.1)\"                                         \n",
        "# purrr         \"R (>= 3.2)\"                                         \n",
        "# R6            \"R (>= 3.0)\"                                         \n",
        "# rappdirs      \"R (>= 3.2)\"                                         \n",
        "# rcmdcheck     NA                                                   \n",
        "# RColorBrewer  \"R (>= 2.0.0)\"                                       \n",
        "# Rcpp          NA                                                   \n",
        "# readr         \"R (>= 3.1)\"                                         \n",
        "# readxl        NA                                                   \n",
        "# rematch       NA                                                   \n",
        "# rematch2      NA                                                   \n",
        "# remotes       \"R (>= 3.0.0)\"                                       \n",
        "# reprex        \"R (>= 3.3)\"                                         \n",
        "# rlang         \"R (>= 3.3.0)\"                                       \n",
        "# rmarkdown     \"R (>= 3.0)\"                                         \n",
        "# roxygen2      \"R (>= 3.2)\"                                         \n",
        "# rprojroot     \"R (>= 3.0.0)\"                                       \n",
        "# rstudioapi    NA                                                   \n",
        "# rversions     NA                                                   \n",
        "# rvest         \"R (>= 3.2)\"                                         \n",
        "# scales        \"R (>= 3.2)\"                                         \n",
        "# selectr       \"R (>= 3.0)\"                                         \n",
        "# sessioninfo   NA                                                   \n",
        "# stringi       \"R (>= 3.1)\"                                         \n",
        "# stringr       \"R (>= 3.1)\"                                         \n",
        "# svglite       \"R (>= 3.0.0)\"                                       \n",
        "# sys           NA                                                   \n",
        "# systemfonts   \"R (>= 3.2.0)\"                                       \n",
        "# testthat      \"R (>= 3.1)\"                                         \n",
        "# tibble        \"R (>= 3.1.0)\"                                       \n",
        "# tidyr         \"R (>= 3.1)\"                                         \n",
        "# tidyselect    \"R (>= 3.2)\"                                         \n",
        "# tidyverse     \"R (>= 3.3)\"                                         \n",
        "# tinytex       NA                                                   \n",
        "# tzdb          \"R (>= 3.3)\"                                         \n",
        "# usethis       \"R (>= 3.2)\"                                         \n",
        "# utf8          \"R (>= 2.10)\"                                        \n",
        "# uuid          \"R (>= 2.9.0)\"                                       \n",
        "# vctrs         \"R (>= 3.3)\"                                         \n",
        "# viridisLite   \"R (>= 2.10)\"                                        \n",
        "# vroom         \"R (>= 3.1)\"                                         \n",
        "# waldo         NA                                                   \n",
        "# whisker       NA                                                   \n",
        "# withr         \"R (>= 3.2.0)\"                                       \n",
        "# xfun          NA                                                   \n",
        "# xml2          \"R (>= 3.1.0)\"                                       \n",
        "# xopen         \"R (>= 3.1)\"                                         \n",
        "# yaml          NA                                                   \n",
        "# zip           NA                                                   \n",
        "# base          NA                                                   \n",
        "# boot          \"R (>= 3.0.0), graphics, stats\"                      \n",
        "# class         \"R (>= 3.0.0), stats, utils\"                         \n",
        "# cluster       \"R (>= 3.4.0)\"                                       \n",
        "# codetools     \"R (>= 2.1)\"                                         \n",
        "# compiler      NA                                                   \n",
        "# datasets      NA                                                   \n",
        "# foreign       \"R (>= 4.0.0)\"                                       \n",
        "# graphics      NA                                                   \n",
        "# grDevices     NA                                                   \n",
        "# grid          NA                                                   \n",
        "# KernSmooth    \"R (>= 2.5.0), stats\"                                \n",
        "# lattice       \"R (>= 3.0.0)\"                                       \n",
        "# MASS          \"R (>= 3.3.0), grDevices, graphics, stats, utils\"    \n",
        "# Matrix        \"R (>= 3.5.0)\"                                       \n",
        "# methods       NA                                                   \n",
        "# mgcv          \"R (>= 3.6.0), nlme (>= 3.1-64)\"                     \n",
        "# nlme          \"R (>= 3.4.0)\"                                       \n",
        "# nnet          \"R (>= 3.0.0), stats, utils\"                         \n",
        "# parallel      NA                                                   \n",
        "# rpart         \"R (>= 2.15.0), graphics, stats, grDevices\"          \n",
        "# spatial       \"R (>= 3.0.0), graphics, stats, utils\"               \n",
        "# splines       NA                                                   \n",
        "# stats         NA                                                   \n",
        "# stats4        NA                                                   \n",
        "# survival      \"R (>= 3.5.0)\"                                       \n",
        "# tcltk         NA                                                   \n",
        "# tools         NA                                                   \n",
        "# utils         NA                                                   \n",
        "#               Imports                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# IRdisplay     \"methods, repr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# IRkernel      \"repr (>= 0.4.99), methods, evaluate (>= 0.10), IRdisplay (>=\\n0.3.0.9999), pbdZMQ (>= 0.2-1), crayon, jsonlite (>= 0.9.6),\\nuuid, digest\"                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# pbdZMQ        NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# repr          \"utils, grDevices, htmltools, jsonlite, pillar (>= 1.4.0),\\nbase64enc\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# askpass       \"sys (>= 2.1)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# assertthat    \"tools\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# backports     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# base64enc     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# bit           NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# bit64         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# blob          \"methods, rlang, vctrs (>= 0.2.1)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# brew          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# brio          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# broom         \"backports, dplyr (>= 1.0.0), ellipsis, generics (>= 0.0.2),\\nglue, methods, purrr, rlang, stringr, tibble (>= 3.0.0), tidyr\\n(>= 1.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# cachem        \"rlang, fastmap\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# callr         \"processx (>= 3.5.0), R6, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# cellranger    \"rematch, tibble\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# cli           \"glue, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# clipr         \"utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# colorspace    \"graphics, grDevices, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# commonmark    NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# cpp11         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# crayon        \"grDevices, methods, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# credentials   \"openssl (>= 1.3), sys (>= 2.1), curl, jsonlite, askpass\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# curl          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# data.table    \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# DBI           NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# dbplyr        \"assertthat (>= 0.2.0), blob (>= 1.2.0), DBI (>= 1.0.0), dplyr\\n(>= 1.0.4), ellipsis, glue (>= 1.2.0), lifecycle (>= 1.0.0),\\nmagrittr, methods, purrr (>= 0.2.5), R6 (>= 2.2.2), rlang (>=\\n0.2.0), tibble (>= 1.4.2), tidyselect (>= 0.2.4), utils, vctrs,\\nwithr\"                                                                                                                                                                                                                                                                                                                     \n",
        "# desc          \"utils, R6, crayon, rprojroot\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# devtools      \"callr (>= 3.6.0), cli (>= 2.4.0), desc (>= 1.3.0), ellipsis\\n(>= 0.3.1), fs (>= 1.5.0), httr (>= 1.4.2), lifecycle (>=\\n1.0.0), memoise (>= 2.0.0), pkgbuild (>= 1.2.0), pkgload (>=\\n1.2.1), rcmdcheck (>= 1.3.3), remotes (>= 2.3.0), rlang (>=\\n0.4.10), roxygen2 (>= 7.1.1), rstudioapi (>= 0.13), rversions\\n(>= 2.0.2), sessioninfo (>= 1.1.1), stats, testthat (>= 3.0.2),\\ntools, utils, withr (>= 2.4.1)\"                                                                                                                                                                      \n",
        "# diffobj       \"crayon (>= 1.3.2), tools, methods, utils, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# digest        \"utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# dplyr         \"ellipsis, generics, glue (>= 1.3.2), lifecycle (>= 1.0.0),\\nmagrittr (>= 1.5), methods, R6, rlang (>= 0.4.10), tibble (>=\\n2.1.3), tidyselect (>= 1.1.0), utils, vctrs (>= 0.3.5), pillar\\n(>= 1.5.1)\"                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# dtplyr        \"crayon, data.table (>= 1.12.4), dplyr (>= 1.0.3), ellipsis,\\nglue, lifecycle, rlang, tibble, tidyselect, vctrs\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# ellipsis      \"rlang (>= 0.3.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# evaluate      \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# fansi         \"grDevices, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# farver        NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# fastmap       NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# forcats       \"ellipsis, magrittr, rlang, tibble\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# fs            \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# gargle        \"cli (>= 3.0.0), fs (>= 1.3.1), glue (>= 1.3.0), httr (>=\\n1.4.0), jsonlite, rappdirs, rlang (>= 0.4.9), rstudioapi,\\nstats, utils, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# generics      \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# gert          \"askpass, credentials (>= 1.2.1), openssl (>= 1.4.1),\\nrstudioapi (>= 0.11), sys, zip (>= 2.1.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# ggplot2       \"digest, glue, grDevices, grid, gtable (>= 0.1.1), isoband,\\nMASS, mgcv, rlang (>= 0.4.10), scales (>= 0.5.0), stats,\\ntibble, withr (>= 2.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# gh            \"cli (>= 2.0.1), gitcreds, httr (>= 1.2), ini, jsonlite\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# gitcreds      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# glue          \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# googledrive   \"cli (>= 3.0.0), gargle (>= 1.2.0), glue (>= 1.4.2), httr,\\njsonlite, lifecycle, magrittr, pillar, purrr (>= 0.2.3), rlang\\n(>= 0.4.9), tibble (>= 2.0.0), utils, uuid, vctrs (>= 0.3.0),\\nwithr\"                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# googlesheets4 \"cellranger, cli (>= 3.0.0), curl, gargle (>= 1.2.0), glue (>=\\n1.3.0), googledrive (>= 2.0.0), httr, ids, magrittr, methods,\\npurrr, rematch2, rlang (>= 0.4.11), tibble (>= 2.1.1), utils,\\nvctrs (>= 0.2.3)\"                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# gtable        \"grid\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# haven         \"forcats (>= 0.2.0), hms, methods, readr (>= 0.1.0), rlang (>=\\n0.4.0), tibble, tidyselect, vctrs (>= 0.3.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# highr         \"xfun (>= 0.18)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# hms           \"ellipsis (>= 0.3.2), lifecycle, methods, pkgconfig, rlang,\\nvctrs (>= 0.3.8)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# htmltools     \"utils, digest, grDevices, base64enc, rlang\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# httr          \"curl (>= 3.0.0), jsonlite, mime, openssl (>= 0.8), R6\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# ids           \"openssl, uuid\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# ini           NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# isoband       \"grid, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# jsonlite      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# knitr         \"evaluate (>= 0.10), highr, methods, markdown, stringr (>=\\n0.6), yaml (>= 2.1.19), xfun (>= 0.21), tools\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# labeling      \"stats, graphics\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# lifecycle     \"glue, rlang (>= 0.4.10)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# lubridate     \"generics, Rcpp (>= 0.12.13)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# magrittr      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# markdown      \"utils, xfun, mime (>= 0.3)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# memoise       \"rlang (>= 0.4.10), cachem\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# mime          \"tools\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# modelr        \"broom, magrittr, purrr (>= 0.2.2), rlang (>= 0.2.0), tibble,\\ntidyr (>= 0.8.0), tidyselect, vctrs\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# munsell       \"colorspace, methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# openssl       \"askpass\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# pillar        \"cli, crayon (>= 1.3.4), ellipsis (>= 0.3.2), fansi, lifecycle,\\nrlang (>= 0.3.0), utf8 (>= 1.1.0), utils, vctrs (>= 0.3.8)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# pkgbuild      \"callr (>= 3.2.0), cli, crayon, desc, prettyunits, R6,\\nrprojroot, withr (>= 2.1.2)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# pkgconfig     \"utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# pkgload       \"cli, crayon, desc, methods, rlang, rprojroot, rstudioapi,\\nutils, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# praise        NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# prettyunits   NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# processx      \"ps (>= 1.2.0), R6, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# progress      \"hms, prettyunits, R6, crayon\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# ps            \"utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# purrr         \"magrittr (>= 1.5), rlang (>= 0.3.1)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# R6            NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# rappdirs      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# rcmdcheck     \"callr (>= 3.1.1.9000), cli (>= 1.1.0), crayon, desc (>=\\n1.2.0), digest, pkgbuild, prettyunits, R6, rprojroot,\\nsessioninfo (>= 1.1.1), utils, withr, xopen\"                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# RColorBrewer  NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# Rcpp          \"methods, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# readr         \"cli, clipr, crayon, hms (>= 0.4.1), methods, rlang, R6,\\ntibble, vroom (>= 1.5.4), utils, lifecycle (>= 0.2.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# readxl        \"cellranger, Rcpp (>= 0.12.18), tibble (>= 1.3.1), utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# rematch       NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# rematch2      \"tibble\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# remotes       \"methods, stats, tools, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# reprex        \"callr (>= 3.6.0), cli (>= 2.3.1), clipr (>= 0.4.0), fs, glue,\\nknitr (>= 1.23), rlang (>= 0.4.0), rmarkdown, rstudioapi,\\nutils, withr (>= 2.3.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# rlang         \"utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# rmarkdown     \"tools, utils, knitr (>= 1.22), yaml (>= 2.1.19), htmltools (>=\\n0.3.5), evaluate (>= 0.13), jsonlite, tinytex (>= 0.31), xfun\\n(>= 0.21), methods, stringr (>= 1.2.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# roxygen2      \"brew, commonmark, desc (>= 1.2.0), digest, knitr, methods,\\npkgload (>= 1.0.2), purrr (>= 0.3.3), R6 (>= 2.1.2), Rcpp (>=\\n0.11.0), rlang, stringi, stringr (>= 1.0.0), utils, xml2\"                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# rprojroot     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# rstudioapi    NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# rversions     \"curl, utils, xml2 (>= 1.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# rvest         \"httr (>= 0.5), lifecycle (>= 1.0.0), magrittr, rlang (>=\\n0.4.10), selectr, tibble, xml2 (>= 1.3)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# scales        \"farver (>= 2.0.3), labeling, lifecycle, munsell (>= 0.5), R6,\\nRColorBrewer, viridisLite\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# selectr       \"methods, stringr, R6\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# sessioninfo   \"cli, tools, utils, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# stringi       \"tools, utils, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# stringr       \"glue (>= 1.2.0), magrittr, stringi (>= 1.1.7)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# svglite       \"systemfonts (>= 1.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# sys           NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# systemfonts   NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# testthat      \"brio, callr (>= 3.5.1), cli (>= 2.2.0), crayon (>= 1.3.4),\\ndesc, digest, ellipsis (>= 0.2.0), evaluate, jsonlite,\\nlifecycle, magrittr, methods, pkgload, praise, processx, ps (>=\\n1.3.4), R6 (>= 2.2.0), rlang (>= 0.4.9), utils, waldo (>=\\n0.2.4), withr (>= 2.3.0)\"                                                                                                                                                                                                                                                                                                               \n",
        "# tibble        \"ellipsis (>= 0.3.2), fansi (>= 0.4.0), lifecycle (>= 0.2.0),\\nmagrittr, methods, pillar (>= 1.6.0), pkgconfig, rlang (>=\\n0.4.3), utils, vctrs (>= 0.3.8)\"                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# tidyr         \"dplyr (>= 0.8.2), ellipsis (>= 0.1.0), glue, lifecycle,\\nmagrittr, purrr, rlang, tibble (>= 2.1.1), tidyselect (>=\\n1.1.0), utils, vctrs (>= 0.3.6)\"                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# tidyselect    \"ellipsis, glue (>= 1.3.0), purrr (>= 0.3.2), rlang (>= 0.4.6),\\nvctrs (>= 0.3.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# tidyverse     \"broom (>= 0.7.6), cli (>= 2.4.0), crayon (>= 1.4.1), dbplyr\\n(>= 2.1.1), dplyr (>= 1.0.5), dtplyr (>= 1.1.0), forcats (>=\\n0.5.1), googledrive (>= 1.0.1), googlesheets4 (>= 0.3.0),\\nggplot2 (>= 3.3.3), haven (>= 2.3.1), hms (>= 1.0.0), httr (>=\\n1.4.2), jsonlite (>= 1.7.2), lubridate (>= 1.7.10), magrittr\\n(>= 2.0.1), modelr (>= 0.1.8), pillar (>= 1.6.0), purrr (>=\\n0.3.4), readr (>= 1.4.0), readxl (>= 1.3.1), reprex (>= 2.0.0),\\nrlang (>= 0.4.10), rstudioapi (>= 0.13), rvest (>= 1.0.0),\\nstringr (>= 1.4.0), tibble (>= 3.1.0), tidyr (>= 1.1.3), xml2\\n(>= 1.3.2)\"\n",
        "# tinytex       \"xfun (>= 0.23)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# tzdb          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# usethis       \"cli, clipr (>= 0.3.0), crayon, curl (>= 2.7), desc, fs (>=\\n1.3.0), gert (>= 1.0.2), gh (>= 1.2.0), glue (>= 1.3.0),\\njsonlite, lifecycle, purrr, rappdirs, rlang (>= 0.4.10),\\nrprojroot (>= 1.2), rstudioapi, stats, utils, whisker, withr\\n(>= 2.3.0), yaml\"                                                                                                                                                                                                                                                                                                                         \n",
        "# utf8          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# uuid          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# vctrs         \"ellipsis (>= 0.2.0), glue, rlang (>= 0.4.10)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# viridisLite   NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# vroom         \"bit64, crayon, cli, glue, hms, lifecycle, methods, rlang (>=\\n0.4.2), stats, tibble (>= 2.0.0), tzdb (>= 0.1.1), vctrs (>=\\n0.2.0), tidyselect, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# waldo         \"cli, diffobj (>= 0.3.4), fansi, glue, methods, rematch2, rlang\\n(>= 0.4.10), tibble\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# whisker       NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# withr         \"graphics, grDevices, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# xfun          \"stats, tools\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# xml2          \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# xopen         \"processx\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# yaml          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# zip           NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# base          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# boot          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# class         \"MASS\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# cluster       \"graphics, grDevices, stats, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# codetools     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# compiler      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# datasets      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# foreign       \"methods, utils, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# graphics      \"grDevices\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# grDevices     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# grid          \"grDevices, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# KernSmooth    NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# lattice       \"grid, grDevices, graphics, stats, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# MASS          \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# Matrix        \"methods, graphics, grid, stats, utils, lattice\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# methods       \"utils, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# mgcv          \"methods, stats, graphics, Matrix, splines, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# nlme          \"graphics, stats, utils, lattice\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# nnet          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# parallel      \"tools, compiler\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# rpart         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# spatial       NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# splines       \"graphics, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# stats         \"utils, grDevices, graphics\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# stats4        \"graphics, methods, stats\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# survival      \"graphics, Matrix, methods, splines, stats, utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# tcltk         \"utils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# tools         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# utils         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "#               LinkingTo                                               \n",
        "# IRdisplay     NA                                                      \n",
        "# IRkernel      NA                                                      \n",
        "# pbdZMQ        NA                                                      \n",
        "# repr          NA                                                      \n",
        "# askpass       NA                                                      \n",
        "# assertthat    NA                                                      \n",
        "# backports     NA                                                      \n",
        "# base64enc     NA                                                      \n",
        "# bit           NA                                                      \n",
        "# bit64         NA                                                      \n",
        "# blob          NA                                                      \n",
        "# brew          NA                                                      \n",
        "# brio          NA                                                      \n",
        "# broom         NA                                                      \n",
        "# cachem        NA                                                      \n",
        "# callr         NA                                                      \n",
        "# cellranger    NA                                                      \n",
        "# cli           NA                                                      \n",
        "# clipr         NA                                                      \n",
        "# colorspace    NA                                                      \n",
        "# commonmark    NA                                                      \n",
        "# cpp11         NA                                                      \n",
        "# crayon        NA                                                      \n",
        "# credentials   NA                                                      \n",
        "# curl          NA                                                      \n",
        "# data.table    NA                                                      \n",
        "# DBI           NA                                                      \n",
        "# dbplyr        NA                                                      \n",
        "# desc          NA                                                      \n",
        "# devtools      NA                                                      \n",
        "# diffobj       NA                                                      \n",
        "# digest        NA                                                      \n",
        "# dplyr         NA                                                      \n",
        "# dtplyr        NA                                                      \n",
        "# ellipsis      NA                                                      \n",
        "# evaluate      NA                                                      \n",
        "# fansi         NA                                                      \n",
        "# farver        NA                                                      \n",
        "# fastmap       NA                                                      \n",
        "# forcats       NA                                                      \n",
        "# fs            NA                                                      \n",
        "# gargle        NA                                                      \n",
        "# generics      NA                                                      \n",
        "# gert          NA                                                      \n",
        "# ggplot2       NA                                                      \n",
        "# gh            NA                                                      \n",
        "# gitcreds      NA                                                      \n",
        "# glue          NA                                                      \n",
        "# googledrive   NA                                                      \n",
        "# googlesheets4 NA                                                      \n",
        "# gtable        NA                                                      \n",
        "# haven         \"cpp11\"                                                 \n",
        "# highr         NA                                                      \n",
        "# hms           NA                                                      \n",
        "# htmltools     NA                                                      \n",
        "# httr          NA                                                      \n",
        "# ids           NA                                                      \n",
        "# ini           NA                                                      \n",
        "# isoband       NA                                                      \n",
        "# jsonlite      NA                                                      \n",
        "# knitr         NA                                                      \n",
        "# labeling      NA                                                      \n",
        "# lifecycle     NA                                                      \n",
        "# lubridate     \"Rcpp\"                                                  \n",
        "# magrittr      NA                                                      \n",
        "# markdown      NA                                                      \n",
        "# memoise       NA                                                      \n",
        "# mime          NA                                                      \n",
        "# modelr        NA                                                      \n",
        "# munsell       NA                                                      \n",
        "# openssl       NA                                                      \n",
        "# pillar        NA                                                      \n",
        "# pkgbuild      NA                                                      \n",
        "# pkgconfig     NA                                                      \n",
        "# pkgload       NA                                                      \n",
        "# praise        NA                                                      \n",
        "# prettyunits   NA                                                      \n",
        "# processx      NA                                                      \n",
        "# progress      NA                                                      \n",
        "# ps            NA                                                      \n",
        "# purrr         NA                                                      \n",
        "# R6            NA                                                      \n",
        "# rappdirs      NA                                                      \n",
        "# rcmdcheck     NA                                                      \n",
        "# RColorBrewer  NA                                                      \n",
        "# Rcpp          NA                                                      \n",
        "# readr         \"cpp11, tzdb (>= 0.1.1)\"                                \n",
        "# readxl        \"progress, Rcpp\"                                        \n",
        "# rematch       NA                                                      \n",
        "# rematch2      NA                                                      \n",
        "# remotes       NA                                                      \n",
        "# reprex        NA                                                      \n",
        "# rlang         NA                                                      \n",
        "# rmarkdown     NA                                                      \n",
        "# roxygen2      \"Rcpp\"                                                  \n",
        "# rprojroot     NA                                                      \n",
        "# rstudioapi    NA                                                      \n",
        "# rversions     NA                                                      \n",
        "# rvest         NA                                                      \n",
        "# scales        NA                                                      \n",
        "# selectr       NA                                                      \n",
        "# sessioninfo   NA                                                      \n",
        "# stringi       NA                                                      \n",
        "# stringr       NA                                                      \n",
        "# svglite       \"cpp11, systemfonts\"                                    \n",
        "# sys           NA                                                      \n",
        "# systemfonts   \"cpp11 (>= 0.2.1)\"                                      \n",
        "# testthat      NA                                                      \n",
        "# tibble        NA                                                      \n",
        "# tidyr         \"cpp11 (>= 0.2.6)\"                                      \n",
        "# tidyselect    NA                                                      \n",
        "# tidyverse     NA                                                      \n",
        "# tinytex       NA                                                      \n",
        "# tzdb          \"cpp11 (>= 0.2.7)\"                                      \n",
        "# usethis       NA                                                      \n",
        "# utf8          NA                                                      \n",
        "# uuid          NA                                                      \n",
        "# vctrs         NA                                                      \n",
        "# viridisLite   NA                                                      \n",
        "# vroom         \"progress (>= 1.2.1), cpp11 (>= 0.2.0), tzdb (>= 0.1.1)\"\n",
        "# waldo         NA                                                      \n",
        "# whisker       NA                                                      \n",
        "# withr         NA                                                      \n",
        "# xfun          NA                                                      \n",
        "# xml2          NA                                                      \n",
        "# xopen         NA                                                      \n",
        "# yaml          NA                                                      \n",
        "# zip           NA                                                      \n",
        "# base          NA                                                      \n",
        "# boot          NA                                                      \n",
        "# class         NA                                                      \n",
        "# cluster       NA                                                      \n",
        "# codetools     NA                                                      \n",
        "# compiler      NA                                                      \n",
        "# datasets      NA                                                      \n",
        "# foreign       NA                                                      \n",
        "# graphics      NA                                                      \n",
        "# grDevices     NA                                                      \n",
        "# grid          NA                                                      \n",
        "# KernSmooth    NA                                                      \n",
        "# lattice       NA                                                      \n",
        "# MASS          NA                                                      \n",
        "# Matrix        NA                                                      \n",
        "# methods       NA                                                      \n",
        "# mgcv          NA                                                      \n",
        "# nlme          NA                                                      \n",
        "# nnet          NA                                                      \n",
        "# parallel      NA                                                      \n",
        "# rpart         NA                                                      \n",
        "# spatial       NA                                                      \n",
        "# splines       NA                                                      \n",
        "# stats         NA                                                      \n",
        "# stats4        NA                                                      \n",
        "# survival      NA                                                      \n",
        "# tcltk         NA                                                      \n",
        "# tools         NA                                                      \n",
        "# utils         NA                                                      \n",
        "#               Suggests                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# IRdisplay     \"testthat, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# IRkernel      \"testthat, roxygen2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# pbdZMQ        NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# repr          \"methods, highr, Cairo, stringr, testthat (>= 3.0.0), leaflet\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# askpass       \"testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# assertthat    \"testthat, covr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# backports     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# base64enc     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# bit           \"testthat (>= 0.11.0), roxygen2, knitr, rmarkdown,\\nmicrobenchmark, bit64 (>= 4.0.0), ff (>= 4.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# bit64         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# blob          \"covr, crayon, pillar (>= 1.2.1), testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# brew          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# brio          \"testthat (>= 2.1.0), covr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# broom         \"AER, akima, AUC, bbmle, betareg, biglm, binGroup, boot,\\nbtergm, car, caret, cluster, cmprsk, coda, covr, drc, e1071,\\nemmeans, epiR, ergm (>= 3.10.4), fixest (>= 0.9.0), gam (>=\\n1.15), gee, geepack, ggplot2, glmnet, glmnetUtils, gmm, Hmisc,\\nirlba, joineRML, Kendall, knitr, ks, Lahman, lavaan, leaps,\\nlfe, lm.beta, lme4, lmodel2, lmtest (>= 0.9.38), lsmeans, maps,\\nmaptools, margins, MASS, Matrix, mclust, mediation, metafor,\\nmfx, mgcv, mlogit, modeldata, modeltests, muhaz, multcomp,\\nnetwork, nnet, orcutt (>= 2.2), ordinal, plm, poLCA, psych,\\nquantreg, rgeos, rmarkdown, robust, robustbase, rsample,\\nsandwich, sp, spdep (>= 1.1), spatialreg, speedglm, spelling,\\nsurvey, survival, systemfit, testthat (>= 2.1.0), tseries,\\nvars, zoo\"\n",
        "# cachem        \"testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# callr         \"cli, covr, ps, rprojroot, spelling, testthat, withr (>=\\n2.3.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# cellranger    \"covr, testthat (>= 1.0.0), knitr, rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# cli           \"callr, covr, grDevices, htmlwidgets, knitr, methods, mockery,\\nprettycode (>= 1.1.0), processx, ps (>= 1.3.4.9000), rlang,\\nrmarkdown, rstudioapi, shiny, testthat, tibble, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# clipr         \"covr, knitr, rmarkdown, rstudioapi (>= 0.5), testthat (>=\\n2.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# colorspace    \"datasets, utils, KernSmooth, MASS, kernlab, mvtnorm, vcd,\\ntcltk, shiny, shinyjs, ggplot2, dplyr, scales, grid, png, jpeg,\\nknitr, rmarkdown, RColorBrewer, rcartocolor, scico, viridis,\\nwesanderson\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# commonmark    \"curl, testthat, xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# cpp11         \"bench, brio, callr, cli, covr, decor, desc, ggplot2, glue,\\nknitr, lobstr, mockery, progress, rmarkdown, scales, Rcpp,\\ntestthat, tibble, utils, vctrs, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# crayon        \"mockery, rstudioapi, testthat, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# credentials   \"testthat, knitr, rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# curl          \"spelling, testthat (>= 1.0.0), knitr, jsonlite, rmarkdown,\\nmagrittr, httpuv (>= 1.4.4), webutils\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# data.table    \"bit64 (>= 4.0.0), bit (>= 4.0.4), curl, R.utils, xts,\\nnanotime, zoo (>= 1.8-1), yaml, knitr, rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# DBI           \"blob, covr, dbplyr, dplyr, glue, hms, knitr, magrittr,\\nrmarkdown, rprojroot, RMariaDB, RSQLite (>= 1.1-2), testthat,\\nxml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# dbplyr        \"bit64, covr, knitr, Lahman, nycflights13, odbc, RMariaDB (>=\\n1.0.2), rmarkdown, RPostgres (>= 1.1.3), RPostgreSQL, RSQLite\\n(>= 2.1.0), testthat (>= 3.0.2), tidyr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# desc          \"covr, testthat, whoami, withr, spelling, gh\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# devtools      \"BiocManager (>= 1.30.12), covr (>= 3.5.1), curl (>= 4.3),\\ndigest (>= 0.6.27), DT (>= 0.17), foghorn (>= 1.3.2), gh (>=\\n1.2.1), gmailr (>= 1.0.0), knitr (>= 1.31), lintr (>= 2.0.1),\\nMASS, mockery (>= 0.4.2), pingr (>= 2.0.1), pkgdown (>= 1.6.1),\\nrhub (>= 1.1.1), rmarkdown (>= 2.7), spelling (>= 2.2)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# diffobj       \"knitr, rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# digest        \"tinytest, knitr, rmarkdown, minidown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# dplyr         \"bench, broom, callr, covr, DBI, dbplyr (>= 1.4.3), knitr,\\nLahman, lobstr, microbenchmark, nycflights13, purrr, rmarkdown,\\nRMySQL, RPostgreSQL, RSQLite, testthat (>= 3.0.2), tidyr, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# dtplyr        \"bench, covr, knitr, rmarkdown, testthat (>= 3.0.0), tidyr (>=\\n1.1.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# ellipsis      \"covr, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# evaluate      \"testthat, lattice, ggplot2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# fansi         \"unitizer, knitr, rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# farver        \"testthat (>= 2.1.0), covr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# fastmap       \"testthat (>= 2.1.1)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# forcats       \"covr, dplyr, ggplot2, knitr, readr, rmarkdown, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# fs            \"testthat, covr, pillar (>= 1.0.0), tibble (>= 1.1.0), crayon,\\nrmarkdown, knitr, withr, spelling, vctrs (>= 0.3.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# gargle        \"aws.ec2metadata, aws.signature, covr, httpuv, knitr, mockr,\\nrmarkdown, sodium, spelling, testthat (>= 3.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# generics      \"covr, pkgload, testthat, tibble\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# gert          \"spelling, knitr, rmarkdown, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# ggplot2       \"covr, ragg, dplyr, ggplot2movies, hexbin, Hmisc, interp,\\nknitr, lattice, mapproj, maps, maptools, multcomp, munsell,\\nnlme, profvis, quantreg, RColorBrewer, rgeos, rmarkdown, rpart,\\nsf (>= 0.7-3), svglite (>= 1.2.0.9001), testthat (>= 2.1.0),\\nvdiffr (>= 1.0.0), xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# gh            \"covr, knitr, mockery, rmarkdown, rprojroot, spelling,\\ntestthat (>= 2.3.2), withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# gitcreds      \"codetools, testthat, knitr, mockery, oskeyring, rmarkdown,\\nwithr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# glue          \"testthat, covr, magrittr, crayon, knitr, rmarkdown, DBI,\\nRSQLite, R.utils, forcats, microbenchmark, rprintf, stringr,\\nggplot2, dplyr, withr, vctrs (>= 0.3.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# googledrive   \"covr, curl, downlit, dplyr (>= 1.0.0), knitr, mockr,\\nrmarkdown, roxygen2, sodium, spelling, testthat (>= 3.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# googlesheets4 \"covr, readr, rmarkdown, sodium, spelling, testthat (>=\\n3.0.0), withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# gtable        \"covr, testthat, knitr, rmarkdown, ggplot2, profvis\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# haven         \"cli, covr, crayon, fs, knitr, pillar (>= 1.4.0), rmarkdown,\\ntestthat (>= 3.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# highr         \"knitr, markdown, testit\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# hms           \"crayon, lubridate, pillar (>= 1.1.0), testthat (>= 3.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# htmltools     \"markdown, testthat, withr, Cairo, ragg, shiny\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# httr          \"covr, httpuv, jpeg, knitr, png, readr, rmarkdown, testthat\\n(>= 0.8.0), xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# ids           \"knitr, rcorpora, rmarkdown, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# ini           \"testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# isoband       \"covr, ggplot2, knitr, magick, microbenchmark, rmarkdown, sf,\\ntestthat, xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# jsonlite      \"httr, curl, plyr, testthat, knitr, rmarkdown, R.rsp, sf\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# knitr         \"formatR, testit, digest, rgl (>= 0.95.1201), codetools,\\nrmarkdown, htmlwidgets (>= 0.7), webshot, tikzDevice (>= 0.10),\\ntinytex, reticulate (>= 1.4), JuliaCall (>= 0.11.1), magick,\\npng, jpeg, gifski, xml2 (>= 1.2.0), httr, DBI (>= 0.4-1),\\nshowtext, tibble, sass, bslib, ragg, styler (>= 1.2.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# labeling      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# lifecycle     \"covr, crayon, knitr, rmarkdown, testthat (>= 3.0.1), tibble\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# lubridate     \"covr, knitr, testthat (>= 2.1.0), vctrs (>= 0.3.0), rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# magrittr      \"covr, knitr, rlang, rmarkdown, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# markdown      \"knitr, RCurl\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# memoise       \"digest, aws.s3, covr, googleAuthR, googleCloudStorageR, httr,\\ntestthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# mime          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# modelr        \"compiler, covr, ggplot2, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# munsell       \"ggplot2, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# openssl       \"testthat (>= 2.1.0), digest, knitr, rmarkdown, jsonlite,\\njose, sodium\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# pillar        \"bit64, debugme, DiagrammeR, dplyr, formattable, ggplot2,\\nknitr, lubridate, nycflights13, palmerpenguins, rmarkdown,\\nscales, stringi, survival, testthat (>= 3.0.2), tibble, units\\n(>= 0.7.2), withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# pkgbuild      \"Rcpp, cpp11, testthat, covr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# pkgconfig     \"covr, testthat, disposables (>= 1.0.3)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# pkgload       \"bitops, covr, pkgbuild, Rcpp, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# praise        \"testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# prettyunits   \"codetools, covr, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# processx      \"callr (>= 3.2.0), cli, codetools, covr, curl, debugme,\\nparallel, testthat, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# progress      \"Rcpp, testthat, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# ps            \"callr, covr, curl, pingr, processx (>= 3.1.0), R6, rlang,\\ntestthat, tibble\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# purrr         \"covr, crayon, dplyr (>= 0.7.8), knitr, rmarkdown, testthat,\\ntibble, tidyselect\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# R6            \"testthat, pryr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# rappdirs      \"roxygen2, testthat (>= 3.0.0), covr, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# rcmdcheck     \"covr, knitr, mockery, rmarkdown, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# RColorBrewer  NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# Rcpp          \"tinytest, inline, rbenchmark, pkgKitten (>= 0.1.2)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# readr         \"covr, curl, dplyr, knitr, rmarkdown, spelling, stringi,\\ntestthat, tzdb (>= 0.1.1), waldo, withr, xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# readxl        \"covr, knitr, rmarkdown, rprojroot (>= 1.1), testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# rematch       \"covr, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# rematch2      \"covr, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# remotes       \"brew, callr, codetools, curl, covr, git2r (>= 0.23.0), knitr,\\nmockery, pkgbuild (>= 1.0.1), pingr, rmarkdown, rprojroot,\\ntestthat, webfakes, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# reprex        \"covr, fortunes, miniUI, mockr, rprojroot, sessioninfo, shiny,\\nspelling, styler (>= 1.2.0), testthat (>= 3.0.2)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# rlang         \"cli, covr, crayon, glue, magrittr, methods, pak, pillar,\\nrmarkdown, testthat (>= 3.0.0), vctrs (>= 0.2.3), withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# rmarkdown     \"shiny (>= 1.6.0), tufte, testthat (>= 3.0.0), digest,\\ndygraphs, vctrs, tibble, fs, rsconnect, withr (>= 2.4.2), bslib\\n(>= 0.2.5.1), sass (>= 0.4.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# roxygen2      \"covr, devtools, rmarkdown, testthat (>= 2.1.0), R.methodsS3,\\nR.oo\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# rprojroot     \"covr, knitr, lifecycle, mockr, rmarkdown, testthat (>=\\n3.0.0), withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# rstudioapi    \"testthat, knitr, rmarkdown, clipr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# rversions     \"mockery, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# rvest         \"covr, glue, knitr, readr, rmarkdown, repurrrsive, spelling,\\nstringi (>= 0.3.1), testthat (>= 3.0.2), webfakes\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# scales        \"bit64, covr, dichromat, ggplot2, hms (>= 0.5.0), testthat (>=\\n2.1.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# selectr       \"testthat, XML, xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# sessioninfo   \"callr, covr, mockery, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# stringi       NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# stringr       \"covr, htmltools, htmlwidgets, knitr, rmarkdown, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# svglite       \"htmltools, testthat, xml2 (>= 1.0.0), covr, fontquiver (>=\\n0.2.0), knitr, rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# sys           \"unix (>= 1.4), spelling, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# systemfonts   \"testthat (>= 2.1.0), covr, knitr, rmarkdown, tools\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# testthat      \"covr, curl (>= 0.9.5), diffviewer (>= 0.1.0), knitr, mockery,\\nrmarkdown, rstudioapi, shiny, usethis, vctrs (>= 0.1.0), xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# tibble        \"bench, bit64, blob, cli, covr, crayon (>= 1.3.4), DiagrammeR,\\ndplyr, evaluate, formattable, ggplot2, hms, htmltools, knitr,\\nlubridate, mockr, nycflights13, purrr, rmarkdown, testthat (>=\\n3.0.2), tidyr, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# tidyr         \"covr, data.table, jsonlite, knitr, readr, repurrrsive (>=\\n1.0.0), rmarkdown, testthat (>= 3.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
        "# tidyselect    \"covr, crayon, dplyr, knitr, magrittr, rmarkdown, testthat (>=\\n2.3.0), tibble (>= 2.1.3), withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# tidyverse     \"covr, feather, glue, knitr, rmarkdown, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# tinytex       \"testit, rstudioapi\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
        "# tzdb          \"covr, testthat (>= 3.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# usethis       \"covr, knitr, magick, mockr, rmarkdown, roxygen2, spelling (>=\\n1.2), styler (>= 1.2.0), testthat (>= 3.0.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# utf8          \"cli, covr, knitr, rlang, rmarkdown, testthat (>= 3.0.0),\\nwithr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# uuid          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# vctrs         \"bit64, covr, crayon, dplyr (>= 0.8.5), generics, knitr,\\npillar (>= 1.4.4), pkgdown, rmarkdown, testthat (>= 2.3.0),\\ntibble, withr, xml2, waldo (>= 0.2.0), zeallot\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# viridisLite   \"hexbin (>= 1.27.0), ggplot2 (>= 1.0.1), testthat, covr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 \n",
        "# vroom         \"archive, bench (>= 1.1.0), covr, curl, dplyr, forcats, fs,\\nggplot2, knitr, patchwork, prettyunits, purrr, rmarkdown,\\nrstudioapi, scales, spelling, testthat (>= 2.1.0), tidyr,\\nwaldo, xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# waldo         \"testthat (>= 3.0.0), covr, R6, xml2\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# whisker       \"markdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# withr         \"covr, DBI, knitr, lattice, methods, rmarkdown, RSQLite,\\ntestthat (>= 2.1.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# xfun          \"testit, parallel, codetools, rstudioapi, tinytex (>= 0.30),\\nmime, markdown, knitr, htmltools, remotes, pak, renv, curl,\\njsonlite, rmarkdown\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# xml2          \"covr, curl, httr, knitr, magrittr, mockery, rmarkdown,\\ntestthat (>= 2.1.0)\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# xopen         \"ps, testthat\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# yaml          \"RUnit\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
        "# zip           \"covr, processx, R6, testthat, withr\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    \n",
        "# base          \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# boot          \"MASS, survival\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# class         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# cluster       \"MASS, Matrix\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           \n",
        "# codetools     NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# compiler      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# datasets      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# foreign       NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# graphics      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# grDevices     \"KernSmooth\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# grid          NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# KernSmooth    \"MASS, carData\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# lattice       \"KernSmooth, MASS, latticeExtra\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
        "# MASS          \"lattice, nlme, nnet, survival\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
        "# Matrix        \"expm, MASS\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# methods       \"codetools\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "# mgcv          \"parallel, survival, MASS\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# nlme          \"Hmisc, MASS\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            \n",
        "# nnet          \"MASS\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# parallel      \"methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
        "# rpart         \"survival\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# spatial       \"MASS\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
        "# splines       \"Matrix, methods\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
        "# stats         \"MASS, Matrix, SuppDists, methods, stats4\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
        "# stats4        NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# survival      NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# tcltk         NA                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       \n",
        "# tools         \"codetools, methods, xml2, curl, commonmark\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             \n",
        "# utils         \"methods, xml2, commonmark\"                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              \n",
        "#               Enhances                                                             \n",
        "# IRdisplay     NA                                                                   \n",
        "# IRkernel      NA                                                                   \n",
        "# pbdZMQ        NA                                                                   \n",
        "# repr          \"data.table, tibble, htmlwidgets, vegalite, plotly, geojsonio\"       \n",
        "# askpass       NA                                                                   \n",
        "# assertthat    NA                                                                   \n",
        "# backports     NA                                                                   \n",
        "# base64enc     \"png\"                                                                \n",
        "# bit           NA                                                                   \n",
        "# bit64         NA                                                                   \n",
        "# blob          NA                                                                   \n",
        "# brew          NA                                                                   \n",
        "# brio          NA                                                                   \n",
        "# broom         NA                                                                   \n",
        "# cachem        NA                                                                   \n",
        "# callr         NA                                                                   \n",
        "# cellranger    NA                                                                   \n",
        "# cli           NA                                                                   \n",
        "# clipr         NA                                                                   \n",
        "# colorspace    NA                                                                   \n",
        "# commonmark    NA                                                                   \n",
        "# cpp11         NA                                                                   \n",
        "# crayon        NA                                                                   \n",
        "# credentials   NA                                                                   \n",
        "# curl          NA                                                                   \n",
        "# data.table    NA                                                                   \n",
        "# DBI           NA                                                                   \n",
        "# dbplyr        NA                                                                   \n",
        "# desc          NA                                                                   \n",
        "# devtools      NA                                                                   \n",
        "# diffobj       NA                                                                   \n",
        "# digest        NA                                                                   \n",
        "# dplyr         NA                                                                   \n",
        "# dtplyr        NA                                                                   \n",
        "# ellipsis      NA                                                                   \n",
        "# evaluate      NA                                                                   \n",
        "# fansi         NA                                                                   \n",
        "# farver        NA                                                                   \n",
        "# fastmap       NA                                                                   \n",
        "# forcats       NA                                                                   \n",
        "# fs            NA                                                                   \n",
        "# gargle        NA                                                                   \n",
        "# generics      NA                                                                   \n",
        "# gert          NA                                                                   \n",
        "# ggplot2       \"sp\"                                                                 \n",
        "# gh            NA                                                                   \n",
        "# gitcreds      NA                                                                   \n",
        "# glue          NA                                                                   \n",
        "# googledrive   NA                                                                   \n",
        "# googlesheets4 NA                                                                   \n",
        "# gtable        NA                                                                   \n",
        "# haven         NA                                                                   \n",
        "# highr         NA                                                                   \n",
        "# hms           NA                                                                   \n",
        "# htmltools     \"knitr\"                                                              \n",
        "# httr          NA                                                                   \n",
        "# ids           NA                                                                   \n",
        "# ini           NA                                                                   \n",
        "# isoband       NA                                                                   \n",
        "# jsonlite      NA                                                                   \n",
        "# knitr         NA                                                                   \n",
        "# labeling      NA                                                                   \n",
        "# lifecycle     NA                                                                   \n",
        "# lubridate     \"chron, timeDate, tis, zoo\"                                          \n",
        "# magrittr      NA                                                                   \n",
        "# markdown      NA                                                                   \n",
        "# memoise       NA                                                                   \n",
        "# mime          NA                                                                   \n",
        "# modelr        NA                                                                   \n",
        "# munsell       NA                                                                   \n",
        "# openssl       NA                                                                   \n",
        "# pillar        NA                                                                   \n",
        "# pkgbuild      NA                                                                   \n",
        "# pkgconfig     NA                                                                   \n",
        "# pkgload       NA                                                                   \n",
        "# praise        NA                                                                   \n",
        "# prettyunits   NA                                                                   \n",
        "# processx      NA                                                                   \n",
        "# progress      NA                                                                   \n",
        "# ps            NA                                                                   \n",
        "# purrr         NA                                                                   \n",
        "# R6            NA                                                                   \n",
        "# rappdirs      NA                                                                   \n",
        "# rcmdcheck     NA                                                                   \n",
        "# RColorBrewer  NA                                                                   \n",
        "# Rcpp          NA                                                                   \n",
        "# readr         NA                                                                   \n",
        "# readxl        NA                                                                   \n",
        "# rematch       NA                                                                   \n",
        "# rematch2      NA                                                                   \n",
        "# remotes       NA                                                                   \n",
        "# reprex        NA                                                                   \n",
        "# rlang         \"winch\"                                                              \n",
        "# rmarkdown     NA                                                                   \n",
        "# roxygen2      NA                                                                   \n",
        "# rprojroot     NA                                                                   \n",
        "# rstudioapi    NA                                                                   \n",
        "# rversions     NA                                                                   \n",
        "# rvest         NA                                                                   \n",
        "# scales        NA                                                                   \n",
        "# selectr       NA                                                                   \n",
        "# sessioninfo   NA                                                                   \n",
        "# stringi       NA                                                                   \n",
        "# stringr       NA                                                                   \n",
        "# svglite       NA                                                                   \n",
        "# sys           NA                                                                   \n",
        "# systemfonts   NA                                                                   \n",
        "# testthat      NA                                                                   \n",
        "# tibble        NA                                                                   \n",
        "# tidyr         NA                                                                   \n",
        "# tidyselect    NA                                                                   \n",
        "# tidyverse     NA                                                                   \n",
        "# tinytex       NA                                                                   \n",
        "# tzdb          NA                                                                   \n",
        "# usethis       NA                                                                   \n",
        "# utf8          NA                                                                   \n",
        "# uuid          NA                                                                   \n",
        "# vctrs         NA                                                                   \n",
        "# viridisLite   NA                                                                   \n",
        "# vroom         NA                                                                   \n",
        "# waldo         NA                                                                   \n",
        "# whisker       NA                                                                   \n",
        "# withr         NA                                                                   \n",
        "# xfun          NA                                                                   \n",
        "# xml2          NA                                                                   \n",
        "# xopen         NA                                                                   \n",
        "# yaml          NA                                                                   \n",
        "# zip           NA                                                                   \n",
        "# base          NA                                                                   \n",
        "# boot          NA                                                                   \n",
        "# class         NA                                                                   \n",
        "# cluster       NA                                                                   \n",
        "# codetools     NA                                                                   \n",
        "# compiler      NA                                                                   \n",
        "# datasets      NA                                                                   \n",
        "# foreign       NA                                                                   \n",
        "# graphics      NA                                                                   \n",
        "# grDevices     NA                                                                   \n",
        "# grid          NA                                                                   \n",
        "# KernSmooth    NA                                                                   \n",
        "# lattice       \"chron\"                                                              \n",
        "# MASS          NA                                                                   \n",
        "# Matrix        \"MatrixModels, graph, SparseM, sfsmisc, igraph, maptools, sp,\\nspdep\"\n",
        "# methods       NA                                                                   \n",
        "# mgcv          NA                                                                   \n",
        "# nlme          NA                                                                   \n",
        "# nnet          NA                                                                   \n",
        "# parallel      \"snow, nws, Rmpi\"                                                    \n",
        "# rpart         NA                                                                   \n",
        "# spatial       NA                                                                   \n",
        "# splines       NA                                                                   \n",
        "# stats         NA                                                                   \n",
        "# stats4        NA                                                                   \n",
        "# survival      NA                                                                   \n",
        "# tcltk         NA                                                                   \n",
        "# tools         NA                                                                   \n",
        "# utils         NA                                                                   \n",
        "#               License                                  License_is_FOSS\n",
        "# IRdisplay     \"MIT + file LICENSE\"                     NA             \n",
        "# IRkernel      \"MIT + file LICENSE\"                     NA             \n",
        "# pbdZMQ        \"GPL-3\"                                  NA             \n",
        "# repr          \"GPL-3\"                                  NA             \n",
        "# askpass       \"MIT + file LICENSE\"                     NA             \n",
        "# assertthat    \"GPL-3\"                                  NA             \n",
        "# backports     \"GPL-2 | GPL-3\"                          NA             \n",
        "# base64enc     \"GPL-2 | GPL-3\"                          NA             \n",
        "# bit           \"GPL-2 | GPL-3\"                          NA             \n",
        "# bit64         \"GPL-2 | GPL-3\"                          NA             \n",
        "# blob          \"MIT + file LICENSE\"                     NA             \n",
        "# brew          \"GPL-2\"                                  NA             \n",
        "# brio          \"MIT + file LICENSE\"                     NA             \n",
        "# broom         \"MIT + file LICENSE\"                     NA             \n",
        "# cachem        \"MIT + file LICENSE\"                     NA             \n",
        "# callr         \"MIT + file LICENSE\"                     NA             \n",
        "# cellranger    \"MIT + file LICENSE\"                     NA             \n",
        "# cli           \"MIT + file LICENSE\"                     NA             \n",
        "# clipr         \"GPL-3\"                                  NA             \n",
        "# colorspace    \"BSD_3_clause + file LICENSE\"            NA             \n",
        "# commonmark    \"BSD_2_clause + file LICENSE\"            NA             \n",
        "# cpp11         \"MIT + file LICENSE\"                     NA             \n",
        "# crayon        \"MIT + file LICENSE\"                     NA             \n",
        "# credentials   \"MIT + file LICENSE\"                     NA             \n",
        "# curl          \"MIT + file LICENSE\"                     NA             \n",
        "# data.table    \"MPL-2.0 | file LICENSE\"                 NA             \n",
        "# DBI           \"LGPL (>= 2.1)\"                          NA             \n",
        "# dbplyr        \"MIT + file LICENSE\"                     NA             \n",
        "# desc          \"MIT + file LICENSE\"                     NA             \n",
        "# devtools      \"MIT + file LICENSE\"                     NA             \n",
        "# diffobj       \"GPL (>= 2)\"                             NA             \n",
        "# digest        \"GPL (>= 2)\"                             NA             \n",
        "# dplyr         \"MIT + file LICENSE\"                     NA             \n",
        "# dtplyr        \"MIT + file LICENSE\"                     NA             \n",
        "# ellipsis      \"MIT + file LICENSE\"                     NA             \n",
        "# evaluate      \"MIT + file LICENSE\"                     NA             \n",
        "# fansi         \"GPL (>= 2)\"                             NA             \n",
        "# farver        \"MIT + file LICENSE\"                     NA             \n",
        "# fastmap       \"MIT + file LICENSE\"                     NA             \n",
        "# forcats       \"MIT + file LICENSE\"                     NA             \n",
        "# fs            \"GPL-3\"                                  NA             \n",
        "# gargle        \"MIT + file LICENSE\"                     NA             \n",
        "# generics      \"MIT + file LICENSE\"                     NA             \n",
        "# gert          \"MIT + file LICENSE\"                     NA             \n",
        "# ggplot2       \"MIT + file LICENSE\"                     NA             \n",
        "# gh            \"MIT + file LICENSE\"                     NA             \n",
        "# gitcreds      \"MIT + file LICENSE\"                     NA             \n",
        "# glue          \"MIT + file LICENSE\"                     NA             \n",
        "# googledrive   \"MIT + file LICENSE\"                     NA             \n",
        "# googlesheets4 \"MIT + file LICENSE\"                     NA             \n",
        "# gtable        \"GPL-2\"                                  NA             \n",
        "# haven         \"MIT + file LICENSE\"                     NA             \n",
        "# highr         \"GPL\"                                    NA             \n",
        "# hms           \"MIT + file LICENSE\"                     NA             \n",
        "# htmltools     \"GPL (>= 2)\"                             NA             \n",
        "# httr          \"MIT + file LICENSE\"                     NA             \n",
        "# ids           \"MIT + file LICENSE\"                     NA             \n",
        "# ini           \"GPL-3\"                                  NA             \n",
        "# isoband       \"MIT + file LICENSE\"                     NA             \n",
        "# jsonlite      \"MIT + file LICENSE\"                     NA             \n",
        "# knitr         \"GPL\"                                    NA             \n",
        "# labeling      \"MIT + file LICENSE | Unlimited\"         NA             \n",
        "# lifecycle     \"MIT + file LICENSE\"                     NA             \n",
        "# lubridate     \"GPL (>= 2)\"                             NA             \n",
        "# magrittr      \"MIT + file LICENSE\"                     NA             \n",
        "# markdown      \"GPL-2\"                                  NA             \n",
        "# memoise       \"MIT + file LICENSE\"                     NA             \n",
        "# mime          \"GPL\"                                    NA             \n",
        "# modelr        \"GPL-3\"                                  NA             \n",
        "# munsell       \"MIT + file LICENSE\"                     NA             \n",
        "# openssl       \"MIT + file LICENSE\"                     NA             \n",
        "# pillar        \"MIT + file LICENSE\"                     NA             \n",
        "# pkgbuild      \"MIT + file LICENSE\"                     NA             \n",
        "# pkgconfig     \"MIT + file LICENSE\"                     NA             \n",
        "# pkgload       \"GPL-3\"                                  NA             \n",
        "# praise        \"MIT + file LICENSE\"                     NA             \n",
        "# prettyunits   \"MIT + file LICENSE\"                     NA             \n",
        "# processx      \"MIT + file LICENSE\"                     NA             \n",
        "# progress      \"MIT + file LICENSE\"                     NA             \n",
        "# ps            \"MIT + file LICENSE\"                     NA             \n",
        "# purrr         \"GPL-3 | file LICENSE\"                   NA             \n",
        "# R6            \"MIT + file LICENSE\"                     NA             \n",
        "# rappdirs      \"MIT + file LICENSE\"                     NA             \n",
        "# rcmdcheck     \"MIT + file LICENSE\"                     NA             \n",
        "# RColorBrewer  \"Apache License 2.0\"                     NA             \n",
        "# Rcpp          \"GPL (>= 2)\"                             NA             \n",
        "# readr         \"MIT + file LICENSE\"                     NA             \n",
        "# readxl        \"GPL-3\"                                  NA             \n",
        "# rematch       \"MIT + file LICENSE\"                     NA             \n",
        "# rematch2      \"MIT + file LICENSE\"                     NA             \n",
        "# remotes       \"MIT + file LICENSE\"                     NA             \n",
        "# reprex        \"MIT + file LICENSE\"                     NA             \n",
        "# rlang         \"MIT + file LICENSE\"                     NA             \n",
        "# rmarkdown     \"GPL-3\"                                  NA             \n",
        "# roxygen2      \"GPL (>= 2)\"                             NA             \n",
        "# rprojroot     \"MIT + file LICENSE\"                     NA             \n",
        "# rstudioapi    \"MIT + file LICENSE\"                     NA             \n",
        "# rversions     \"MIT + file LICENSE\"                     NA             \n",
        "# rvest         \"MIT + file LICENSE\"                     NA             \n",
        "# scales        \"MIT + file LICENSE\"                     NA             \n",
        "# selectr       \"BSD_3_clause + file LICENCE\"            NA             \n",
        "# sessioninfo   \"GPL-2\"                                  NA             \n",
        "# stringi       \"file LICENSE\"                           \"yes\"          \n",
        "# stringr       \"GPL-2 | file LICENSE\"                   NA             \n",
        "# svglite       \"GPL (>= 2)\"                             NA             \n",
        "# sys           \"MIT + file LICENSE\"                     NA             \n",
        "# systemfonts   \"MIT + file LICENSE\"                     NA             \n",
        "# testthat      \"MIT + file LICENSE\"                     NA             \n",
        "# tibble        \"MIT + file LICENSE\"                     NA             \n",
        "# tidyr         \"MIT + file LICENSE\"                     NA             \n",
        "# tidyselect    \"MIT + file LICENSE\"                     NA             \n",
        "# tidyverse     \"MIT + file LICENSE\"                     NA             \n",
        "# tinytex       \"MIT + file LICENSE\"                     NA             \n",
        "# tzdb          \"MIT + file LICENSE\"                     NA             \n",
        "# usethis       \"MIT + file LICENSE\"                     NA             \n",
        "# utf8          \"Apache License (== 2.0) | file LICENSE\" NA             \n",
        "# uuid          \"MIT + file LICENSE\"                     NA             \n",
        "# vctrs         \"MIT + file LICENSE\"                     NA             \n",
        "# viridisLite   \"MIT + file LICENSE\"                     NA             \n",
        "# vroom         \"MIT + file LICENSE\"                     NA             \n",
        "# waldo         \"MIT + file LICENSE\"                     NA             \n",
        "# whisker       \"GPL-3\"                                  NA             \n",
        "# withr         \"MIT + file LICENSE\"                     NA             \n",
        "# xfun          \"MIT + file LICENSE\"                     NA             \n",
        "# xml2          \"GPL (>= 2)\"                             NA             \n",
        "# xopen         \"MIT + file LICENSE\"                     NA             \n",
        "# yaml          \"BSD_3_clause + file LICENSE\"            NA             \n",
        "# zip           \"MIT + file LICENSE\"                     NA             \n",
        "# base          \"Part of R 4.1.1\"                        NA             \n",
        "# boot          \"Unlimited\"                              NA             \n",
        "# class         \"GPL-2 | GPL-3\"                          NA             \n",
        "# cluster       \"GPL (>= 2)\"                             NA             \n",
        "# codetools     \"GPL\"                                    NA             \n",
        "# compiler      \"Part of R 4.1.1\"                        NA             \n",
        "# datasets      \"Part of R 4.1.1\"                        NA             \n",
        "# foreign       \"GPL (>= 2)\"                             NA             \n",
        "# graphics      \"Part of R 4.1.1\"                        NA             \n",
        "# grDevices     \"Part of R 4.1.1\"                        NA             \n",
        "# grid          \"Part of R 4.1.1\"                        NA             \n",
        "# KernSmooth    \"Unlimited\"                              NA             \n",
        "# lattice       \"GPL (>= 2)\"                             NA             \n",
        "# MASS          \"GPL-2 | GPL-3\"                          NA             \n",
        "# Matrix        \"GPL (>= 2) | file LICENCE\"              NA             \n",
        "# methods       \"Part of R 4.1.1\"                        NA             \n",
        "# mgcv          \"GPL (>= 2)\"                             NA             \n",
        "# nlme          \"GPL (>= 2) | file LICENCE\"              NA             \n",
        "# nnet          \"GPL-2 | GPL-3\"                          NA             \n",
        "# parallel      \"Part of R 4.1.1\"                        NA             \n",
        "# rpart         \"GPL-2 | GPL-3\"                          NA             \n",
        "# spatial       \"GPL-2 | GPL-3\"                          NA             \n",
        "# splines       \"Part of R 4.1.1\"                        NA             \n",
        "# stats         \"Part of R 4.1.1\"                        NA             \n",
        "# stats4        \"Part of R 4.1.1\"                        NA             \n",
        "# survival      \"LGPL (>= 2)\"                            NA             \n",
        "# tcltk         \"Part of R 4.1.1\"                        NA             \n",
        "# tools         \"Part of R 4.1.1\"                        NA             \n",
        "# utils         \"Part of R 4.1.1\"                        NA             \n",
        "#               License_restricts_use OS_type MD5sum NeedsCompilation Built  \n",
        "# IRdisplay     NA                    NA      NA     \"no\"             \"4.1.1\"\n",
        "# IRkernel      NA                    NA      NA     \"no\"             \"4.1.1\"\n",
        "# pbdZMQ        NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# repr          NA                    NA      NA     \"no\"             \"4.1.1\"\n",
        "# askpass       NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# assertthat    NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# backports     NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# base64enc     NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# bit           NA                    NA      NA     \"yes\"            \"4.0.2\"\n",
        "# bit64         NA                    NA      NA     \"yes\"            \"4.0.2\"\n",
        "# blob          NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# brew          NA                    NA      NA     NA               \"4.0.0\"\n",
        "# brio          NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# broom         NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# cachem        NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# callr         NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# cellranger    NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# cli           NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# clipr         NA                    NA      NA     \"no\"             \"4.0.2\"\n",
        "# colorspace    NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# commonmark    NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# cpp11         NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# crayon        NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# credentials   NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# curl          NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# data.table    NA                    NA      NA     \"yes\"            \"4.0.4\"\n",
        "# DBI           NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# dbplyr        NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# desc          NA                    NA      NA     \"no\"             \"4.0.4\"\n",
        "# devtools      NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# diffobj       NA                    NA      NA     \"yes\"            \"4.0.4\"\n",
        "# digest        NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# dplyr         NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# dtplyr        NA                    NA      NA     \"no\"             \"4.0.4\"\n",
        "# ellipsis      NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# evaluate      NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# fansi         NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# farver        NA                    NA      NA     \"yes\"            \"4.0.4\"\n",
        "# fastmap       NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# forcats       NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# fs            NA                    NA      NA     \"yes\"            \"4.0.2\"\n",
        "# gargle        NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# generics      NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# gert          NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# ggplot2       NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# gh            NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# gitcreds      NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# glue          NA                    NA      NA     \"yes\"            \"4.0.2\"\n",
        "# googledrive   NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# googlesheets4 NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# gtable        NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# haven         NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# highr         NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# hms           NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# htmltools     NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# httr          NA                    NA      NA     \"no\"             \"4.0.2\"\n",
        "# ids           NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# ini           NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# isoband       NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# jsonlite      NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# knitr         NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# labeling      NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# lifecycle     NA                    NA      NA     \"no\"             \"4.0.4\"\n",
        "# lubridate     NA                    NA      NA     \"yes\"            \"4.0.4\"\n",
        "# magrittr      NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# markdown      NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# memoise       NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# mime          NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# modelr        NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# munsell       NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# openssl       NA                    NA      NA     \"yes\"            \"4.0.2\"\n",
        "# pillar        NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# pkgbuild      NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# pkgconfig     NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# pkgload       NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# praise        NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# prettyunits   NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# processx      NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# progress      NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# ps            NA                    NA      NA     \"yes\"            \"4.0.4\"\n",
        "# purrr         NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# R6            NA                    NA      NA     \"no\"             \"4.1.1\"\n",
        "# rappdirs      NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# rcmdcheck     NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# RColorBrewer  NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# Rcpp          NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# readr         NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# readxl        NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# rematch       NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# rematch2      NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# remotes       NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# reprex        NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# rlang         NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# rmarkdown     NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# roxygen2      NA                    NA      NA     \"yes\"            \"4.0.2\"\n",
        "# rprojroot     NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# rstudioapi    NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# rversions     NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# rvest         NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# scales        NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# selectr       NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# sessioninfo   NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# stringi       NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# stringr       NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# svglite       NA                    NA      NA     \"yes\"            \"4.0.4\"\n",
        "# sys           NA                    NA      NA     \"yes\"            \"4.0.2\"\n",
        "# systemfonts   NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# testthat      NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# tibble        NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# tidyr         NA                    NA      NA     \"yes\"            \"4.0.4\"\n",
        "# tidyselect    NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# tidyverse     NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# tinytex       NA                    NA      NA     \"no\"             \"4.1.0\"\n",
        "# tzdb          NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# usethis       NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# utf8          NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# uuid          NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# vctrs         NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# viridisLite   NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# vroom         NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# waldo         NA                    NA      NA     \"no\"             \"4.1.1\"\n",
        "# whisker       NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# withr         NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# xfun          NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# xml2          NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# xopen         NA                    NA      NA     \"no\"             \"4.0.0\"\n",
        "# yaml          NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# zip           NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# base          NA                    NA      NA     NA               \"4.1.1\"\n",
        "# boot          NA                    NA      NA     \"no\"             \"4.0.5\"\n",
        "# class         NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# cluster       NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# codetools     NA                    NA      NA     \"no\"             \"4.0.3\"\n",
        "# compiler      NA                    NA      NA     NA               \"4.1.1\"\n",
        "# datasets      NA                    NA      NA     NA               \"4.1.1\"\n",
        "# foreign       NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# graphics      NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# grDevices     NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# grid          NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# KernSmooth    NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# lattice       NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# MASS          NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# Matrix        NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# methods       NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# mgcv          NA                    NA      NA     \"yes\"            \"4.1.0\"\n",
        "# nlme          NA                    NA      NA     \"yes\"            \"4.0.3\"\n",
        "# nnet          NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# parallel      NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# rpart         NA                    NA      NA     \"yes\"            \"4.0.0\"\n",
        "# spatial       NA                    NA      NA     \"yes\"            \"4.0.5\"\n",
        "# splines       NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# stats         NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# stats4        NA                    NA      NA     NA               \"4.1.1\"\n",
        "# survival      NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# tcltk         NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# tools         NA                    NA      NA     \"yes\"            \"4.1.1\"\n",
        "# utils         NA                    NA      NA     \"yes\"            \"4.1.1\""
      ],
      "execution_count": null,
      "outputs": []
    }
  ]
}